{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "This notebook performs specific alignments on primary contigs and their associated haplotigs. Adapted from an original script by Benjamin Schwessinger."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO \n",
    "import shutil\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define the PATH for variable parameters\n",
    "\n",
    "GENOME_VERSION = 'v04'\n",
    "\n",
    "BASE_OUT_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/%s' % GENOME_VERSION\n",
    "GENOME_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/genome_%s/' % GENOME_VERSION\n",
    "GENOME  = 'DK_0911_%s' % GENOME_VERSION\n",
    "OUT_PATH_NUCMER = os.path.join(BASE_OUT_PATH, 'nucmer_analysis/')\n",
    "OUT_PATH_ASSEMBLYTICS = os.path.join(BASE_OUT_PATH, 'Assemblytics/')\n",
    "MUMMER_PATH_PREFIX = '/home/benjamin/anaconda3/bin/'\n",
    "ASSEMBLYTICS_PATH = '/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics'\n",
    "py2_env = 'py27' #python 2 environment in conda \n",
    "if not os.path.isdir(BASE_OUT_PATH):\n",
    "    os.mkdir(BASE_OUT_PATH)\n",
    "if not os.path.isdir(OUT_PATH_NUCMER):\n",
    "    os.mkdir(OUT_PATH_NUCMER)\n",
    "if not os.path.isdir(OUT_PATH_ASSEMBLYTICS):\n",
    "    os.mkdir(OUT_PATH_ASSEMBLYTICS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "P_GENOME = GENOME + '_p_ctg'\n",
    "H_GENOME = GENOME + '_h_ctg'\n",
    "\n",
    "for x in (x + '.fa' for x in [P_GENOME, H_GENOME]):\n",
    "    shutil.copy2(GENOME_PATH + x, OUT_PATH_NUCMER)\n",
    "    shutil.copy2(GENOME_PATH + x, OUT_PATH_ASSEMBLYTICS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "#define the scripts to generate\n",
    "bash_script_q= GENOME+\"_ph_ctg_qmapping.sh\"\n",
    "bash_script_g=GENOME+\"_ph_ctg_gmapping.sh\"\n",
    "bash_script_nucmer_assemblytics = GENOME +\"_nucmer_assemblytics_mapping.sh\"\n",
    "bash_script_assemblytics = GENOME + '_assemblytics.sh'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfq = open(os.path.join(OUT_PATH_NUCMER, bash_script_q), 'w')\n",
    "outfq.write('#!/bin/bash\\n')\n",
    "outfg = open(os.path.join(OUT_PATH_NUCMER,bash_script_g), 'w')\n",
    "outfg.write('#!/bin/bash\\n') #parsing out P and corresponding h contigs and writing a short nucmer script that aligns them against each other\n",
    "outfna = open(os.path.join(OUT_PATH_ASSEMBLYTICS,bash_script_nucmer_assemblytics), 'w')\n",
    "outfna.write('#!/bin/bash\\n')\n",
    "\n",
    "for pseq_record in SeqIO.parse(OUT_PATH_ASSEMBLYTICS+'/'+GENOME+'_p_ctg.fa', 'fasta'):\n",
    "    p_acontigs = []\n",
    "    p_contig = pseq_record.id.split(\"_\")[0]+\"_\"+pseq_record.id.split(\"_\")[1]\n",
    "    suffix = GENOME+\"_\"+p_contig+\"_php\"\n",
    "    p_file = GENOME+\"_\"+p_contig+'.fa'\n",
    "    SeqIO.write(pseq_record, OUT_PATH_ASSEMBLYTICS+'/'+ p_file, 'fasta')\n",
    "    SeqIO.write(pseq_record,OUT_PATH_NUCMER+'/'+ p_file, 'fasta')\n",
    "    for aseq_record in SeqIO.parse(OUT_PATH_ASSEMBLYTICS+'/'+GENOME+'_h_ctg.fa', 'fasta'):\n",
    "        if aseq_record.id.split(\"_\")[1]  == pseq_record.id.split(\"_\")[1]:\n",
    "            p_acontigs.append(aseq_record)\n",
    "    a_file = GENOME +\"_\"+pseq_record.id.split(\"_\")[0]+\"_\"+pseq_record.id.split(\"_\")[1]+'_h_ctgs.fa'\n",
    "    #if we have alternative contigs save those too\n",
    "    if p_acontigs != []:\n",
    "        outfq.write('cd ' + OUT_PATH_NUCMER + '\\n')\n",
    "        SeqIO.write(p_acontigs, OUT_PATH_ASSEMBLYTICS+'/'+  a_file, 'fasta')\n",
    "        SeqIO.write(p_acontigs, OUT_PATH_NUCMER+'/'+  a_file, 'fasta')\n",
    "        outfq.write(MUMMER_PATH_PREFIX +'/nucmer '+p_file+' '+a_file+\" > \"+'out.delta\\n')\n",
    "        outfq.write(MUMMER_PATH_PREFIX +'/delta-filter -q '+'out.delta'+\" > \"+suffix+\"_qfiltered.delta\\n\")\n",
    "        outfq.write(MUMMER_PATH_PREFIX +'/show-coords -T '+suffix+\"_qfiltered.delta > \"+suffix+\".qcoords\\n\")\n",
    "        outfq.write(MUMMER_PATH_PREFIX +'/mummerplot -p '+suffix+'_qfiltered --png '+suffix+\"_qfiltered.delta\\n\")\n",
    "        outfq.write(MUMMER_PATH_PREFIX +'/mummerplot -c -p '+suffix+'_qfiltered_cov --png '+suffix+\"_qfiltered.delta\\n\")\n",
    "        #for g_file bash script\n",
    "        outfg.write('cd ' + OUT_PATH_NUCMER + '\\n')\n",
    "        outfg.write(MUMMER_PATH_PREFIX +'/nucmer '+p_file+' '+a_file+\" > \"+'out.delta\\n')\n",
    "        outfg.write(MUMMER_PATH_PREFIX +'/delta-filter -g '+'out.delta'+\" > \"+suffix+\"_gfiltered.delta\\n\")\n",
    "        outfg.write(MUMMER_PATH_PREFIX +'/show-coords -T '+suffix+\"_gfiltered.delta > \"+suffix+\".gcoords\\n\")\n",
    "        outfg.write(MUMMER_PATH_PREFIX +'/mummerplot -p '+suffix+'_gfiltered --png '+suffix+\"_gfiltered.delta\\n\")\n",
    "        outfg.write(MUMMER_PATH_PREFIX +'/mummerplot -c -p  '+suffix+'_gfiltered_cov --png '+suffix+\"_gfiltered.delta\\n\")\n",
    "        #for nucmer assemblytics out\n",
    "        outfna.write('cd ' + OUT_PATH_ASSEMBLYTICS + '\\n')\n",
    "        outfna.write(MUMMER_PATH_PREFIX +'/nucmer -maxmatch -l 100 -c 500 '+p_file+' '+a_file+' -prefix ' + suffix +'\\n')\n",
    "outfna.close()\n",
    "outfq.close()\n",
    "outfg.close()  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run the scripts and check if there are errors\n",
    "\n",
    "bash_script_q_stderr = subprocess.check_output('bash %s' %os.path.join(OUT_PATH_NUCMER, bash_script_q), shell=True, stderr=subprocess.STDOUT)\n",
    "bash_script_g_stderr = subprocess.check_output('bash %s' %os.path.join(OUT_PATH_NUCMER, bash_script_g), shell=True, stderr=subprocess.STDOUT)\n",
    "bash_script_assemblytics_stderr = subprocess.check_output('bash %s' %os.path.join(OUT_PATH_ASSEMBLYTICS, bash_script_nucmer_assemblytics), shell=True, stderr=subprocess.STDOUT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#write the Assemblytics script\n",
    "\n",
    "folder_locs = []\n",
    "\n",
    "outfnarun = open(os.path.join(OUT_PATH_ASSEMBLYTICS, bash_script_assemblytics), 'w')\n",
    "outfnarun.write('#!/bin/bash\\n')\n",
    "delta_files = [x for x in os.listdir(OUT_PATH_ASSEMBLYTICS) if x.endswith('delta')]\n",
    "outfnarun.write('export PATH=\"%s\":$PATH\\n'% ASSEMBLYTICS_PATH)\n",
    "outfnarun.write('#Assemblytics delta_file output_prefix unique_anchor_length maximum_feature_length path_to_R_scripts\\n')\n",
    "outfnarun.write('source activate %s\\n' %py2_env)\n",
    "outfnarun.write('cd %s\\n' %OUT_PATH_ASSEMBLYTICS)\n",
    "\n",
    "for delta in delta_files:\n",
    "    folder_name = OUT_PATH_ASSEMBLYTICS + delta[:-6] + '_8kbp'\n",
    "    \n",
    "    folder_locs.append(folder_name)\n",
    "    \n",
    "    output_prefix = delta[:-6] + '_8kbp_50kp'\n",
    "    outfnarun.write(\"mkdir %s\\ncp %s %s\\ncd %s\\n\" % (folder_name, delta, folder_name, folder_name))\n",
    "    outfnarun.write(\"Assemblytics %s %s 8000 50000 %s\\n\" % (delta, output_prefix, ASSEMBLYTICS_PATH) )\n",
    "    output_prefix = delta[:-6] + '_8kbp_10kp'\n",
    "    outfnarun.write(\"Assemblytics %s %s 8000 10000 %s\\ncd ..\\n\" % (delta, output_prefix, ASSEMBLYTICS_PATH) )\n",
    "outfnarun.write('source deactivate\\n')\n",
    "outfnarun.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Input delta file: DK_0911_v04_pcontig_083_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_083_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_083_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_083.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_083_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 252 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_083_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_083_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_083_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_083.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_083_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 252 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_008_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_008_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_008_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_008.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_008_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 39 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 39 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5114 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 61 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_008_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_008_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_008_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_008.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_008_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 39 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 39 queries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5114 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 61 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_094_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_094_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_094_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_094.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_094_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 35 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_094_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_094_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_094_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_094.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_094_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 35 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_093_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_093_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_093_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_093.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_093_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 250 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_093_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_093_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_093_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_093.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_093_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 250 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_058_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_058_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_058_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_058.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_058_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1276 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 17 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_058_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_058_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_058_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_058.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_058_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1276 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 17 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_075_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_075_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_075_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_075.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_075_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 0 queries\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 0 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 0 total lines in file\n",
      "/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2889: RuntimeWarning: Mean of empty slice.\n",
      "  out=out, **kwargs)\n",
      "/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/_methods.py:80: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics/Assemblytics_uniq_anchor.py\", line 463, in <module>\n",
      "    main()\n",
      "  File \"/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics/Assemblytics_uniq_anchor.py\", line 460, in main\n",
      "    args.func(args)\n",
      "  File \"/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics/Assemblytics_uniq_anchor.py\", line 250, in run\n",
      "    f_stats_out.write( \"Min: %s\\n\" % gig_meg(np.min(ref_lengths)))\n",
      "  File \"/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/fromnumeric.py\", line 2352, in amin\n",
      "    out=out, **kwargs)\n",
      "  File \"/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/_methods.py\", line 29, in _amin\n",
      "    return umr_minimum(a, axis, None, out, keepdims)\n",
      "ValueError: zero-size array to reduction operation minimum which has no identity\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 0 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Error in revalue(bed$type, c(Repeat_expansion = \"Repeat expansion\", Repeat_contraction = \"Repeat contraction\",  : \n",
      "  x is not a factor or a character vector.\n",
      "Execution halted\n",
      "Error in order(ref.loc.of.longest.alignment.by.query) : \n",
      "  argument 1 is not a vector\n",
      "Execution halted\n",
      "Error in read.table(file = file, header = header, sep = sep, quote = quote,  : \n",
      "  no lines available in input\n",
      "Calls: read.csv -> read.table\n",
      "Execution halted\n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_075_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_075_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_075_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_075.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_075_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 0 queries\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 0 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 0 total lines in file\n",
      "/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2889: RuntimeWarning: Mean of empty slice.\n",
      "  out=out, **kwargs)\n",
      "/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/_methods.py:80: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics/Assemblytics_uniq_anchor.py\", line 463, in <module>\n",
      "    main()\n",
      "  File \"/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics/Assemblytics_uniq_anchor.py\", line 460, in main\n",
      "    args.func(args)\n",
      "  File \"/home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics/Assemblytics_uniq_anchor.py\", line 250, in run\n",
      "    f_stats_out.write( \"Min: %s\\n\" % gig_meg(np.min(ref_lengths)))\n",
      "  File \"/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/fromnumeric.py\", line 2352, in amin\n",
      "    out=out, **kwargs)\n",
      "  File \"/home/gamran/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/_methods.py\", line 29, in _amin\n",
      "    return umr_minimum(a, axis, None, out, keepdims)\n",
      "ValueError: zero-size array to reduction operation minimum which has no identity\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 0 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Error in revalue(bed$type, c(Repeat_expansion = \"Repeat expansion\", Repeat_contraction = \"Repeat contraction\",  : \n",
      "  x is not a factor or a character vector.\n",
      "Execution halted\n",
      "Error in order(ref.loc.of.longest.alignment.by.query) : \n",
      "  argument 1 is not a vector\n",
      "Execution halted\n",
      "Error in read.table(file = file, header = header, sep = sep, quote = quote,  : \n",
      "  no lines available in input\n",
      "Calls: read.csv -> read.table\n",
      "Execution halted\n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_088_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_088_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_088_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_088.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_088_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 270 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_088_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_088_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_088_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_088.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_088_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 270 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_018_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_018_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_018_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_018.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_018_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 22 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 22 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3046 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 36 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_018_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_018_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_018_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_018.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_018_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 22 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 22 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3046 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 36 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_024_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_024_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_024_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_024.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_024_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 20 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 20 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2864 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 28 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_024_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_024_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_024_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_024.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_024_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 20 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 20 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2864 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 28 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_026_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_026_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_026_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_026.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_026_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 18 queries\n",
      "Progress: 5%\n",
      "Progress: 11%\n",
      "Progress: 16%\n",
      "Progress: 22%\n",
      "Progress: 27%\n",
      "Progress: 33%\n",
      "Progress: 38%\n",
      "Progress: 44%\n",
      "Progress: 50%\n",
      "Progress: 55%\n",
      "Progress: 61%\n",
      "Progress: 66%\n",
      "Progress: 72%\n",
      "Progress: 77%\n",
      "Progress: 83%\n",
      "Progress: 88%\n",
      "Progress: 94%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 18 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2000 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 30 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_026_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_026_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_026_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_026.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_026_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 18 queries\n",
      "Progress: 5%\n",
      "Progress: 11%\n",
      "Progress: 16%\n",
      "Progress: 22%\n",
      "Progress: 27%\n",
      "Progress: 33%\n",
      "Progress: 38%\n",
      "Progress: 44%\n",
      "Progress: 50%\n",
      "Progress: 55%\n",
      "Progress: 61%\n",
      "Progress: 66%\n",
      "Progress: 72%\n",
      "Progress: 77%\n",
      "Progress: 83%\n",
      "Progress: 88%\n",
      "Progress: 94%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 18 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2000 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 30 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_012_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_012_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_012_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_012.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_012_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 25 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 24%\n",
      "Progress: 32%\n",
      "Progress: 40%\n",
      "Progress: 48%\n",
      "Progress: 56%\n",
      "Progress: 64%\n",
      "Progress: 72%\n",
      "Progress: 80%\n",
      "Progress: 88%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 25 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3807 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 55 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_012_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_012_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_012_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_012.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_012_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 25 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 24%\n",
      "Progress: 32%\n",
      "Progress: 40%\n",
      "Progress: 48%\n",
      "Progress: 56%\n",
      "Progress: 64%\n",
      "Progress: 72%\n",
      "Progress: 80%\n",
      "Progress: 88%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 25 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3807 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 55 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_042_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_042_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_042_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_042.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_042_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 13 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 13 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1615 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 22 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_042_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_042_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_042_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_042.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_042_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 13 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 13 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1615 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 22 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_048_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_048_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_048_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_048.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_048_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1180 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 17 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_048_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_048_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_048_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_048.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_048_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1180 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 17 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_053_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_053_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_053_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_053.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_053_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 730 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 14 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_053_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_053_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_053_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_053.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_053_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 730 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 14 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_017_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_017_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_017_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_017.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_017_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 22 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 22 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3749 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 43 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_017_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_017_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_017_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_017.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_017_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 22 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 22 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3749 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 43 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_009_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_009_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_009_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_009.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_009_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 27 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 22%\n",
      "Progress: 29%\n",
      "Progress: 37%\n",
      "Progress: 44%\n",
      "Progress: 51%\n",
      "Progress: 59%\n",
      "Progress: 66%\n",
      "Progress: 74%\n",
      "Progress: 81%\n",
      "Progress: 88%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 27 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5169 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 61 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_009_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_009_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_009_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_009.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_009_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 27 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 22%\n",
      "Progress: 29%\n",
      "Progress: 37%\n",
      "Progress: 44%\n",
      "Progress: 51%\n",
      "Progress: 59%\n",
      "Progress: 66%\n",
      "Progress: 74%\n",
      "Progress: 81%\n",
      "Progress: 88%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 27 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5169 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 61 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_005_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_005_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_005_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_005.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_005_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 37 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 24%\n",
      "Progress: 32%\n",
      "Progress: 40%\n",
      "Progress: 48%\n",
      "Progress: 56%\n",
      "Progress: 64%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 89%\n",
      "Progress: 97%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 37 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5957 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 79 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_005_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_005_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_005_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_005.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_005_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 37 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 24%\n",
      "Progress: 32%\n",
      "Progress: 40%\n",
      "Progress: 48%\n",
      "Progress: 56%\n",
      "Progress: 64%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 89%\n",
      "Progress: 97%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 37 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5957 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 79 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_072_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_072_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_072_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_072.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_072_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 323 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 7 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_072_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_072_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_072_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_072.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_072_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 323 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 7 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_050_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_050_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_050_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_050.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_050_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 557 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 14 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_050_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_050_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_050_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_050.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_050_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 557 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 14 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_095_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_095_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_095_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_095.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_095_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 150 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_095_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_095_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_095_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_095.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_095_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 150 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_051_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_051_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_051_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_051.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_051_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1668 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 20 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_051_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_051_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_051_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_051.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_051_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1668 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 20 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_076_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_076_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_076_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_076.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_076_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 460 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 4 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_076_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_076_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_076_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_076.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_076_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 460 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 4 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_001_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_001_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_001_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_001.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_001_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 40 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 40 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5380 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 80 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_001_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_001_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_001_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_001.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_001_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 40 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 40 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 5380 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 80 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_037_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_037_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_037_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_037.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_037_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 20 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 20 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1954 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 25 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_037_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_037_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_037_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_037.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_037_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 20 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 20 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1954 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 25 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_031_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_031_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_031_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_031.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_031_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 16 queries\n",
      "Progress: 6%\n",
      "Progress: 12%\n",
      "Progress: 18%\n",
      "Progress: 25%\n",
      "Progress: 31%\n",
      "Progress: 37%\n",
      "Progress: 43%\n",
      "Progress: 50%\n",
      "Progress: 56%\n",
      "Progress: 62%\n",
      "Progress: 68%\n",
      "Progress: 75%\n",
      "Progress: 81%\n",
      "Progress: 87%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 16 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1433 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 22 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_031_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_031_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_031_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_031.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_031_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 16 queries\n",
      "Progress: 6%\n",
      "Progress: 12%\n",
      "Progress: 18%\n",
      "Progress: 25%\n",
      "Progress: 31%\n",
      "Progress: 37%\n",
      "Progress: 43%\n",
      "Progress: 50%\n",
      "Progress: 56%\n",
      "Progress: 62%\n",
      "Progress: 68%\n",
      "Progress: 75%\n",
      "Progress: 81%\n",
      "Progress: 87%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 16 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1433 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 22 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_070_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_070_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_070_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_070.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_070_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1053 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_070_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_070_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_070_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_070.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_070_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1053 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_054_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_054_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_054_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_054.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_054_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1596 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 14 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_054_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_054_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_054_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_054.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_054_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1596 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 14 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_000_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_000_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_000_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_000.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_000_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 67 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 26%\n",
      "Progress: 35%\n",
      "Progress: 44%\n",
      "Progress: 53%\n",
      "Progress: 62%\n",
      "Progress: 71%\n",
      "Progress: 80%\n",
      "Progress: 89%\n",
      "Progress: 98%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 67 queries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading file and recording all the entries we decided to keep: 0 seconds for 11565 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 123 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_000_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_000_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_000_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_000.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_000_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 67 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 26%\n",
      "Progress: 35%\n",
      "Progress: 44%\n",
      "Progress: 53%\n",
      "Progress: 62%\n",
      "Progress: 71%\n",
      "Progress: 80%\n",
      "Progress: 89%\n",
      "Progress: 98%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 67 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 11565 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 123 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_069_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_069_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_069_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_069.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_069_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 340 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_069_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_069_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_069_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_069.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_069_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 340 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_084_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_084_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_084_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_084.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_084_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 376 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_084_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_084_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_084_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_084.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_084_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 376 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 3 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_081_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_081_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_081_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_081.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_081_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 5 queries\n",
      "Progress: 20%\n",
      "Progress: 40%\n",
      "Progress: 60%\n",
      "Progress: 80%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 5 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 733 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 5 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_081_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_081_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_081_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_081.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_081_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 5 queries\n",
      "Progress: 20%\n",
      "Progress: 40%\n",
      "Progress: 60%\n",
      "Progress: 80%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 5 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 733 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 5 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_019_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_019_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_019_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_019.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_019_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 24 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 24 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3292 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 40 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_019_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_019_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_019_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_019.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_019_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 24 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 24 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3292 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 40 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_064_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_064_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_064_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_064.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_064_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 41 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_064_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_064_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_064_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_064.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_064_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 41 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_004_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_004_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_004_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_004.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_004_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 41 queries\n",
      "Progress: 9%\n",
      "Progress: 19%\n",
      "Progress: 29%\n",
      "Progress: 39%\n",
      "Progress: 48%\n",
      "Progress: 58%\n",
      "Progress: 68%\n",
      "Progress: 78%\n",
      "Progress: 87%\n",
      "Progress: 97%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 41 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 6197 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 69 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_004_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_004_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_004_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_004.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_004_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 41 queries\n",
      "Progress: 9%\n",
      "Progress: 19%\n",
      "Progress: 29%\n",
      "Progress: 39%\n",
      "Progress: 48%\n",
      "Progress: 58%\n",
      "Progress: 68%\n",
      "Progress: 78%\n",
      "Progress: 87%\n",
      "Progress: 97%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 41 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 6197 total lines in file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 69 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_029_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_029_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_029_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_029.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_029_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 13 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 13 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2004 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 34 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_029_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_029_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_029_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_029.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_029_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 13 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 13 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2004 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 34 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_063_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_063_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_063_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_063.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_063_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 20 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_063_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_063_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_063_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_063.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_063_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 20 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_035_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_035_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_035_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_035.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_035_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 15 queries\n",
      "Progress: 6%\n",
      "Progress: 13%\n",
      "Progress: 20%\n",
      "Progress: 26%\n",
      "Progress: 33%\n",
      "Progress: 40%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 60%\n",
      "Progress: 66%\n",
      "Progress: 73%\n",
      "Progress: 80%\n",
      "Progress: 86%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 15 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2645 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 32 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_035_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_035_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_035_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_035.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_035_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 15 queries\n",
      "Progress: 6%\n",
      "Progress: 13%\n",
      "Progress: 20%\n",
      "Progress: 26%\n",
      "Progress: 33%\n",
      "Progress: 40%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 60%\n",
      "Progress: 66%\n",
      "Progress: 73%\n",
      "Progress: 80%\n",
      "Progress: 86%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 15 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2645 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 32 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_020_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_020_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_020_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_020.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_020_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 21 queries\n",
      "Progress: 9%\n",
      "Progress: 19%\n",
      "Progress: 28%\n",
      "Progress: 38%\n",
      "Progress: 47%\n",
      "Progress: 57%\n",
      "Progress: 66%\n",
      "Progress: 76%\n",
      "Progress: 85%\n",
      "Progress: 95%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 21 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2690 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 40 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_020_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_020_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_020_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_020.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_020_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 21 queries\n",
      "Progress: 9%\n",
      "Progress: 19%\n",
      "Progress: 28%\n",
      "Progress: 38%\n",
      "Progress: 47%\n",
      "Progress: 57%\n",
      "Progress: 66%\n",
      "Progress: 76%\n",
      "Progress: 85%\n",
      "Progress: 95%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 21 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2690 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 40 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_078_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_078_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_078_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_078.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_078_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 372 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 4 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_078_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_078_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_078_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_078.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_078_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 372 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 4 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_049_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_049_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_049_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_049.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_049_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 9 queries\n",
      "Progress: 11%\n",
      "Progress: 22%\n",
      "Progress: 33%\n",
      "Progress: 44%\n",
      "Progress: 55%\n",
      "Progress: 66%\n",
      "Progress: 77%\n",
      "Progress: 88%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 9 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1221 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_049_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_049_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_049_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_049.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_049_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 9 queries\n",
      "Progress: 11%\n",
      "Progress: 22%\n",
      "Progress: 33%\n",
      "Progress: 44%\n",
      "Progress: 55%\n",
      "Progress: 66%\n",
      "Progress: 77%\n",
      "Progress: 88%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 9 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1221 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_041_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_041_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_041_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_041.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_041_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 10 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 10 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1028 total lines in file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 15 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_041_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_041_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_041_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_041.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_041_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 10 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 10 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1028 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 15 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_066_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_066_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_066_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_066.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_066_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 5 queries\n",
      "Progress: 20%\n",
      "Progress: 40%\n",
      "Progress: 60%\n",
      "Progress: 80%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 5 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 322 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 8 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_066_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_066_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_066_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_066.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_066_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 5 queries\n",
      "Progress: 20%\n",
      "Progress: 40%\n",
      "Progress: 60%\n",
      "Progress: 80%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 5 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 322 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 8 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_101_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_101_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_101_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_101.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_101_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 130 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_101_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_101_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_101_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_101.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_101_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 130 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_015_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_015_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_015_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_015.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_015_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 27 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 22%\n",
      "Progress: 29%\n",
      "Progress: 37%\n",
      "Progress: 44%\n",
      "Progress: 51%\n",
      "Progress: 59%\n",
      "Progress: 66%\n",
      "Progress: 74%\n",
      "Progress: 81%\n",
      "Progress: 88%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 27 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 4036 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 53 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_015_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_015_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_015_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_015.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_015_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 27 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 22%\n",
      "Progress: 29%\n",
      "Progress: 37%\n",
      "Progress: 44%\n",
      "Progress: 51%\n",
      "Progress: 59%\n",
      "Progress: 66%\n",
      "Progress: 74%\n",
      "Progress: 81%\n",
      "Progress: 88%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 27 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 4036 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 53 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_011_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_011_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_011_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_011.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_011_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 31 queries\n",
      "Progress: 9%\n",
      "Progress: 19%\n",
      "Progress: 29%\n",
      "Progress: 38%\n",
      "Progress: 48%\n",
      "Progress: 58%\n",
      "Progress: 67%\n",
      "Progress: 77%\n",
      "Progress: 87%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 31 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3912 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 58 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_011_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_011_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_011_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_011.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_011_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 31 queries\n",
      "Progress: 9%\n",
      "Progress: 19%\n",
      "Progress: 29%\n",
      "Progress: 38%\n",
      "Progress: 48%\n",
      "Progress: 58%\n",
      "Progress: 67%\n",
      "Progress: 77%\n",
      "Progress: 87%\n",
      "Progress: 96%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 31 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3912 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 58 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_034_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_034_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_034_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_034.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_034_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 16 queries\n",
      "Progress: 6%\n",
      "Progress: 12%\n",
      "Progress: 18%\n",
      "Progress: 25%\n",
      "Progress: 31%\n",
      "Progress: 37%\n",
      "Progress: 43%\n",
      "Progress: 50%\n",
      "Progress: 56%\n",
      "Progress: 62%\n",
      "Progress: 68%\n",
      "Progress: 75%\n",
      "Progress: 81%\n",
      "Progress: 87%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 16 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2521 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 37 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_034_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_034_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_034_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_034.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_034_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 16 queries\n",
      "Progress: 6%\n",
      "Progress: 12%\n",
      "Progress: 18%\n",
      "Progress: 25%\n",
      "Progress: 31%\n",
      "Progress: 37%\n",
      "Progress: 43%\n",
      "Progress: 50%\n",
      "Progress: 56%\n",
      "Progress: 62%\n",
      "Progress: 68%\n",
      "Progress: 75%\n",
      "Progress: 81%\n",
      "Progress: 87%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 16 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2521 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 37 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_046_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_046_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_046_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_046.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_046_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 792 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_046_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_046_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_046_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_046.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_046_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 7 queries\n",
      "Progress: 14%\n",
      "Progress: 28%\n",
      "Progress: 42%\n",
      "Progress: 57%\n",
      "Progress: 71%\n",
      "Progress: 85%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 7 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 792 total lines in file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_032_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_032_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_032_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_032.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_032_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 11 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 11 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2910 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 27 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_032_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_032_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_032_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_032.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_032_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 11 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 11 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2910 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 27 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_021_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_021_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_021_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_021.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_021_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 24 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 24 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3295 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 45 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_021_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_021_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_021_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_021.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_021_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 24 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 24 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3295 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 45 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_023_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_023_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_023_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_023.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_023_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 23 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 26%\n",
      "Progress: 34%\n",
      "Progress: 43%\n",
      "Progress: 52%\n",
      "Progress: 60%\n",
      "Progress: 69%\n",
      "Progress: 78%\n",
      "Progress: 86%\n",
      "Progress: 95%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 23 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3230 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 46 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_023_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_023_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_023_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_023.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_023_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 23 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 26%\n",
      "Progress: 34%\n",
      "Progress: 43%\n",
      "Progress: 52%\n",
      "Progress: 60%\n",
      "Progress: 69%\n",
      "Progress: 78%\n",
      "Progress: 86%\n",
      "Progress: 95%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 23 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3230 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 46 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_006_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_006_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_006_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_006.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_006_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 36 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 36 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 4925 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 59 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_006_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_006_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_006_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_006.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_006_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 36 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 36 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 4925 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 59 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_073_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_073_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_073_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_073.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_073_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 222 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_073_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_073_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_073_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_073.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_073_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 222 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_068_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_068_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_068_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_068.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_068_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 5 queries\n",
      "Progress: 20%\n",
      "Progress: 40%\n",
      "Progress: 60%\n",
      "Progress: 80%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 5 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 578 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_068_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_068_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_068_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_068.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_068_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 5 queries\n",
      "Progress: 20%\n",
      "Progress: 40%\n",
      "Progress: 60%\n",
      "Progress: 80%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 5 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 578 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_043_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_043_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_043_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_043.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_043_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 61 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_043_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_043_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_043_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_043.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_043_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 61 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_089_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_089_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_089_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_089.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_089_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 163 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_089_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_089_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_089_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_089.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_089_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 163 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_047_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_047_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_047_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_047.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_047_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1012 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 9 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_047_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_047_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_047_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_047.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_047_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1012 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 9 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_003_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_003_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_003_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_003.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_003_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 45 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 26%\n",
      "Progress: 35%\n",
      "Progress: 44%\n",
      "Progress: 53%\n",
      "Progress: 62%\n",
      "Progress: 71%\n",
      "Progress: 80%\n",
      "Progress: 88%\n",
      "Progress: 97%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 45 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 6762 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 83 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_003_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_003_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_003_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_003.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_003_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 45 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 26%\n",
      "Progress: 35%\n",
      "Progress: 44%\n",
      "Progress: 53%\n",
      "Progress: 62%\n",
      "Progress: 71%\n",
      "Progress: 80%\n",
      "Progress: 88%\n",
      "Progress: 97%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 45 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 6762 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 83 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_074_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_074_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_074_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_074.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_074_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 168 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 7 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_074_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_074_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_074_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_074.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_074_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 168 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 7 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_038_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_038_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_038_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_038.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_038_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 10 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 10 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1061 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_038_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_038_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_038_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_038.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_038_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 10 queries\n",
      "Progress: 10%\n",
      "Progress: 20%\n",
      "Progress: 30%\n",
      "Progress: 40%\n",
      "Progress: 50%\n",
      "Progress: 60%\n",
      "Progress: 70%\n",
      "Progress: 80%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 10 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1061 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_013_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_013_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_013_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_013.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_013_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 28 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 21%\n",
      "Progress: 28%\n",
      "Progress: 35%\n",
      "Progress: 42%\n",
      "Progress: 50%\n",
      "Progress: 57%\n",
      "Progress: 64%\n",
      "Progress: 71%\n",
      "Progress: 78%\n",
      "Progress: 85%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 28 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 4623 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 65 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_013_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_013_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_013_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_013.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_013_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 28 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 21%\n",
      "Progress: 28%\n",
      "Progress: 35%\n",
      "Progress: 42%\n",
      "Progress: 50%\n",
      "Progress: 57%\n",
      "Progress: 64%\n",
      "Progress: 71%\n",
      "Progress: 78%\n",
      "Progress: 85%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 28 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 4623 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 65 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_036_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_036_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_036_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_036.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_036_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 11 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 11 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1077 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 19 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_036_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_036_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_036_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_036.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_036_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 11 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 11 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1077 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 19 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_039_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_039_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_039_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_039.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_039_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 14 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 21%\n",
      "Progress: 28%\n",
      "Progress: 35%\n",
      "Progress: 42%\n",
      "Progress: 50%\n",
      "Progress: 57%\n",
      "Progress: 64%\n",
      "Progress: 71%\n",
      "Progress: 78%\n",
      "Progress: 85%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 14 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2069 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 26 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_039_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_039_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_039_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_039.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_039_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 14 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 21%\n",
      "Progress: 28%\n",
      "Progress: 35%\n",
      "Progress: 42%\n",
      "Progress: 50%\n",
      "Progress: 57%\n",
      "Progress: 64%\n",
      "Progress: 71%\n",
      "Progress: 78%\n",
      "Progress: 85%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 14 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2069 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 26 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_055_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_055_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_055_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_055.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_055_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 765 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_055_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_055_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_055_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_055.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_055_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 3 queries\n",
      "Progress: 33%\n",
      "Progress: 66%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 3 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 765 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_027_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_027_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_027_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_027.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_027_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 741 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 15 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_027_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_027_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_027_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_027.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_027_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 741 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 15 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_010_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_010_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_010_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_010.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_010_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 32 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 28%\n",
      "Progress: 37%\n",
      "Progress: 46%\n",
      "Progress: 56%\n",
      "Progress: 65%\n",
      "Progress: 75%\n",
      "Progress: 84%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 32 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3778 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 46 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_010_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_010_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_010_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_010.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_010_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 32 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 28%\n",
      "Progress: 37%\n",
      "Progress: 46%\n",
      "Progress: 56%\n",
      "Progress: 65%\n",
      "Progress: 75%\n",
      "Progress: 84%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 32 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 3778 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 46 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_067_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_067_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_067_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_067.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_067_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 484 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 6 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_067_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_067_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_067_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_067.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_067_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 484 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 6 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_033_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_033_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_033_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_033.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_033_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 14 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 21%\n",
      "Progress: 28%\n",
      "Progress: 35%\n",
      "Progress: 42%\n",
      "Progress: 50%\n",
      "Progress: 57%\n",
      "Progress: 64%\n",
      "Progress: 71%\n",
      "Progress: 78%\n",
      "Progress: 85%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 14 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2032 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 31 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_033_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_033_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_033_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_033.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_033_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 14 queries\n",
      "Progress: 7%\n",
      "Progress: 14%\n",
      "Progress: 21%\n",
      "Progress: 28%\n",
      "Progress: 35%\n",
      "Progress: 42%\n",
      "Progress: 50%\n",
      "Progress: 57%\n",
      "Progress: 64%\n",
      "Progress: 71%\n",
      "Progress: 78%\n",
      "Progress: 85%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 14 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2032 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 31 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_082_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_082_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_082_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_082.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_082_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 304 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 6 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_082_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_082_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_082_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_082.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_082_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 304 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 6 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_107_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_107_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_107_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_107.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_107_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 91 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_107_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_107_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_107_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_107.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_107_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 91 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_052_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_052_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_052_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_052.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_052_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 688 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 10 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_052_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_052_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_052_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_052.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_052_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 688 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 10 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_079_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_079_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_079_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_079.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_079_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 316 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 4 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_079_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_079_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_079_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_079.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_079_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 4 queries\n",
      "Progress: 25%\n",
      "Progress: 50%\n",
      "Progress: 75%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 4 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 316 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 4 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_060_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_060_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_060_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_060.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_060_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 788 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_060_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_060_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_060_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_060.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_060_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 788 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 11 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_090_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_090_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_090_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_090.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_090_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 82 total lines in file\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_090_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_090_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_090_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_090.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_090_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 82 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_040_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_040_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_040_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_040.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_040_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 15 queries\n",
      "Progress: 6%\n",
      "Progress: 13%\n",
      "Progress: 20%\n",
      "Progress: 26%\n",
      "Progress: 33%\n",
      "Progress: 40%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 60%\n",
      "Progress: 66%\n",
      "Progress: 73%\n",
      "Progress: 80%\n",
      "Progress: 86%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 15 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2071 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 23 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_040_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_040_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_040_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_040.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_040_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 15 queries\n",
      "Progress: 6%\n",
      "Progress: 13%\n",
      "Progress: 20%\n",
      "Progress: 26%\n",
      "Progress: 33%\n",
      "Progress: 40%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 60%\n",
      "Progress: 66%\n",
      "Progress: 73%\n",
      "Progress: 80%\n",
      "Progress: 86%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 15 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2071 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 23 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_087_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_087_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_087_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_087.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_087_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 51 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_087_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_087_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_087_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_087.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_087_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 2 queries\n",
      "Progress: 50%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 2 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 51 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 2 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_025_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_025_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_025_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_025.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_025_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 7 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_025_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_025_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_025_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_025.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_025_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 7 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_106_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_106_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_106_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_106.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_106_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 15 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 50000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_106_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_106_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_106_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_106.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_106_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 1 queries\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 1 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 15 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 1 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction, Tandem_expansion, Tandem_contraction\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 10\"\n",
      "[1] \"max_var= 50\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 50\"\n",
      "[1] \"max_var= 500\"\n",
      "[1] \"No variants in plot:\"\n",
      "[1] \"min_var= 500\"\n",
      "[1] \"max_var= 10000\"\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_016_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_016_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_016_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_016.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_016_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 24 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 24 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2815 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 43 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_016_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_016_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_016_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_016.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_016_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 24 queries\n",
      "Progress: 8%\n",
      "Progress: 16%\n",
      "Progress: 25%\n",
      "Progress: 33%\n",
      "Progress: 41%\n",
      "Progress: 50%\n",
      "Progress: 58%\n",
      "Progress: 66%\n",
      "Progress: 75%\n",
      "Progress: 83%\n",
      "Progress: 91%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 24 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2815 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 43 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_002_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_002_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_002_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_002.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_002_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 47 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 25%\n",
      "Progress: 34%\n",
      "Progress: 42%\n",
      "Progress: 51%\n",
      "Progress: 59%\n",
      "Progress: 68%\n",
      "Progress: 76%\n",
      "Progress: 85%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 47 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 7500 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 90 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_002_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_002_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_002_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_002.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_002_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 47 queries\n",
      "Progress: 8%\n",
      "Progress: 17%\n",
      "Progress: 25%\n",
      "Progress: 34%\n",
      "Progress: 42%\n",
      "Progress: 51%\n",
      "Progress: 59%\n",
      "Progress: 68%\n",
      "Progress: 76%\n",
      "Progress: 85%\n",
      "Progress: 93%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 47 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 7500 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 90 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_045_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_045_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_045_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_045.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_045_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2233 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 26 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_045_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_045_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_045_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_045.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_045_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2233 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 26 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_057_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_057_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_057_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_057.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_057_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 983 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_057_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_057_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_057_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_057.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_057_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 8 queries\n",
      "Progress: 12%\n",
      "Progress: 25%\n",
      "Progress: 37%\n",
      "Progress: 50%\n",
      "Progress: 62%\n",
      "Progress: 75%\n",
      "Progress: 87%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 8 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 983 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 13 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_022_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_022_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_022_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_022.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_022_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 17 queries\n",
      "Progress: 5%\n",
      "Progress: 11%\n",
      "Progress: 17%\n",
      "Progress: 23%\n",
      "Progress: 29%\n",
      "Progress: 35%\n",
      "Progress: 41%\n",
      "Progress: 47%\n",
      "Progress: 52%\n",
      "Progress: 58%\n",
      "Progress: 64%\n",
      "Progress: 70%\n",
      "Progress: 76%\n",
      "Progress: 82%\n",
      "Progress: 88%\n",
      "Progress: 94%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 17 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1735 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 27 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_022_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_022_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_022_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_022.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_022_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 17 queries\n",
      "Progress: 5%\n",
      "Progress: 11%\n",
      "Progress: 17%\n",
      "Progress: 23%\n",
      "Progress: 29%\n",
      "Progress: 35%\n",
      "Progress: 41%\n",
      "Progress: 47%\n",
      "Progress: 52%\n",
      "Progress: 58%\n",
      "Progress: 64%\n",
      "Progress: 70%\n",
      "Progress: 76%\n",
      "Progress: 82%\n",
      "Progress: 88%\n",
      "Progress: 94%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 17 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1735 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 27 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_065_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_065_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_065_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_065.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_065_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1103 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 9 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_065_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_065_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_065_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_065.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_065_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 6 queries\n",
      "Progress: 16%\n",
      "Progress: 33%\n",
      "Progress: 50%\n",
      "Progress: 66%\n",
      "Progress: 83%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 6 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1103 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 9 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion, Repeat_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_030_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_030_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_030_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_030.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_030_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 17 queries\n",
      "Progress: 5%\n",
      "Progress: 11%\n",
      "Progress: 17%\n",
      "Progress: 23%\n",
      "Progress: 29%\n",
      "Progress: 35%\n",
      "Progress: 41%\n",
      "Progress: 47%\n",
      "Progress: 52%\n",
      "Progress: 58%\n",
      "Progress: 64%\n",
      "Progress: 70%\n",
      "Progress: 76%\n",
      "Progress: 82%\n",
      "Progress: 88%\n",
      "Progress: 94%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 17 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2505 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 31 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_030_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_030_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_030_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_030.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_030_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 17 queries\n",
      "Progress: 5%\n",
      "Progress: 11%\n",
      "Progress: 17%\n",
      "Progress: 23%\n",
      "Progress: 29%\n",
      "Progress: 35%\n",
      "Progress: 41%\n",
      "Progress: 47%\n",
      "Progress: 52%\n",
      "Progress: 58%\n",
      "Progress: 64%\n",
      "Progress: 70%\n",
      "Progress: 76%\n",
      "Progress: 82%\n",
      "Progress: 88%\n",
      "Progress: 94%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 17 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2505 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 31 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_028_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_028_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_028_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_028.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_028_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 13 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 13 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2081 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 20 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_028_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_028_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_028_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_028.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_028_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 13 queries\n",
      "Progress: 7%\n",
      "Progress: 15%\n",
      "Progress: 23%\n",
      "Progress: 30%\n",
      "Progress: 38%\n",
      "Progress: 46%\n",
      "Progress: 53%\n",
      "Progress: 61%\n",
      "Progress: 69%\n",
      "Progress: 76%\n",
      "Progress: 84%\n",
      "Progress: 92%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 13 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 2081 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 20 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Tandem_expansion, Tandem_contraction\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_014_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_014_php_8kbp_50kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 50000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_014_php_8kbp_50kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_014.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_014_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 11 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 11 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1228 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 16 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n",
      "Input delta file: DK_0911_v04_pcontig_014_php.delta\n",
      "Output prefix: DK_0911_v04_pcontig_014_php_8kbp_10kp\n",
      "Unique anchor length: 8000\n",
      "Maximum feature length: 10000\n",
      "Path to R scripts: /home/benjamin/genome_assembly/PST79/FALCON/p_assemblies/v9_1/downstream_analysis_2017/scripts/Assemblytics\n",
      "DK_0911_v04_pcontig_014_php_8kbp_10kp\n",
      "STARTING,DONE,Starting unique anchor filtering.\n",
      "1. Filter delta file\n",
      "Keeping fully unique alignments even if they are below the unique anchor length of 8000 bp\n",
      "header:\n",
      "/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_014.fa /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_014_h_ctgs.fa\n",
      "NUCMER\n",
      "First read through the file: 0 seconds for 0 query-reference combinations\n",
      "Filtering alignments of 11 queries\n",
      "Progress: 9%\n",
      "Progress: 18%\n",
      "Progress: 27%\n",
      "Progress: 36%\n",
      "Progress: 45%\n",
      "Progress: 54%\n",
      "Progress: 63%\n",
      "Progress: 72%\n",
      "Progress: 81%\n",
      "Progress: 90%\n",
      "Progress: 100%\n",
      "Progress: 100%\n",
      "Deciding which alignments to keep: 0 seconds for 11 queries\n",
      "Reading file and recording all the entries we decided to keep: 0 seconds for 1228 total lines in file\n",
      "UNIQFILTER,DONE,Step 1: Assemblytics_uniq_anchor.py completed successfully. Now finding variants between alignments.\n",
      "2. Finding variants between alignments\n",
      "Loaded 16 alignments\n",
      "BETWEEN,DONE,Step 2: Assemblytics_between_alignments.pl completed successfully. Now finding variants within alignments.\n",
      "3. Finding variants within alignments\n",
      "WITHIN,DONE,Step 3: Assemblytics_within_alignment.py completed successfully. Now combining the two sets of variants together.\n",
      "4. Combine variants between and within alignments\n",
      "COMBINE,DONE,Step 4: Variants combined successfully. Now generating figures and summary statistics.\n",
      "The following `from` values were not present in `x`: Repeat_expansion\n",
      "Warning messages:\n",
      "1: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "2: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "3: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "4: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "5: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "6: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "7: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "8: `geom_bar()` no longer has a `binwidth` parameter. Please use `geom_histogram()` instead. \n",
      "null device \n",
      "          1 \n",
      "SUMMARY,DONE,Step 5: Assemblytics_summary.py completed successfully\n"
     ]
    }
   ],
   "source": [
    "# bash_script_assemblytics_stderr = subprocess.check_output('bash %s' %os.path.join(OUT_PATH_ASSEMBLYTICS, bash_script_assemblytics), shell=True, stderr=subprocess.STDOUT)\n",
    "\n",
    "!bash {os.path.join(OUT_PATH_ASSEMBLYTICS, bash_script_assemblytics)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COMPARING ALL FOLDERS TO REF: DK_0911_v04_pcontig_055_php_8kbp\n",
      "Reference Counts: {'tab': 2, 'csv': 8, 'genome': 4, 'txt': 2, 'summary': 2, 'gz': 2, 'delta': 1, 'index': 4, 'bed': 6}\n",
      "matches: 86 discrepancies: 0 \n",
      "match %: 100.0\n"
     ]
    }
   ],
   "source": [
    "# CHECK IF MAPPING WENT CORRECTLY\n",
    "\n",
    "os.chdir('/home/gamran/genome_analysis/Warrior/Richard/scripts')\n",
    "%run file_counting.ipynb\n",
    "\n",
    "def checkMapping(folder_locs, ref = 62):\n",
    "    '''Checks whether the mapping process in above cells worked consistently across all files'''\n",
    "\n",
    "    '''N.B. working reference count on 6/12/17 was:\n",
    "    COMPARING ALL FOLDERS TO REF: DK_0911_v03_pcontig_018_php_8kbp\n",
    "    Reference Counts: {'tab': 2, 'index': 4, 'df': 2, 'genome': 4, 'summary': 2, 'csv': 8, 'txt': 2, 'gz': 2, 'bed': 6, 'delta': 1}\n",
    "    matches: 86 discrepancies: 0 \n",
    "    match %: 100.0'''\n",
    "\n",
    "    ref_folder = folder_locs[ref]\n",
    "    print('COMPARING ALL FOLDERS TO REF: %s' %getFolderName(folder_locs[ref]))\n",
    "    # ignore .pdf and .png, as folders are expected to have different numbers of these\n",
    "    refDict = getExtensionCounts(ref_folder, ['pdf', 'png'])\n",
    "    print('Reference Counts:', refDict)\n",
    "    \n",
    "    matches = 0\n",
    "    problems = 0\n",
    "    for folder_loc in folder_locs:\n",
    "        # ignore .pdf and .png, as folders are expected to have different numbers of these\n",
    "        discrepancies = getDiscrepancies(folder_loc, refDict)\n",
    "        if discrepancies == '':\n",
    "            matches += 1\n",
    "        else:\n",
    "            problems += 1\n",
    "            print(discrepancies)\n",
    "\n",
    "    print(\"matches: %i discrepancies: %i \" %(matches, problems))\n",
    "    print(\"match %:\", matches/(matches + problems)*100)\n",
    "    \n",
    "    return\n",
    "\n",
    "checkMapping(folder_locs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
