{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Defining Alleles v02"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook is aimed at doing allelic comparison between primary contigs and hapltotigs. This was used to generate DataFrames used during the analysis and evolved during the analysis. So it might be a bit more complicated than it needs to be, but it works.\n",
    "\n",
    "This notebook does NOT use proteinortho in its identification of allelic pairs\n",
    "\n",
    "Based on original code by Benjamin Schwessinger."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook defines alleles based on a Falcon-Unzip assembly where relational information between two contigs are available.\n",
    "\n",
    "#### The input as follows:\n",
    "* Assemblytics alingment of all haplotigs onto their corresponding primary contigs. The *.oriented_coords.csv will be converted to a gff file of haplotig alingments onto their respective primary contigs\n",
    "* gff file of all features. This will be filtered down to genes only and separated out for each contig\n",
    "* protein and gene fa files for reciprocal blast of haplotig sequences onto primary sequences\n",
    "\n",
    "#### What happens:\n",
    "* for each primary gene/protein the the overlapping haplotig is pulled out and added to the blast dataframe\n",
    "* this blast dataframe of primary proteins onto haplotig proteins is than used for filtering to provide the following files\n",
    "* best blast hits are first filtered on e-value minimums followed by BitScore maximum\n",
    "\n",
    "#### What comes out in the alleles folder:\n",
    "##### for primary genes/proteins\n",
    "* a set of files that contain two proteins IDs identifying the two alleles.  \n",
    "* *_p_ctg.h_contig_overlap.alleles contains best blast hits of the primary protein on haplotig proteins of the overlapping haplotig (truely linked alleles).\n",
    "* *_p_ctg.no_respective_h_contig_overlap.alleles contains all primary protein that don't have a blast hit on the overlapping haplotig but another haplotig associated with its primary contig.\n",
    "* *_p_ctg.no_specific_h_contig_overlap.alleles contains all primary proteins that don't have a blast hit on an associated haplotig but another haplotig.\n",
    "* *_p_ctg.no_alleles contains all primary proteins without blast hit on haplotigs.\n",
    "\n",
    "##### for haplotig genes/proteins\n",
    "* *_h_ctg.no.alleles contains all haplotig proteins that are not associated with a primary allele in the above analysis.\n",
    "* *_h_ctg.best_p_hits.no_alleles contains all haplotig proteins that are not associated with a primary allele in the above analysis, but have a blast hit (h on p). Only the best blast hit for each haplotig is recorded in this file. This might be duplications in the haplotigs.\n",
    "* *_h_ctg.no_p_hits.alleles contains all haplotig proteins that have no blast hit.\n",
    "\n",
    "##### filtering of blast output\n",
    "* filtering of blast output is possible on Query coverage of the blast hit [Qcov] and the percentage identity of the aligenment [PctID]. This generates additional outfiles with *.QcovXX.PctIDYY.alleles. The default values are set to Qcov80 and PctID70.\n",
    "\n",
    "#### What else to consider:\n",
    "* the script was tested for outputing the right number of protein sequences for primary and haplotig sequences.\n",
    "* this is a script and not a program. No warrenties.\n",
    "* Feedback always welcome.\n",
    "* Path and other variables are defined at the top of the script.\n",
    "* It assumes that primary contig sequences are labled as e.g. pcontig_000 and associated haplotigs as hcontig_000_0xx.\n",
    "\n",
    "#### Downstream considerations:\n",
    "* Why have some proteins no blast hits? Is it simple that their gene models are missing? Seen this and working on this.\n",
    "* In case of primary genes without blast hits do those lie in homozygous coverage areas? Working on this.\n",
    "* Are some of the h genes/proteins without alleles but with blast hits duplications that diverget?\n",
    "* Do some of the primary proteins/genes without blast hit on haplotigs have a good blast hit on other primary contigs? Is this reciprocal?\n",
    "* Can we use some of this information to talk about linkage of contigs or does this go to far without fully phased genomes using loooong reads or single nucleii sequencing? Would say yes so far.\n",
    "* Some primary proteins have two equal got hits are those duplications in the haplotig?\n",
    "* If we perform downstream genetic variation analysis between alleles do we see any GO, KEGG, effector or other functional domains enriched?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "import pysam\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from pybedtools import BedTool\n",
    "import numpy as np\n",
    "import pybedtools\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import sys\n",
    "import subprocess\n",
    "import shutil\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define the PATH\n",
    "\n",
    "GENOME_VERSION = 'v04'\n",
    "\n",
    "BASE_OUT_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/%s_no_proteinortho' % GENOME_VERSION\n",
    "GENOME_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/genome_%s/' % GENOME_VERSION\n",
    "ASSEMBLYTICS_IN_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/%s/Assemblytics/' % GENOME_VERSION\n",
    "\n",
    "BLAST_DB = os.path.join(BASE_OUT_PATH, 'blast_DB')\n",
    "OUT_PATH = os.path.join(BASE_OUT_PATH, 'allele_analysis')\n",
    "OUT_tmp = os.path.join(OUT_PATH, 'tmp')\n",
    "PROTEINORTHO_OUT_PATH = os.path.join(BASE_OUT_PATH, 'proteinortho')\n",
    "    \n",
    "#renamed the allele path to reflect the proteinortho results\n",
    "ALLELE_PATH = os.path.join(OUT_PATH, 'alleles_proteinortho_graph516')\n",
    "if not os.path.exists(BASE_OUT_PATH):\n",
    "    os.mkdir(BASE_OUT_PATH)\n",
    "if not os.path.isdir(OUT_PATH):\n",
    "    os.mkdir(OUT_PATH)\n",
    "if not os.path.isdir(OUT_tmp):\n",
    "    os.mkdir(OUT_tmp)\n",
    "if not os.path.exists(ALLELE_PATH):\n",
    "    os.mkdir(ALLELE_PATH)\n",
    "if not os.path.exists(BLAST_DB):\n",
    "    os.mkdir(BLAST_DB)\n",
    "if not os.path.exists(PROTEINORTHO_OUT_PATH):\n",
    "    os.mkdir(PROTEINORTHO_OUT_PATH)\n",
    "\n",
    "#Define your p and h genome and move it into the allele analysis folder\n",
    "P_GENOME = 'DK_0911_%s_p_ctg' % GENOME_VERSION\n",
    "H_GENOME = 'DK_0911_%s_h_ctg' % GENOME_VERSION\n",
    "for x in (x + '.fa' for x in [P_GENOME, H_GENOME]):\n",
    "    shutil.copy2(GENOME_PATH+'/'+x, OUT_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define ENV parameters for blast hits and threads used in blast analysis\n",
    "n_threads = 16\n",
    "e_value = 1e-3\n",
    "Qcov_cut_off = 0 #this defines the mimimum coverage of the Query to be required for filtering. Will become part of name.\n",
    "PctID_cut_off = 70 #this defines the mimimum PctID accross the alignment to be required for filtering. Will become part of name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def att_column_p(x, y, z, a, b):\n",
    "    '''Generate attribute column of h on p p_gff dataframe.\n",
    "    '''\n",
    "    string = 'h_contig=%s;query_start=%s;query_stop=%s;query_length=%s;query_aln_ln=%s' % (x, str(y), str(z), str(a), str(b))\n",
    "    return string\n",
    "\n",
    "def att_column_h(x, y, z, a, b):\n",
    "    '''Generate attribute column of h on p h_gff dataframe.\n",
    "    '''\n",
    "    string = 'p_contig=%s;ref_start=%s;ref_stop=%s;ref_length=%s;ref_aln_ln=%s' % (x, str(y), str(z), str(a), str(b))\n",
    "    return string\n",
    "\n",
    "def gene_contig_subset(df, contig, feature):\n",
    "    '''Input the gff and subset the feature columne by gene.\n",
    "        Return data frame subset by contig and gene'''\n",
    "    tmp_df = df[(df[0].str.contains(contig)) & (df[2]==feature)]\n",
    "    tmp_df.sort_values(3,inplace=True)\n",
    "    tmp_df.reset_index(drop=True, inplace=True)\n",
    "    return tmp_df\n",
    "\n",
    "def same_contig_blast(x,y):\n",
    "    '''Function that checks if the blast hit in column y is on the same contig as the the query sequence in\n",
    "    column y.\n",
    "    '''\n",
    "    q_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', x).group(1)\n",
    "    if y != 'NaN':\n",
    "        hit_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', y).group(1)\n",
    "    else:\n",
    "        hit_contig = 'NaN'\n",
    "    return q_contig == hit_contig\n",
    "\n",
    "def target_on_mapped_haplotig(t_contig, h_contig_overlap):\n",
    "    '''Simple function that checks if the target contig is in the list of h overlapping contigs'''\n",
    "    if t_contig == False or h_contig_overlap == False:\n",
    "        return False\n",
    "    return t_contig in h_contig_overlap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Considerations for blast analysis\n",
    "Do gene blast and protein blast. Initially go off protein blast results. They should be more imformative as it also includes the coding region and the frame.\n",
    "> Pull this in as a dataframe and filter it down as well to each contig. Do blasts both ways as well.\n",
    "> Copy over primary & haplotig .protein.fa and .gene.fa files from the GENOME_PATH, if not already present."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "makeblastdb -in DK_0911_v04_h_ctg.gene.fa -dbtype nucl is done!\n",
      "makeblastdb -in DK_0911_v04_p_ctg.gene.fa -dbtype nucl is done!\n",
      "makeblastdb -in DK_0911_v04_p_ctg.protein.fa -dbtype prot is done!\n",
      "makeblastdb -in DK_0911_v04_h_ctg.protein.fa -dbtype prot is done!\n",
      "All databases generated and ready to go!\n"
     ]
    }
   ],
   "source": [
    "#generate the blast databases if not already present\n",
    "os.chdir(BLAST_DB)\n",
    "# copy over .protein.fa and .gene.fa files from the GENOME_PATH folder\n",
    "# if not already present, for both primary and haplotigs\n",
    "shutil.copy2(os.path.join(GENOME_PATH, P_GENOME + '.protein.fa'), BLAST_DB)\n",
    "shutil.copy2(os.path.join(GENOME_PATH, P_GENOME + '.gene.fa'), BLAST_DB)\n",
    "shutil.copy2(os.path.join(GENOME_PATH, H_GENOME + '.protein.fa'), BLAST_DB)\n",
    "shutil.copy2(os.path.join(GENOME_PATH, H_GENOME + '.gene.fa'), BLAST_DB)\n",
    "\n",
    "blast_dir_content = os.listdir(BLAST_DB)\n",
    "for x in blast_dir_content:\n",
    "    if x.endswith('.fa') and ({os.path.isfile(x + e) for e in ['.psq', '.phr', '.pin'] } != {True}\\\n",
    "           and {os.path.isfile(x + e) for e in ['.nin', '.nhr', '.nsq'] } != {True} ):\n",
    "\n",
    "        make_DB_options = ['-in']\n",
    "        make_DB_options.append(x)\n",
    "        make_DB_options.append('-dbtype')\n",
    "        if 'protein' in x:\n",
    "            make_DB_options.append('prot')\n",
    "        else:\n",
    "            make_DB_options.append('nucl')\n",
    "        make_DB_command = 'makeblastdb %s' % ' '.join(make_DB_options)\n",
    "        make_DB_stderr = subprocess.check_output(make_DB_command, shell=True, stderr=subprocess.STDOUT)\n",
    "        print('%s is done!' % make_DB_command)\n",
    "print(\"All databases generated and ready to go!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Blasting DK_0911_v04_p_ctg.gene.fa ...\n",
      "Blast command generated:\n",
      "blastn -query DK_0911_v04_h_ctg.gene.fa -db DK_0911_v04_p_ctg.gene.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/v04_no_proteinortho/allele_analysis/DK_0911_v04_h_ctg.DK_0911_v04_p_ctg.0.001.blastn.outfmt6\n",
      "Executing new blast command...\n",
      "New blast executed!\n",
      "\n",
      "Blasting DK_0911_v04_h_ctg.gene.fa ...\n",
      "Blast command generated:\n",
      "blastn -query DK_0911_v04_p_ctg.gene.fa -db DK_0911_v04_h_ctg.gene.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/v04_no_proteinortho/allele_analysis/DK_0911_v04_p_ctg.DK_0911_v04_h_ctg.0.001.blastn.outfmt6\n",
      "Executing new blast command...\n",
      "New blast executed!\n",
      "\n",
      "Blasting DK_0911_v04_h_ctg.protein.fa ...\n",
      "Blast command generated:\n",
      "blastp -query DK_0911_v04_p_ctg.protein.fa -db DK_0911_v04_h_ctg.protein.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/v04_no_proteinortho/allele_analysis/DK_0911_v04_p_ctg.DK_0911_v04_h_ctg.0.001.blastp.outfmt6\n",
      "Executing new blast command...\n",
      "New blast executed!\n",
      "\n",
      "Blasting DK_0911_v04_p_ctg.protein.fa ...\n",
      "Blast command generated:\n",
      "blastp -query DK_0911_v04_h_ctg.protein.fa -db DK_0911_v04_p_ctg.protein.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/v04_no_proteinortho/allele_analysis/DK_0911_v04_h_ctg.DK_0911_v04_p_ctg.0.001.blastp.outfmt6\n",
      "Executing new blast command...\n",
      "New blast executed!\n"
     ]
    }
   ],
   "source": [
    "blast_dict = {}\n",
    "blast_dict['gene_fa'] = [x for x  in os.listdir(BLAST_DB) if x.endswith('gene.fa') and 'ph_ctg' not in x]\n",
    "blast_dict['protein_fa'] = [x for x  in os.listdir(BLAST_DB) if x.endswith('protein.fa') and 'ph_ctg' not in x]\n",
    "blast_stderr_dict = {}\n",
    "for key in blast_dict.keys():\n",
    "    for n, query in enumerate(blast_dict[key]):\n",
    "        tmp_list = blast_dict[key][:]\n",
    "        tmp_stderr_list = []\n",
    "        #this loops through all remaining files and does blast against all those\n",
    "        del tmp_list[n]\n",
    "        for db in tmp_list:\n",
    "            print(\"\\nBlasting %s ...\" %db)\n",
    "            blast_options = ['-query']\n",
    "            blast_options.append(query)\n",
    "            blast_options.append('-db')\n",
    "\n",
    "            blast_options.append(db)\n",
    "            blast_options.append('-outfmt 6')\n",
    "            blast_options.append('-evalue')\n",
    "            blast_options.append(str(e_value))\n",
    "            blast_options.append('-num_threads')\n",
    "            blast_options.append(str(n_threads))\n",
    "            blast_options.append('>')\n",
    "            if 'gene' in query and 'gene' in db:\n",
    "                out_name_list = [ query.split('.')[0], db.split('.')[0], str(e_value), 'blastn.outfmt6']\n",
    "                out_name = os.path.join(OUT_PATH,'.'.join(out_name_list))\n",
    "                blast_options.append(out_name)\n",
    "                blast_command = 'blastn %s' % ' '.join(blast_options)\n",
    "            elif 'protein' in query and 'protein' in db:\n",
    "                out_name_list = [ query.split('.')[0], db.split('.')[0], str(e_value), 'blastp.outfmt6']\n",
    "                out_name = os.path.join(OUT_PATH,'.'.join(out_name_list))\n",
    "                blast_options.append(out_name)\n",
    "                blast_command = 'blastp %s' % ' '.join(blast_options)\n",
    "            print(\"Blast command generated:\\n%s\" % blast_command)\n",
    "            if not os.path.exists(out_name):\n",
    "                print(\"Executing new blast command...\")\n",
    "                blast_stderr_dict[blast_command] = subprocess.check_output(blast_command, shell=True, stderr=subprocess.STDOUT)\n",
    "                print(\"New blast executed!\")\n",
    "            else:\n",
    "                blast_stderr_dict[blast_command] = 'Previously done already!'\n",
    "                print(\"Blast command previously completed!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make a sequence length dict for all genes and proteins for which blast was performed\n",
    "seq_list = []\n",
    "length_list =[]\n",
    "for key in blast_dict.keys():\n",
    "    for file in blast_dict[key]:\n",
    "        for seq in SeqIO.parse(open(file), 'fasta'):\n",
    "            seq_list.append(seq.id)\n",
    "            length_list.append(len(seq.seq))\n",
    "length_dict = dict(zip(seq_list, length_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get assemblytics folders\n",
    "ass_folders = [os.path.join(ASSEMBLYTICS_IN_PATH, x) for x in os.listdir(ASSEMBLYTICS_IN_PATH) if x.endswith('_php_8kbp')]\n",
    "ass_folders.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def same_contig_blast(x,y):\n",
    "    '''Function that checks if the blast hit in columne y is on the same contig as the the query sequence in\n",
    "    column y.\n",
    "    '''\n",
    "    try:\n",
    "        q_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', x).group(1)\n",
    "        if y != 'NaN':\n",
    "            hit_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', y).group(1)\n",
    "        else:\n",
    "            hit_contig = 'NaN'\n",
    "        return q_contig == hit_contig\n",
    "    except AttributeError:\n",
    "        print(\"AttributeError\\nx: %s\\ny: %s\" % (x, y))\n",
    "        sys.exit()\n",
    "    except TypeError:\n",
    "        print(\"x: %s\\ny: %s\" % (x, y))\n",
    "        print(\"TypeError\\ntype(x): %s\\ntype(y): %s\" % (type(x), type(y)))\n",
    "        sys.exit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in the gff files\n",
    "p_gff = pd.read_csv(GENOME_PATH+'/'+P_GENOME+'.anno.gff3', sep='\\t', header=None)\n",
    "h_gff = pd.read_csv(GENOME_PATH+'/'+H_GENOME+'.anno.gff3', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in the blast df and add QLgth and QCov columns\n",
    "blast_out_dict = {}\n",
    "blast_header = ['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery', 'StopQuery', 'StartTarget',\\\n",
    "              'StopTarget', 'e-value','BitScore']\n",
    "for key in blast_stderr_dict.keys():\n",
    "    file_loc = key.split('>')[-1][1:]\n",
    "    # file_loc = .../output/defining_alleles/allele_analysis/DK_0911_v03_h_ctg.DK_0911_v03_p_ctg.0.001.blastp.outfmt6\n",
    "    \n",
    "    file_name = file_loc.split('/')[-1]\n",
    "    # DK_0911_v03_h_ctg.DK_0911_v03_p_ctg.0.001.blastp.outfmt6\n",
    "    \n",
    "    tmp_df = pd.read_csv(file_loc, sep='\\t', header=None, names=blast_header)\n",
    "    \n",
    "    tmp_df['QLgth'] = tmp_df['Query'].apply(lambda x: length_dict[x])\n",
    "    tmp_df['QCov'] = tmp_df['AlnLgth']/tmp_df['QLgth']*100\n",
    "    \n",
    "    tmp_df.sort_values(by=['Query', 'e-value','BitScore', ],ascending=[True, True, False], inplace=True)\n",
    "    \n",
    "    # assert(len(tmp_df.loc[tmp_df['Query'] == 'pcontig_000:10023-10824']) == 0)\n",
    "    \n",
    "    #now make sure to add proteins/genes without blast hit to the DataFrame\n",
    "    #check if gene blast or protein blast and pull out respective query file\n",
    "    if file_loc.split('.')[-2] == 'blastp':\n",
    "        tmp_blast_seq = [x for x in blast_dict['protein_fa'] if x.startswith(file_name.split('.')[0])][0]\n",
    "    elif file_loc.split('.')[-2] == 'blastn':\n",
    "        tmp_blast_seq = [x for x in blast_dict['gene_fa'] if x.startswith(file_name.split('.')[0])][0]\n",
    "    tmp_all_blast_seq = []\n",
    "    #make list of all query sequences\n",
    "    for seq in SeqIO.parse(os.path.join(BLAST_DB, tmp_blast_seq), 'fasta'):\n",
    "        tmp_all_blast_seq.append(str(seq.id))\n",
    "    tmp_all_queries_w_hit = tmp_df[\"Query\"].unique()\n",
    "    tmp_queries_no_hit = set(tmp_all_blast_seq) - set(tmp_all_queries_w_hit)\n",
    "    no_hit_list = []\n",
    "    \n",
    "    #loop over the quieres with no hit and make list of list out of them the first element being the query id\n",
    "    for x in tmp_queries_no_hit:\n",
    "        NA_list = ['NaN'] * len(tmp_df.columns)\n",
    "        NA_list[0] = x\n",
    "        no_hit_list.append(NA_list)\n",
    "    \n",
    "    tmp_no_hit_df = pd.DataFrame(no_hit_list)\n",
    "    tmp_no_hit_df.columns = tmp_df.columns\n",
    "    tmp_no_hit_df['QLgth'] = tmp_no_hit_df.Query.apply(lambda x: length_dict[x])\n",
    "    #map stuff at the tmp level\n",
    "    \n",
    "    tmp_df = tmp_df.append(tmp_no_hit_df)\n",
    "    assert(len(tmp_df.loc[tmp_df['Query'] == 'NaN']) == 0)\n",
    "    \n",
    "    \n",
    "    tmp_df['q_contig'] = tmp_df['Query'].str.extract(r'([p|h][a-z]*_[^.]*).?', expand = False)\n",
    "    tmp_df['t_contig'] = tmp_df['Target'].str.extract(r'([p|h][a-z]*_[^.]*).?', expand = False)\n",
    "    # print(tmp_df['q_contig'])\n",
    "    \n",
    "    #fix that if you don't extract anything return False and not 'nan'\n",
    "    tmp_df['t_contig'].fillna(False, inplace=True)\n",
    "    \n",
    "    # print(tmp_df.apply(lambda row: print(row['Query'])), axis = 1)\n",
    "    \n",
    "    tmp_df['q_contig == t_contig'] = tmp_df.apply(lambda row: same_contig_blast(row['Query'], row['Target']), axis = 1)\n",
    "    tmp_df.reset_index(inplace=True, drop=True)\n",
    "    blast_out_dict[file_name] = tmp_df.iloc[:,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Check on /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/v04/Assemblytics/DK_0911_v04_pcontig_075_php_8kbp assemblytics\n"
     ]
    }
   ],
   "source": [
    "for folder in ass_folders:\n",
    "    '''\n",
    "    From oriented_coords.csv files generate gff files with the following set up.\n",
    "    h_contig overlap\n",
    "    'query', 'ref', 'h_feature', 'query_start', 'query_end', 'query_aln_len', 'strand', 'frame', 'attribute_h'\n",
    "    p_contig overlap\n",
    "    'ref', 'query', 'p_feature','ref_start','ref_end', 'alignment_length', 'strand', 'frame', 'attribute_p'.\n",
    "    Save those file in a new folder for further downstream analysis. The same folder should include the contig and gene filtered\n",
    "    gffs. \n",
    "    '''\n",
    "    orient_coords_file = [os.path.join(folder, x) for x in os.listdir(folder) if x.endswith('oriented_coords.csv')][0]\n",
    "    #load in df and generate several additions columns\n",
    "    tmp_df = pd.read_csv(orient_coords_file, sep=',')\n",
    "    #check if there is any resonable alignment if not go to the next one\n",
    "    if len(tmp_df) == 0:\n",
    "        print('Check on %s assemblytics' % (folder))\n",
    "        continue\n",
    "    tmp_df['p_feature'] = \"haplotig\"\n",
    "    tmp_df['h_feature'] ='primary_contig'\n",
    "    tmp_df['strand'] = \"+\"\n",
    "    tmp_df['frame'] = 0\n",
    "    tmp_df['query_aln_len'] = abs(tmp_df['query_end']-tmp_df['query_start'])\n",
    "    tmp_df['alignment_length'] = abs(tmp_df['ref_end'] - tmp_df['ref_start'])\n",
    "    tmp_df.reset_index(drop=True, inplace=True)\n",
    "    tmp_df.sort_values('query', inplace=True)\n",
    "    tmp_df['attribute_p'] = tmp_df.apply(lambda row: att_column_p(row['query'],row['query_start'], row['query_end'],\\\n",
    "                                                         row['query_length'], row['query_aln_len']), axis=1)\n",
    "    \n",
    "    tmp_df['attribute_h'] = tmp_df.apply(lambda row: att_column_h(row['ref'], row['ref_start'], row['ref_end'],\\\n",
    "                                                                  row['ref_length'], row['alignment_length']), axis=1)\n",
    "    \n",
    "    #generate tmp gff dataframe\n",
    "    tmp_p_gff_df = tmp_df.loc[:, ['ref', 'query', 'p_feature','ref_start','ref_end', 'alignment_length', 'strand', 'frame', 'attribute_p']]\n",
    "    #now filter added if start < stop swap round. And if start == 0. This was mostly?! only? the case for haplotigs\n",
    "    tmp_p_gff_df['comp'] = tmp_p_gff_df['ref_start'] - tmp_p_gff_df['ref_end']\n",
    "    tmp_swap_index  = tmp_p_gff_df['comp'] > 0\n",
    "    tmp_p_gff_df.loc[tmp_swap_index, 'ref_start'] , tmp_p_gff_df.loc[tmp_swap_index, 'ref_end'] = tmp_p_gff_df['ref_end'], tmp_p_gff_df['ref_start']\n",
    "    #and if 'ref_start' == g\n",
    "    is_null_index  = tmp_p_gff_df['ref_start'] == 0\n",
    "    tmp_p_gff_df.loc[is_null_index, 'ref_start'] = 1\n",
    "    tmp_p_gff_df = tmp_p_gff_df.iloc[:, 0:9]\n",
    "    tmp_p_gff_df.sort_values('ref_start',inplace=True)\n",
    "    tmp_h_gff_df = tmp_df.loc[:, ['query', 'ref', 'h_feature', 'query_start', 'query_end', 'query_aln_len', 'strand', 'frame', 'attribute_h']]\n",
    "    #same fix for tmp_h_gff\n",
    "    tmp_h_gff_df['comp'] = tmp_h_gff_df['query_start'] - tmp_h_gff_df['query_end']\n",
    "    tmp_swap_index  = tmp_h_gff_df['comp'] > 0\n",
    "    tmp_h_gff_df.loc[tmp_swap_index, 'query_start'] , tmp_h_gff_df.loc[tmp_swap_index, 'query_end'] = tmp_h_gff_df['query_end'], tmp_h_gff_df['query_start']\n",
    "    #and if 'query_start' == g\n",
    "    is_null_index  = tmp_h_gff_df['query_start'] == 0\n",
    "    tmp_h_gff_df.loc[is_null_index, 'query_start'] = 1\n",
    "    tmp_h_gff_df = tmp_h_gff_df.iloc[:, 0:9]\n",
    "    tmp_h_gff_df.sort_values('query_start', inplace=True)\n",
    "    #make the outfolder and save gff files of overlap and filtered gene files\n",
    "    folder_suffix = folder.split('/')[-1]\n",
    "    #get the contig\n",
    "    contig = re.search(r'p([a-z]*_[0-9]*)_', folder_suffix).group(1)\n",
    "    out_folder = os.path.join(OUT_PATH, folder_suffix)\n",
    "    if not os.path.exists(out_folder):\n",
    "        os.mkdir(out_folder)\n",
    "    out_name_p = os.path.join(out_folder, folder_suffix )\n",
    "    tmp_p_gff_df.to_csv(out_name_p+'.p_by_h_cov.gff', sep='\\t' ,header=None, index =None)\n",
    "    out_name_h =  os.path.join(out_folder, folder_suffix)\n",
    "    tmp_h_gff_df.to_csv(out_name_h+'.h_by_p_cov.gff' , sep='\\t' ,header=None, index =None)\n",
    "    #write those out to new folder for php together with the p and h gff of genes\n",
    "    p_gene_gff = gene_contig_subset(p_gff, contig, 'gene')\n",
    "    p_gene_gff.to_csv(out_name_p+'.p_gene.gff', sep='\\t' ,header=None, index =None)\n",
    "    h_gene_gff = gene_contig_subset(h_gff, contig, 'gene')\n",
    "    h_gene_gff.to_csv(out_name_h+'.h_gene.gff', sep='\\t' ,header=None, index =None)\n",
    "    \n",
    "    #next would be to do overlap between the gene gff and the corresponding alignment gff. Write this out. Dict for each gene and its corresponding h_contig/p_contig\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now loop over the outfolders\n",
    "out_folders = [os.path.join(OUT_PATH, x) for x in os.listdir(OUT_PATH) if x.endswith('_php_8kbp')]\n",
    "out_folders.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 661\n",
      "This is the length of the intersect 403\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 451\n",
      "This is the length of the intersect 407\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 382\n",
      "This is the length of the intersect 380\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 295\n",
      "This is the length of the intersect 297\n",
      "This is the length of the intersect 347\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 249\n",
      "This is the length of the intersect 271\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 277\n",
      "This is the length of the intersect 324\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 78\n",
      "This is the length of the intersect 303\n",
      "This is the length of the intersect 232\n",
      "This is the length of the intersect 247\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 186\n",
      "This is the length of the intersect 204\n",
      "This is the length of the intersect 240\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 218\n",
      "This is the length of the intersect 111\n",
      "This is the length of the intersect 214\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 167\n",
      "This is the length of the intersect 0\n",
      "This is the length of the intersect 168\n",
      "This is the length of the intersect 79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 100\n",
      "This is the length of the intersect 151\n",
      "This is the length of the intersect 173\n",
      "This is the length of the intersect 86\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 159\n",
      "This is the length of the intersect 162\n",
      "This is the length of the intersect 185\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 126\n",
      "This is the length of the intersect 161\n",
      "This is the length of the intersect 122\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 66\n",
      "This is the length of the intersect 130\n",
      "This is the length of the intersect 135\n",
      "This is the length of the intersect 103\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 98\n",
      "This is the length of the intersect 5\n",
      "This is the length of the intersect 116\n",
      "This is the length of the intersect 67\n",
      "This is the length of the intersect 58\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 73\n",
      "This is the length of the intersect 69\n",
      "This is the length of the intersect 70\n",
      "This is the length of the intersect 80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 35\n",
      "This is the length of the intersect 62\n",
      "This is the length of the intersect 87\n",
      "This is the length of the intersect 75\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 91\n",
      "This is the length of the intersect 74\n",
      "This is the length of the intersect 60\n",
      "This is the length of the intersect 0\n",
      "This is the length of the intersect 3\n",
      "This is the length of the intersect 42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 38\n",
      "This is the length of the intersect 33\n",
      "This is the length of the intersect 39\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 60\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 25\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 19\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 20\n",
      "This is the length of the intersect 21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 28\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 14\n",
      "This is the length of the intersect 20\n",
      "This is the length of the intersect 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 7\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 7\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 5\n",
      "This is the length of the intersect 22\n",
      "This is the length of the intersect 4\n",
      "This is the length of the intersect 0\n",
      "This is the length of the intersect 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    }
   ],
   "source": [
    "p_gene_h_contig_overlap_dict = {}\n",
    "for out_folder in out_folders:\n",
    "    tmp_folder_content = os.listdir(out_folder)\n",
    "    tmp_p_gene_gff = [os.path.join(out_folder, x) for x in tmp_folder_content if x.endswith('p_gene.gff') ][0]\n",
    "    tmp_p_gene_bed = pybedtools.BedTool(tmp_p_gene_gff).remove_invalid().saveas()\n",
    "    tmp_p_by_h_gff = [os.path.join(out_folder, x) for x in tmp_folder_content if x.endswith('p_by_h_cov.gff') ][0]\n",
    "    tmp_p_by_h_bed = pybedtools.BedTool(tmp_p_by_h_gff).remove_invalid().saveas()\n",
    "    #generate a bed intersect\n",
    "    p_gene_and_p_by_h = tmp_p_gene_bed.intersect(tmp_p_by_h_bed, wb=True)\n",
    "    #check the length of the obtained intersect is \n",
    "    print(\"This is the length of the intersect %i\" %len(p_gene_and_p_by_h ))\n",
    "    if len(p_gene_and_p_by_h) == 0:\n",
    "        continue\n",
    "    #turn this into a dataframe \n",
    "    p_gene_and_p_by_h_df = p_gene_and_p_by_h.to_dataframe()\n",
    "    p_gene_and_p_by_h_df['p_gene'] = p_gene_and_p_by_h_df[8].str.extract(r'ID=([^;]*);', expand=False)\n",
    "    p_gene_and_p_by_h_df['p_protein'] = p_gene_and_p_by_h_df['p_gene'].str.replace('TU', 'model')\n",
    "    #save this in the same folder as dataframe\n",
    "    p_gene_and_p_by_h_df.to_csv(tmp_p_gene_gff.replace('p_gene.gff', 'gene_haplotig_intersect.df'), sep='\\t', index=None)\n",
    "    p_gene_and_p_by_h_df_2 = p_gene_and_p_by_h_df.loc[:, [10, 'p_protein']]\n",
    "    #make a dict for the overlap of each gene with it's correspoding haplotig\n",
    "    tmp_p_gene_h_overlap_df = p_gene_and_p_by_h_df_2.groupby('p_protein')[10].apply(list)\n",
    "    tmp_p_gene_h_overlap_dict = dict(zip(tmp_p_gene_h_overlap_df.index, tmp_p_gene_h_overlap_df.values))\n",
    "    p_gene_h_contig_overlap_dict.update(tmp_p_gene_h_overlap_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get out the p_protein on h_protein blast out dataframe and append the h_contig overlap for each protein\n",
    "_key = [x for x in blast_out_dict.keys() if x.startswith(P_GENOME) and x.split('.')[-2] == 'blastp' and x.endswith('outfmt6')][0]\n",
    "p_gene_h_contig_overlap_df = pd.DataFrame([p_gene_h_contig_overlap_dict.keys(), p_gene_h_contig_overlap_dict.values()],  index = ['p_protein', 'h_contig_overlap']).T\n",
    "_tmp_df = blast_out_dict[_key]\n",
    "_tmp_df = _tmp_df.merge(p_gene_h_contig_overlap_df, how='outer' ,left_on='Query', right_on='p_protein')\n",
    "_tmp_df['h_contig_overlap'].fillna(False, inplace=True)\n",
    "#check if the protein hit resites on the overlapping haplotig\n",
    "_tmp_df['t_contig == h_contig_overlap'] = _tmp_df.apply(lambda row: target_on_mapped_haplotig(row['t_contig'], row['h_contig_overlap']), axis =1 )\n",
    "#blast_out_dict[_key] = _tmp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# p_gene_h_contig_overlap_df['p_protein'] = p_gene_h_contig_overlap_df['p_protein'].apply(lambda x: x.replace('model', 'TU'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write the dataframe again\n",
    "_tmp_df.to_csv(os.path.join(OUT_PATH, _key+'.allele_analysis'), index=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now mangle the data a bit\n",
    "#write out t_contig == h_contig_overlap best hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No cut off defined! Do you want to define it. Please go ahead above.\n"
     ]
    }
   ],
   "source": [
    "#Qcov_cut_off = ''\n",
    "#PctID_cut_off = ''\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    pass\n",
    "else: \n",
    "    print('No cut off defined! Do you want to define it. Please go ahead above.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "# Finding:\n",
    "# • (htg & pCtg on same contig and overlap)\n",
    "\n",
    "_tmp_grouped = _tmp_df[(_tmp_df['t_contig == h_contig_overlap'] == True)].groupby('Query')\n",
    "# .groupby('Query') creates a GroupBy object that can be thought of mini-DataFrames,\n",
    "# each with the same Query. This groups are then reduced according to the\n",
    "# minimum e-value and maximum BitScore of each group (i.e. retaining the\n",
    "# best hit for each Query).\n",
    "\n",
    "# filter on the e-value, then on the BitScore\n",
    "best_hits_t_contig_and_h_contig = _tmp_grouped.apply(lambda g: g[(g['e-value'] == g['e-value'].min())])\n",
    "best_hits_t_contig_and_h_contig = best_hits_t_contig_and_h_contig.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "\n",
    "out_name = P_GENOME + '.h_contig_overlap.alleles'\n",
    "\n",
    "best_hits_t_contig_and_h_contig.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#now save including the filter as well\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    out_name = P_GENOME + '.h_contig_overlap.Qcov%s.PctID%s.alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    best_hits_t_contig_and_h_contig[(best_hits_t_contig_and_h_contig['QCov'] >= Qcov_cut_off) & (best_hits_t_contig_and_h_contig['PctID'] >= PctID_cut_off)]\\\n",
    "    .loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering is working; proceed.\n"
     ]
    }
   ],
   "source": [
    "# test if filtering like this works\n",
    "if set(_tmp_df[(_tmp_df['t_contig == h_contig_overlap'] == True)]['Query'].unique()) == set(best_hits_t_contig_and_h_contig['Query'].unique().tolist()):\n",
    "    print(\"Filtering is working; proceed.\")\n",
    "else:\n",
    "    print('The filter threw an error. Not the expected output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop everything that has already been matched well\n",
    "# already matched:\n",
    "# • (htg belongs to pCtg and matching sequence overlaps)\n",
    "\n",
    "query_ids_to_drop = best_hits_t_contig_and_h_contig['Query'].unique().tolist()\n",
    "target_ids_to_drop = best_hits_t_contig_and_h_contig['Target'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "# finding\n",
    "# • (htg belongs to pCtg but matching sequence does not overlap)\n",
    "_tmp_groupbed = _tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop))&\\\n",
    "                        (_tmp_df['q_contig == t_contig'] == True)].groupby('Query')\n",
    "best_hits_same_contig_no_overlap = _tmp_groupbed.apply(lambda g: g[(g['e-value'] == g['e-value'].min())]) \n",
    "best_hits_same_contig_no_overlap = best_hits_same_contig_no_overlap.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "\n",
    "out_name = P_GENOME + '.no_specific_h_contig_overlap.alleles'\n",
    "best_hits_same_contig_no_overlap.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#now save including the filter as well\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    out_name = P_GENOME + '.no_specific_h_contig_overlap.Qcov%s.PctID%s.alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    best_hits_same_contig_no_overlap[(best_hits_same_contig_no_overlap['QCov'] >= Qcov_cut_off) & (best_hits_same_contig_no_overlap['PctID'] >= PctID_cut_off)]\\\n",
    "    .loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All good please proceed!\n"
     ]
    }
   ],
   "source": [
    "#test if filtering like this works\n",
    "if set(_tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop)) & (_tmp_df['q_contig == t_contig'] == True)]['Query'].unique())\\\n",
    "        == set(best_hits_same_contig_no_overlap['Query'].unique().tolist()):\n",
    "    print(\"All good please proceed!\")\n",
    "else:\n",
    "    print('The filter threw an error. Not the expected output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop everything that has already been matched well\n",
    "# already matched:\n",
    "# • (htg belongs to pCtg and matching sequence overlaps)\n",
    "# • (htg belongs to pCtg but matching sequence does not overlap)\n",
    "\n",
    "query_ids_to_drop = query_ids_to_drop + best_hits_same_contig_no_overlap['Query'].unique().tolist()\n",
    "target_ids_to_drop = target_ids_to_drop + best_hits_same_contig_no_overlap['Target'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "# now get all remaining matches, i.e.\n",
    "# • (htg does not belong to pCtg)\n",
    "\n",
    "_tmp_groupbed = _tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop))&\\\n",
    "                        (_tmp_df['q_contig == t_contig'] == False)\\\n",
    "                        & (_tmp_df['Target'] != 'NaN' )].groupby('Query')\n",
    "best_hits_different_contig = _tmp_groupbed.apply(lambda g: g[(g['e-value'] == g['e-value'].min()) ])\n",
    "best_hits_different_contig = best_hits_different_contig .reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "out_name = P_GENOME + '.no_respective_h_contig_overlap.alleles'\n",
    "best_hits_different_contig.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#now save including the filter as well\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    out_name = P_GENOME + '.no_respective_h_contig_overlap.Qcov%s.PctID%s.alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    best_hits_different_contig[(best_hits_different_contig['QCov'] >= Qcov_cut_off) & (best_hits_different_contig['PctID'] >= PctID_cut_off)]\\\n",
    "    .loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All good please proceed!\n"
     ]
    }
   ],
   "source": [
    "if set(_tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop))& (_tmp_df['q_contig == t_contig'] == False) &\\\n",
    "               (_tmp_df['Target'] != 'NaN' )]['Query'].unique().tolist())\\\n",
    "        == set(best_hits_different_contig['Query'].unique().tolist()):\n",
    "    print(\"All good please proceed!\")\n",
    "else:\n",
    "    print('The filter threw an error. Not the expected output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop everything that has already been matched well\n",
    "# already matched:\n",
    "# • (htg belongs to pCtg and matching sequence overlaps)\n",
    "# • (htg belongs to pCtg but matching sequence does not overlap)\n",
    "# • (htg does not belong to pCtg)\n",
    "\n",
    "query_ids_to_drop = query_ids_to_drop + best_hits_different_contig['Query'].unique().tolist()\n",
    "target_ids_to_drop = target_ids_to_drop + best_hits_different_contig['Target'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "#now print out the p_proteins without alleles (i.e. no good match)\n",
    "out_name = P_GENOME + '.no_alleles'\n",
    "_tmp_df[~_tmp_df['Query'].isin(query_ids_to_drop)].drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#for filtering different query and targets to drop are required to be dropped\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    query_ids_to_drop_filtered = \\\n",
    "    best_hits_t_contig_and_h_contig[(best_hits_t_contig_and_h_contig['QCov'] >= Qcov_cut_off) & (best_hits_t_contig_and_h_contig['PctID']>= PctID_cut_off)]['Query'].unique().tolist() \\\n",
    "    + best_hits_same_contig_no_overlap[(best_hits_same_contig_no_overlap['QCov'] >= Qcov_cut_off) & (best_hits_same_contig_no_overlap['PctID'] >= PctID_cut_off)]['Query'].unique().tolist()\\\n",
    "    + best_hits_different_contig[(best_hits_different_contig['QCov'] >= Qcov_cut_off) & (best_hits_different_contig['PctID'] >= PctID_cut_off)]['Query'].unique().tolist()\n",
    "    \n",
    "    target_ids_to_drop_filtered = \\\n",
    "    best_hits_t_contig_and_h_contig[(best_hits_t_contig_and_h_contig['QCov'] >= Qcov_cut_off) & (best_hits_t_contig_and_h_contig['PctID']>= PctID_cut_off)]['Target'].unique().tolist()\\\n",
    "    + best_hits_same_contig_no_overlap[(best_hits_same_contig_no_overlap['QCov'] >= Qcov_cut_off) & (best_hits_same_contig_no_overlap['PctID'] >= PctID_cut_off)]['Target'].unique().tolist() \\\n",
    "    + best_hits_different_contig[(best_hits_different_contig['QCov'] >= Qcov_cut_off) & (best_hits_different_contig['PctID'] >= PctID_cut_off)]['Target'].unique().tolist()\n",
    "    out_name = P_GENOME + '.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    #save the dataframe\n",
    "    _tmp_df[~_tmp_df['Query'].isin(query_ids_to_drop_filtered)].drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)    \n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")                                                                                                                             "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save back the dataframe with '.analyzed' added to the key\n",
    "blast_out_dict[_key+'.analyzed'] = _tmp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pull out the h on p blast dataframe\n",
    "_key = [x for x in blast_out_dict.keys() if x.startswith(H_GENOME) and x.split('.')[-2] == 'blastp' and x.endswith('outfmt6')][0]\n",
    "_tmp_df = blast_out_dict[_key]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pull out all h_contigs without alleles\n",
    "_tmp_df_no_alleles = _tmp_df[~_tmp_df['Query'].isin(target_ids_to_drop)]\n",
    "out_name = H_GENOME + '.no_alleles'\n",
    "_tmp_df_no_alleles.drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now pull out all h_proteins that have a blast hit but were left behind with selection \n",
    "#based on proteinortho and best blast hits above. No QCov and PctID filtering\n",
    "_tmp_df_no_alleles_grouped = _tmp_df_no_alleles[_tmp_df_no_alleles[\"Target\"] != 'NaN'].groupby('Query')\n",
    "h_protein_no_alleles = _tmp_df_no_alleles_grouped.apply(lambda g: g[(g['e-value'] == g['e-value'].min())]) \n",
    "h_protein_no_alleles = h_protein_no_alleles.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "out_name = H_GENOME + '.best_p_hits.no_alleles'\n",
    "h_protein_no_alleles.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "#write done the hits with no blast hit at all. Meaning blasting h on p gave no results\n",
    "out_name = H_GENOME + '.no_p_hits.no_alleles'\n",
    "_tmp_df_no_alleles[_tmp_df_no_alleles[\"Target\"] == 'NaN']['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "#now pull out all h_proteins that have a blast hit but were left behind with selection of best blast hits above. With QCov and PctID filtering.\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "#pull out all h proteins without alleles when filtering was applied\n",
    "    _tmp_df_no_alleles_filtered = _tmp_df[~_tmp_df['Query'].isin(target_ids_to_drop_filtered)]\n",
    "    out_name = H_GENOME + '.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    _tmp_df_no_alleles_filtered.drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#pull out all the h proteins with blast hit when filtering was applied and filter those as well.\n",
    "    #first get ride of all NAs in future version maybe convert QCov and PctID to floats or ints so this is not neccessary anymore\n",
    "    _tmp_df_no_alleles_filtered_no_NA = _tmp_df_no_alleles_filtered[(_tmp_df_no_alleles_filtered[\"Target\"] != 'NaN')]\n",
    "    _tmp_df_no_alleles_filtered_grouped = \\\n",
    "        _tmp_df_no_alleles_filtered_no_NA[(_tmp_df_no_alleles_filtered_no_NA['QCov'] >= Qcov_cut_off) &\\\n",
    "                                    (_tmp_df_no_alleles_filtered_no_NA['PctID'] >= PctID_cut_off)].groupby('Query') \n",
    "\n",
    "    h_protein_no_alleles_filtered = _tmp_df_no_alleles_filtered_grouped.apply(lambda g: g[(g['e-value'] == g['e-value'].min())]) \n",
    "    h_protein_no_alleles_filtered = h_protein_no_alleles_filtered.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "    out_name = H_GENOME + '.best_p_hits.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    h_protein_no_alleles_filtered.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "    target_ids_to_drop_filtered = target_ids_to_drop_filtered + h_protein_no_alleles_filtered['Query'].unique().tolist()\n",
    "    out_name = H_GENOME + '.no_p_hits.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    _tmp_df_no_alleles_filtered[~_tmp_df_no_alleles_filtered['Query'].isin(target_ids_to_drop_filtered)].drop_duplicates(subset='Query')['Query']\\\n",
    "    .to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "#write the dataframe again\n",
    "_tmp_df.to_csv(os.path.join(OUT_PATH, _key+'.allele_analysis'), index=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery',\n",
       "       'StopQuery', 'StartTarget', 'StopTarget', 'e-value', 'BitScore',\n",
       "       'QLgth', 'QCov', 'q_contig', 't_contig', 'q_contig == t_contig'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_tmp_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery',\n",
       "       'StopQuery', 'StartTarget', 'StopTarget', 'e-value', 'BitScore',\n",
       "       'QLgth', 'QCov', 'q_contig', 't_contig', 'q_contig == t_contig',\n",
       "       'p_protein', 'h_contig_overlap', 't_contig == h_contig_overlap'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#might be good to write these analyzed dataframes out as well to make life a bit easier\n",
    "blast_out_dict['%s.%s.0.001.blastp.outfmt6.analyzed' % (P_GENOME, H_GENOME)].columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1803 haplotig proteins have no allele\n",
      "Of these haplotig proteins, 1516 have a BLAST hit\n",
      "Thus, 287 haplotig proteins have both no allele and no BLAST hit\n",
      "\n",
      "For each haplotig protein with no allele, if it has a BLAST hit (blast h on p) we take the best BLAST hit. If the %ID of ths hit is >=90%, we assign it as an allele.\n",
      "Using this method, we reassigned 941 haplotig proteins as alleles\n"
     ]
    }
   ],
   "source": [
    "htgOnPctgBlast = pd.read_csv(os.path.join(OUT_PATH, '%s.%s.0.001.blastp.outfmt6' % (H_GENOME, P_GENOME)), sep='\\t', index_col=None, header=None, names=blast_header)\n",
    "htgNoAllelesDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.no_alleles' % H_GENOME), names=['Query'])\n",
    "\n",
    "htgNoAllelesBlastDf = htgOnPctgBlast[htgOnPctgBlast['Query'].isin(htgNoAllelesDf.Query)]\n",
    "print('%s haplotig proteins have no allele' % len(htgNoAllelesDf))\n",
    "print('Of these haplotig proteins, %s have a BLAST hit' % len(htgNoAllelesBlastDf['Query'].unique()))\n",
    "print('Thus, %s haplotig proteins have both no allele and no BLAST hit' % (len(htgNoAllelesDf['Query']) - len(htgNoAllelesBlastDf['Query'].unique())))\n",
    "\n",
    "# Look at haplotig on primary contig blast for haplotig proteins with no allele,\n",
    "# and take the best hit (based on e-value and bitscore. If it is tied, then all best hits will\n",
    "# be taken. If the best hit(s) is >=90%ID, assign the haplotig.\n",
    "\n",
    "print('\\nFor each haplotig protein with no allele, if it has a BLAST hit (blast h on p) we take the best BLAST hit. If the %ID of ths hit is >=90%, we assign it as an allele.')\n",
    "htgNoAllelesBlastDf_grouped = htgNoAllelesBlastDf.groupby('Query')\n",
    "htgNoAllelesBlastDf = htgNoAllelesBlastDf_grouped.apply(lambda g: g[g['PctID'] == g['PctID'].max()])\n",
    "\n",
    "htgNoAllelesBlastDf = htgNoAllelesBlastDf[htgNoAllelesBlastDf['PctID'] >= 90]\n",
    "out_name = H_GENOME + '.manual_assigned.alleles'\n",
    "htgNoAllelesBlastDf.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "print('Using this method, we reassigned %s haplotig proteins as alleles' % len(htgNoAllelesBlastDf))\n",
    "\n",
    "htgNoAllelesDf = htgNoAllelesDf[~htgNoAllelesDf['Query'].isin(htgNoAllelesBlastDf.Query)]\n",
    "out_name = H_GENOME + '.no_alleles'\n",
    "htgNoAllelesDf.to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>e-value</th>\n",
       "      <th>BitScore</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Query</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_001.8</th>\n",
       "      <th>158696</th>\n",
       "      <td>evm.model.hcontig_000_001.8</td>\n",
       "      <td>evm.model.pcontig_000.1077</td>\n",
       "      <td>100.00</td>\n",
       "      <td>167</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "      <td>4.000000e-120</td>\n",
       "      <td>351.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_007.8</th>\n",
       "      <th>103152</th>\n",
       "      <td>evm.model.hcontig_000_007.8</td>\n",
       "      <td>evm.model.pcontig_000.501</td>\n",
       "      <td>95.40</td>\n",
       "      <td>261</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>261</td>\n",
       "      <td>899</td>\n",
       "      <td>1159</td>\n",
       "      <td>1.000000e-172</td>\n",
       "      <td>509.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_013.2</th>\n",
       "      <th>7275</th>\n",
       "      <td>evm.model.hcontig_000_013.2</td>\n",
       "      <td>evm.model.pcontig_000.916</td>\n",
       "      <td>100.00</td>\n",
       "      <td>327</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>92</td>\n",
       "      <td>418</td>\n",
       "      <td>2</td>\n",
       "      <td>328</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>667.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_013.4</th>\n",
       "      <th>7277</th>\n",
       "      <td>evm.model.hcontig_000_013.4</td>\n",
       "      <td>evm.model.pcontig_000.917</td>\n",
       "      <td>100.00</td>\n",
       "      <td>74</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>101</td>\n",
       "      <td>109</td>\n",
       "      <td>182</td>\n",
       "      <td>5.000000e-49</td>\n",
       "      <td>155.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_014.1</th>\n",
       "      <th>48440</th>\n",
       "      <td>evm.model.hcontig_000_014.1</td>\n",
       "      <td>evm.model.pcontig_000.106</td>\n",
       "      <td>99.79</td>\n",
       "      <td>477</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>477</td>\n",
       "      <td>1</td>\n",
       "      <td>477</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>979.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_015.3</th>\n",
       "      <th>86815</th>\n",
       "      <td>evm.model.hcontig_000_015.3</td>\n",
       "      <td>evm.model.pcontig_000.873</td>\n",
       "      <td>99.75</td>\n",
       "      <td>408</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>90</td>\n",
       "      <td>497</td>\n",
       "      <td>109</td>\n",
       "      <td>516</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>836.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_015.4</th>\n",
       "      <th>86816</th>\n",
       "      <td>evm.model.hcontig_000_015.4</td>\n",
       "      <td>evm.model.pcontig_000.873</td>\n",
       "      <td>100.00</td>\n",
       "      <td>263</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>263</td>\n",
       "      <td>617</td>\n",
       "      <td>879</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_016.2</th>\n",
       "      <th>92327</th>\n",
       "      <td>evm.model.hcontig_000_016.2</td>\n",
       "      <td>evm.model.pcontig_004.580</td>\n",
       "      <td>94.74</td>\n",
       "      <td>57</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>78</td>\n",
       "      <td>77</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000e-31</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_016.5</th>\n",
       "      <th>92334</th>\n",
       "      <td>evm.model.hcontig_000_016.5</td>\n",
       "      <td>evm.model.pcontig_000.186</td>\n",
       "      <td>100.00</td>\n",
       "      <td>654</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>654</td>\n",
       "      <td>120</td>\n",
       "      <td>773</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1332.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_016.6</th>\n",
       "      <th>92337</th>\n",
       "      <td>evm.model.hcontig_000_016.6</td>\n",
       "      <td>evm.model.pcontig_000.186</td>\n",
       "      <td>100.00</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>81</td>\n",
       "      <td>1</td>\n",
       "      <td>81</td>\n",
       "      <td>3.000000e-49</td>\n",
       "      <td>165.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_019.13</th>\n",
       "      <th>125852</th>\n",
       "      <td>evm.model.hcontig_000_019.13</td>\n",
       "      <td>evm.model.pcontig_004.580</td>\n",
       "      <td>94.74</td>\n",
       "      <td>57</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>78</td>\n",
       "      <td>77</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000e-31</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_019.41</th>\n",
       "      <th>126259</th>\n",
       "      <td>evm.model.hcontig_000_019.41</td>\n",
       "      <td>evm.model.pcontig_000.214</td>\n",
       "      <td>100.00</td>\n",
       "      <td>94</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>94</td>\n",
       "      <td>1</td>\n",
       "      <td>94</td>\n",
       "      <td>6.000000e-59</td>\n",
       "      <td>190.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_021.2</th>\n",
       "      <th>210812</th>\n",
       "      <td>evm.model.hcontig_000_021.2</td>\n",
       "      <td>evm.model.pcontig_000.998</td>\n",
       "      <td>100.00</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>93</td>\n",
       "      <td>351</td>\n",
       "      <td>443</td>\n",
       "      <td>1.000000e-57</td>\n",
       "      <td>189.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_023.2</th>\n",
       "      <th>98825</th>\n",
       "      <td>evm.model.hcontig_000_023.2</td>\n",
       "      <td>evm.model.pcontig_000.695</td>\n",
       "      <td>100.00</td>\n",
       "      <td>252</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>252</td>\n",
       "      <td>386</td>\n",
       "      <td>637</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>523.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_025.3</th>\n",
       "      <th>10767</th>\n",
       "      <td>evm.model.hcontig_000_025.3</td>\n",
       "      <td>evm.model.pcontig_000.77</td>\n",
       "      <td>100.00</td>\n",
       "      <td>149</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>149</td>\n",
       "      <td>121</td>\n",
       "      <td>269</td>\n",
       "      <td>7.000000e-107</td>\n",
       "      <td>308.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_028.1</th>\n",
       "      <th>129821</th>\n",
       "      <td>evm.model.hcontig_000_028.1</td>\n",
       "      <td>evm.model.pcontig_000.240</td>\n",
       "      <td>99.20</td>\n",
       "      <td>125</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>125</td>\n",
       "      <td>2.000000e-84</td>\n",
       "      <td>246.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_030.3</th>\n",
       "      <th>9490</th>\n",
       "      <td>evm.model.hcontig_000_030.3</td>\n",
       "      <td>evm.model.pcontig_000.818</td>\n",
       "      <td>99.35</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>2.000000e-108</td>\n",
       "      <td>308.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_030.4</th>\n",
       "      <th>9498</th>\n",
       "      <td>evm.model.hcontig_000_030.4</td>\n",
       "      <td>evm.model.pcontig_000.819</td>\n",
       "      <td>99.49</td>\n",
       "      <td>389</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>389</td>\n",
       "      <td>1</td>\n",
       "      <td>389</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>786.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_031.30</th>\n",
       "      <th>5461</th>\n",
       "      <td>evm.model.hcontig_000_031.30</td>\n",
       "      <td>evm.model.pcontig_000.465</td>\n",
       "      <td>100.00</td>\n",
       "      <td>401</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>401</td>\n",
       "      <td>1</td>\n",
       "      <td>401</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>833.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_033.1</th>\n",
       "      <th>107444</th>\n",
       "      <td>evm.model.hcontig_000_033.1</td>\n",
       "      <td>evm.model.pcontig_000.516</td>\n",
       "      <td>99.41</td>\n",
       "      <td>170</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>170</td>\n",
       "      <td>1</td>\n",
       "      <td>170</td>\n",
       "      <td>4.000000e-120</td>\n",
       "      <td>352.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_036.13</th>\n",
       "      <th>130099</th>\n",
       "      <td>evm.model.hcontig_000_036.13</td>\n",
       "      <td>evm.model.pcontig_000.919</td>\n",
       "      <td>97.04</td>\n",
       "      <td>135</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>135</td>\n",
       "      <td>1</td>\n",
       "      <td>135</td>\n",
       "      <td>6.000000e-80</td>\n",
       "      <td>251.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_037.8</th>\n",
       "      <th>125325</th>\n",
       "      <td>evm.model.hcontig_000_037.8</td>\n",
       "      <td>evm.model.pcontig_000.51</td>\n",
       "      <td>97.30</td>\n",
       "      <td>889</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>866</td>\n",
       "      <td>1</td>\n",
       "      <td>889</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1793.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_039.3</th>\n",
       "      <th>11187</th>\n",
       "      <td>evm.model.hcontig_000_039.3</td>\n",
       "      <td>evm.model.pcontig_000.108</td>\n",
       "      <td>100.00</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>158</td>\n",
       "      <td>329</td>\n",
       "      <td>471</td>\n",
       "      <td>2.000000e-98</td>\n",
       "      <td>294.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_039.4</th>\n",
       "      <th>11197</th>\n",
       "      <td>evm.model.hcontig_000_039.4</td>\n",
       "      <td>evm.model.pcontig_000.108</td>\n",
       "      <td>98.84</td>\n",
       "      <td>172</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>172</td>\n",
       "      <td>1</td>\n",
       "      <td>172</td>\n",
       "      <td>1.000000e-104</td>\n",
       "      <td>311.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_046.13</th>\n",
       "      <th>102438</th>\n",
       "      <td>evm.model.hcontig_000_046.13</td>\n",
       "      <td>evm.model.pcontig_000.60</td>\n",
       "      <td>100.00</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>65</td>\n",
       "      <td>81</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>3.000000e-05</td>\n",
       "      <td>39.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_046.14</th>\n",
       "      <th>102439</th>\n",
       "      <td>evm.model.hcontig_000_046.14</td>\n",
       "      <td>evm.model.pcontig_021.5</td>\n",
       "      <td>91.53</td>\n",
       "      <td>118</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>118</td>\n",
       "      <td>233</td>\n",
       "      <td>350</td>\n",
       "      <td>2.000000e-73</td>\n",
       "      <td>229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_046.15</th>\n",
       "      <th>102450</th>\n",
       "      <td>evm.model.hcontig_000_046.15</td>\n",
       "      <td>evm.model.pcontig_021.5</td>\n",
       "      <td>94.44</td>\n",
       "      <td>108</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>108</td>\n",
       "      <td>394</td>\n",
       "      <td>501</td>\n",
       "      <td>4.000000e-66</td>\n",
       "      <td>209.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_046.19</th>\n",
       "      <th>102462</th>\n",
       "      <td>evm.model.hcontig_000_046.19</td>\n",
       "      <td>evm.model.pcontig_021.5</td>\n",
       "      <td>90.24</td>\n",
       "      <td>82</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>82</td>\n",
       "      <td>409</td>\n",
       "      <td>490</td>\n",
       "      <td>3.000000e-47</td>\n",
       "      <td>158.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_046.26</th>\n",
       "      <th>102506</th>\n",
       "      <td>evm.model.hcontig_000_046.26</td>\n",
       "      <td>evm.model.pcontig_000.74</td>\n",
       "      <td>93.75</td>\n",
       "      <td>48</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>177</td>\n",
       "      <td>224</td>\n",
       "      <td>1</td>\n",
       "      <td>48</td>\n",
       "      <td>3.000000e-22</td>\n",
       "      <td>88.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_046.3</th>\n",
       "      <th>102264</th>\n",
       "      <td>evm.model.hcontig_000_046.3</td>\n",
       "      <td>evm.model.pcontig_000.50</td>\n",
       "      <td>99.47</td>\n",
       "      <td>190</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>190</td>\n",
       "      <td>1</td>\n",
       "      <td>190</td>\n",
       "      <td>3.000000e-128</td>\n",
       "      <td>361.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_076_003.2</th>\n",
       "      <th>10109</th>\n",
       "      <td>evm.model.hcontig_076_003.2</td>\n",
       "      <td>evm.model.pcontig_013.279</td>\n",
       "      <td>90.41</td>\n",
       "      <td>73</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>73</td>\n",
       "      <td>840</td>\n",
       "      <td>912</td>\n",
       "      <td>3.000000e-39</td>\n",
       "      <td>138.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_076_004.6</th>\n",
       "      <th>135085</th>\n",
       "      <td>evm.model.hcontig_076_004.6</td>\n",
       "      <td>evm.model.pcontig_076.34</td>\n",
       "      <td>100.00</td>\n",
       "      <td>122</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>122</td>\n",
       "      <td>239</td>\n",
       "      <td>360</td>\n",
       "      <td>7.000000e-84</td>\n",
       "      <td>251.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_078_001.1</th>\n",
       "      <th>141481</th>\n",
       "      <td>evm.model.hcontig_078_001.1</td>\n",
       "      <td>evm.model.pcontig_078.19</td>\n",
       "      <td>97.25</td>\n",
       "      <td>1202</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1169</td>\n",
       "      <td>507</td>\n",
       "      <td>1708</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2381.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_079_003.13</th>\n",
       "      <th>111991</th>\n",
       "      <td>evm.model.hcontig_079_003.13</td>\n",
       "      <td>evm.model.pcontig_058.7</td>\n",
       "      <td>97.51</td>\n",
       "      <td>442</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>438</td>\n",
       "      <td>1</td>\n",
       "      <td>442</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>876.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_079_003.14</th>\n",
       "      <th>111992</th>\n",
       "      <td>evm.model.hcontig_079_003.14</td>\n",
       "      <td>evm.model.pcontig_058.7</td>\n",
       "      <td>100.00</td>\n",
       "      <td>194</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>194</td>\n",
       "      <td>510</td>\n",
       "      <td>703</td>\n",
       "      <td>2.000000e-136</td>\n",
       "      <td>406.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_081_002.1</th>\n",
       "      <th>140222</th>\n",
       "      <td>evm.model.hcontig_081_002.1</td>\n",
       "      <td>evm.model.pcontig_081.6</td>\n",
       "      <td>95.45</td>\n",
       "      <td>132</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "      <td>132</td>\n",
       "      <td>1.000000e-83</td>\n",
       "      <td>253.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_081_004.4</th>\n",
       "      <th>185</th>\n",
       "      <td>evm.model.hcontig_081_004.4</td>\n",
       "      <td>evm.model.pcontig_070.58</td>\n",
       "      <td>90.67</td>\n",
       "      <td>418</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>407</td>\n",
       "      <td>1</td>\n",
       "      <td>410</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>741.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_081_005.1</th>\n",
       "      <th>5872</th>\n",
       "      <td>evm.model.hcontig_081_005.1</td>\n",
       "      <td>evm.model.pcontig_070.58</td>\n",
       "      <td>90.27</td>\n",
       "      <td>401</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>390</td>\n",
       "      <td>1</td>\n",
       "      <td>393</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>705.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_081_005.5</th>\n",
       "      <th>5887</th>\n",
       "      <td>evm.model.hcontig_081_005.5</td>\n",
       "      <td>evm.model.pcontig_058.18</td>\n",
       "      <td>93.30</td>\n",
       "      <td>179</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>179</td>\n",
       "      <td>686</td>\n",
       "      <td>864</td>\n",
       "      <td>6.000000e-113</td>\n",
       "      <td>347.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_081_005.6</th>\n",
       "      <th>5890</th>\n",
       "      <td>evm.model.hcontig_081_005.6</td>\n",
       "      <td>evm.model.pcontig_058.18</td>\n",
       "      <td>95.12</td>\n",
       "      <td>82</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100</td>\n",
       "      <td>891</td>\n",
       "      <td>972</td>\n",
       "      <td>2.000000e-46</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_084_001.19</th>\n",
       "      <th>72572</th>\n",
       "      <td>evm.model.hcontig_084_001.19</td>\n",
       "      <td>evm.model.pcontig_084.19</td>\n",
       "      <td>100.00</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>64</td>\n",
       "      <td>174</td>\n",
       "      <td>232</td>\n",
       "      <td>6.000000e-34</td>\n",
       "      <td>119.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_084_001.21</th>\n",
       "      <th>72574</th>\n",
       "      <td>evm.model.hcontig_084_001.21</td>\n",
       "      <td>evm.model.pcontig_084.20</td>\n",
       "      <td>97.44</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>416</td>\n",
       "      <td>454</td>\n",
       "      <td>2.000000e-21</td>\n",
       "      <td>85.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_085_001.12</th>\n",
       "      <th>36892</th>\n",
       "      <td>evm.model.hcontig_085_001.12</td>\n",
       "      <td>evm.model.pcontig_006.374</td>\n",
       "      <td>98.12</td>\n",
       "      <td>160</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>160</td>\n",
       "      <td>1</td>\n",
       "      <td>160</td>\n",
       "      <td>7.000000e-101</td>\n",
       "      <td>297.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">evm.model.hcontig_085_001.3</th>\n",
       "      <th>36493</th>\n",
       "      <td>evm.model.hcontig_085_001.3</td>\n",
       "      <td>evm.model.pcontig_041.10</td>\n",
       "      <td>100.00</td>\n",
       "      <td>295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>295</td>\n",
       "      <td>326</td>\n",
       "      <td>620</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>614.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36494</th>\n",
       "      <td>evm.model.hcontig_085_001.3</td>\n",
       "      <td>evm.model.pcontig_015.347</td>\n",
       "      <td>100.00</td>\n",
       "      <td>295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>295</td>\n",
       "      <td>353</td>\n",
       "      <td>647</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>614.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36495</th>\n",
       "      <td>evm.model.hcontig_085_001.3</td>\n",
       "      <td>evm.model.pcontig_009.518</td>\n",
       "      <td>100.00</td>\n",
       "      <td>295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>295</td>\n",
       "      <td>295</td>\n",
       "      <td>589</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>613.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_085_001.4</th>\n",
       "      <th>36552</th>\n",
       "      <td>evm.model.hcontig_085_001.4</td>\n",
       "      <td>evm.model.pcontig_008.5</td>\n",
       "      <td>99.38</td>\n",
       "      <td>160</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>160</td>\n",
       "      <td>1</td>\n",
       "      <td>160</td>\n",
       "      <td>1.000000e-110</td>\n",
       "      <td>320.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">evm.model.hcontig_085_001.5</th>\n",
       "      <th>36599</th>\n",
       "      <td>evm.model.hcontig_085_001.5</td>\n",
       "      <td>evm.model.pcontig_008.4</td>\n",
       "      <td>99.54</td>\n",
       "      <td>650</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>650</td>\n",
       "      <td>1</td>\n",
       "      <td>650</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1363.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36600</th>\n",
       "      <td>evm.model.hcontig_085_001.5</td>\n",
       "      <td>evm.model.pcontig_041.11</td>\n",
       "      <td>99.54</td>\n",
       "      <td>651</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>651</td>\n",
       "      <td>1</td>\n",
       "      <td>651</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1356.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_085_001.6</th>\n",
       "      <th>36701</th>\n",
       "      <td>evm.model.hcontig_085_001.6</td>\n",
       "      <td>evm.model.pcontig_008.4</td>\n",
       "      <td>99.70</td>\n",
       "      <td>333</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>333</td>\n",
       "      <td>668</td>\n",
       "      <td>1000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>678.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_085_003.1</th>\n",
       "      <th>6871</th>\n",
       "      <td>evm.model.hcontig_085_003.1</td>\n",
       "      <td>evm.model.pcontig_015.347</td>\n",
       "      <td>99.69</td>\n",
       "      <td>647</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>647</td>\n",
       "      <td>1</td>\n",
       "      <td>647</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1343.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">evm.model.hcontig_085_003.2</th>\n",
       "      <th>6937</th>\n",
       "      <td>evm.model.hcontig_085_003.2</td>\n",
       "      <td>evm.model.pcontig_008.4</td>\n",
       "      <td>99.69</td>\n",
       "      <td>650</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>650</td>\n",
       "      <td>1</td>\n",
       "      <td>650</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1366.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6938</th>\n",
       "      <td>evm.model.hcontig_085_003.2</td>\n",
       "      <td>evm.model.pcontig_041.11</td>\n",
       "      <td>99.69</td>\n",
       "      <td>651</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>651</td>\n",
       "      <td>1</td>\n",
       "      <td>651</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1360.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_085_004.5</th>\n",
       "      <th>164308</th>\n",
       "      <td>evm.model.hcontig_085_004.5</td>\n",
       "      <td>evm.model.pcontig_048.2</td>\n",
       "      <td>98.06</td>\n",
       "      <td>103</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>114</td>\n",
       "      <td>108</td>\n",
       "      <td>210</td>\n",
       "      <td>2.000000e-70</td>\n",
       "      <td>212.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_091_003.1</th>\n",
       "      <th>130417</th>\n",
       "      <td>evm.model.hcontig_091_003.1</td>\n",
       "      <td>evm.model.pcontig_029.3</td>\n",
       "      <td>97.70</td>\n",
       "      <td>174</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>66</td>\n",
       "      <td>239</td>\n",
       "      <td>15</td>\n",
       "      <td>188</td>\n",
       "      <td>6.000000e-120</td>\n",
       "      <td>347.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_091_003.2</th>\n",
       "      <th>130422</th>\n",
       "      <td>evm.model.hcontig_091_003.2</td>\n",
       "      <td>evm.model.pcontig_029.2</td>\n",
       "      <td>97.06</td>\n",
       "      <td>68</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>502</td>\n",
       "      <td>568</td>\n",
       "      <td>4.000000e-35</td>\n",
       "      <td>125.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_093_002.4</th>\n",
       "      <th>183334</th>\n",
       "      <td>evm.model.hcontig_093_002.4</td>\n",
       "      <td>evm.model.pcontig_093.2</td>\n",
       "      <td>99.55</td>\n",
       "      <td>440</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>440</td>\n",
       "      <td>1</td>\n",
       "      <td>440</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>907.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_095_001.2</th>\n",
       "      <th>6863</th>\n",
       "      <td>evm.model.hcontig_095_001.2</td>\n",
       "      <td>evm.model.pcontig_095.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>88</td>\n",
       "      <td>141</td>\n",
       "      <td>228</td>\n",
       "      <td>1.000000e-55</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_095_002.1</th>\n",
       "      <th>35977</th>\n",
       "      <td>evm.model.hcontig_095_002.1</td>\n",
       "      <td>evm.model.pcontig_095.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>88</td>\n",
       "      <td>141</td>\n",
       "      <td>228</td>\n",
       "      <td>1.000000e-55</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_095_002.11</th>\n",
       "      <th>36000</th>\n",
       "      <td>evm.model.hcontig_095_002.11</td>\n",
       "      <td>evm.model.pcontig_095.10</td>\n",
       "      <td>99.16</td>\n",
       "      <td>119</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>119</td>\n",
       "      <td>1</td>\n",
       "      <td>119</td>\n",
       "      <td>1.000000e-81</td>\n",
       "      <td>244.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>941 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            Query  \\\n",
       "Query                                                               \n",
       "evm.model.hcontig_000_001.8  158696   evm.model.hcontig_000_001.8   \n",
       "evm.model.hcontig_000_007.8  103152   evm.model.hcontig_000_007.8   \n",
       "evm.model.hcontig_000_013.2  7275     evm.model.hcontig_000_013.2   \n",
       "evm.model.hcontig_000_013.4  7277     evm.model.hcontig_000_013.4   \n",
       "evm.model.hcontig_000_014.1  48440    evm.model.hcontig_000_014.1   \n",
       "evm.model.hcontig_000_015.3  86815    evm.model.hcontig_000_015.3   \n",
       "evm.model.hcontig_000_015.4  86816    evm.model.hcontig_000_015.4   \n",
       "evm.model.hcontig_000_016.2  92327    evm.model.hcontig_000_016.2   \n",
       "evm.model.hcontig_000_016.5  92334    evm.model.hcontig_000_016.5   \n",
       "evm.model.hcontig_000_016.6  92337    evm.model.hcontig_000_016.6   \n",
       "evm.model.hcontig_000_019.13 125852  evm.model.hcontig_000_019.13   \n",
       "evm.model.hcontig_000_019.41 126259  evm.model.hcontig_000_019.41   \n",
       "evm.model.hcontig_000_021.2  210812   evm.model.hcontig_000_021.2   \n",
       "evm.model.hcontig_000_023.2  98825    evm.model.hcontig_000_023.2   \n",
       "evm.model.hcontig_000_025.3  10767    evm.model.hcontig_000_025.3   \n",
       "evm.model.hcontig_000_028.1  129821   evm.model.hcontig_000_028.1   \n",
       "evm.model.hcontig_000_030.3  9490     evm.model.hcontig_000_030.3   \n",
       "evm.model.hcontig_000_030.4  9498     evm.model.hcontig_000_030.4   \n",
       "evm.model.hcontig_000_031.30 5461    evm.model.hcontig_000_031.30   \n",
       "evm.model.hcontig_000_033.1  107444   evm.model.hcontig_000_033.1   \n",
       "evm.model.hcontig_000_036.13 130099  evm.model.hcontig_000_036.13   \n",
       "evm.model.hcontig_000_037.8  125325   evm.model.hcontig_000_037.8   \n",
       "evm.model.hcontig_000_039.3  11187    evm.model.hcontig_000_039.3   \n",
       "evm.model.hcontig_000_039.4  11197    evm.model.hcontig_000_039.4   \n",
       "evm.model.hcontig_000_046.13 102438  evm.model.hcontig_000_046.13   \n",
       "evm.model.hcontig_000_046.14 102439  evm.model.hcontig_000_046.14   \n",
       "evm.model.hcontig_000_046.15 102450  evm.model.hcontig_000_046.15   \n",
       "evm.model.hcontig_000_046.19 102462  evm.model.hcontig_000_046.19   \n",
       "evm.model.hcontig_000_046.26 102506  evm.model.hcontig_000_046.26   \n",
       "evm.model.hcontig_000_046.3  102264   evm.model.hcontig_000_046.3   \n",
       "...                                                           ...   \n",
       "evm.model.hcontig_076_003.2  10109    evm.model.hcontig_076_003.2   \n",
       "evm.model.hcontig_076_004.6  135085   evm.model.hcontig_076_004.6   \n",
       "evm.model.hcontig_078_001.1  141481   evm.model.hcontig_078_001.1   \n",
       "evm.model.hcontig_079_003.13 111991  evm.model.hcontig_079_003.13   \n",
       "evm.model.hcontig_079_003.14 111992  evm.model.hcontig_079_003.14   \n",
       "evm.model.hcontig_081_002.1  140222   evm.model.hcontig_081_002.1   \n",
       "evm.model.hcontig_081_004.4  185      evm.model.hcontig_081_004.4   \n",
       "evm.model.hcontig_081_005.1  5872     evm.model.hcontig_081_005.1   \n",
       "evm.model.hcontig_081_005.5  5887     evm.model.hcontig_081_005.5   \n",
       "evm.model.hcontig_081_005.6  5890     evm.model.hcontig_081_005.6   \n",
       "evm.model.hcontig_084_001.19 72572   evm.model.hcontig_084_001.19   \n",
       "evm.model.hcontig_084_001.21 72574   evm.model.hcontig_084_001.21   \n",
       "evm.model.hcontig_085_001.12 36892   evm.model.hcontig_085_001.12   \n",
       "evm.model.hcontig_085_001.3  36493    evm.model.hcontig_085_001.3   \n",
       "                             36494    evm.model.hcontig_085_001.3   \n",
       "                             36495    evm.model.hcontig_085_001.3   \n",
       "evm.model.hcontig_085_001.4  36552    evm.model.hcontig_085_001.4   \n",
       "evm.model.hcontig_085_001.5  36599    evm.model.hcontig_085_001.5   \n",
       "                             36600    evm.model.hcontig_085_001.5   \n",
       "evm.model.hcontig_085_001.6  36701    evm.model.hcontig_085_001.6   \n",
       "evm.model.hcontig_085_003.1  6871     evm.model.hcontig_085_003.1   \n",
       "evm.model.hcontig_085_003.2  6937     evm.model.hcontig_085_003.2   \n",
       "                             6938     evm.model.hcontig_085_003.2   \n",
       "evm.model.hcontig_085_004.5  164308   evm.model.hcontig_085_004.5   \n",
       "evm.model.hcontig_091_003.1  130417   evm.model.hcontig_091_003.1   \n",
       "evm.model.hcontig_091_003.2  130422   evm.model.hcontig_091_003.2   \n",
       "evm.model.hcontig_093_002.4  183334   evm.model.hcontig_093_002.4   \n",
       "evm.model.hcontig_095_001.2  6863     evm.model.hcontig_095_001.2   \n",
       "evm.model.hcontig_095_002.1  35977    evm.model.hcontig_095_002.1   \n",
       "evm.model.hcontig_095_002.11 36000   evm.model.hcontig_095_002.11   \n",
       "\n",
       "                                                         Target   PctID  \\\n",
       "Query                                                                     \n",
       "evm.model.hcontig_000_001.8  158696  evm.model.pcontig_000.1077  100.00   \n",
       "evm.model.hcontig_000_007.8  103152   evm.model.pcontig_000.501   95.40   \n",
       "evm.model.hcontig_000_013.2  7275     evm.model.pcontig_000.916  100.00   \n",
       "evm.model.hcontig_000_013.4  7277     evm.model.pcontig_000.917  100.00   \n",
       "evm.model.hcontig_000_014.1  48440    evm.model.pcontig_000.106   99.79   \n",
       "evm.model.hcontig_000_015.3  86815    evm.model.pcontig_000.873   99.75   \n",
       "evm.model.hcontig_000_015.4  86816    evm.model.pcontig_000.873  100.00   \n",
       "evm.model.hcontig_000_016.2  92327    evm.model.pcontig_004.580   94.74   \n",
       "evm.model.hcontig_000_016.5  92334    evm.model.pcontig_000.186  100.00   \n",
       "evm.model.hcontig_000_016.6  92337    evm.model.pcontig_000.186  100.00   \n",
       "evm.model.hcontig_000_019.13 125852   evm.model.pcontig_004.580   94.74   \n",
       "evm.model.hcontig_000_019.41 126259   evm.model.pcontig_000.214  100.00   \n",
       "evm.model.hcontig_000_021.2  210812   evm.model.pcontig_000.998  100.00   \n",
       "evm.model.hcontig_000_023.2  98825    evm.model.pcontig_000.695  100.00   \n",
       "evm.model.hcontig_000_025.3  10767     evm.model.pcontig_000.77  100.00   \n",
       "evm.model.hcontig_000_028.1  129821   evm.model.pcontig_000.240   99.20   \n",
       "evm.model.hcontig_000_030.3  9490     evm.model.pcontig_000.818   99.35   \n",
       "evm.model.hcontig_000_030.4  9498     evm.model.pcontig_000.819   99.49   \n",
       "evm.model.hcontig_000_031.30 5461     evm.model.pcontig_000.465  100.00   \n",
       "evm.model.hcontig_000_033.1  107444   evm.model.pcontig_000.516   99.41   \n",
       "evm.model.hcontig_000_036.13 130099   evm.model.pcontig_000.919   97.04   \n",
       "evm.model.hcontig_000_037.8  125325    evm.model.pcontig_000.51   97.30   \n",
       "evm.model.hcontig_000_039.3  11187    evm.model.pcontig_000.108  100.00   \n",
       "evm.model.hcontig_000_039.4  11197    evm.model.pcontig_000.108   98.84   \n",
       "evm.model.hcontig_000_046.13 102438    evm.model.pcontig_000.60  100.00   \n",
       "evm.model.hcontig_000_046.14 102439     evm.model.pcontig_021.5   91.53   \n",
       "evm.model.hcontig_000_046.15 102450     evm.model.pcontig_021.5   94.44   \n",
       "evm.model.hcontig_000_046.19 102462     evm.model.pcontig_021.5   90.24   \n",
       "evm.model.hcontig_000_046.26 102506    evm.model.pcontig_000.74   93.75   \n",
       "evm.model.hcontig_000_046.3  102264    evm.model.pcontig_000.50   99.47   \n",
       "...                                                         ...     ...   \n",
       "evm.model.hcontig_076_003.2  10109    evm.model.pcontig_013.279   90.41   \n",
       "evm.model.hcontig_076_004.6  135085    evm.model.pcontig_076.34  100.00   \n",
       "evm.model.hcontig_078_001.1  141481    evm.model.pcontig_078.19   97.25   \n",
       "evm.model.hcontig_079_003.13 111991     evm.model.pcontig_058.7   97.51   \n",
       "evm.model.hcontig_079_003.14 111992     evm.model.pcontig_058.7  100.00   \n",
       "evm.model.hcontig_081_002.1  140222     evm.model.pcontig_081.6   95.45   \n",
       "evm.model.hcontig_081_004.4  185       evm.model.pcontig_070.58   90.67   \n",
       "evm.model.hcontig_081_005.1  5872      evm.model.pcontig_070.58   90.27   \n",
       "evm.model.hcontig_081_005.5  5887      evm.model.pcontig_058.18   93.30   \n",
       "evm.model.hcontig_081_005.6  5890      evm.model.pcontig_058.18   95.12   \n",
       "evm.model.hcontig_084_001.19 72572     evm.model.pcontig_084.19  100.00   \n",
       "evm.model.hcontig_084_001.21 72574     evm.model.pcontig_084.20   97.44   \n",
       "evm.model.hcontig_085_001.12 36892    evm.model.pcontig_006.374   98.12   \n",
       "evm.model.hcontig_085_001.3  36493     evm.model.pcontig_041.10  100.00   \n",
       "                             36494    evm.model.pcontig_015.347  100.00   \n",
       "                             36495    evm.model.pcontig_009.518  100.00   \n",
       "evm.model.hcontig_085_001.4  36552      evm.model.pcontig_008.5   99.38   \n",
       "evm.model.hcontig_085_001.5  36599      evm.model.pcontig_008.4   99.54   \n",
       "                             36600     evm.model.pcontig_041.11   99.54   \n",
       "evm.model.hcontig_085_001.6  36701      evm.model.pcontig_008.4   99.70   \n",
       "evm.model.hcontig_085_003.1  6871     evm.model.pcontig_015.347   99.69   \n",
       "evm.model.hcontig_085_003.2  6937       evm.model.pcontig_008.4   99.69   \n",
       "                             6938      evm.model.pcontig_041.11   99.69   \n",
       "evm.model.hcontig_085_004.5  164308     evm.model.pcontig_048.2   98.06   \n",
       "evm.model.hcontig_091_003.1  130417     evm.model.pcontig_029.3   97.70   \n",
       "evm.model.hcontig_091_003.2  130422     evm.model.pcontig_029.2   97.06   \n",
       "evm.model.hcontig_093_002.4  183334     evm.model.pcontig_093.2   99.55   \n",
       "evm.model.hcontig_095_001.2  6863       evm.model.pcontig_095.1  100.00   \n",
       "evm.model.hcontig_095_002.1  35977      evm.model.pcontig_095.1  100.00   \n",
       "evm.model.hcontig_095_002.11 36000     evm.model.pcontig_095.10   99.16   \n",
       "\n",
       "                                     AlnLgth  NumMis  NumGap  StartQuery  \\\n",
       "Query                                                                      \n",
       "evm.model.hcontig_000_001.8  158696      167       0       0           1   \n",
       "evm.model.hcontig_000_007.8  103152      261      12       0           1   \n",
       "evm.model.hcontig_000_013.2  7275        327       0       0          92   \n",
       "evm.model.hcontig_000_013.4  7277         74       0       0          28   \n",
       "evm.model.hcontig_000_014.1  48440       477       1       0           1   \n",
       "evm.model.hcontig_000_015.3  86815       408       1       0          90   \n",
       "evm.model.hcontig_000_015.4  86816       263       0       0           1   \n",
       "evm.model.hcontig_000_016.2  92327        57       3       0          22   \n",
       "evm.model.hcontig_000_016.5  92334       654       0       0           1   \n",
       "evm.model.hcontig_000_016.6  92337        81       0       0           1   \n",
       "evm.model.hcontig_000_019.13 125852       57       3       0          22   \n",
       "evm.model.hcontig_000_019.41 126259       94       0       0           1   \n",
       "evm.model.hcontig_000_021.2  210812       93       0       0           1   \n",
       "evm.model.hcontig_000_023.2  98825       252       0       0           1   \n",
       "evm.model.hcontig_000_025.3  10767       149       0       0           1   \n",
       "evm.model.hcontig_000_028.1  129821      125       1       0          30   \n",
       "evm.model.hcontig_000_030.3  9490        154       1       0           1   \n",
       "evm.model.hcontig_000_030.4  9498        389       2       0           1   \n",
       "evm.model.hcontig_000_031.30 5461        401       0       0           1   \n",
       "evm.model.hcontig_000_033.1  107444      170       1       0           1   \n",
       "evm.model.hcontig_000_036.13 130099      135       4       0           1   \n",
       "evm.model.hcontig_000_037.8  125325      889       1       1           1   \n",
       "evm.model.hcontig_000_039.3  11187       143       0       0          16   \n",
       "evm.model.hcontig_000_039.4  11197       172       2       0           1   \n",
       "evm.model.hcontig_000_046.13 102438       17       0       0          65   \n",
       "evm.model.hcontig_000_046.14 102439      118      10       0           1   \n",
       "evm.model.hcontig_000_046.15 102450      108       6       0           1   \n",
       "evm.model.hcontig_000_046.19 102462       82       8       0           1   \n",
       "evm.model.hcontig_000_046.26 102506       48       3       0         177   \n",
       "evm.model.hcontig_000_046.3  102264      190       1       0           1   \n",
       "...                                      ...     ...     ...         ...   \n",
       "evm.model.hcontig_076_003.2  10109        73       7       0           1   \n",
       "evm.model.hcontig_076_004.6  135085      122       0       0           1   \n",
       "evm.model.hcontig_078_001.1  141481     1202       0       1           1   \n",
       "evm.model.hcontig_079_003.13 111991      442       7       1           1   \n",
       "evm.model.hcontig_079_003.14 111992      194       0       0           1   \n",
       "evm.model.hcontig_081_002.1  140222      132       6       0           1   \n",
       "evm.model.hcontig_081_004.4  185         418      20       3           1   \n",
       "evm.model.hcontig_081_005.1  5872        401      20       3           1   \n",
       "evm.model.hcontig_081_005.5  5887        179      12       0           1   \n",
       "evm.model.hcontig_081_005.6  5890         82       4       0          19   \n",
       "evm.model.hcontig_084_001.19 72572        59       0       0           6   \n",
       "evm.model.hcontig_084_001.21 72574        39       1       0           1   \n",
       "evm.model.hcontig_085_001.12 36892       160       3       0           1   \n",
       "evm.model.hcontig_085_001.3  36493       295       0       0           1   \n",
       "                             36494       295       0       0           1   \n",
       "                             36495       295       0       0           1   \n",
       "evm.model.hcontig_085_001.4  36552       160       1       0           1   \n",
       "evm.model.hcontig_085_001.5  36599       650       3       0           1   \n",
       "                             36600       651       3       0           1   \n",
       "evm.model.hcontig_085_001.6  36701       333       1       0           1   \n",
       "evm.model.hcontig_085_003.1  6871        647       2       0           1   \n",
       "evm.model.hcontig_085_003.2  6937        650       2       0           1   \n",
       "                             6938        651       2       0           1   \n",
       "evm.model.hcontig_085_004.5  164308      103       2       0          12   \n",
       "evm.model.hcontig_091_003.1  130417      174       4       0          66   \n",
       "evm.model.hcontig_091_003.2  130422       68       1       1           1   \n",
       "evm.model.hcontig_093_002.4  183334      440       2       0           1   \n",
       "evm.model.hcontig_095_001.2  6863         88       0       0           1   \n",
       "evm.model.hcontig_095_002.1  35977        88       0       0           1   \n",
       "evm.model.hcontig_095_002.11 36000       119       1       0           1   \n",
       "\n",
       "                                     StopQuery  StartTarget  StopTarget  \\\n",
       "Query                                                                     \n",
       "evm.model.hcontig_000_001.8  158696        167            1         167   \n",
       "evm.model.hcontig_000_007.8  103152        261          899        1159   \n",
       "evm.model.hcontig_000_013.2  7275          418            2         328   \n",
       "evm.model.hcontig_000_013.4  7277          101          109         182   \n",
       "evm.model.hcontig_000_014.1  48440         477            1         477   \n",
       "evm.model.hcontig_000_015.3  86815         497          109         516   \n",
       "evm.model.hcontig_000_015.4  86816         263          617         879   \n",
       "evm.model.hcontig_000_016.2  92327          78           77         133   \n",
       "evm.model.hcontig_000_016.5  92334         654          120         773   \n",
       "evm.model.hcontig_000_016.6  92337          81            1          81   \n",
       "evm.model.hcontig_000_019.13 125852         78           77         133   \n",
       "evm.model.hcontig_000_019.41 126259         94            1          94   \n",
       "evm.model.hcontig_000_021.2  210812         93          351         443   \n",
       "evm.model.hcontig_000_023.2  98825         252          386         637   \n",
       "evm.model.hcontig_000_025.3  10767         149          121         269   \n",
       "evm.model.hcontig_000_028.1  129821        154            1         125   \n",
       "evm.model.hcontig_000_030.3  9490          154            1         154   \n",
       "evm.model.hcontig_000_030.4  9498          389            1         389   \n",
       "evm.model.hcontig_000_031.30 5461          401            1         401   \n",
       "evm.model.hcontig_000_033.1  107444        170            1         170   \n",
       "evm.model.hcontig_000_036.13 130099        135            1         135   \n",
       "evm.model.hcontig_000_037.8  125325        866            1         889   \n",
       "evm.model.hcontig_000_039.3  11187         158          329         471   \n",
       "evm.model.hcontig_000_039.4  11197         172            1         172   \n",
       "evm.model.hcontig_000_046.13 102438         81            1          17   \n",
       "evm.model.hcontig_000_046.14 102439        118          233         350   \n",
       "evm.model.hcontig_000_046.15 102450        108          394         501   \n",
       "evm.model.hcontig_000_046.19 102462         82          409         490   \n",
       "evm.model.hcontig_000_046.26 102506        224            1          48   \n",
       "evm.model.hcontig_000_046.3  102264        190            1         190   \n",
       "...                                        ...          ...         ...   \n",
       "evm.model.hcontig_076_003.2  10109          73          840         912   \n",
       "evm.model.hcontig_076_004.6  135085        122          239         360   \n",
       "evm.model.hcontig_078_001.1  141481       1169          507        1708   \n",
       "evm.model.hcontig_079_003.13 111991        438            1         442   \n",
       "evm.model.hcontig_079_003.14 111992        194          510         703   \n",
       "evm.model.hcontig_081_002.1  140222        132            1         132   \n",
       "evm.model.hcontig_081_004.4  185           407            1         410   \n",
       "evm.model.hcontig_081_005.1  5872          390            1         393   \n",
       "evm.model.hcontig_081_005.5  5887          179          686         864   \n",
       "evm.model.hcontig_081_005.6  5890          100          891         972   \n",
       "evm.model.hcontig_084_001.19 72572          64          174         232   \n",
       "evm.model.hcontig_084_001.21 72574          39          416         454   \n",
       "evm.model.hcontig_085_001.12 36892         160            1         160   \n",
       "evm.model.hcontig_085_001.3  36493         295          326         620   \n",
       "                             36494         295          353         647   \n",
       "                             36495         295          295         589   \n",
       "evm.model.hcontig_085_001.4  36552         160            1         160   \n",
       "evm.model.hcontig_085_001.5  36599         650            1         650   \n",
       "                             36600         651            1         651   \n",
       "evm.model.hcontig_085_001.6  36701         333          668        1000   \n",
       "evm.model.hcontig_085_003.1  6871          647            1         647   \n",
       "evm.model.hcontig_085_003.2  6937          650            1         650   \n",
       "                             6938          651            1         651   \n",
       "evm.model.hcontig_085_004.5  164308        114          108         210   \n",
       "evm.model.hcontig_091_003.1  130417        239           15         188   \n",
       "evm.model.hcontig_091_003.2  130422         68          502         568   \n",
       "evm.model.hcontig_093_002.4  183334        440            1         440   \n",
       "evm.model.hcontig_095_001.2  6863           88          141         228   \n",
       "evm.model.hcontig_095_002.1  35977          88          141         228   \n",
       "evm.model.hcontig_095_002.11 36000         119            1         119   \n",
       "\n",
       "                                           e-value  BitScore  \n",
       "Query                                                         \n",
       "evm.model.hcontig_000_001.8  158696  4.000000e-120     351.0  \n",
       "evm.model.hcontig_000_007.8  103152  1.000000e-172     509.0  \n",
       "evm.model.hcontig_000_013.2  7275     0.000000e+00     667.0  \n",
       "evm.model.hcontig_000_013.4  7277     5.000000e-49     155.0  \n",
       "evm.model.hcontig_000_014.1  48440    0.000000e+00     979.0  \n",
       "evm.model.hcontig_000_015.3  86815    0.000000e+00     836.0  \n",
       "evm.model.hcontig_000_015.4  86816    0.000000e+00     555.0  \n",
       "evm.model.hcontig_000_016.2  92327    1.000000e-31     112.0  \n",
       "evm.model.hcontig_000_016.5  92334    0.000000e+00    1332.0  \n",
       "evm.model.hcontig_000_016.6  92337    3.000000e-49     165.0  \n",
       "evm.model.hcontig_000_019.13 125852   1.000000e-31     112.0  \n",
       "evm.model.hcontig_000_019.41 126259   6.000000e-59     190.0  \n",
       "evm.model.hcontig_000_021.2  210812   1.000000e-57     189.0  \n",
       "evm.model.hcontig_000_023.2  98825    0.000000e+00     523.0  \n",
       "evm.model.hcontig_000_025.3  10767   7.000000e-107     308.0  \n",
       "evm.model.hcontig_000_028.1  129821   2.000000e-84     246.0  \n",
       "evm.model.hcontig_000_030.3  9490    2.000000e-108     308.0  \n",
       "evm.model.hcontig_000_030.4  9498     0.000000e+00     786.0  \n",
       "evm.model.hcontig_000_031.30 5461     0.000000e+00     833.0  \n",
       "evm.model.hcontig_000_033.1  107444  4.000000e-120     352.0  \n",
       "evm.model.hcontig_000_036.13 130099   6.000000e-80     251.0  \n",
       "evm.model.hcontig_000_037.8  125325   0.000000e+00    1793.0  \n",
       "evm.model.hcontig_000_039.3  11187    2.000000e-98     294.0  \n",
       "evm.model.hcontig_000_039.4  11197   1.000000e-104     311.0  \n",
       "evm.model.hcontig_000_046.13 102438   3.000000e-05      39.7  \n",
       "evm.model.hcontig_000_046.14 102439   2.000000e-73     229.0  \n",
       "evm.model.hcontig_000_046.15 102450   4.000000e-66     209.0  \n",
       "evm.model.hcontig_000_046.19 102462   3.000000e-47     158.0  \n",
       "evm.model.hcontig_000_046.26 102506   3.000000e-22      88.6  \n",
       "evm.model.hcontig_000_046.3  102264  3.000000e-128     361.0  \n",
       "...                                            ...       ...  \n",
       "evm.model.hcontig_076_003.2  10109    3.000000e-39     138.0  \n",
       "evm.model.hcontig_076_004.6  135085   7.000000e-84     251.0  \n",
       "evm.model.hcontig_078_001.1  141481   0.000000e+00    2381.0  \n",
       "evm.model.hcontig_079_003.13 111991   0.000000e+00     876.0  \n",
       "evm.model.hcontig_079_003.14 111992  2.000000e-136     406.0  \n",
       "evm.model.hcontig_081_002.1  140222   1.000000e-83     253.0  \n",
       "evm.model.hcontig_081_004.4  185      0.000000e+00     741.0  \n",
       "evm.model.hcontig_081_005.1  5872     0.000000e+00     705.0  \n",
       "evm.model.hcontig_081_005.5  5887    6.000000e-113     347.0  \n",
       "evm.model.hcontig_081_005.6  5890     2.000000e-46     160.0  \n",
       "evm.model.hcontig_084_001.19 72572    6.000000e-34     119.0  \n",
       "evm.model.hcontig_084_001.21 72574    2.000000e-21      85.1  \n",
       "evm.model.hcontig_085_001.12 36892   7.000000e-101     297.0  \n",
       "evm.model.hcontig_085_001.3  36493    0.000000e+00     614.0  \n",
       "                             36494    0.000000e+00     614.0  \n",
       "                             36495    0.000000e+00     613.0  \n",
       "evm.model.hcontig_085_001.4  36552   1.000000e-110     320.0  \n",
       "evm.model.hcontig_085_001.5  36599    0.000000e+00    1363.0  \n",
       "                             36600    0.000000e+00    1356.0  \n",
       "evm.model.hcontig_085_001.6  36701    0.000000e+00     678.0  \n",
       "evm.model.hcontig_085_003.1  6871     0.000000e+00    1343.0  \n",
       "evm.model.hcontig_085_003.2  6937     0.000000e+00    1366.0  \n",
       "                             6938     0.000000e+00    1360.0  \n",
       "evm.model.hcontig_085_004.5  164308   2.000000e-70     212.0  \n",
       "evm.model.hcontig_091_003.1  130417  6.000000e-120     347.0  \n",
       "evm.model.hcontig_091_003.2  130422   4.000000e-35     125.0  \n",
       "evm.model.hcontig_093_002.4  183334   0.000000e+00     907.0  \n",
       "evm.model.hcontig_095_001.2  6863     1.000000e-55     173.0  \n",
       "evm.model.hcontig_095_002.1  35977    1.000000e-55     173.0  \n",
       "evm.model.hcontig_095_002.11 36000    1.000000e-81     244.0  \n",
       "\n",
       "[941 rows x 12 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "htgNoAllelesBlastDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
