{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DK_0911_post_allele_analysis_v02"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Based on original code by Benjamin Schwessinger."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Inputs: output from `DK_0911_defining_alleles_v02` & primary+haplotig (ph) protein/gene/cds .*fasta* files from `DK_0911_generate_fasta_files_from_gff3`.\n",
    "- Programs: **MUSCLE**, **PAML**\n",
    "- Purpose: generate and save a DataFrame containing dN/dS information (number of nonsynonymous substitutions per non-synonymous site to the number of synonymous substitutions per synonymous site), as well as Hamming & Levenshtein distances (measures of % identity). Also provides visualisations of some of this data.\n",
    "\n",
    "#### Overview\n",
    "1. Reads in the large allele DataFrames generated in `DK_0911_defining_alleles_v02` (i.e. proteinortho hits OR best blast hit) - see description header cell in that notebook for more information on which alleles are included in that DataFrame.\n",
    "2. Filters the allele DataFrames based on %ID and %QCov (this can be set to filter only BLAST-identified alleles or both BLAST- and proteinortho-identified alleles) so that distance information is not calculated on an unnecessarily large number of alleles.\n",
    "3. Calculates distance & dN/dS information, and saves this to an output file so that it does not have to be re-calculated (if for whatever reason, the inputs change so that dN/dS or distance information should change, this output file (`DK_0911_v0x_analysed_alleles.df`) should be deleted so that it can be re-generated.\n",
    "4. Plots graphs of allele-type distribution (pie chart) and allele-type Levenshtein distances (measures of similarity) for different levels of allele-filtering (QCov/TCov/%ID/Levenshtein similarity).\n",
    "\n",
    "NB:\n",
    "- dN/dS information is currently not utilised in this script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:11.535137Z",
     "start_time": "2018-02-26T13:06:10.986516Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.371468Z",
     "start_time": "2018-02-26T13:06:11.545224Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "from Bio import SeqIO\n",
    "from Bio import AlignIO\n",
    "import distance\n",
    "import editdistance\n",
    "import math\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import re\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib as mpl\n",
    "import collections\n",
    "import pybedtools\n",
    "from sklearn.externals.joblib import Parallel, delayed\n",
    "import itertools as it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.438815Z",
     "start_time": "2018-02-26T13:06:13.374933Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#define variables here\n",
    "GENOME_VERSION = 'v04'\n",
    "\n",
    "BASE_PATH = '/home/benjamin/genome_assembly/Warrior/allele_analysis/%s/' % GENOME_VERSION\n",
    "\n",
    "YN00_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/post_analysis/yn00.ctl'\n",
    "BASE_OUT_PATH = os.path.join(BASE_PATH, 'post_analysis/')\n",
    "ALLELE_PATH = os.path.join(BASE_PATH, 'allele_analysis/alleles_proteinortho_graph516/')\n",
    "UNFILTERED_DF_PATH = os.path.join(BASE_PATH, \\\n",
    "    'allele_analysis/DK_0911_%s_p_ctg.DK_0911_%s_h_ctg.0.001.blastp.outfmt6.allele_analysis' \\\n",
    "                                  % (GENOME_VERSION, GENOME_VERSION))\n",
    "GENOME_PATH = '/home/benjamin/genome_assembly/Warrior/genome_%s/' % GENOME_VERSION\n",
    "FIGURE_PATH = os.path.join(BASE_OUT_PATH, 'figures')\n",
    "\n",
    "GENOME = 'DK_0911_%s' % GENOME_VERSION\n",
    "P_GENOME = GENOME + '_p_ctg'\n",
    "H_GENOME = GENOME + '_h_ctg'\n",
    "\n",
    "threads = 8\n",
    "\n",
    "# Base filtering so that distance calculations are not performed on all allele pairs.\n",
    "# Distance calculations will only be performed on allele pairs above the defined cutoffs.\n",
    "# Note that proteinortho alleles will not be affected (this can be changed in the filterAlleleDf function).\n",
    "BASE_QCOV_CUTOFF = 0\n",
    "BASE_TCOV_CUTOFF = 0\n",
    "BASE_PCTID_CUTOFF = 0\n",
    "\n",
    "P_PROTEINS_FASTA = os.path.join(GENOME_PATH, P_GENOME + '.protein.fa')\n",
    "\n",
    "PAML_PATH = os.path.join(BASE_OUT_PATH, 'paml')\n",
    "if not os.path.exists(BASE_OUT_PATH):\n",
    "    os.mkdir(BASE_OUT_PATH)\n",
    "if not os.path.exists(FIGURE_PATH):\n",
    "    os.mkdir(FIGURE_PATH)\n",
    "if not os.path.exists(PAML_PATH):\n",
    "    os.mkdir(PAML_PATH)\n",
    "    shutil.copy2(YN00_PATH, PAML_PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.458087Z",
     "start_time": "2018-02-26T13:06:13.449039Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#for homozygous region analysis\n",
    "COV_PATH = '/home/benjamin/genome_assembly/Warrior/COV'\n",
    "homo_bed_fh = os.path.join(COV_PATH, 'DK0911_v04_ph_ctg.ph_p_homo_cov.bed')\n",
    "anno_gff_p_fh = os.path.join(GENOME_PATH, 'DK_0911_v04_p_ctg.anno.gff3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.480608Z",
     "start_time": "2018-02-26T13:06:13.465530Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "PH_PROTEIN_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.protein.fa')\n",
    "PH_GENE_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.gene.fa')\n",
    "PH_CDS_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.cds.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.504764Z",
     "start_time": "2018-02-26T13:06:13.487664Z"
    }
   },
   "outputs": [],
   "source": [
    "#for finalizing the protein ortho single analysis\n",
    "proortho_fh = os.path.join(BASE_PATH, 'proteinortho', 'ph_ctg_516.proteinortho')\n",
    "proorthograph_fh = os.path.join(BASE_PATH, 'proteinortho', 'ph_ctg_516.proteinortho-graph')\n",
    "poff_graph_fh = os.path.join(BASE_PATH, 'proteinortho', 'ph_ctg_516.poff-graph')\n",
    "assert(os.path.exists(proortho_fh))\n",
    "assert(os.path.exists(proorthograph_fh))\n",
    "assert(os.path.exists(poff_graph_fh))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.556737Z",
     "start_time": "2018-02-26T13:06:13.524758Z"
    }
   },
   "outputs": [],
   "source": [
    "def getInterHaplotypeParaloges(proorthograph_fh,poff_graph_fh):\n",
    "    \"\"\"This function compares the proteinorth (po) and po synteny output. \n",
    "    It returns all Target+Query pairs that are only found in po as list. \n",
    "    It also checks if there are pairs only found in poff.\"\"\"\n",
    "    graph_header = ['Target', 'Query', 'evalue_ab', 'bitscore_ab', 'evalue_ba', 'bitscore_ba', 'same_strand' , 'simscore']\n",
    "    poff_df = pd.read_csv(poff_graph_fh, sep='\\t', header=None, names=graph_header, comment='#' )\n",
    "    po_df = pd.read_csv(proorthograph_fh, sep='\\t', header=None, names=graph_header, comment='#' )\n",
    "    poff_paring = (poff_df.Target+ '+' + poff_df.Query)\n",
    "    po_paring = (po_df.Target+ '+' + po_df.Query)\n",
    "    interhaplotypeparaloges_l = list(set(po_paring) - set(poff_paring))\n",
    "    poff_only_pairs = list(set(poff_paring) - set(po_paring))\n",
    "    if len(poff_only_pairs) > 0:\n",
    "        print(\"Check poff and proteinortho output as conceptually all poff pairs should be in included in the proteinortho output.\\nThese are the pairs:\")\n",
    "        print(poff_only_pairs)\n",
    "    return interhaplotypeparaloges_l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.579685Z",
     "start_time": "2018-02-26T13:06:13.560170Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assign_INP(aTargetplusaQuery, proorthograph_fh = proorthograph_fh,poff_graph_fh = poff_graph_fh ):\n",
    "    \"\"\"\n",
    "    Returns a series of of True/False values if a aTarget+aQuery pair are element identified as \n",
    "    Inter haplotype paraloge.\n",
    "    Input:\n",
    "        proorthograh_fh, proteinortho graph output file handle that has been run including the single flag\n",
    "        queries, series of protein queries used for the blast search.\n",
    "        poff_graph_fh, proteinortho graph output file handle that has been run including the synteny and\n",
    "        single flag\n",
    "        aTargetplusaQuery, series of protein aTargets + aQuiers used for the blast search.\n",
    "    Outupt:\n",
    "        Returns a True/False series with the aTargets + aQuiers as index.\n",
    "    \n",
    "    \"\"\"\n",
    "\n",
    "    interhaplotypeparaloges_l = getInterHaplotypeParaloges(proorthograph_fh,poff_graph_fh)\n",
    "    interhaplotypeparaloges_s = pd.Series([False]*len(aTargetplusaQuery), index=aTargetplusaQuery)\n",
    "    interhaplotypeparaloges_s.loc[interhaplotypeparaloges_s.index.isin(interhaplotypeparaloges_l)] = True\n",
    "    #print(len(Unphasedgenes))\n",
    "    return interhaplotypeparaloges_s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.600770Z",
     "start_time": "2018-02-26T13:06:13.582132Z"
    }
   },
   "outputs": [],
   "source": [
    "def getProOrthsingles(proortho_fh):\n",
    "    \"\"\"Parses out all proteinortho singles from a proteinortho output file.\n",
    "    Returns a list of these singles.\"\"\"\n",
    "    with open(proortho_fh) as file:\n",
    "        id_pattern = r'(evm\\S*)'\n",
    "        regex = re.compile(id_pattern)\n",
    "        po_single_list = []\n",
    "        for line in file:\n",
    "            if '*' in line:\n",
    "                po_single_list.append(regex.search(line).groups()[0])\n",
    "            else:\n",
    "                continue\n",
    "    return po_single_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.624444Z",
     "start_time": "2018-02-26T13:06:13.607693Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assign_PO_single(queries, proortho_fh=proortho_fh):\n",
    "    \"\"\"\n",
    "    Returns a series of of True/False values if a query element identified as single.\n",
    "    Input:\n",
    "        proortho_fh, proteinortho output file handle that has been run including the single flag\n",
    "        queries, series of protein queries used for the blast search.\n",
    "    Outupt:\n",
    "        Returns a True/False series with the queries as index.\n",
    "    \n",
    "    \"\"\"\n",
    "\n",
    "    PO_singles = getProOrthsingles(proortho_fh)\n",
    "    PO_singles_s = pd.Series([False]*len(queries), index=queries)\n",
    "    PO_singles_s.loc[PO_singles_s.index.isin(PO_singles)] = True\n",
    "    #print(len(Unphasedgenes))\n",
    "    return PO_singles_s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.649232Z",
     "start_time": "2018-02-26T13:06:13.637651Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def allgffTogenegff(gff_fh, write_out=True):\n",
    "    '''Converts at complete gff to a gene gff only and writtes it out.'''\n",
    "    gene_gff = pd.read_csv(gff_fh, sep='\\t', header=None)\n",
    "    gene_gff = gene_gff[gene_gff[2] == 'gene']\n",
    "    gene_gff.reset_index(drop=True, inplace=True)\n",
    "    gene_gff.to_csv(gff_fh.replace('anno', 'gene'), sep='\\t', header=False, index=False)\n",
    "    return gene_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.671504Z",
     "start_time": "2018-02-26T13:06:13.651889Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def col_8_id(x):\n",
    "    '''Function that pulls out the ID from the 9th column of a df.'''\n",
    "    pattern = r'ID=([a-zA-Z0-9_.]*);'\n",
    "    regex = re.compile(pattern)  \n",
    "    m = regex.search(x)\n",
    "    match = m.groups()[0].replace('TU', 'model')\n",
    "    if match.startswith('cds.'):\n",
    "        match = match[4:]\n",
    "    if 'exon' in match:\n",
    "        _list = match.split('.')\n",
    "        match = '.'.join(_list[:-1])\n",
    "    return match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T03:29:57.693406Z",
     "start_time": "2018-02-27T03:29:57.619359Z"
    }
   },
   "outputs": [],
   "source": [
    "def assignMatchType(allele_source, overlap, no_overlap):\n",
    "    if allele_source == np.nan:\n",
    "        return allele_source\n",
    "    \n",
    "    s = allele_source + '_'\n",
    "    \n",
    "    if overlap ==  True:\n",
    "        s += 'overlap'\n",
    "    elif no_overlap == True:\n",
    "        s += 'no_overlap'\n",
    "    else: # different_pcontig\n",
    "        s += 'unlinked'\n",
    "    return s\n",
    "\n",
    "def reduceGroups(g):\n",
    "    '''returns the best hit based on e-value and BitScore per group'''\n",
    "    if len(g) == 1:\n",
    "        return g\n",
    "    tmp_g = g[g['e-value'] == g['e-value'].min()]\n",
    "    if len(tmp_g) == 1:\n",
    "        return tmp_g\n",
    "    return tmp_g[tmp_g['BitScore'] == tmp_g['BitScore'].max()]\n",
    "\n",
    "def filterAlleleDf(alleleDf, qCov, tCov, pctId, levSim, leavePO=False):\n",
    "    if leavePO:\n",
    "        no_PO_df = alleleDf[(alleleDf['allele_source'] == 'h_rBLAST') | (alleleDf['allele_source'] == 'BLAST')]\n",
    "        PO_df = alleleDf[alleleDf['allele_source'] == 'PO']\n",
    "\n",
    "        filtered_no_PO_df = filterAlleleDf(no_PO_df, qCov, tCov, pctId, levSim)\n",
    "        return filtered_no_PO_df.append(PO_df, ignore_index=True)\n",
    "    \n",
    "    if qCov:\n",
    "        alleleDf = alleleDf[alleleDf['QCov'] > qCov]\n",
    "    if tCov:\n",
    "        alleleDf = alleleDf[alleleDf['TCov'] > tCov]\n",
    "    if pctId:\n",
    "        alleleDf = alleleDf[alleleDf['PctID'] > pctId]\n",
    "    if levSim:\n",
    "        levDist = (100-levSim)/100.0\n",
    "        alleleDf = alleleDf[alleleDf['protein_levenshtein'] < levDist]\n",
    "\n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.793099Z",
     "start_time": "2018-02-26T13:06:13.780669Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def geneUnphased(Gene_gff_fh, Homo_cov_bed_fh ):\n",
    "    \"\"\"\n",
    "    Returns a list of all genes that are unphased.\n",
    "    \n",
    "    Input: * Fh for annotation gff file\n",
    "           * Fh for Homo_cov_bed_fh\n",
    "    Output: A set of gene IDs that are unphases\n",
    "    \"\"\"\n",
    "    geneGff_bed = pybedtools.BedTool(Gene_gff_fh)\n",
    "    homo_p_bed = pybedtools.BedTool(Homo_cov_bed_fh)\n",
    "    gene_ids_ph_p_homo = []\n",
    "    for x in geneGff_bed.intersect(homo_p_bed, f=0.4):\n",
    "        y = col_8_id(x[8])\n",
    "        gene_ids_ph_p_homo.append(y)\n",
    "    gene_ids_ph_p_homo = set(gene_ids_ph_p_homo)\n",
    "    return gene_ids_ph_p_homo\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.811802Z",
     "start_time": "2018-02-26T13:06:13.795533Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assign_unphased(queries, gff_fh=anno_gff_p_fh, homo_bed_fh=homo_bed_fh):\n",
    "    \"\"\"\n",
    "    Takes a series of gene models and checks if they overlap with the\n",
    "    homo_bed dataframe. The minimum overlap is f=0.4 with the intersect function.\n",
    "    \n",
    "    TO DO: For now the homo_bed only includes primary gene models. Should be changed.\n",
    "    Input:  \n",
    "        queries, pd.Series of gene names. \n",
    "        gff_fh, file handle of gff file.\n",
    "        homo_bed_dfh, file handle of bed file that represents non-phased regions.\n",
    "    \n",
    "    \"\"\"\n",
    "    if '.gene' not in gff_fh: \n",
    "        _ = allgffTogenegff(gff_fh)\n",
    "    Unphasedgenes = geneUnphased(gff_fh.replace('anno', 'gene'), homo_bed_fh)\n",
    "    unphased_s = pd.Series([False]*len(queries), index=queries)\n",
    "    unphased_s.loc[unphased_s.index.isin(Unphasedgenes)] = True\n",
    "    #print(len(Unphasedgenes))\n",
    "    return unphased_s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.821390Z",
     "start_time": "2018-02-26T13:06:13.814399Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def getFastaDict(fastaFile):\n",
    "    d = {}\n",
    "    for gene in SeqIO.parse(fastaFile, 'fasta'):\n",
    "        d[gene.id] = gene\n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.879906Z",
     "start_time": "2018-02-26T13:06:13.823766Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def writeAllelicFasta(alleleOne, alleleTwo, alleleType, outPath):\n",
    "    '''writes fasta file containing fasta information for two alleles\n",
    "    in the outPath'''\n",
    "    assert(alleleType.upper() in ['CDS', 'GENE', 'PROTEIN'])\n",
    "    \n",
    "    seqRecordDict = globals()['SEQRECORD_' + alleleType.upper() + '_DICT']\n",
    "    try:\n",
    "        alleleSeqRecords = [seqRecordDict[alleleOne], seqRecordDict[alleleTwo]]\n",
    "    except KeyError:\n",
    "        print(alleleOne)\n",
    "        print(alleleTwo)\n",
    "        print(alleleType)\n",
    "        sys.exit()\n",
    "    with open(os.path.join(outPath, alleleType.lower() + '.fa'), 'w') as outFile:\n",
    "        SeqIO.write(alleleSeqRecords, outFile, 'fasta')\n",
    "    return True\n",
    "\n",
    "def writeAlignmentScript(alleleOutPath, scriptLoc = os.path.join(PAML_PATH, 'paml_script.sh')):\n",
    "    with open(scriptLoc, 'a') as outFile:\n",
    "        print('cd %s' % alleleOutPath, file=outFile)\n",
    "        print('/home/gamran/anaconda3/muscle3.8.31_i86linux64 -clwstrict -in protein.fa -out protein.aln', file=outFile)\n",
    "        print('perl /home/gamran/anaconda3/pal2nal.v14/pal2nal.pl -output paml protein.aln cds.fa > cds_codon.aln', file=outFile)\n",
    "        print('perl /home/gamran/anaconda3/pal2nal.v14/pal2nal.pl protein.aln cds.fa > cds_codon.clustal', file=outFile)\n",
    "        print('cp %s/yn00.ctl ./' % PAML_PATH, file=outFile)\n",
    "        print('/home/gamran/anaconda3/paml4.9g/bin/yn00', file=outFile)\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.911216Z",
     "start_time": "2018-02-26T13:06:13.882673Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def prepareAlignmentBashScript(scriptLoc = os.path.join(PAML_PATH, 'paml_script.sh')):\n",
    "    with open(scriptLoc, 'w') as pamlScript:\n",
    "        print('#!/bin/bash', file=pamlScript)\n",
    "\n",
    "    for index, [Query, Target] in alleleDf.iloc[:, :2].iterrows():\n",
    "        #if we don't have a blast hit skip.\n",
    "        if pd.isnull(Target):\n",
    "            continue\n",
    "        else:\n",
    "            alleleOutPath = os.path.join(PAML_PATH, '%s_%s' % (Query, Target))\n",
    "            if not os.path.exists(alleleOutPath):\n",
    "                os.mkdir(os.path.join(PAML_PATH, '%s_%s' % (Query, Target)))\n",
    "\n",
    "            writeAllelicFasta(Query, Target, 'CDS', alleleOutPath)\n",
    "            writeAllelicFasta(Query, Target, 'PROTEIN', alleleOutPath)\n",
    "\n",
    "            writeAlignmentScript(alleleOutPath, os.path.join(PAML_PATH, 'paml_script.sh'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.063321Z",
     "start_time": "2018-02-26T13:06:13.914458Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assignDistancesToAlleles(folder, alignmentFile, alleleType):\n",
    "    '''Adds Hamming and Levenshtein distance columns to an allele pair\n",
    "    (indexed by 'folder' name) in df'''\n",
    "    #print(folder)\n",
    "    if pd.isnull(folder):\n",
    "        return np.nan, np.nan\n",
    "    assert(alleleType.upper() in ['PROTEIN', 'CDS', 'GENE'])\n",
    "    seq1, seq2 = AlignIO.read(open(alignmentFile, 'r'), format='clustal', seq_count=2)\n",
    "    seq1 = str(seq1.seq).upper()\n",
    "    seq2 = str(seq2.seq).upper()\n",
    "    assert(len(seq1) == len(seq2))\n",
    "    return editdistance.eval(seq1, seq2)/len(seq1), distance.hamming(seq1, seq2, normalized=True)\n",
    "\n",
    "def assignDistancesToAllAlleles(df_folder_index, all_folders, tmp_path, suffix):\n",
    "    \"\"\"\n",
    "    Reads in the index that contains the folder pairings for the alignements.\n",
    "    Returns a protein_df and CDS_df that contain the hamming and levenshtein distance each.\n",
    "    \"\"\"\n",
    "    cleaned_index = [x for x in df_folder_index if x in all_folders]\n",
    "    count = 0\n",
    "    total = len(df_folder_index)\n",
    "    percentDone = 0\n",
    "    protein_lev_dict = {}\n",
    "    protein_ham_dict = {}\n",
    "    CDS_lev_dict = {}\n",
    "    CDS_ham_dict = {}\n",
    "    \n",
    "    #print(\"Calculating distances and adding them to the allele DataFrame...\")\n",
    "    \n",
    "    for folder in cleaned_index:\n",
    "        if pd.isnull(folder):\n",
    "            proteinAlignmentFile = ''\n",
    "            cdsAlignmentFile = ''\n",
    "        else:\n",
    "            proteinAlignmentFile = os.path.join(PAML_PATH, folder, 'protein.aln')\n",
    "            cdsAlignmentFile = os.path.join(PAML_PATH, folder, 'cds_codon.clustal')\n",
    "        #here the nan get overwritten. This doesn't matter though as they are all\n",
    "        #nan anyway.\n",
    "        protein_lev_dict[folder], protein_ham_dict[folder]  = \\\n",
    "        assignDistancesToAlleles(folder, proteinAlignmentFile, 'PROTEIN')\n",
    "        CDS_lev_dict[folder], CDS_ham_dict[folder]  = \\\n",
    "        assignDistancesToAlleles(folder, cdsAlignmentFile, 'CDS')\n",
    "\n",
    "        count += 1\n",
    "        #if round(count/total * 100) > percentDone:\n",
    "            #percentDone = round(count/total * 100)\n",
    "            #print(\"%s%% complete\" % percentDone)\n",
    "            \n",
    "    newdf_columns=['protein_hamming', 'protein_levenshtein', 'cds_hamming',\n",
    "       'cds_levenshtein']\n",
    "    if len(protein_ham_dict) > 0:\n",
    "        df = pd.DataFrame([protein_ham_dict,protein_lev_dict,CDS_ham_dict,CDS_lev_dict]).T\n",
    "        df.rename(columns=dict(zip(df.columns,newdf_columns)),inplace=True)\n",
    "        out_name = os.path.join(tmp_path, '%s_%s.%s' % (df.index[0],df.index[-1],suffix))\n",
    "        df.round(4).to_csv(out_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.255606Z",
     "start_time": "2018-02-26T13:06:14.071708Z"
    }
   },
   "outputs": [],
   "source": [
    "def parse_dNdS_to_df(line, folder):\n",
    "    \"\"\"\n",
    "    Function that parses out dN and dS of a yn00 file and calls the \n",
    "    assign_dNdS function. Therefore returns a single element\n",
    "    pd.Series with folder name as index.\n",
    "    \"\"\"\n",
    "    dN = re.findall(r'dN = [-| ]?(.*) w', line)[0]\n",
    "    dS = re.findall(r'dS = [-| ]?(.*) dN', line)[0]\n",
    "    return assign_dNdS(dN, dS, folder)\n",
    "\n",
    "def assign_dNdS(dN, dS, folder):\n",
    "    '''\n",
    "    Function that cacluates the dN/dS ratio an returns it as a series usind the folder as index.\n",
    "    Input: dN, dS, folder(name)\n",
    "    Output: single element pd.Series with folder name as index.\n",
    "    '''\n",
    "    if float(dS) > 0:\n",
    "        series = pd.Series([float(dN)/float(dS)], index=[folder])\n",
    "    else:\n",
    "        series = pd.Series([np.nan], index=[folder])\n",
    "    return series\n",
    "\n",
    "def assign_dNdS_to_all_alleles(folder_index, all_folders, tmp_path, suffix):\n",
    "    \"\"\"Function that parses out the different dN/dS ratios from a yn.out file for a list/index\n",
    "    of folders that contain these yn.out files. The output dataframe is saved to tmp folder using\n",
    "    the suffix.\n",
    "    Input:  folder_index, list or index where to find the yn.out for each pairing.\n",
    "            all_folders, are all possible folders. This is used to filter out nan and so from\n",
    "                paralizing etc.\n",
    "            tmp_path, is the path were \n",
    "    Output: Saved out tmp df with the suffix as file ending.\n",
    "    \n",
    "    \"\"\"\n",
    "    cleaned_index = [x for x in folder_index if x in all_folders]\n",
    "    #print(cleaned_index)\n",
    "    yn00_s = pd.Series([], name='yn00_dN/dS')\n",
    "    LWL85_s = pd.Series([], name='LWL85_dN/dS')\n",
    "    LWL85m_s = pd.Series([], name='LWL85m_dN/dS')\n",
    "    LPB93_s = pd.Series([], name='LPB93_dN/dS')\n",
    "    #header = ['folder','yn00_dN/dS', 'LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS']\n",
    "    #append these list\n",
    "    for folder in cleaned_index:\n",
    "        alleleYn = os.path.join(PAML_PATH, folder,'yn.out')\n",
    "        with open(alleleYn, 'r') as ynOut:\n",
    "            #now loop over the lines and parse out stuff\n",
    "            for i, line in enumerate(ynOut):\n",
    "                if line.startswith('seq. seq. ') and i > 0:\n",
    "                    next(ynOut) # we want the line that is two after the line starting with 'seq. seq '\n",
    "                    dataLine = next(ynOut)\n",
    "                    dN = dataLine.split('+-')[0].rstrip().split(' ')[-1]\n",
    "                    dS = dataLine.split('+-')[1].rstrip().split(' ')[-1]\n",
    "                    yn00_s = yn00_s.append(assign_dNdS(dN, dS, folder))\n",
    "                elif line.startswith('LWL85:') and 'nan' not in line:\n",
    "                    LWL85_s = LWL85_s.append(parse_dNdS_to_df(line, folder))\n",
    "                elif line.startswith('LWL85m:') and 'nan' not in line:\n",
    "                    LWL85m_s= LWL85m_s.append(parse_dNdS_to_df(line, folder))\n",
    "                elif line.startswith('LPB93:') and 'nan' not in line:\n",
    "                    LPB93_s =LPB93_s.append(parse_dNdS_to_df(line, folder))\n",
    "                else:\n",
    "                    continue\n",
    "    out_df = pd.concat([yn00_s.round(4),LWL85_s.round(4),\\\n",
    "                        LWL85m_s.round(4), LPB93_s.round(4)],axis =1)\n",
    "    new_columns = ['yn00_dN/dS','LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS' ]\n",
    "    if len(out_df) > 0:\n",
    "        out_df.rename(columns=dict(zip(out_df.columns, new_columns)), inplace=True)\n",
    "        out_name = os.path.join(tmp_path, '%s_%s.%s' % (out_df.index[0],out_df.index[-1],suffix))\n",
    "        out_df.to_csv(out_name, sep='\\t')\n",
    "    #return out_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.287985Z",
     "start_time": "2018-02-26T13:06:14.260411Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def checkPamlFilesExist(alleleDf):\n",
    "    '''loops through all folder names in alleleDf.index to check if their PAML files have\n",
    "    all been generated in those folders. refDict is based on the contents of a folder\n",
    "    that was known to be run successfully.'''\n",
    "    refDict = {'aln': 2,\n",
    "     'clustal': 1,\n",
    "     'ctl': 1,\n",
    "     'dN': 1,\n",
    "     'dS': 1,\n",
    "     'fa': 2,\n",
    "     'out': 1,\n",
    "     'rst': 1,\n",
    "     'rst1': 1,\n",
    "     'rub': 1,\n",
    "     't': 1}\n",
    "    for file in (x for x in alleleDf.index if not pd.isnull(x)):\n",
    "        if not os.path.exists(os.path.join(PAML_PATH, file)):\n",
    "            return False\n",
    "        discrepancies = getDiscrepancies(os.path.join(PAML_PATH, file), refDict)\n",
    "        if discrepancies != '':\n",
    "            print(discrepancies)\n",
    "            return False\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.314801Z",
     "start_time": "2018-02-26T13:06:14.290615Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def grouper(iterable, n, fillvalue=None):\n",
    "    \"Collect data into fixed-length chunks or blocks\"\n",
    "    # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx\"\n",
    "    args = [iter(iterable)] * n\n",
    "    return it.zip_longest(*args, fillvalue=fillvalue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:27:17.161541Z",
     "start_time": "2018-02-26T13:27:17.132641Z"
    }
   },
   "outputs": [],
   "source": [
    "def combineTmpToDf(header, suffix, tmp_path, clean=True):\n",
    "    \"\"\"Combines the files of a temporary folder into a dataframe based on the tmp files\n",
    "    suffix. Returns combined dataframe. And cleans up if needed.\"\"\"\n",
    "    tmp_assigneddfs_fh = [os.path.join(tmp_path, file) for file in os.listdir(tmp_path)\\\n",
    "                         if file.endswith(suffix) ]\n",
    "    print(len(tmp_assigneddfs_fh))\n",
    "    tmp_df = pd.DataFrame(columns=header)\n",
    "    for df_fh in tmp_assigneddfs_fh:\n",
    "        tmp_df = pd.concat([tmp_df, pd.read_csv(df_fh, index_col = 0, sep='\\t')])\n",
    "    if clean == True:\n",
    "        #now clean up again\n",
    "        for file in tmp_assigneddfs_fh:\n",
    "            os.remove(file)\n",
    "    return tmp_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T03:30:05.961270Z",
     "start_time": "2018-02-27T03:30:04.011284Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "(\"unsupported operand type(s) for +: 'float' and 'str'\", 'occurred at index 10793')",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-207-7ad7bce8d1ca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mhFullAlleleDf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mALLELE_PATH\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'%s.full_df.alleles'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mH_GENOME\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'matchType'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnan\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'matchType'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0massignMatchType\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele_source'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m't_contig == h_contig_overlap'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'q_contig == t_contig'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0mpFullAlleleDf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mALLELE_PATH\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'%s.full_df.alleles'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mP_GENOME\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mpFullAlleleDf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'matchType'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnan\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.5/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, axis, broadcast, raw, reduce, args, **kwds)\u001b[0m\n\u001b[1;32m   4260\u001b[0m                         \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4261\u001b[0m                         \u001b[0mreduce\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mreduce\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4262\u001b[0;31m                         ignore_failures=ignore_failures)\n\u001b[0m\u001b[1;32m   4263\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4264\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_apply_broadcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.5/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_apply_standard\u001b[0;34m(self, func, axis, ignore_failures, reduce)\u001b[0m\n\u001b[1;32m   4356\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4357\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mv\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries_gen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4358\u001b[0;31m                     \u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4359\u001b[0m                     \u001b[0mkeys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4360\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-207-7ad7bce8d1ca>\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(row)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mhFullAlleleDf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mALLELE_PATH\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'%s.full_df.alleles'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mH_GENOME\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'matchType'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnan\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'matchType'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0massignMatchType\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele_source'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m't_contig == h_contig_overlap'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'q_contig == t_contig'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0mpFullAlleleDf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mALLELE_PATH\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'%s.full_df.alleles'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mP_GENOME\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mpFullAlleleDf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'matchType'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnan\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpFullAlleleDf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-206-a0b0286d8d8b>\u001b[0m in \u001b[0;36massignMatchType\u001b[0;34m(allele_source, overlap, no_overlap)\u001b[0m\n\u001b[1;32m      3\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mallele_source\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mallele_source\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'_'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0moverlap\u001b[0m \u001b[0;34m==\u001b[0m  \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: (\"unsupported operand type(s) for +: 'float' and 'str'\", 'occurred at index 10793')"
     ]
    }
   ],
   "source": [
    "hFullAlleleDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.full_df.alleles' % H_GENOME), header=0, sep='\\t')\n",
    "hFullAlleleDf['matchType'] = pd.Series([np.nan]*len(hFullAlleleDf.index), index=hFullAlleleDf.index)\n",
    "hFullAlleleDf['matchType'] = hFullAlleleDf.apply(lambda row: assignMatchType(row['allele_source'], row['t_contig == h_contig_overlap'], row['q_contig == t_contig']), axis=1)\n",
    "pFullAlleleDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.full_df.alleles' % P_GENOME), header=0, sep='\\t')\n",
    "pFullAlleleDf['matchType'] = pd.Series([np.nan]*len(pFullAlleleDf.index), index=pFullAlleleDf.index)\n",
    "pFullAlleleDf['matchType'] = pFullAlleleDf.apply(lambda row: assignMatchType(row['allele_source'], row['t_contig == h_contig_overlap'], row['q_contig == t_contig']), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T04:25:12.848461Z",
     "start_time": "2018-02-27T04:25:12.751462Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>TLgth</th>\n",
       "      <th>TCov</th>\n",
       "      <th>q_contig</th>\n",
       "      <th>t_contig</th>\n",
       "      <th>q_contig == t_contig</th>\n",
       "      <th>p_protein</th>\n",
       "      <th>h_contig_overlap</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "      <th>allele_source</th>\n",
       "      <th>matchType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>25.00</td>\n",
       "      <td>240.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>330.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>...</td>\n",
       "      <td>768.0</td>\n",
       "      <td>31.250000</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_072_002</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>['hcontig_000_058']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>70.27</td>\n",
       "      <td>37.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>...</td>\n",
       "      <td>90.0</td>\n",
       "      <td>41.111111</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_029_007</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>96.88</td>\n",
       "      <td>128.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>210.0</td>\n",
       "      <td>60.952381</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>96.95</td>\n",
       "      <td>164.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>...</td>\n",
       "      <td>508.0</td>\n",
       "      <td>32.283465</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Query                        Target  PctID  AlnLgth  \\\n",
       "0    evm.model.pcontig_000.10                           NaN    NaN      NaN   \n",
       "1   evm.model.pcontig_000.100   evm.model.hcontig_072_002.3  25.00    240.0   \n",
       "2  evm.model.pcontig_000.1001   evm.model.hcontig_029_007.1  70.27     37.0   \n",
       "3  evm.model.pcontig_000.1002  evm.model.hcontig_003_006.10  96.88    128.0   \n",
       "4  evm.model.pcontig_000.1003   evm.model.hcontig_003_006.9  96.95    164.0   \n",
       "\n",
       "   NumMis  NumGap  StartQuery  StopQuery  StartTarget  StopTarget    ...     \\\n",
       "0     NaN     NaN         NaN        NaN          NaN         NaN    ...      \n",
       "1   155.0     8.0       110.0      330.0        125.0       358.0    ...      \n",
       "2    11.0     0.0       130.0      166.0         52.0        88.0    ...      \n",
       "3     4.0     0.0         1.0      128.0         83.0       210.0    ...      \n",
       "4     5.0     0.0         1.0      164.0        345.0       508.0    ...      \n",
       "\n",
       "   TLgth       TCov     q_contig         t_contig  q_contig == t_contig  \\\n",
       "0    NaN        NaN  pcontig_000            False                 False   \n",
       "1  768.0  31.250000  pcontig_000  hcontig_072_002                 False   \n",
       "2   90.0  41.111111  pcontig_000  hcontig_029_007                 False   \n",
       "3  210.0  60.952381  pcontig_000  hcontig_003_006                 False   \n",
       "4  508.0  32.283465  pcontig_000  hcontig_003_006                 False   \n",
       "\n",
       "                    p_protein     h_contig_overlap  \\\n",
       "0                         NaN                False   \n",
       "1   evm.model.pcontig_000.100  ['hcontig_000_058']   \n",
       "2  evm.model.pcontig_000.1001  ['hcontig_000_078']   \n",
       "3  evm.model.pcontig_000.1002  ['hcontig_000_078']   \n",
       "4  evm.model.pcontig_000.1003  ['hcontig_000_078']   \n",
       "\n",
       "  t_contig == h_contig_overlap  allele_source matchType  \n",
       "0                        False          BLAST     BLAST  \n",
       "1                        False          BLAST     BLAST  \n",
       "2                        False          BLAST     BLAST  \n",
       "3                        False          BLAST     BLAST  \n",
       "4                        False          BLAST     BLAST  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pFullAlleleDf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:53.774191Z",
     "start_time": "2018-02-26T13:25:53.326398Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# filter out haplotig proteins that already have alleles identified by BLAST or proteinortho.\n",
    "hFullAlleleDf = hFullAlleleDf[(~hFullAlleleDf['Query'].isin(pFullAlleleDf['Target']))]\n",
    "pFullAlleleDf['aQuery'] = pFullAlleleDf['Query']\n",
    "pFullAlleleDf['aTarget'] = pFullAlleleDf['Target']\n",
    "hFullAlleleDf['aQuery'] = hFullAlleleDf['Target']\n",
    "hFullAlleleDf['aTarget'] = hFullAlleleDf['Query']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:53.799101Z",
     "start_time": "2018-02-26T13:25:53.776713Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "phFullAlleleDf = pFullAlleleDf.append(hFullAlleleDf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:57.635871Z",
     "start_time": "2018-02-26T13:25:54.686464Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "SEQRECORD_PROTEIN_DICT = getFastaDict(PH_PROTEIN_FASTA)\n",
    "SEQRECORD_GENE_DICT = getFastaDict(PH_GENE_FASTA)\n",
    "SEQRECORD_CDS_DICT = getFastaDict(PH_CDS_FASTA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:46:54.038594Z",
     "start_time": "2018-02-26T13:46:54.008517Z"
    }
   },
   "outputs": [],
   "source": [
    "alleleDf = phFullAlleleDf.copy()\n",
    "alleleDf['folder'] = alleleDf.Query + '_' + alleleDf.Target\n",
    "alleleDf.set_index('folder', inplace=True)\n",
    "# assert(len(alleleDf) == len(overlapDf) + len(noOverlapDf) + len(diffContigDf) + len(manualAssignDf))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:35:19.297460Z",
     "start_time": "2018-02-26T13:35:19.242464Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>q_contig</th>\n",
       "      <th>t_contig</th>\n",
       "      <th>q_contig == t_contig</th>\n",
       "      <th>p_protein</th>\n",
       "      <th>h_contig_overlap</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "      <th>allele_source</th>\n",
       "      <th>matchType</th>\n",
       "      <th>aQuery</th>\n",
       "      <th>aTarget</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>folder</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.100_evm.model.hcontig_072_002.3</th>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>25.00</td>\n",
       "      <td>240.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>330.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_072_002</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>['hcontig_000_058']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.1001_evm.model.hcontig_029_007.1</th>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>70.27</td>\n",
       "      <td>37.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_029_007</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.1002_evm.model.hcontig_003_006.10</th>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>96.88</td>\n",
       "      <td>128.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.1003_evm.model.hcontig_003_006.9</th>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>96.95</td>\n",
       "      <td>164.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         Query  \\\n",
       "folder                                                                           \n",
       "NaN                                                   evm.model.pcontig_000.10   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.pcontig_000.100   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  evm.model.pcontig_000.1001   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.pcontig_000.1002   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  evm.model.pcontig_000.1003   \n",
       "\n",
       "                                                                          Target  \\\n",
       "folder                                                                             \n",
       "NaN                                                                          NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.hcontig_072_002.3   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...   evm.model.hcontig_029_007.1   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.hcontig_003_006.10   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...   evm.model.hcontig_003_006.9   \n",
       "\n",
       "                                                    PctID  AlnLgth  NumMis  \\\n",
       "folder                                                                       \n",
       "NaN                                                   NaN      NaN     NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  25.00    240.0   155.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  70.27     37.0    11.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  96.88    128.0     4.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  96.95    164.0     5.0   \n",
       "\n",
       "                                                    NumGap  StartQuery  \\\n",
       "folder                                                                   \n",
       "NaN                                                    NaN         NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...     8.0       110.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...     0.0       130.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...     0.0         1.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...     0.0         1.0   \n",
       "\n",
       "                                                    StopQuery  StartTarget  \\\n",
       "folder                                                                       \n",
       "NaN                                                       NaN          NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...      330.0        125.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...      166.0         52.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...      128.0         83.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...      164.0        345.0   \n",
       "\n",
       "                                                    StopTarget  \\\n",
       "folder                                                           \n",
       "NaN                                                        NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...       358.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...        88.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...       210.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...       508.0   \n",
       "\n",
       "                                                                ...               \\\n",
       "folder                                                          ...                \n",
       "NaN                                                             ...                \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...              ...                \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...              ...                \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...              ...                \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...              ...                \n",
       "\n",
       "                                                       q_contig  \\\n",
       "folder                                                            \n",
       "NaN                                                 pcontig_000   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  pcontig_000   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  pcontig_000   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  pcontig_000   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  pcontig_000   \n",
       "\n",
       "                                                           t_contig  \\\n",
       "folder                                                                \n",
       "NaN                                                           False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  hcontig_072_002   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  hcontig_029_007   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  hcontig_003_006   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  hcontig_003_006   \n",
       "\n",
       "                                                    q_contig == t_contig  \\\n",
       "folder                                                                     \n",
       "NaN                                                                False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...                 False   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...                 False   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...                 False   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...                 False   \n",
       "\n",
       "                                                                     p_protein  \\\n",
       "folder                                                                           \n",
       "NaN                                                                        NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.pcontig_000.100   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  evm.model.pcontig_000.1001   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.pcontig_000.1002   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  evm.model.pcontig_000.1003   \n",
       "\n",
       "                                                       h_contig_overlap  \\\n",
       "folder                                                                    \n",
       "NaN                                                               False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  ['hcontig_000_058']   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  ['hcontig_000_078']   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  ['hcontig_000_078']   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  ['hcontig_000_078']   \n",
       "\n",
       "                                                    t_contig == h_contig_overlap  \\\n",
       "folder                                                                             \n",
       "NaN                                                                        False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...                         False   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...                         False   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...                         False   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...                         False   \n",
       "\n",
       "                                                   allele_source matchType  \\\n",
       "folder                                                                       \n",
       "NaN                                                        BLAST     BLAST   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...         BLAST     BLAST   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...         BLAST     BLAST   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...         BLAST     BLAST   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...         BLAST     BLAST   \n",
       "\n",
       "                                                                        aQuery  \\\n",
       "folder                                                                           \n",
       "NaN                                                   evm.model.pcontig_000.10   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.pcontig_000.100   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  evm.model.pcontig_000.1001   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.pcontig_000.1002   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  evm.model.pcontig_000.1003   \n",
       "\n",
       "                                                                         aTarget  \n",
       "folder                                                                            \n",
       "NaN                                                                          NaN  \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.hcontig_072_002.3  \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...   evm.model.hcontig_029_007.1  \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.hcontig_003_006.10  \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...   evm.model.hcontig_003_006.9  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:36:11.484438Z",
     "start_time": "2018-02-26T13:36:11.367440Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "os.chdir('/home/gamran/genome_analysis/Warrior/Richard/scripts')\n",
    "%run file_counting.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T14:15:06.312304Z",
     "start_time": "2018-02-26T14:15:06.016952Z"
    }
   },
   "outputs": [],
   "source": [
    "def main(alleleDf = alleleDf):\n",
    "    prepareAlignmentBashScript(os.path.join(PAML_PATH, 'paml_script.sh'))\n",
    "    \n",
    "    # if already run before, comment out this line\n",
    "    print(\"Checking whether all PAML files already exist in %s...\" % PAML_PATH)\n",
    "    if checkPamlFilesExist(alleleDf):\n",
    "        print('PAML appears to have been run to completion previously. Therefore, it will not be run this time.')\n",
    "    else:\n",
    "        'Not all files generated by PAML appear to exist. Running PAML now (this may take some time)...'\n",
    "        !bash {os.path.join(PAML_PATH, 'paml_script.sh')}\n",
    "        print('PAML finished running.')\n",
    "\n",
    "    analysedAllelesPath = os.path.join(BASE_OUT_PATH, GENOME+'_analysed_alleles.df')    \n",
    "    alleleDf.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    #dataframe where the index is not 'NaN'\n",
    "    noNANdf = alleleDf.loc[~alleleDf.index.isnull(),:].copy()\n",
    "    all_folders = noNANdf.index\n",
    "    #generate a tmp folder for the parallized analysis\n",
    "    tmp_path = os.path.join(BASE_OUT_PATH, 'tmp')\n",
    "    if not os.path.exists(tmp_path):\n",
    "        os.mkdir(tmp_path)\n",
    "    #assign the distances\n",
    "    dist_suffix = 'distdf_tmp'\n",
    "    #do parallized analysi\n",
    "    Parallel(n_jobs=threads)(delayed(assignDistancesToAllAlleles)(list(folder_index_list),all_folders,tmp_path, dist_suffix)\\\n",
    "                       for folder_index_list in grouper(noNANdf.index, 100, np.nan))\n",
    "    distdf_header = ['protein_hamming', 'protein_levenshtein', 'cds_hamming',\n",
    "       'cds_levenshtein']\n",
    "    distdf = combineTmpToDf(distdf_header, dist_suffix, tmp_path, clean=True)\n",
    "    distdf['Index'] = distdf.index\n",
    "    noNANdf['Index'] = noNANdf.index\n",
    "    tmp_df = pd.merge(noNANdf, distdf,how='inner')\n",
    "    tmp_df.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    \n",
    "    print(\"Done with caculate pairwise alignment distances.\")\n",
    "    #pd.util.testing.assert_frame_equal(alleleDf, pd.read_csv(analysedAllelesPath, sep='\\t', index_col=0))\n",
    "    print(\"Starting to parse dN/dS ratios from yn.out file.\")\n",
    "    dNdS_suffix = 'dNdSdf_tmp'\n",
    "    #now assign the dNdS ratios\n",
    "    Parallel(n_jobs=threads)(delayed(assign_dNdS_to_all_alleles)(list(folder_index_list),all_folders,tmp_path, dNdS_suffix)\\\n",
    "                       for folder_index_list in grouper(noNANdf.index, 100, np.nan))\n",
    "    dNdS_header = ['yn00_dN/dS','LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS' ]\n",
    "    dNdSdf = combineTmpToDf(dNdS_header, dNdS_suffix, tmp_path, clean=True)\n",
    "    dNdSdf['Index'] = dNdSdf.index\n",
    "    tmp_df = pd.merge(tmp_df, dNdSdf,how='inner')\n",
    "    tmp_df.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    print('Finish to parse dN/dS ratios.')\n",
    "    print('Combining Dataframe of potential allele pairs with Dataframe of blast none hits.')\n",
    "    #now pull together the dataframes of hits and no-hits blast again.\n",
    "    pre_col_order = tmp_df.columns\n",
    "    alleleDf = pd.concat([tmp_df, alleleDf.loc[alleleDf.index.isnull(), :]], axis=0, ignore_index=True)\n",
    "    alleleDf = alleleDf.loc[:,pre_col_order]\n",
    "    \n",
    "    print(\"Assigning the unphased genes.\")\n",
    "    unphased_s = assign_unphased(alleleDf.Query)\n",
    "    assert(any(unphased_s.index == alleleDf.Query))\n",
    "    alleleDf['unphased'] = unphased_s.values\n",
    "    \n",
    "    print(\"Assining the proteinortho single genes.\")\n",
    "    PO_single_s = assign_PO_single(alleleDf.Query)\n",
    "    assert(any(PO_single_s.index == alleleDf.Query))\n",
    "    alleleDf['PO_single'] = PO_single_s.values\n",
    "    \n",
    "    print(\"Assining the proteinortho inter haplome paralogoues by comparing\\\n",
    "    proteinortho (po) and its synteny mode output (poff)\")\n",
    "    alleleDf['aTarget+aQuery'] = alleleDf.aTarget + '+' + alleleDf.aQuery\n",
    "    interhaplotypeparaloges_s = assign_INP(alleleDf['aTarget+aQuery'])\n",
    "    assert(any(interhaplotypeparaloges_s.index == alleleDf['aTarget+aQuery'] ))\n",
    "    alleleDf['PO_interhaplotype_paralogs'] = interhaplotypeparaloges_s.values\n",
    "    alleleDf.drop('aTarget+aQuery', axis=1, inplace=True)\n",
    "    alleleDf.reset_index(drop=True, inplace=True)\n",
    "    alleleDf.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T14:20:34.198884Z",
     "start_time": "2018-02-26T14:18:35.337098Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking whether all PAML files already exist in /home/benjamin/genome_assembly/Warrior/allele_analysis/v04/post_analysis/paml...\n",
      "PAML appears to have been run to completion previously. Therefore, it will not be run this time.\n",
      "203\n",
      "Done with caculate pairwise alignment distances.\n",
      "Starting to parse dN/dS ratios from yn.out file.\n",
      "203\n",
      "Finish to parse dN/dS ratios.\n",
      "Combining Dataframe of potential allele pairs with Dataframe of blast none hits.\n",
      "Assigning the unphased genes.\n",
      "Assining the proteinortho single genes.\n",
      "Assining the proteinortho inter haplome paralogoues by comparing    proteinortho (po) and its synteny mode output (poff)\n",
      "Check poff and proteinortho output as conceptually all poff pairs should be in included in the proteinortho output.\n",
      "These are the pairs:\n",
      "['evm.model.hcontig_056_002.2+evm.model.pcontig_058.1', 'evm.model.hcontig_050_006.10+evm.model.pcontig_050.56', 'evm.model.hcontig_036_004.26+evm.model.pcontig_036.110']\n"
     ]
    }
   ],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    alleleDf_out = main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T14:21:32.381084Z",
     "start_time": "2018-02-26T14:21:32.369692Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "assert(len(alleleDf) == len(alleleDf_out))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T02:59:35.022027Z",
     "start_time": "2018-02-27T02:59:34.915214Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>protein_levenshtein</th>\n",
       "      <th>cds_hamming</th>\n",
       "      <th>cds_levenshtein</th>\n",
       "      <th>yn00_dN/dS</th>\n",
       "      <th>LWL85_dN/dS</th>\n",
       "      <th>LWL85m_dN/dS</th>\n",
       "      <th>LPB93_dN/dS</th>\n",
       "      <th>unphased</th>\n",
       "      <th>PO_single</th>\n",
       "      <th>PO_interhaplotype_paralogs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>evm.model.pcontig_000.1068</td>\n",
       "      <td>evm.model.hcontig_000_054.9</td>\n",
       "      <td>79.67</td>\n",
       "      <td>246.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>220.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8336</td>\n",
       "      <td>0.8245</td>\n",
       "      <td>0.8245</td>\n",
       "      <td>0.4342</td>\n",
       "      <td>0.4009</td>\n",
       "      <td>0.8530</td>\n",
       "      <td>0.7245</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>evm.model.pcontig_000.121</td>\n",
       "      <td>evm.model.hcontig_014_010.9</td>\n",
       "      <td>27.85</td>\n",
       "      <td>237.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>284.0</td>\n",
       "      <td>507.0</td>\n",
       "      <td>2274.0</td>\n",
       "      <td>2494.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9094</td>\n",
       "      <td>0.8784</td>\n",
       "      <td>0.8513</td>\n",
       "      <td>0.1855</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>evm.model.pcontig_000.124</td>\n",
       "      <td>evm.model.hcontig_009_030.33</td>\n",
       "      <td>40.38</td>\n",
       "      <td>52.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8234</td>\n",
       "      <td>0.7542</td>\n",
       "      <td>0.6929</td>\n",
       "      <td>0.2376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>evm.model.pcontig_000.226</td>\n",
       "      <td>evm.model.hcontig_040_007.41</td>\n",
       "      <td>27.90</td>\n",
       "      <td>534.0</td>\n",
       "      <td>318.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>552.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>489.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8321</td>\n",
       "      <td>0.7624</td>\n",
       "      <td>0.7114</td>\n",
       "      <td>0.1878</td>\n",
       "      <td>0.3450</td>\n",
       "      <td>0.3083</td>\n",
       "      <td>0.3876</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>evm.model.pcontig_000.228</td>\n",
       "      <td>evm.model.hcontig_000_048.30</td>\n",
       "      <td>57.37</td>\n",
       "      <td>190.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7387</td>\n",
       "      <td>0.6999</td>\n",
       "      <td>0.6809</td>\n",
       "      <td>0.5743</td>\n",
       "      <td>0.4815</td>\n",
       "      <td>0.5134</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>evm.model.pcontig_000.339</td>\n",
       "      <td>evm.model.hcontig_038_008.14</td>\n",
       "      <td>85.94</td>\n",
       "      <td>128.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1603</td>\n",
       "      <td>0.1323</td>\n",
       "      <td>0.1323</td>\n",
       "      <td>0.4089</td>\n",
       "      <td>0.2594</td>\n",
       "      <td>0.4050</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>evm.model.pcontig_000.340</td>\n",
       "      <td>evm.model.hcontig_000_031.19</td>\n",
       "      <td>28.98</td>\n",
       "      <td>245.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>361.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>430.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8427</td>\n",
       "      <td>0.7720</td>\n",
       "      <td>0.7191</td>\n",
       "      <td>0.2170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>evm.model.pcontig_000.350</td>\n",
       "      <td>evm.model.hcontig_008_038.9</td>\n",
       "      <td>76.39</td>\n",
       "      <td>72.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>230.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9417</td>\n",
       "      <td>0.9350</td>\n",
       "      <td>0.9346</td>\n",
       "      <td>0.4800</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.4339</td>\n",
       "      <td>0.4172</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>evm.model.pcontig_000.351</td>\n",
       "      <td>evm.model.hcontig_001_024.7</td>\n",
       "      <td>29.82</td>\n",
       "      <td>218.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>172.0</td>\n",
       "      <td>362.0</td>\n",
       "      <td>2778.0</td>\n",
       "      <td>2992.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9534</td>\n",
       "      <td>0.9366</td>\n",
       "      <td>0.9232</td>\n",
       "      <td>0.1858</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>evm.model.pcontig_000.356</td>\n",
       "      <td>evm.model.hcontig_058_009.51</td>\n",
       "      <td>50.24</td>\n",
       "      <td>422.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>505.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>458.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7040</td>\n",
       "      <td>0.6764</td>\n",
       "      <td>0.6451</td>\n",
       "      <td>0.1077</td>\n",
       "      <td>0.1513</td>\n",
       "      <td>0.2024</td>\n",
       "      <td>0.1919</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>evm.model.pcontig_000.373</td>\n",
       "      <td>evm.model.hcontig_000_078.36</td>\n",
       "      <td>43.15</td>\n",
       "      <td>248.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>346.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7102</td>\n",
       "      <td>0.6405</td>\n",
       "      <td>0.5859</td>\n",
       "      <td>0.2262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>evm.model.pcontig_000.376</td>\n",
       "      <td>evm.model.hcontig_053_002.6</td>\n",
       "      <td>54.48</td>\n",
       "      <td>402.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>401.0</td>\n",
       "      <td>332.0</td>\n",
       "      <td>728.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7007</td>\n",
       "      <td>0.6794</td>\n",
       "      <td>0.6644</td>\n",
       "      <td>0.0898</td>\n",
       "      <td>0.1365</td>\n",
       "      <td>0.1461</td>\n",
       "      <td>0.1483</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>evm.model.pcontig_000.376</td>\n",
       "      <td>evm.model.hcontig_013_020.13</td>\n",
       "      <td>54.48</td>\n",
       "      <td>402.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>401.0</td>\n",
       "      <td>332.0</td>\n",
       "      <td>728.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7007</td>\n",
       "      <td>0.6794</td>\n",
       "      <td>0.6644</td>\n",
       "      <td>0.0898</td>\n",
       "      <td>0.1365</td>\n",
       "      <td>0.1461</td>\n",
       "      <td>0.1483</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>evm.model.pcontig_000.377</td>\n",
       "      <td>evm.model.hcontig_047_002.1</td>\n",
       "      <td>26.32</td>\n",
       "      <td>209.0</td>\n",
       "      <td>147.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>323.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8174</td>\n",
       "      <td>0.7575</td>\n",
       "      <td>0.6826</td>\n",
       "      <td>0.2663</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>evm.model.pcontig_000.379</td>\n",
       "      <td>evm.model.hcontig_008_037.10</td>\n",
       "      <td>83.85</td>\n",
       "      <td>260.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>260.0</td>\n",
       "      <td>2327.0</td>\n",
       "      <td>2586.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9235</td>\n",
       "      <td>0.9185</td>\n",
       "      <td>0.9180</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>0.3130</td>\n",
       "      <td>0.3325</td>\n",
       "      <td>0.3525</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>evm.model.pcontig_000.380</td>\n",
       "      <td>evm.model.hcontig_008_037.10</td>\n",
       "      <td>68.97</td>\n",
       "      <td>145.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>2120.0</td>\n",
       "      <td>2262.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9628</td>\n",
       "      <td>0.9564</td>\n",
       "      <td>0.9557</td>\n",
       "      <td>0.6659</td>\n",
       "      <td>0.4305</td>\n",
       "      <td>0.5830</td>\n",
       "      <td>0.6186</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>evm.model.pcontig_000.381</td>\n",
       "      <td>evm.model.hcontig_023_013.8</td>\n",
       "      <td>37.89</td>\n",
       "      <td>161.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>150.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>374.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8403</td>\n",
       "      <td>0.7836</td>\n",
       "      <td>0.7653</td>\n",
       "      <td>0.1587</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>evm.model.pcontig_000.382</td>\n",
       "      <td>evm.model.hcontig_001_032.19</td>\n",
       "      <td>38.19</td>\n",
       "      <td>144.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>185.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6784</td>\n",
       "      <td>0.5544</td>\n",
       "      <td>0.5059</td>\n",
       "      <td>0.3363</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>evm.model.pcontig_000.388</td>\n",
       "      <td>evm.model.hcontig_000_002.2</td>\n",
       "      <td>85.15</td>\n",
       "      <td>101.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8005</td>\n",
       "      <td>0.7859</td>\n",
       "      <td>0.7852</td>\n",
       "      <td>0.7371</td>\n",
       "      <td>0.5006</td>\n",
       "      <td>0.6001</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>evm.model.pcontig_000.395</td>\n",
       "      <td>evm.model.hcontig_074_103.5</td>\n",
       "      <td>63.10</td>\n",
       "      <td>84.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8678</td>\n",
       "      <td>0.8687</td>\n",
       "      <td>0.8628</td>\n",
       "      <td>0.2043</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>evm.model.pcontig_000.396</td>\n",
       "      <td>evm.model.hcontig_002_017.2</td>\n",
       "      <td>84.21</td>\n",
       "      <td>266.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>252.0</td>\n",
       "      <td>1014.0</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9055</td>\n",
       "      <td>0.9050</td>\n",
       "      <td>0.9045</td>\n",
       "      <td>0.2436</td>\n",
       "      <td>0.1796</td>\n",
       "      <td>0.2323</td>\n",
       "      <td>0.2367</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>evm.model.pcontig_000.401</td>\n",
       "      <td>evm.model.hcontig_017_021.4</td>\n",
       "      <td>82.00</td>\n",
       "      <td>200.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5765</td>\n",
       "      <td>0.5655</td>\n",
       "      <td>0.5595</td>\n",
       "      <td>0.2707</td>\n",
       "      <td>0.2282</td>\n",
       "      <td>0.3322</td>\n",
       "      <td>0.3264</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>evm.model.pcontig_000.407</td>\n",
       "      <td>evm.model.hcontig_001_010.17</td>\n",
       "      <td>71.63</td>\n",
       "      <td>571.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>507.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>571.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4737</td>\n",
       "      <td>0.4968</td>\n",
       "      <td>0.4934</td>\n",
       "      <td>0.0901</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>0.1356</td>\n",
       "      <td>0.1307</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>evm.model.pcontig_000.410</td>\n",
       "      <td>evm.model.hcontig_011_029.49</td>\n",
       "      <td>60.76</td>\n",
       "      <td>79.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>429.0</td>\n",
       "      <td>507.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9031</td>\n",
       "      <td>0.8839</td>\n",
       "      <td>0.8766</td>\n",
       "      <td>0.2808</td>\n",
       "      <td>0.2908</td>\n",
       "      <td>0.2605</td>\n",
       "      <td>0.3183</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>evm.model.pcontig_000.431</td>\n",
       "      <td>evm.model.hcontig_000_031.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>220.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>630.0</td>\n",
       "      <td>849.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>220.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7417</td>\n",
       "      <td>0.7417</td>\n",
       "      <td>0.7417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>evm.model.pcontig_000.544</td>\n",
       "      <td>evm.model.hcontig_000_003.3</td>\n",
       "      <td>24.36</td>\n",
       "      <td>312.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>315.0</td>\n",
       "      <td>596.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>767.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8086</td>\n",
       "      <td>0.7288</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>0.2272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>evm.model.pcontig_000.550</td>\n",
       "      <td>evm.model.hcontig_013_020.19</td>\n",
       "      <td>57.14</td>\n",
       "      <td>63.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.4262</td>\n",
       "      <td>0.3882</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.4383</td>\n",
       "      <td>0.4112</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>evm.model.pcontig_000.561</td>\n",
       "      <td>evm.model.hcontig_004_054.12</td>\n",
       "      <td>50.78</td>\n",
       "      <td>128.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>168.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6140</td>\n",
       "      <td>0.5556</td>\n",
       "      <td>0.5380</td>\n",
       "      <td>0.1833</td>\n",
       "      <td>0.2257</td>\n",
       "      <td>0.2526</td>\n",
       "      <td>0.2551</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>evm.model.pcontig_000.568</td>\n",
       "      <td>evm.model.hcontig_011_019.1</td>\n",
       "      <td>84.62</td>\n",
       "      <td>78.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>714.0</td>\n",
       "      <td>791.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9204</td>\n",
       "      <td>0.9168</td>\n",
       "      <td>0.9168</td>\n",
       "      <td>0.1535</td>\n",
       "      <td>0.2091</td>\n",
       "      <td>0.2569</td>\n",
       "      <td>0.2466</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>evm.model.pcontig_000.569</td>\n",
       "      <td>evm.model.hcontig_001_017.47</td>\n",
       "      <td>90.20</td>\n",
       "      <td>102.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3333</td>\n",
       "      <td>0.3116</td>\n",
       "      <td>0.3116</td>\n",
       "      <td>0.4175</td>\n",
       "      <td>0.2935</td>\n",
       "      <td>0.5472</td>\n",
       "      <td>0.4902</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22247</th>\n",
       "      <td>evm.model.pcontig_064.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22248</th>\n",
       "      <td>evm.model.pcontig_064.45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22249</th>\n",
       "      <td>evm.model.pcontig_064.47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22251</th>\n",
       "      <td>evm.model.pcontig_064.59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22262</th>\n",
       "      <td>evm.model.pcontig_065.29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22264</th>\n",
       "      <td>evm.model.pcontig_066.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22265</th>\n",
       "      <td>evm.model.pcontig_066.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22267</th>\n",
       "      <td>evm.model.pcontig_066.14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22269</th>\n",
       "      <td>evm.model.pcontig_066.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22270</th>\n",
       "      <td>evm.model.pcontig_066.37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22277</th>\n",
       "      <td>evm.model.pcontig_066.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22282</th>\n",
       "      <td>evm.model.pcontig_069.21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22284</th>\n",
       "      <td>evm.model.pcontig_069.31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22285</th>\n",
       "      <td>evm.model.pcontig_069.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22287</th>\n",
       "      <td>evm.model.pcontig_069.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22292</th>\n",
       "      <td>evm.model.pcontig_072.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22296</th>\n",
       "      <td>evm.model.pcontig_072.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22299</th>\n",
       "      <td>evm.model.pcontig_072.27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22300</th>\n",
       "      <td>evm.model.pcontig_072.28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22306</th>\n",
       "      <td>evm.model.pcontig_073.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22309</th>\n",
       "      <td>evm.model.pcontig_073.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22311</th>\n",
       "      <td>evm.model.pcontig_074.27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22315</th>\n",
       "      <td>evm.model.pcontig_074.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22317</th>\n",
       "      <td>evm.model.pcontig_075.16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22318</th>\n",
       "      <td>evm.model.pcontig_075.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22329</th>\n",
       "      <td>evm.model.pcontig_077.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22331</th>\n",
       "      <td>evm.model.pcontig_077.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22332</th>\n",
       "      <td>evm.model.pcontig_078.16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22345</th>\n",
       "      <td>evm.model.pcontig_083.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22354</th>\n",
       "      <td>evm.model.pcontig_106.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1639 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Query                        Target   PctID  \\\n",
       "39     evm.model.pcontig_000.1068   evm.model.hcontig_000_054.9   79.67   \n",
       "61      evm.model.pcontig_000.121   evm.model.hcontig_014_010.9   27.85   \n",
       "62      evm.model.pcontig_000.124  evm.model.hcontig_009_030.33   40.38   \n",
       "131     evm.model.pcontig_000.226  evm.model.hcontig_040_007.41   27.90   \n",
       "133     evm.model.pcontig_000.228  evm.model.hcontig_000_048.30   57.37   \n",
       "213     evm.model.pcontig_000.339  evm.model.hcontig_038_008.14   85.94   \n",
       "215     evm.model.pcontig_000.340  evm.model.hcontig_000_031.19   28.98   \n",
       "219     evm.model.pcontig_000.350   evm.model.hcontig_008_038.9   76.39   \n",
       "220     evm.model.pcontig_000.351   evm.model.hcontig_001_024.7   29.82   \n",
       "223     evm.model.pcontig_000.356  evm.model.hcontig_058_009.51   50.24   \n",
       "232     evm.model.pcontig_000.373  evm.model.hcontig_000_078.36   43.15   \n",
       "234     evm.model.pcontig_000.376   evm.model.hcontig_053_002.6   54.48   \n",
       "235     evm.model.pcontig_000.376  evm.model.hcontig_013_020.13   54.48   \n",
       "236     evm.model.pcontig_000.377   evm.model.hcontig_047_002.1   26.32   \n",
       "237     evm.model.pcontig_000.379  evm.model.hcontig_008_037.10   83.85   \n",
       "238     evm.model.pcontig_000.380  evm.model.hcontig_008_037.10   68.97   \n",
       "239     evm.model.pcontig_000.381   evm.model.hcontig_023_013.8   37.89   \n",
       "240     evm.model.pcontig_000.382  evm.model.hcontig_001_032.19   38.19   \n",
       "241     evm.model.pcontig_000.388   evm.model.hcontig_000_002.2   85.15   \n",
       "245     evm.model.pcontig_000.395   evm.model.hcontig_074_103.5   63.10   \n",
       "246     evm.model.pcontig_000.396   evm.model.hcontig_002_017.2   84.21   \n",
       "251     evm.model.pcontig_000.401   evm.model.hcontig_017_021.4   82.00   \n",
       "255     evm.model.pcontig_000.407  evm.model.hcontig_001_010.17   71.63   \n",
       "257     evm.model.pcontig_000.410  evm.model.hcontig_011_029.49   60.76   \n",
       "266     evm.model.pcontig_000.431   evm.model.hcontig_000_031.1  100.00   \n",
       "343     evm.model.pcontig_000.544   evm.model.hcontig_000_003.3   24.36   \n",
       "345     evm.model.pcontig_000.550  evm.model.hcontig_013_020.19   57.14   \n",
       "350     evm.model.pcontig_000.561  evm.model.hcontig_004_054.12   50.78   \n",
       "352     evm.model.pcontig_000.568   evm.model.hcontig_011_019.1   84.62   \n",
       "353     evm.model.pcontig_000.569  evm.model.hcontig_001_017.47   90.20   \n",
       "...                           ...                           ...     ...   \n",
       "22247     evm.model.pcontig_064.3                           NaN     NaN   \n",
       "22248    evm.model.pcontig_064.45                           NaN     NaN   \n",
       "22249    evm.model.pcontig_064.47                           NaN     NaN   \n",
       "22251    evm.model.pcontig_064.59                           NaN     NaN   \n",
       "22262    evm.model.pcontig_065.29                           NaN     NaN   \n",
       "22264    evm.model.pcontig_066.10                           NaN     NaN   \n",
       "22265    evm.model.pcontig_066.11                           NaN     NaN   \n",
       "22267    evm.model.pcontig_066.14                           NaN     NaN   \n",
       "22269     evm.model.pcontig_066.3                           NaN     NaN   \n",
       "22270    evm.model.pcontig_066.37                           NaN     NaN   \n",
       "22277     evm.model.pcontig_066.6                           NaN     NaN   \n",
       "22282    evm.model.pcontig_069.21                           NaN     NaN   \n",
       "22284    evm.model.pcontig_069.31                           NaN     NaN   \n",
       "22285    evm.model.pcontig_069.33                           NaN     NaN   \n",
       "22287     evm.model.pcontig_069.9                           NaN     NaN   \n",
       "22292    evm.model.pcontig_072.13                           NaN     NaN   \n",
       "22296    evm.model.pcontig_072.23                           NaN     NaN   \n",
       "22299    evm.model.pcontig_072.27                           NaN     NaN   \n",
       "22300    evm.model.pcontig_072.28                           NaN     NaN   \n",
       "22306     evm.model.pcontig_073.1                           NaN     NaN   \n",
       "22309     evm.model.pcontig_073.4                           NaN     NaN   \n",
       "22311    evm.model.pcontig_074.27                           NaN     NaN   \n",
       "22315     evm.model.pcontig_074.6                           NaN     NaN   \n",
       "22317    evm.model.pcontig_075.16                           NaN     NaN   \n",
       "22318    evm.model.pcontig_075.17                           NaN     NaN   \n",
       "22329    evm.model.pcontig_077.13                           NaN     NaN   \n",
       "22331     evm.model.pcontig_077.6                           NaN     NaN   \n",
       "22332    evm.model.pcontig_078.16                           NaN     NaN   \n",
       "22345     evm.model.pcontig_083.4                           NaN     NaN   \n",
       "22354     evm.model.pcontig_106.2                           NaN     NaN   \n",
       "\n",
       "       AlnLgth  NumMis  NumGap  StartQuery  StopQuery  StartTarget  \\\n",
       "39       246.0    24.0     1.0         1.0      246.0          1.0   \n",
       "61       237.0   142.0    10.0       284.0      507.0       2274.0   \n",
       "62        52.0    30.0     1.0        94.0      145.0         39.0   \n",
       "131      534.0   318.0    15.0        38.0      552.0          4.0   \n",
       "133      190.0    62.0     5.0       205.0      394.0          8.0   \n",
       "213      128.0    15.0     2.0         1.0      126.0          1.0   \n",
       "215      245.0   127.0    10.0       154.0      361.0        196.0   \n",
       "219       72.0    17.0     0.0         1.0       72.0        159.0   \n",
       "220      218.0   123.0     5.0       172.0      362.0       2778.0   \n",
       "223      422.0   159.0     8.0        96.0      505.0         76.0   \n",
       "232      248.0   123.0     4.0         2.0      235.0        103.0   \n",
       "234      402.0   176.0     2.0         2.0      401.0        332.0   \n",
       "235      402.0   176.0     2.0         2.0      401.0        332.0   \n",
       "236      209.0   147.0     4.0         1.0      205.0        118.0   \n",
       "237      260.0    42.0     0.0         1.0      260.0       2327.0   \n",
       "238      145.0    43.0     1.0         1.0      145.0       2120.0   \n",
       "239      161.0    89.0     1.0         1.0      150.0        214.0   \n",
       "240      144.0    85.0     3.0        48.0      188.0         43.0   \n",
       "241      101.0    15.0     0.0         1.0      101.0         55.0   \n",
       "245       84.0    30.0     1.0         1.0       84.0        112.0   \n",
       "246      266.0    28.0     3.0         1.0      252.0       1014.0   \n",
       "251      200.0    24.0     1.0         1.0      200.0          1.0   \n",
       "255      571.0    98.0     3.0         1.0      507.0          1.0   \n",
       "257       79.0    31.0     0.0         1.0       79.0        429.0   \n",
       "266      220.0     0.0     0.0       630.0      849.0          1.0   \n",
       "343      312.0   187.0    13.0       315.0      596.0        475.0   \n",
       "345       63.0    27.0     0.0         5.0       67.0          1.0   \n",
       "350      128.0    61.0     1.0        41.0      168.0          1.0   \n",
       "352       78.0    12.0     0.0         1.0       78.0        714.0   \n",
       "353      102.0    10.0     0.0         1.0      102.0          1.0   \n",
       "...        ...     ...     ...         ...        ...          ...   \n",
       "22247      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22248      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22249      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22251      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22262      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22264      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22265      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22267      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22269      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22270      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22277      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22282      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22284      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22285      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22287      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22292      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22296      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22299      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22300      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22306      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22309      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22311      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22315      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22317      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22318      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22329      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22331      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22332      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22345      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "22354      NaN     NaN     NaN         NaN        NaN          NaN   \n",
       "\n",
       "       StopTarget            ...              protein_levenshtein  \\\n",
       "39          220.0            ...                           0.8336   \n",
       "61         2494.0            ...                           0.9094   \n",
       "62           89.0            ...                           0.8234   \n",
       "131         489.0            ...                           0.8321   \n",
       "133         178.0            ...                           0.7387   \n",
       "213         127.0            ...                           0.1603   \n",
       "215         430.0            ...                           0.8427   \n",
       "219         230.0            ...                           0.9417   \n",
       "220        2992.0            ...                           0.9534   \n",
       "223         458.0            ...                           0.7040   \n",
       "232         346.0            ...                           0.7102   \n",
       "234         728.0            ...                           0.7007   \n",
       "235         728.0            ...                           0.7007   \n",
       "236         323.0            ...                           0.8174   \n",
       "237        2586.0            ...                           0.9235   \n",
       "238        2262.0            ...                           0.9628   \n",
       "239         374.0            ...                           0.8403   \n",
       "240         185.0            ...                           0.6784   \n",
       "241         155.0            ...                           0.8005   \n",
       "245         194.0            ...                           0.8678   \n",
       "246        1279.0            ...                           0.9055   \n",
       "251         188.0            ...                           0.5765   \n",
       "255         571.0            ...                           0.4737   \n",
       "257         507.0            ...                           0.9031   \n",
       "266         220.0            ...                           0.7417   \n",
       "343         767.0            ...                           0.8086   \n",
       "345          63.0            ...                           0.5190   \n",
       "350         126.0            ...                           0.6140   \n",
       "352         791.0            ...                           0.9204   \n",
       "353         102.0            ...                           0.3333   \n",
       "...           ...            ...                              ...   \n",
       "22247         NaN            ...                              NaN   \n",
       "22248         NaN            ...                              NaN   \n",
       "22249         NaN            ...                              NaN   \n",
       "22251         NaN            ...                              NaN   \n",
       "22262         NaN            ...                              NaN   \n",
       "22264         NaN            ...                              NaN   \n",
       "22265         NaN            ...                              NaN   \n",
       "22267         NaN            ...                              NaN   \n",
       "22269         NaN            ...                              NaN   \n",
       "22270         NaN            ...                              NaN   \n",
       "22277         NaN            ...                              NaN   \n",
       "22282         NaN            ...                              NaN   \n",
       "22284         NaN            ...                              NaN   \n",
       "22285         NaN            ...                              NaN   \n",
       "22287         NaN            ...                              NaN   \n",
       "22292         NaN            ...                              NaN   \n",
       "22296         NaN            ...                              NaN   \n",
       "22299         NaN            ...                              NaN   \n",
       "22300         NaN            ...                              NaN   \n",
       "22306         NaN            ...                              NaN   \n",
       "22309         NaN            ...                              NaN   \n",
       "22311         NaN            ...                              NaN   \n",
       "22315         NaN            ...                              NaN   \n",
       "22317         NaN            ...                              NaN   \n",
       "22318         NaN            ...                              NaN   \n",
       "22329         NaN            ...                              NaN   \n",
       "22331         NaN            ...                              NaN   \n",
       "22332         NaN            ...                              NaN   \n",
       "22345         NaN            ...                              NaN   \n",
       "22354         NaN            ...                              NaN   \n",
       "\n",
       "       cds_hamming  cds_levenshtein  yn00_dN/dS  LWL85_dN/dS  LWL85m_dN/dS  \\\n",
       "39          0.8245           0.8245      0.4342       0.4009        0.8530   \n",
       "61          0.8784           0.8513      0.1855          NaN           NaN   \n",
       "62          0.7542           0.6929      0.2376          NaN           NaN   \n",
       "131         0.7624           0.7114      0.1878       0.3450        0.3083   \n",
       "133         0.6999           0.6809      0.5743       0.4815        0.5134   \n",
       "213         0.1323           0.1323      0.4089       0.2594        0.4050   \n",
       "215         0.7720           0.7191      0.2170          NaN           NaN   \n",
       "219         0.9350           0.9346      0.4800       0.2859        0.4339   \n",
       "220         0.9366           0.9232      0.1858          NaN           NaN   \n",
       "223         0.6764           0.6451      0.1077       0.1513        0.2024   \n",
       "232         0.6405           0.5859      0.2262          NaN           NaN   \n",
       "234         0.6794           0.6644      0.0898       0.1365        0.1461   \n",
       "235         0.6794           0.6644      0.0898       0.1365        0.1461   \n",
       "236         0.7575           0.6826      0.2663          NaN           NaN   \n",
       "237         0.9185           0.9180      0.4067       0.3130        0.3325   \n",
       "238         0.9564           0.9557      0.6659       0.4305        0.5830   \n",
       "239         0.7836           0.7653      0.1587          NaN           NaN   \n",
       "240         0.5544           0.5059      0.3363          NaN           NaN   \n",
       "241         0.7859           0.7852      0.7371       0.5006        0.6001   \n",
       "245         0.8687           0.8628      0.2043          NaN           NaN   \n",
       "246         0.9050           0.9045      0.2436       0.1796        0.2323   \n",
       "251         0.5655           0.5595      0.2707       0.2282        0.3322   \n",
       "255         0.4968           0.4934      0.0901       0.1020        0.1356   \n",
       "257         0.8839           0.8766      0.2808       0.2908        0.2605   \n",
       "266         0.7417           0.7417         NaN          NaN           NaN   \n",
       "343         0.7288           0.6396      0.2272          NaN           NaN   \n",
       "345         0.4262           0.3882      0.3671       0.4383        0.4112   \n",
       "350         0.5556           0.5380      0.1833       0.2257        0.2526   \n",
       "352         0.9168           0.9168      0.1535       0.2091        0.2569   \n",
       "353         0.3116           0.3116      0.4175       0.2935        0.5472   \n",
       "...            ...              ...         ...          ...           ...   \n",
       "22247          NaN              NaN         NaN          NaN           NaN   \n",
       "22248          NaN              NaN         NaN          NaN           NaN   \n",
       "22249          NaN              NaN         NaN          NaN           NaN   \n",
       "22251          NaN              NaN         NaN          NaN           NaN   \n",
       "22262          NaN              NaN         NaN          NaN           NaN   \n",
       "22264          NaN              NaN         NaN          NaN           NaN   \n",
       "22265          NaN              NaN         NaN          NaN           NaN   \n",
       "22267          NaN              NaN         NaN          NaN           NaN   \n",
       "22269          NaN              NaN         NaN          NaN           NaN   \n",
       "22270          NaN              NaN         NaN          NaN           NaN   \n",
       "22277          NaN              NaN         NaN          NaN           NaN   \n",
       "22282          NaN              NaN         NaN          NaN           NaN   \n",
       "22284          NaN              NaN         NaN          NaN           NaN   \n",
       "22285          NaN              NaN         NaN          NaN           NaN   \n",
       "22287          NaN              NaN         NaN          NaN           NaN   \n",
       "22292          NaN              NaN         NaN          NaN           NaN   \n",
       "22296          NaN              NaN         NaN          NaN           NaN   \n",
       "22299          NaN              NaN         NaN          NaN           NaN   \n",
       "22300          NaN              NaN         NaN          NaN           NaN   \n",
       "22306          NaN              NaN         NaN          NaN           NaN   \n",
       "22309          NaN              NaN         NaN          NaN           NaN   \n",
       "22311          NaN              NaN         NaN          NaN           NaN   \n",
       "22315          NaN              NaN         NaN          NaN           NaN   \n",
       "22317          NaN              NaN         NaN          NaN           NaN   \n",
       "22318          NaN              NaN         NaN          NaN           NaN   \n",
       "22329          NaN              NaN         NaN          NaN           NaN   \n",
       "22331          NaN              NaN         NaN          NaN           NaN   \n",
       "22332          NaN              NaN         NaN          NaN           NaN   \n",
       "22345          NaN              NaN         NaN          NaN           NaN   \n",
       "22354          NaN              NaN         NaN          NaN           NaN   \n",
       "\n",
       "      LPB93_dN/dS unphased PO_single PO_interhaplotype_paralogs  \n",
       "39         0.7245     True      True                      False  \n",
       "61            NaN     True      True                      False  \n",
       "62            NaN     True      True                      False  \n",
       "131        0.3876     True      True                      False  \n",
       "133        0.5319     True      True                      False  \n",
       "213        0.3666     True      True                      False  \n",
       "215           NaN     True      True                      False  \n",
       "219        0.4172     True      True                      False  \n",
       "220           NaN     True      True                      False  \n",
       "223        0.1919     True      True                      False  \n",
       "232           NaN     True      True                      False  \n",
       "234        0.1483     True      True                      False  \n",
       "235        0.1483     True      True                      False  \n",
       "236           NaN     True      True                      False  \n",
       "237        0.3525     True      True                      False  \n",
       "238        0.6186     True      True                      False  \n",
       "239           NaN     True      True                      False  \n",
       "240           NaN     True      True                      False  \n",
       "241        0.5578     True      True                      False  \n",
       "245           NaN     True      True                      False  \n",
       "246        0.2367     True      True                      False  \n",
       "251        0.3264     True      True                      False  \n",
       "255        0.1307     True      True                      False  \n",
       "257        0.3183     True      True                      False  \n",
       "266           NaN     True      True                      False  \n",
       "343           NaN     True      True                      False  \n",
       "345        0.5508     True      True                      False  \n",
       "350        0.2551     True      True                      False  \n",
       "352        0.2466     True      True                      False  \n",
       "353        0.4902     True      True                      False  \n",
       "...           ...      ...       ...                        ...  \n",
       "22247         NaN     True      True                      False  \n",
       "22248         NaN     True      True                      False  \n",
       "22249         NaN     True      True                      False  \n",
       "22251         NaN     True      True                      False  \n",
       "22262         NaN     True      True                      False  \n",
       "22264         NaN     True      True                      False  \n",
       "22265         NaN     True      True                      False  \n",
       "22267         NaN     True      True                      False  \n",
       "22269         NaN     True      True                      False  \n",
       "22270         NaN     True      True                      False  \n",
       "22277         NaN     True      True                      False  \n",
       "22282         NaN     True      True                      False  \n",
       "22284         NaN     True      True                      False  \n",
       "22285         NaN     True      True                      False  \n",
       "22287         NaN     True      True                      False  \n",
       "22292         NaN     True      True                      False  \n",
       "22296         NaN     True      True                      False  \n",
       "22299         NaN     True      True                      False  \n",
       "22300         NaN     True      True                      False  \n",
       "22306         NaN     True      True                      False  \n",
       "22309         NaN     True      True                      False  \n",
       "22311         NaN     True      True                      False  \n",
       "22315         NaN     True      True                      False  \n",
       "22317         NaN     True      True                      False  \n",
       "22318         NaN     True      True                      False  \n",
       "22329         NaN     True      True                      False  \n",
       "22331         NaN     True      True                      False  \n",
       "22332         NaN     True      True                      False  \n",
       "22345         NaN     True      True                      False  \n",
       "22354         NaN     True      True                      False  \n",
       "\n",
       "[1639 rows x 38 columns]"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf_out[(alleleDf_out.unphased == True) & (alleleDf_out.PO_single == True)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T03:01:36.794067Z",
     "start_time": "2018-02-27T03:01:36.707233Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6404"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf_out[(alleleDf_out.unphased == False) & (alleleDf_out.PO_single == True)]['Query'].unique().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T03:11:12.535376Z",
     "start_time": "2018-02-27T03:11:12.528724Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['BLAST', 'PO', 'h_rBLAST', nan], dtype=object)"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf_out.allele_source.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T03:13:19.826046Z",
     "start_time": "2018-02-27T03:13:19.811305Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery',\n",
       "       'StopQuery', 'StartTarget', 'StopTarget', 'e-value', 'BitScore',\n",
       "       'QLgth', 'QCov', 'TLgth', 'TCov', 'q_contig', 't_contig',\n",
       "       'q_contig == t_contig', 'p_protein', 'h_contig_overlap',\n",
       "       't_contig == h_contig_overlap', 'allele_source', 'matchType', 'aQuery',\n",
       "       'aTarget'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phFullAlleleDf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T03:12:21.672184Z",
     "start_time": "2018-02-27T03:12:21.635557Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11188    PO\n",
       "11189    PO\n",
       "11190    PO\n",
       "11191    PO\n",
       "11192    PO\n",
       "11193    PO\n",
       "11194    PO\n",
       "11195    PO\n",
       "11196    PO\n",
       "11197    PO\n",
       "11198    PO\n",
       "11199    PO\n",
       "11200    PO\n",
       "11201    PO\n",
       "11202    PO\n",
       "11203    PO\n",
       "11204    PO\n",
       "11205    PO\n",
       "11206    PO\n",
       "11207    PO\n",
       "11208    PO\n",
       "11209    PO\n",
       "11210    PO\n",
       "11211    PO\n",
       "11212    PO\n",
       "11213    PO\n",
       "11214    PO\n",
       "11215    PO\n",
       "11216    PO\n",
       "11217    PO\n",
       "         ..\n",
       "19300    PO\n",
       "19301    PO\n",
       "19302    PO\n",
       "19303    PO\n",
       "19304    PO\n",
       "19305    PO\n",
       "19306    PO\n",
       "19307    PO\n",
       "19308    PO\n",
       "19309    PO\n",
       "19310    PO\n",
       "19311    PO\n",
       "19312    PO\n",
       "19313    PO\n",
       "19314    PO\n",
       "19315    PO\n",
       "19316    PO\n",
       "19317    PO\n",
       "19318    PO\n",
       "19319    PO\n",
       "19320    PO\n",
       "19321    PO\n",
       "19322    PO\n",
       "19323    PO\n",
       "19324    PO\n",
       "19325    PO\n",
       "19326    PO\n",
       "19327    PO\n",
       "19328    PO\n",
       "19329    PO\n",
       "Name: matchType, Length: 8142, dtype: object"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf_out[alleleDf_out.allele_source == 'PO'].matchType"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-20T02:49:47.072255Z",
     "start_time": "2018-02-20T02:49:46.220172Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "########## FIGURE PLOTTING ##########\n",
    "\n",
    "def make_autopct(values):\n",
    "    def my_autopct(pct):\n",
    "        total = sum(values)\n",
    "        val = int(round(pct*total/100.0))\n",
    "        return '{p:.2f}%\\n({v:d})'.format(p=pct,v=val)\n",
    "    return my_autopct\n",
    "\n",
    "def autolabel(rects, labels, ax, fontsize):\n",
    "    \"\"\"\n",
    "    Attach a text label above each bar displaying its height\n",
    "    \"\"\"\n",
    "    for i, rect in enumerate(rects):\n",
    "        height = rect.get_height()\n",
    "        ax.text(rect.get_x() + rect.get_width()/2., height, str(labels[i]), ha='center', va='bottom', fontsize=fontsize)\n",
    "\n",
    "def getNumNoAlleles(pProteinFastaFile, alleleDf):\n",
    "    with open(pProteinFastaFile) as pProteinFasta:\n",
    "        pProteinList = []\n",
    "        for line in pProteinFasta:\n",
    "            if line.startswith('>'):\n",
    "                pProteinList.append(line[1:].strip())\n",
    "\n",
    "    assert(len(pProteinList) == len(set(pProteinList)))\n",
    "\n",
    "\n",
    "    pairedPProteinList = list(alleleDf['Query'])\n",
    "    pairedPProteinList += list(alleleDf['Target'])\n",
    "    pairedPProteinList = set(pairedPProteinList)\n",
    "    \n",
    "    for pairedPProtein in pairedPProteinList:\n",
    "        if pairedPProtein in pProteinList:\n",
    "            pProteinList.remove(pairedPProtein)\n",
    "    \n",
    "    return len(pProteinList)\n",
    "\n",
    "def plotAlleleTypesPie(ax, alleleDf, colors, includeNoAlleles=True):\n",
    "    '''Plots a pie chart of allele types, with the option of also including \n",
    "    primary proteins with no alleles. Strictly, this is not an accurate representation\n",
    "    of the distribution of primary proteins as the reciprocal BLAST-identified (h on p) alleles\n",
    "    may result in double-counting of primary proteins.\n",
    "    '''\n",
    "    # OrderedDict to preserve order, so that plots are coloured with same key as the distance \n",
    "    # bar graphs. This is a bit of a hack-fix; must enter these by hand again in the same order \n",
    "    # as 'matchType' occurs in the alleleAveragesByMatchType DataFrame.\n",
    "    alleleTypeCountDict = collections.OrderedDict()\n",
    "    \n",
    "    for matchType in alleleDf['matchType'].unique():\n",
    "        alleleTypeCountDict[matchType] = len(alleleDf[alleleDf['matchType'] == matchType])\n",
    "    \n",
    "    if includeNoAlleles==True:\n",
    "        numNoAlleles = getNumNoAlleles(P_PROTEINS_FASTA, alleleDf)\n",
    "        alleleTypeCountDict['no_allele'] = numNoAlleles\n",
    "\n",
    "    patches, texts, autotexts = ax.pie(list(alleleTypeCountDict.values()), labels=alleleTypeCountDict.keys(), autopct=make_autopct(list(alleleTypeCountDict.values())), colors=colors)\n",
    "    ax.axis('equal')\n",
    "    ax.set_title('Allele Types', loc='center', fontsize=TITLE_SIZE, position=(0.5, 1.1))\n",
    "\n",
    "def plotLevenshteinBar(alleleAverages, ax, colors):\n",
    "    '''Plots a bar graph of normalised Levenshtein distances on ax from DataFrame alleleAverages.'''\n",
    "    \n",
    "    ind = np.arange(len(alleleAverages.protein_levenshtein))\n",
    "    rects = ax.bar(ind, alleleAverages.protein_levenshtein, 0.35, color=colors, align='center') \n",
    "    \n",
    "    sns.despine(top=True, right=True)\n",
    "    \n",
    "    barLabels = []\n",
    "    for levDist in alleleAverages.protein_levenshtein:\n",
    "        barLabels.append(str(int((1-levDist)*100)) + '%')\n",
    "    autolabel(rects, barLabels, ax, INLINE_LABEL_SIZE)\n",
    "\n",
    "    ax.set_xticks(ind)\n",
    "    ax.set_xticklabels(alleleAverages.index, rotation=45)\n",
    "\n",
    "    # ax.set_xlabel('Allele Types', fontsize=AXIS_LABEL_SIZE)\n",
    "    ax.set_ylabel('Normalised Levenshtein Distance', fontsize=AXIS_LABEL_SIZE)\n",
    "\n",
    "    ax.tick_params(axis='both', which='major', labelsize=AXIS_TICK_SIZE, pad=3)\n",
    "\n",
    "    for tick in ax.get_xaxis().get_major_ticks():\n",
    "        tick.set_pad(2*tick.get_pad())\n",
    "        tick.label1 = tick._get_text1()\n",
    "        \n",
    "def plotAlleles(alleleDf, qCovFilters, tCovFilters, pctIdFilters, levSimFilters, leavePO):\n",
    "    '''Makes a 3x2 plot with normalised Levenshtein distance plots in column 1 and\n",
    "    a pie chart representing the distribution of allele types in column 2.\n",
    "    Each row shows different levels of filtering.\n",
    "    \n",
    "    leavePO is a boolean that determines whether only BLAST hits will be filtered (leavePO=True)\n",
    "    or both BLAST and PO alleles should be filtered (leavePO=False)'''\n",
    "    cmap = plt.cm.Greens\n",
    "    colors = cmap(np.linspace(0.0, 0.6, len(alleleDf['matchType'].unique())))\n",
    "    \n",
    "    assert(len(qCovFilters) == len(pctIdFilters) == len(levSimFilters))\n",
    "    \n",
    "    fig, ax = plt.subplots(len(qCovFilters), 2, figsize=(30, 12*len(qCovFilters)))\n",
    "    \n",
    "    for i in range(len(qCovFilters)):\n",
    "        \n",
    "        filteredAlleleDf = filterAlleleDf(alleleDf, qCovFilters[i], tCovFilters[i], pctIdFilters[i], levSimFilters[i], True)\n",
    "        print(filteredAlleleDf[filteredAlleleDf.allele_source == 'PO'].shape[0])\n",
    "        # levenshtein distance plot\n",
    "        alleleAveragesByMatchType = filteredAlleleDf.groupby(['matchType']).mean()\n",
    "        plotLevenshteinBar(alleleAveragesByMatchType, ax[i, 0], colors)\n",
    "        ax[i, 0].set_xticklabels(alleleAveragesByMatchType.index, rotation=45, ha='right')\n",
    "        \n",
    "        # pie plot\n",
    "        plotAlleleTypesPie(ax[i, 1], filteredAlleleDf, colors)\n",
    "        \n",
    "        # include filtering criteria in title\n",
    "        qCovFilter = qCovFilters[i]\n",
    "        tCovFilter = tCovFilters[i]\n",
    "        pctIdFilter = pctIdFilters[i]\n",
    "        levSimFilter = levSimFilters[i]\n",
    "        if qCovFilter < BASE_QCOV_CUTOFF:\n",
    "            qCovFilter = BASE_QCOV_CUTOFF\n",
    "            print('Base QCov cut-off is 70%; if you desire to filter below this value, decrease BASE_QCOV_CUTOFF.')\n",
    "        if tCovFilter < BASE_TCOV_CUTOFF:\n",
    "            tCovFilter = BASE_TCOV_CUTOFF\n",
    "            print('Base TCov cut-off is 70%; if you desire to filter below this value, decrease BASE_TCOV_CUTOFF.')\n",
    "        if pctIdFilter < BASE_PCTID_CUTOFF:\n",
    "            pctIdFilter = BASE_PCTID_CUTOFF\n",
    "            print('Base %ID cut-off is 70%; if you desire to filter below this value, decrease BASE_PCTID_CUTOFF.')\n",
    "        if not levSimFilter:\n",
    "            levSimFilter = 0\n",
    "    \n",
    "        ax[i, 0].set_title('QCov > %s%%, TCov > %s%%, ID > %s%%, L. sim. > %s%%, PO Filtered: %s' % (qCovFilter, tCovFilter, pctIdFilter, levSimFilter, not leavePO), position=(0.5, 0.85))\n",
    "        \n",
    "    fig.tight_layout()\n",
    "    fig.savefig(os.path.join(FIGURE_PATH, 'fig'), bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-20T02:50:17.346988Z",
     "start_time": "2018-02-20T02:49:49.617825Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Used in the pie chart for all text except title\n",
    "# the ax.pie plotting interface is weird - cannot set other font sizes properly?\n",
    "mpl.rcParams['font.size'] = 24\n",
    "\n",
    "TITLE_SIZE = 32\n",
    "AXIS_LABEL_SIZE = 28\n",
    "AXIS_TICK_SIZE = 24\n",
    "INLINE_LABEL_SIZE = 24\n",
    "\n",
    "# These lists must all be the same length\n",
    "qCovFilters = [False, 80, 90, 95]\n",
    "pctIdFilters = [False, 80, 90, 95]\n",
    "tCovFilters = [False, 80, 90, 95]\n",
    "levSimFilters = [False, False, False, False]\n",
    "\n",
    "plotAlleles(alleleDf, qCovFilters, tCovFilters, pctIdFilters, levSimFilters, True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
