{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DK_0911_post_allele_analysis_v02"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Based on original code by Benjamin Schwessinger."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Inputs: output from `DK_0911_defining_alleles_v02` & primary+haplotig (ph) protein/gene/cds .*fasta* files from `DK_0911_generate_fasta_files_from_gff3`.\n",
    "- Programs: **MUSCLE**, **PAML**\n",
    "- Purpose: generate and save a DataFrame containing dN/dS information (number of nonsynonymous substitutions per non-synonymous site to the number of synonymous substitutions per synonymous site), as well as Hamming & Levenshtein distances (measures of % identity). Also provides visualisations of some of this data.\n",
    "\n",
    "#### Overview\n",
    "1. Reads in the large allele DataFrames generated in `DK_0911_defining_alleles_v02` (i.e. proteinortho hits OR best blast hit) - see description header cell in that notebook for more information on which alleles are included in that DataFrame.\n",
    "2. Filters the allele DataFrames based on %ID and %QCov (this can be set to filter only BLAST-identified alleles or both BLAST- and proteinortho-identified alleles) so that distance information is not calculated on an unnecessarily large number of alleles.\n",
    "3. Calculates distance & dN/dS information, and saves this to an output file so that it does not have to be re-calculated (if for whatever reason, the inputs change so that dN/dS or distance information should change, this output file (`DK_0911_v0x_analysed_alleles.df`) should be deleted so that it can be re-generated.\n",
    "4. Plots graphs of allele-type distribution (pie chart) and allele-type Levenshtein distances (measures of similarity) for different levels of allele-filtering (QCov/TCov/%ID/Levenshtein similarity).\n",
    "\n",
    "NB:\n",
    "- dN/dS information is currently not utilised in this script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:11.535137Z",
     "start_time": "2018-02-26T13:06:10.986516Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.371468Z",
     "start_time": "2018-02-26T13:06:11.545224Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "from Bio import SeqIO\n",
    "from Bio import AlignIO\n",
    "import distance\n",
    "import editdistance\n",
    "import math\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import re\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib as mpl\n",
    "import collections\n",
    "import pybedtools\n",
    "from sklearn.externals.joblib import Parallel, delayed\n",
    "import itertools as it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.438815Z",
     "start_time": "2018-02-26T13:06:13.374933Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#define variables here\n",
    "GENOME_VERSION = 'v04'\n",
    "\n",
    "BASE_PATH = '/home/benjamin/genome_assembly/Warrior/allele_analysis/%s/' % GENOME_VERSION\n",
    "\n",
    "YN00_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/post_analysis/yn00.ctl'\n",
    "BASE_OUT_PATH = os.path.join(BASE_PATH, 'post_analysis/')\n",
    "ALLELE_PATH = os.path.join(BASE_PATH, 'allele_analysis/alleles_proteinortho_graph516/')\n",
    "UNFILTERED_DF_PATH = os.path.join(BASE_PATH, \\\n",
    "    'allele_analysis/DK_0911_%s_p_ctg.DK_0911_%s_h_ctg.0.001.blastp.outfmt6.allele_analysis' \\\n",
    "                                  % (GENOME_VERSION, GENOME_VERSION))\n",
    "GENOME_PATH = '/home/benjamin/genome_assembly/Warrior/genome_%s/' % GENOME_VERSION\n",
    "FIGURE_PATH = os.path.join(BASE_OUT_PATH, 'figures')\n",
    "\n",
    "GENOME = 'DK_0911_%s' % GENOME_VERSION\n",
    "P_GENOME = GENOME + '_p_ctg'\n",
    "H_GENOME = GENOME + '_h_ctg'\n",
    "\n",
    "threads = 8\n",
    "\n",
    "# Base filtering so that distance calculations are not performed on all allele pairs.\n",
    "# Distance calculations will only be performed on allele pairs above the defined cutoffs.\n",
    "# Note that proteinortho alleles will not be affected (this can be changed in the filterAlleleDf function).\n",
    "BASE_QCOV_CUTOFF = 0\n",
    "BASE_TCOV_CUTOFF = 0\n",
    "BASE_PCTID_CUTOFF = 0\n",
    "\n",
    "P_PROTEINS_FASTA = os.path.join(GENOME_PATH, P_GENOME + '.protein.fa')\n",
    "\n",
    "PAML_PATH = os.path.join(BASE_OUT_PATH, 'paml')\n",
    "if not os.path.exists(BASE_OUT_PATH):\n",
    "    os.mkdir(BASE_OUT_PATH)\n",
    "if not os.path.exists(FIGURE_PATH):\n",
    "    os.mkdir(FIGURE_PATH)\n",
    "if not os.path.exists(PAML_PATH):\n",
    "    os.mkdir(PAML_PATH)\n",
    "    shutil.copy2(YN00_PATH, PAML_PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.458087Z",
     "start_time": "2018-02-26T13:06:13.449039Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#for homozygous region analysis\n",
    "COV_PATH = '/home/benjamin/genome_assembly/Warrior/COV'\n",
    "homo_bed_fh = os.path.join(COV_PATH, 'DK0911_v04_ph_ctg.ph_p_homo_cov.bed')\n",
    "anno_gff_p_fh = os.path.join(GENOME_PATH, 'DK_0911_v04_p_ctg.anno.gff3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.480608Z",
     "start_time": "2018-02-26T13:06:13.465530Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "PH_PROTEIN_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.protein.fa')\n",
    "PH_GENE_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.gene.fa')\n",
    "PH_CDS_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.cds.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.504764Z",
     "start_time": "2018-02-26T13:06:13.487664Z"
    }
   },
   "outputs": [],
   "source": [
    "#for finalizing the protein ortho single analysis\n",
    "proortho_fh = os.path.join(BASE_PATH, 'proteinortho', 'ph_ctg_516.proteinortho')\n",
    "proorthograph_fh = os.path.join(BASE_PATH, 'proteinortho', 'ph_ctg_516.proteinortho-graph')\n",
    "poff_graph_fh = os.path.join(BASE_PATH, 'proteinortho', 'ph_ctg_516.poff-graph')\n",
    "assert(os.path.exists(proortho_fh))\n",
    "assert(os.path.exists(proorthograph_fh))\n",
    "assert(os.path.exists(poff_graph_fh))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.556737Z",
     "start_time": "2018-02-26T13:06:13.524758Z"
    }
   },
   "outputs": [],
   "source": [
    "def getInterHaplotypeParaloges(proorthograph_fh,poff_graph_fh):\n",
    "    \"\"\"This function compares the proteinorth (po) and po synteny output. \n",
    "    It returns all Target+Query pairs that are only found in po as list. \n",
    "    It also checks if there are pairs only found in poff.\"\"\"\n",
    "    graph_header = ['Target', 'Query', 'evalue_ab', 'bitscore_ab', 'evalue_ba', 'bitscore_ba', 'same_strand' , 'simscore']\n",
    "    poff_df = pd.read_csv(poff_graph_fh, sep='\\t', header=None, names=graph_header, comment='#' )\n",
    "    po_df = pd.read_csv(proorthograph_fh, sep='\\t', header=None, names=graph_header, comment='#' )\n",
    "    poff_paring = (poff_df.Target+ '+' + poff_df.Query)\n",
    "    po_paring = (po_df.Target+ '+' + po_df.Query)\n",
    "    interhaplotypeparaloges_l = list(set(po_paring) - set(poff_paring))\n",
    "    poff_only_pairs = list(set(poff_paring) - set(po_paring))\n",
    "    if len(poff_only_pairs) > 0:\n",
    "        print(\"Check poff and proteinortho output as conceptually all poff pairs should be in included in the proteinortho output.\\nThese are the pairs:\")\n",
    "        print(poff_only_pairs)\n",
    "    return interhaplotypeparaloges_l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.579685Z",
     "start_time": "2018-02-26T13:06:13.560170Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assign_INP(aTargetplusaQuery, proorthograph_fh = proorthograph_fh,poff_graph_fh = poff_graph_fh ):\n",
    "    \"\"\"\n",
    "    Returns a series of of True/False values if a aTarget+aQuery pair are element identified as \n",
    "    Inter haplotype paraloge.\n",
    "    Input:\n",
    "        proorthograh_fh, proteinortho graph output file handle that has been run including the single flag\n",
    "        queries, series of protein queries used for the blast search.\n",
    "        poff_graph_fh, proteinortho graph output file handle that has been run including the synteny and\n",
    "        single flag\n",
    "        aTargetplusaQuery, series of protein aTargets + aQuiers used for the blast search.\n",
    "    Outupt:\n",
    "        Returns a True/False series with the aTargets + aQuiers as index.\n",
    "    \n",
    "    \"\"\"\n",
    "\n",
    "    interhaplotypeparaloges_l = getInterHaplotypeParaloges(proorthograph_fh,poff_graph_fh)\n",
    "    interhaplotypeparaloges_s = pd.Series([False]*len(aTargetplusaQuery), index=aTargetplusaQuery)\n",
    "    interhaplotypeparaloges_s.loc[interhaplotypeparaloges_s.index.isin(interhaplotypeparaloges_l)] = True\n",
    "    #print(len(Unphasedgenes))\n",
    "    return interhaplotypeparaloges_s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.600770Z",
     "start_time": "2018-02-26T13:06:13.582132Z"
    }
   },
   "outputs": [],
   "source": [
    "def getProOrthsingles(proortho_fh):\n",
    "    \"\"\"Parses out all proteinortho singles from a proteinortho output file.\n",
    "    Returns a list of these singles.\"\"\"\n",
    "    with open(proortho_fh) as file:\n",
    "        id_pattern = r'(evm\\S*)'\n",
    "        regex = re.compile(id_pattern)\n",
    "        po_single_list = []\n",
    "        for line in file:\n",
    "            if '*' in line:\n",
    "                po_single_list.append(regex.search(line).groups()[0])\n",
    "            else:\n",
    "                continue\n",
    "    return po_single_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.624444Z",
     "start_time": "2018-02-26T13:06:13.607693Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assign_PO_single(queries, proortho_fh=proortho_fh):\n",
    "    \"\"\"\n",
    "    Returns a series of of True/False values if a query element identified as single.\n",
    "    Input:\n",
    "        proortho_fh, proteinortho output file handle that has been run including the single flag\n",
    "        queries, series of protein queries used for the blast search.\n",
    "    Outupt:\n",
    "        Returns a True/False series with the queries as index.\n",
    "    \n",
    "    \"\"\"\n",
    "\n",
    "    PO_singles = getProOrthsingles(proortho_fh)\n",
    "    PO_singles_s = pd.Series([False]*len(queries), index=queries)\n",
    "    PO_singles_s.loc[PO_singles_s.index.isin(PO_singles)] = True\n",
    "    #print(len(Unphasedgenes))\n",
    "    return PO_singles_s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.649232Z",
     "start_time": "2018-02-26T13:06:13.637651Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def allgffTogenegff(gff_fh, write_out=True):\n",
    "    '''Converts at complete gff to a gene gff only and writtes it out.'''\n",
    "    gene_gff = pd.read_csv(gff_fh, sep='\\t', header=None)\n",
    "    gene_gff = gene_gff[gene_gff[2] == 'gene']\n",
    "    gene_gff.reset_index(drop=True, inplace=True)\n",
    "    gene_gff.to_csv(gff_fh.replace('anno', 'gene'), sep='\\t', header=False, index=False)\n",
    "    return gene_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.671504Z",
     "start_time": "2018-02-26T13:06:13.651889Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def col_8_id(x):\n",
    "    '''Function that pulls out the ID from the 9th column of a df.'''\n",
    "    pattern = r'ID=([a-zA-Z0-9_.]*);'\n",
    "    regex = re.compile(pattern)  \n",
    "    m = regex.search(x)\n",
    "    match = m.groups()[0].replace('TU', 'model')\n",
    "    if match.startswith('cds.'):\n",
    "        match = match[4:]\n",
    "    if 'exon' in match:\n",
    "        _list = match.split('.')\n",
    "        match = '.'.join(_list[:-1])\n",
    "    return match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.766555Z",
     "start_time": "2018-02-26T13:06:13.674577Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assignMatchType(allele_source, overlap, no_overlap):\n",
    "    if not allele_source == np.nan:\n",
    "        return allele_source\n",
    "    \n",
    "    s = allele_source + '_'\n",
    "    \n",
    "    if overlap:\n",
    "        s += 'overlap'\n",
    "    elif no_overlap:\n",
    "        s += 'no_overlap'\n",
    "    else: # different_pcontig\n",
    "        s += 'unlinked'\n",
    "    return s\n",
    "\n",
    "def reduceGroups(g):\n",
    "    '''returns the best hit based on e-value and BitScore per group'''\n",
    "    if len(g) == 1:\n",
    "        return g\n",
    "    tmp_g = g[g['e-value'] == g['e-value'].min()]\n",
    "    if len(tmp_g) == 1:\n",
    "        return tmp_g\n",
    "    return tmp_g[tmp_g['BitScore'] == tmp_g['BitScore'].max()]\n",
    "\n",
    "def filterAlleleDf(alleleDf, qCov, tCov, pctId, levSim, leavePO=False):\n",
    "    if leavePO:\n",
    "        no_PO_df = alleleDf[(alleleDf['allele_source'] == 'h_rBLAST') | (alleleDf['allele_source'] == 'BLAST')]\n",
    "        PO_df = alleleDf[alleleDf['allele_source'] == 'PO']\n",
    "\n",
    "        filtered_no_PO_df = filterAlleleDf(no_PO_df, qCov, tCov, pctId, levSim)\n",
    "        return filtered_no_PO_df.append(PO_df, ignore_index=True)\n",
    "    \n",
    "    if qCov:\n",
    "        alleleDf = alleleDf[alleleDf['QCov'] > qCov]\n",
    "    if tCov:\n",
    "        alleleDf = alleleDf[alleleDf['TCov'] > tCov]\n",
    "    if pctId:\n",
    "        alleleDf = alleleDf[alleleDf['PctID'] > pctId]\n",
    "    if levSim:\n",
    "        levDist = (100-levSim)/100.0\n",
    "        alleleDf = alleleDf[alleleDf['protein_levenshtein'] < levDist]\n",
    "\n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.793099Z",
     "start_time": "2018-02-26T13:06:13.780669Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def geneUnphased(Gene_gff_fh, Homo_cov_bed_fh ):\n",
    "    \"\"\"\n",
    "    Returns a list of all genes that are unphased.\n",
    "    \n",
    "    Input: * Fh for annotation gff file\n",
    "           * Fh for Homo_cov_bed_fh\n",
    "    Output: A set of gene IDs that are unphases\n",
    "    \"\"\"\n",
    "    geneGff_bed = pybedtools.BedTool(Gene_gff_fh)\n",
    "    homo_p_bed = pybedtools.BedTool(Homo_cov_bed_fh)\n",
    "    gene_ids_ph_p_homo = []\n",
    "    for x in geneGff_bed.intersect(homo_p_bed, f=0.4):\n",
    "        y = col_8_id(x[8])\n",
    "        gene_ids_ph_p_homo.append(y)\n",
    "    gene_ids_ph_p_homo = set(gene_ids_ph_p_homo)\n",
    "    return gene_ids_ph_p_homo\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.811802Z",
     "start_time": "2018-02-26T13:06:13.795533Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assign_unphased(queries, gff_fh=anno_gff_p_fh, homo_bed_fh=homo_bed_fh):\n",
    "    \"\"\"\n",
    "    Takes a series of gene models and checks if they overlap with the\n",
    "    homo_bed dataframe. The minimum overlap is f=0.4 with the intersect function.\n",
    "    \n",
    "    TO DO: For now the homo_bed only includes primary gene models. Should be changed.\n",
    "    Input:  \n",
    "        queries, pd.Series of gene names. \n",
    "        gff_fh, file handle of gff file.\n",
    "        homo_bed_dfh, file handle of bed file that represents non-phased regions.\n",
    "    \n",
    "    \"\"\"\n",
    "    if '.gene' not in gff_fh: \n",
    "        _ = allgffTogenegff(gff_fh)\n",
    "    Unphasedgenes = geneUnphased(gff_fh.replace('anno', 'gene'), homo_bed_fh)\n",
    "    unphased_s = pd.Series([False]*len(queries), index=queries)\n",
    "    unphased_s.loc[unphased_s.index.isin(Unphasedgenes)] = True\n",
    "    #print(len(Unphasedgenes))\n",
    "    return unphased_s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.821390Z",
     "start_time": "2018-02-26T13:06:13.814399Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def getFastaDict(fastaFile):\n",
    "    d = {}\n",
    "    for gene in SeqIO.parse(fastaFile, 'fasta'):\n",
    "        d[gene.id] = gene\n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.879906Z",
     "start_time": "2018-02-26T13:06:13.823766Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def writeAllelicFasta(alleleOne, alleleTwo, alleleType, outPath):\n",
    "    '''writes fasta file containing fasta information for two alleles\n",
    "    in the outPath'''\n",
    "    assert(alleleType.upper() in ['CDS', 'GENE', 'PROTEIN'])\n",
    "    \n",
    "    seqRecordDict = globals()['SEQRECORD_' + alleleType.upper() + '_DICT']\n",
    "    try:\n",
    "        alleleSeqRecords = [seqRecordDict[alleleOne], seqRecordDict[alleleTwo]]\n",
    "    except KeyError:\n",
    "        print(alleleOne)\n",
    "        print(alleleTwo)\n",
    "        print(alleleType)\n",
    "        sys.exit()\n",
    "    with open(os.path.join(outPath, alleleType.lower() + '.fa'), 'w') as outFile:\n",
    "        SeqIO.write(alleleSeqRecords, outFile, 'fasta')\n",
    "    return True\n",
    "\n",
    "def writeAlignmentScript(alleleOutPath, scriptLoc = os.path.join(PAML_PATH, 'paml_script.sh')):\n",
    "    with open(scriptLoc, 'a') as outFile:\n",
    "        print('cd %s' % alleleOutPath, file=outFile)\n",
    "        print('/home/gamran/anaconda3/muscle3.8.31_i86linux64 -clwstrict -in protein.fa -out protein.aln', file=outFile)\n",
    "        print('perl /home/gamran/anaconda3/pal2nal.v14/pal2nal.pl -output paml protein.aln cds.fa > cds_codon.aln', file=outFile)\n",
    "        print('perl /home/gamran/anaconda3/pal2nal.v14/pal2nal.pl protein.aln cds.fa > cds_codon.clustal', file=outFile)\n",
    "        print('cp %s/yn00.ctl ./' % PAML_PATH, file=outFile)\n",
    "        print('/home/gamran/anaconda3/paml4.9g/bin/yn00', file=outFile)\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:13.911216Z",
     "start_time": "2018-02-26T13:06:13.882673Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def prepareAlignmentBashScript(scriptLoc = os.path.join(PAML_PATH, 'paml_script.sh')):\n",
    "    with open(scriptLoc, 'w') as pamlScript:\n",
    "        print('#!/bin/bash', file=pamlScript)\n",
    "\n",
    "    for index, [Query, Target] in alleleDf.iloc[:, :2].iterrows():\n",
    "        #if we don't have a blast hit skip.\n",
    "        if pd.isnull(Target):\n",
    "            continue\n",
    "        else:\n",
    "            alleleOutPath = os.path.join(PAML_PATH, '%s_%s' % (Query, Target))\n",
    "            if not os.path.exists(alleleOutPath):\n",
    "                os.mkdir(os.path.join(PAML_PATH, '%s_%s' % (Query, Target)))\n",
    "\n",
    "            writeAllelicFasta(Query, Target, 'CDS', alleleOutPath)\n",
    "            writeAllelicFasta(Query, Target, 'PROTEIN', alleleOutPath)\n",
    "\n",
    "            writeAlignmentScript(alleleOutPath, os.path.join(PAML_PATH, 'paml_script.sh'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.063321Z",
     "start_time": "2018-02-26T13:06:13.914458Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def assignDistancesToAlleles(folder, alignmentFile, alleleType):\n",
    "    '''Adds Hamming and Levenshtein distance columns to an allele pair\n",
    "    (indexed by 'folder' name) in df'''\n",
    "    #print(folder)\n",
    "    if pd.isnull(folder):\n",
    "        return np.nan, np.nan\n",
    "    assert(alleleType.upper() in ['PROTEIN', 'CDS', 'GENE'])\n",
    "    seq1, seq2 = AlignIO.read(open(alignmentFile, 'r'), format='clustal', seq_count=2)\n",
    "    seq1 = str(seq1.seq).upper()\n",
    "    seq2 = str(seq2.seq).upper()\n",
    "    assert(len(seq1) == len(seq2))\n",
    "    return editdistance.eval(seq1, seq2)/len(seq1), distance.hamming(seq1, seq2, normalized=True)\n",
    "\n",
    "def assignDistancesToAllAlleles(df_folder_index, all_folders, tmp_path, suffix):\n",
    "    \"\"\"\n",
    "    Reads in the index that contains the folder pairings for the alignements.\n",
    "    Returns a protein_df and CDS_df that contain the hamming and levenshtein distance each.\n",
    "    \"\"\"\n",
    "    cleaned_index = [x for x in df_folder_index if x in all_folders]\n",
    "    count = 0\n",
    "    total = len(df_folder_index)\n",
    "    percentDone = 0\n",
    "    protein_lev_dict = {}\n",
    "    protein_ham_dict = {}\n",
    "    CDS_lev_dict = {}\n",
    "    CDS_ham_dict = {}\n",
    "    \n",
    "    #print(\"Calculating distances and adding them to the allele DataFrame...\")\n",
    "    \n",
    "    for folder in cleaned_index:\n",
    "        if pd.isnull(folder):\n",
    "            proteinAlignmentFile = ''\n",
    "            cdsAlignmentFile = ''\n",
    "        else:\n",
    "            proteinAlignmentFile = os.path.join(PAML_PATH, folder, 'protein.aln')\n",
    "            cdsAlignmentFile = os.path.join(PAML_PATH, folder, 'cds_codon.clustal')\n",
    "        #here the nan get overwritten. This doesn't matter though as they are all\n",
    "        #nan anyway.\n",
    "        protein_lev_dict[folder], protein_ham_dict[folder]  = \\\n",
    "        assignDistancesToAlleles(folder, proteinAlignmentFile, 'PROTEIN')\n",
    "        CDS_lev_dict[folder], CDS_ham_dict[folder]  = \\\n",
    "        assignDistancesToAlleles(folder, cdsAlignmentFile, 'CDS')\n",
    "\n",
    "        count += 1\n",
    "        #if round(count/total * 100) > percentDone:\n",
    "            #percentDone = round(count/total * 100)\n",
    "            #print(\"%s%% complete\" % percentDone)\n",
    "            \n",
    "    newdf_columns=['protein_hamming', 'protein_levenshtein', 'cds_hamming',\n",
    "       'cds_levenshtein']\n",
    "    if len(protein_ham_dict) > 0:\n",
    "        df = pd.DataFrame([protein_ham_dict,protein_lev_dict,CDS_ham_dict,CDS_lev_dict]).T\n",
    "        df.rename(columns=dict(zip(df.columns,newdf_columns)),inplace=True)\n",
    "        out_name = os.path.join(tmp_path, '%s_%s.%s' % (df.index[0],df.index[-1],suffix))\n",
    "        df.round(4).to_csv(out_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.255606Z",
     "start_time": "2018-02-26T13:06:14.071708Z"
    }
   },
   "outputs": [],
   "source": [
    "def parse_dNdS_to_df(line, folder):\n",
    "    \"\"\"\n",
    "    Function that parses out dN and dS of a yn00 file and calls the \n",
    "    assign_dNdS function. Therefore returns a single element\n",
    "    pd.Series with folder name as index.\n",
    "    \"\"\"\n",
    "    dN = re.findall(r'dN = [-| ]?(.*) w', line)[0]\n",
    "    dS = re.findall(r'dS = [-| ]?(.*) dN', line)[0]\n",
    "    return assign_dNdS(dN, dS, folder)\n",
    "\n",
    "def assign_dNdS(dN, dS, folder):\n",
    "    '''\n",
    "    Function that cacluates the dN/dS ratio an returns it as a series usind the folder as index.\n",
    "    Input: dN, dS, folder(name)\n",
    "    Output: single element pd.Series with folder name as index.\n",
    "    '''\n",
    "    if float(dS) > 0:\n",
    "        series = pd.Series([float(dN)/float(dS)], index=[folder])\n",
    "    else:\n",
    "        series = pd.Series([np.nan], index=[folder])\n",
    "    return series\n",
    "\n",
    "def assign_dNdS_to_all_alleles(folder_index, all_folders, tmp_path, suffix):\n",
    "    \"\"\"Function that parses out the different dN/dS ratios from a yn.out file for a list/index\n",
    "    of folders that contain these yn.out files. The output dataframe is saved to tmp folder using\n",
    "    the suffix.\n",
    "    Input:  folder_index, list or index where to find the yn.out for each pairing.\n",
    "            all_folders, are all possible folders. This is used to filter out nan and so from\n",
    "                paralizing etc.\n",
    "            tmp_path, is the path were \n",
    "    Output: Saved out tmp df with the suffix as file ending.\n",
    "    \n",
    "    \"\"\"\n",
    "    cleaned_index = [x for x in folder_index if x in all_folders]\n",
    "    #print(cleaned_index)\n",
    "    yn00_s = pd.Series([], name='yn00_dN/dS')\n",
    "    LWL85_s = pd.Series([], name='LWL85_dN/dS')\n",
    "    LWL85m_s = pd.Series([], name='LWL85m_dN/dS')\n",
    "    LPB93_s = pd.Series([], name='LPB93_dN/dS')\n",
    "    #header = ['folder','yn00_dN/dS', 'LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS']\n",
    "    #append these list\n",
    "    for folder in cleaned_index:\n",
    "        alleleYn = os.path.join(PAML_PATH, folder,'yn.out')\n",
    "        with open(alleleYn, 'r') as ynOut:\n",
    "            #now loop over the lines and parse out stuff\n",
    "            for i, line in enumerate(ynOut):\n",
    "                if line.startswith('seq. seq. ') and i > 0:\n",
    "                    next(ynOut) # we want the line that is two after the line starting with 'seq. seq '\n",
    "                    dataLine = next(ynOut)\n",
    "                    dN = dataLine.split('+-')[0].rstrip().split(' ')[-1]\n",
    "                    dS = dataLine.split('+-')[1].rstrip().split(' ')[-1]\n",
    "                    yn00_s = yn00_s.append(assign_dNdS(dN, dS, folder))\n",
    "                elif line.startswith('LWL85:') and 'nan' not in line:\n",
    "                    LWL85_s = LWL85_s.append(parse_dNdS_to_df(line, folder))\n",
    "                elif line.startswith('LWL85m:') and 'nan' not in line:\n",
    "                    LWL85m_s= LWL85m_s.append(parse_dNdS_to_df(line, folder))\n",
    "                elif line.startswith('LPB93:') and 'nan' not in line:\n",
    "                    LPB93_s =LPB93_s.append(parse_dNdS_to_df(line, folder))\n",
    "                else:\n",
    "                    continue\n",
    "    out_df = pd.concat([yn00_s.round(4),LWL85_s.round(4),\\\n",
    "                        LWL85m_s.round(4), LPB93_s.round(4)],axis =1)\n",
    "    new_columns = ['yn00_dN/dS','LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS' ]\n",
    "    if len(out_df) > 0:\n",
    "        out_df.rename(columns=dict(zip(out_df.columns, new_columns)), inplace=True)\n",
    "        out_name = os.path.join(tmp_path, '%s_%s.%s' % (out_df.index[0],out_df.index[-1],suffix))\n",
    "        out_df.to_csv(out_name, sep='\\t')\n",
    "    #return out_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.287985Z",
     "start_time": "2018-02-26T13:06:14.260411Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def checkPamlFilesExist(alleleDf):\n",
    "    '''loops through all folder names in alleleDf.index to check if their PAML files have\n",
    "    all been generated in those folders. refDict is based on the contents of a folder\n",
    "    that was known to be run successfully.'''\n",
    "    refDict = {'aln': 2,\n",
    "     'clustal': 1,\n",
    "     'ctl': 1,\n",
    "     'dN': 1,\n",
    "     'dS': 1,\n",
    "     'fa': 2,\n",
    "     'out': 1,\n",
    "     'rst': 1,\n",
    "     'rst1': 1,\n",
    "     'rub': 1,\n",
    "     't': 1}\n",
    "    for file in (x for x in alleleDf.index if not pd.isnull(x)):\n",
    "        if not os.path.exists(os.path.join(PAML_PATH, file)):\n",
    "            return False\n",
    "        discrepancies = getDiscrepancies(os.path.join(PAML_PATH, file), refDict)\n",
    "        if discrepancies != '':\n",
    "            print(discrepancies)\n",
    "            return False\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:06:14.314801Z",
     "start_time": "2018-02-26T13:06:14.290615Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def grouper(iterable, n, fillvalue=None):\n",
    "    \"Collect data into fixed-length chunks or blocks\"\n",
    "    # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx\"\n",
    "    args = [iter(iterable)] * n\n",
    "    return it.zip_longest(*args, fillvalue=fillvalue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:27:17.161541Z",
     "start_time": "2018-02-26T13:27:17.132641Z"
    }
   },
   "outputs": [],
   "source": [
    "def combineTmpToDf(header, suffix, tmp_path, clean=True):\n",
    "    \"\"\"Combines the files of a temporary folder into a dataframe based on the tmp files\n",
    "    suffix. Returns combined dataframe. And cleans up if needed.\"\"\"\n",
    "    tmp_assigneddfs_fh = [os.path.join(tmp_path, file) for file in os.listdir(tmp_path)\\\n",
    "                         if file.endswith(suffix) ]\n",
    "    print(len(tmp_assigneddfs_fh))\n",
    "    tmp_df = pd.DataFrame(columns=header)\n",
    "    for df_fh in tmp_assigneddfs_fh:\n",
    "        tmp_df = pd.concat([tmp_df, pd.read_csv(df_fh, index_col = 0, sep='\\t')])\n",
    "    if clean == True:\n",
    "        #now clean up again\n",
    "        for file in tmp_assigneddfs_fh:\n",
    "            os.remove(file)\n",
    "    return tmp_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:53.323820Z",
     "start_time": "2018-02-26T13:25:51.688067Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hFullAlleleDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.full_df.alleles' % H_GENOME), header=0, sep='\\t')\n",
    "hFullAlleleDf['matchType'] = pd.Series(index=hFullAlleleDf.index)\n",
    "hFullAlleleDf['matchType'] = hFullAlleleDf.apply(lambda row: assignMatchType(row['allele_source'], row['t_contig == h_contig_overlap'], row['q_contig == t_contig']), axis=1)\n",
    "pFullAlleleDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.full_df.alleles' % P_GENOME), header=0, sep='\\t')\n",
    "pFullAlleleDf['matchType'] = pd.Series(index=pFullAlleleDf.index)\n",
    "pFullAlleleDf['matchType'] = pFullAlleleDf.apply(lambda row: assignMatchType(row['allele_source'], row['t_contig == h_contig_overlap'], row['q_contig == t_contig']), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:53.774191Z",
     "start_time": "2018-02-26T13:25:53.326398Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# filter out haplotig proteins that already have alleles identified by BLAST or proteinortho.\n",
    "hFullAlleleDf = hFullAlleleDf[(~hFullAlleleDf['Query'].isin(pFullAlleleDf['Target']))]\n",
    "pFullAlleleDf['aQuery'] = pFullAlleleDf['Query']\n",
    "pFullAlleleDf['aTarget'] = pFullAlleleDf['Target']\n",
    "hFullAlleleDf['aQuery'] = hFullAlleleDf['Target']\n",
    "hFullAlleleDf['aTarget'] = hFullAlleleDf['Query']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:53.799101Z",
     "start_time": "2018-02-26T13:25:53.776713Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "phFullAlleleDf = pFullAlleleDf.append(hFullAlleleDf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:25:57.635871Z",
     "start_time": "2018-02-26T13:25:54.686464Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "SEQRECORD_PROTEIN_DICT = getFastaDict(PH_PROTEIN_FASTA)\n",
    "SEQRECORD_GENE_DICT = getFastaDict(PH_GENE_FASTA)\n",
    "SEQRECORD_CDS_DICT = getFastaDict(PH_CDS_FASTA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:46:54.038594Z",
     "start_time": "2018-02-26T13:46:54.008517Z"
    }
   },
   "outputs": [],
   "source": [
    "alleleDf = phFullAlleleDf.copy()\n",
    "alleleDf['folder'] = alleleDf.Query + '_' + alleleDf.Target\n",
    "alleleDf.set_index('folder', inplace=True)\n",
    "# assert(len(alleleDf) == len(overlapDf) + len(noOverlapDf) + len(diffContigDf) + len(manualAssignDf))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:35:19.297460Z",
     "start_time": "2018-02-26T13:35:19.242464Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>q_contig</th>\n",
       "      <th>t_contig</th>\n",
       "      <th>q_contig == t_contig</th>\n",
       "      <th>p_protein</th>\n",
       "      <th>h_contig_overlap</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "      <th>allele_source</th>\n",
       "      <th>matchType</th>\n",
       "      <th>aQuery</th>\n",
       "      <th>aTarget</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>folder</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.100_evm.model.hcontig_072_002.3</th>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>25.00</td>\n",
       "      <td>240.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>330.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_072_002</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>['hcontig_000_058']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.1001_evm.model.hcontig_029_007.1</th>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>70.27</td>\n",
       "      <td>37.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_029_007</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.1002_evm.model.hcontig_003_006.10</th>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>96.88</td>\n",
       "      <td>128.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.pcontig_000.1003_evm.model.hcontig_003_006.9</th>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>96.95</td>\n",
       "      <td>164.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         Query  \\\n",
       "folder                                                                           \n",
       "NaN                                                   evm.model.pcontig_000.10   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.pcontig_000.100   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  evm.model.pcontig_000.1001   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.pcontig_000.1002   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  evm.model.pcontig_000.1003   \n",
       "\n",
       "                                                                          Target  \\\n",
       "folder                                                                             \n",
       "NaN                                                                          NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.hcontig_072_002.3   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...   evm.model.hcontig_029_007.1   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.hcontig_003_006.10   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...   evm.model.hcontig_003_006.9   \n",
       "\n",
       "                                                    PctID  AlnLgth  NumMis  \\\n",
       "folder                                                                       \n",
       "NaN                                                   NaN      NaN     NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  25.00    240.0   155.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  70.27     37.0    11.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  96.88    128.0     4.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  96.95    164.0     5.0   \n",
       "\n",
       "                                                    NumGap  StartQuery  \\\n",
       "folder                                                                   \n",
       "NaN                                                    NaN         NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...     8.0       110.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...     0.0       130.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...     0.0         1.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...     0.0         1.0   \n",
       "\n",
       "                                                    StopQuery  StartTarget  \\\n",
       "folder                                                                       \n",
       "NaN                                                       NaN          NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...      330.0        125.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...      166.0         52.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...      128.0         83.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...      164.0        345.0   \n",
       "\n",
       "                                                    StopTarget  \\\n",
       "folder                                                           \n",
       "NaN                                                        NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...       358.0   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...        88.0   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...       210.0   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...       508.0   \n",
       "\n",
       "                                                                ...               \\\n",
       "folder                                                          ...                \n",
       "NaN                                                             ...                \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...              ...                \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...              ...                \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...              ...                \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...              ...                \n",
       "\n",
       "                                                       q_contig  \\\n",
       "folder                                                            \n",
       "NaN                                                 pcontig_000   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  pcontig_000   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  pcontig_000   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  pcontig_000   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  pcontig_000   \n",
       "\n",
       "                                                           t_contig  \\\n",
       "folder                                                                \n",
       "NaN                                                           False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  hcontig_072_002   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  hcontig_029_007   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  hcontig_003_006   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  hcontig_003_006   \n",
       "\n",
       "                                                    q_contig == t_contig  \\\n",
       "folder                                                                     \n",
       "NaN                                                                False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...                 False   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...                 False   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...                 False   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...                 False   \n",
       "\n",
       "                                                                     p_protein  \\\n",
       "folder                                                                           \n",
       "NaN                                                                        NaN   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.pcontig_000.100   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  evm.model.pcontig_000.1001   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.pcontig_000.1002   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  evm.model.pcontig_000.1003   \n",
       "\n",
       "                                                       h_contig_overlap  \\\n",
       "folder                                                                    \n",
       "NaN                                                               False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...  ['hcontig_000_058']   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  ['hcontig_000_078']   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  ['hcontig_000_078']   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  ['hcontig_000_078']   \n",
       "\n",
       "                                                    t_contig == h_contig_overlap  \\\n",
       "folder                                                                             \n",
       "NaN                                                                        False   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...                         False   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...                         False   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...                         False   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...                         False   \n",
       "\n",
       "                                                   allele_source matchType  \\\n",
       "folder                                                                       \n",
       "NaN                                                        BLAST     BLAST   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...         BLAST     BLAST   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...         BLAST     BLAST   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...         BLAST     BLAST   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...         BLAST     BLAST   \n",
       "\n",
       "                                                                        aQuery  \\\n",
       "folder                                                                           \n",
       "NaN                                                   evm.model.pcontig_000.10   \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.pcontig_000.100   \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...  evm.model.pcontig_000.1001   \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.pcontig_000.1002   \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...  evm.model.pcontig_000.1003   \n",
       "\n",
       "                                                                         aTarget  \n",
       "folder                                                                            \n",
       "NaN                                                                          NaN  \n",
       "evm.model.pcontig_000.100_evm.model.hcontig_072...   evm.model.hcontig_072_002.3  \n",
       "evm.model.pcontig_000.1001_evm.model.hcontig_02...   evm.model.hcontig_029_007.1  \n",
       "evm.model.pcontig_000.1002_evm.model.hcontig_00...  evm.model.hcontig_003_006.10  \n",
       "evm.model.pcontig_000.1003_evm.model.hcontig_00...   evm.model.hcontig_003_006.9  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:36:11.484438Z",
     "start_time": "2018-02-26T13:36:11.367440Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "os.chdir('/home/gamran/genome_analysis/Warrior/Richard/scripts')\n",
    "%run file_counting.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T14:03:58.993140Z",
     "start_time": "2018-02-26T14:03:58.706643Z"
    }
   },
   "outputs": [],
   "source": [
    "def main(alleleDf = alleleDf):\n",
    "    prepareAlignmentBashScript(os.path.join(PAML_PATH, 'paml_script.sh'))\n",
    "    \n",
    "    # if already run before, comment out this line\n",
    "    print(\"Checking whether all PAML files already exist in %s...\" % PAML_PATH)\n",
    "    if checkPamlFilesExist(alleleDf):\n",
    "        print('PAML appears to have been run to completion previously. Therefore, it will not be run this time.')\n",
    "    else:\n",
    "        'Not all files generated by PAML appear to exist. Running PAML now (this may take some time)...'\n",
    "        !bash {os.path.join(PAML_PATH, 'paml_script.sh')}\n",
    "        print('PAML finished running.')\n",
    "\n",
    "    analysedAllelesPath = os.path.join(BASE_OUT_PATH, GENOME+'_analysed_alleles.df')    \n",
    "    alleleDf.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    #dataframe where the index is not 'NaN'\n",
    "    noNANdf = alleleDf.loc[~alleleDf.index.isnull(),:].copy()\n",
    "    all_folders = noNANdf.index\n",
    "    #generate a tmp folder for the parallized analysis\n",
    "    tmp_path = os.path.join(BASE_OUT_PATH, 'tmp')\n",
    "    if not os.path.exists(tmp_path):\n",
    "        os.mkdir(tmp_path)\n",
    "    #assign the distances\n",
    "    dist_suffix = 'distdf_tmp'\n",
    "    #do parallized analysi\n",
    "    Parallel(n_jobs=threads)(delayed(assignDistancesToAllAlleles)(list(folder_index_list),all_folders,tmp_path, dist_suffix)\\\n",
    "                       for folder_index_list in grouper(noNANdf.index, 100, np.nan))\n",
    "    distdf_header = ['protein_hamming', 'protein_levenshtein', 'cds_hamming',\n",
    "       'cds_levenshtein']\n",
    "    distdf = combineTmpToDf(distdf_header, dist_suffix, tmp_path, clean=False)\n",
    "    distdf['Index'] = distdf.index\n",
    "    noNANdf['Index'] = noNANdf.index\n",
    "    tmp_df = pd.merge(noNANdf, distdf,how='inner')\n",
    "    tmp_df.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    \n",
    "    print(\"Done with caculate pairwise alignment distances.\")\n",
    "    #pd.util.testing.assert_frame_equal(alleleDf, pd.read_csv(analysedAllelesPath, sep='\\t', index_col=0))\n",
    "    print(\"Starting to parse dN/dS ratios from yn.out file.\")\n",
    "    dNdS_suffix = 'dNdSdf_tmp'\n",
    "    #now assign the dNdS ratios\n",
    "    Parallel(n_jobs=threads)(delayed(assign_dNdS_to_all_alleles)(list(folder_index_list),all_folders,tmp_path, dNdS_suffix)\\\n",
    "                       for folder_index_list in grouper(noNANdf.index, 100, np.nan))\n",
    "    dNdS_header = ['yn00_dN/dS','LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS' ]\n",
    "    dNdSdf = combineTmpToDf(dNdS_header, dNdS_suffix, tmp_path, clean=False)\n",
    "    dNdSdf['Index'] = dNdSdf.index\n",
    "    tmp_df = pd.merge(tmp_df, dNdSdf,how='inner')\n",
    "    tmp_df.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    print('Finish to parse dN/dS ratios.')\n",
    "    print('Combining Dataframe of potential allele pairs with Dataframe of blast none hits.')\n",
    "    #now pull together the dataframes of hits and no-hits blast again.\n",
    "    pre_col_order = tmp_df.columns\n",
    "    allele_df = pd.concat([tmp_df, alleleDf.loc[alleleDf.index.isnull(), :]], axis=0, ignore_index=True)\n",
    "    allele_df = allele_df.loc[pre_col_order, :]\n",
    "    \n",
    "    print(\"Assigning the unphased genes.\")\n",
    "    unphased_s = assign_unphased(alleleDf.Query)\n",
    "    assert(any(unphased_s.index == alleleDf.Query))\n",
    "    alleleDf['unphased'] = unphased_s.values\n",
    "    \n",
    "    print(\"Assining the proteinortho single genes.\")\n",
    "    PO_single_s = assign_PO_single(alleleDf.Query)\n",
    "    assert(any(PO_single_s.index == alleleDf.Query))\n",
    "    alleleDf['PO_single'] = PO_single_s.values\n",
    "    \n",
    "    print(\"Assining the proteinortho inter haplome paralogoues by comparing\\\n",
    "    proteinortho (po) and its synteny mode output (poff)\")\n",
    "    alleleDf['aTarget+aQuery'] = alleleDf.aTarget + '+' + alleleDf.aQuery\n",
    "    interhaplotypeparaloges_s = assign_INP(alleleDf['aTarget+aQuery'])\n",
    "    assert(any(interhaplotypeparaloges_s.index == alleleDf['aTarget+aQuery'] ))\n",
    "    alleleDf['PO_interhaplotype_paralogs'] = interhaplotypeparaloges_s.values\n",
    "    alleleDf.drop('aTarget+aQuery', axis=1, inplace=True)\n",
    "    alleleDf.reset_index(drop=True, inplace=True)\n",
    "    alleleDf.to_csv(analysedAllelesPath, sep='\\t')\n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-26T14:04:02.313Z"
    }
   },
   "outputs": [],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    alleleDf_test = main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:55:59.575121Z",
     "start_time": "2018-02-26T13:55:58.911891Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "203\n"
     ]
    }
   ],
   "source": [
    "alleleDf = phFullAlleleDf.copy()\n",
    "alleleDf['folder'] = alleleDf.Query + '_' + alleleDf.Target\n",
    "alleleDf.set_index('folder', inplace=True)\n",
    "noNANdf = alleleDf.loc[~alleleDf.index.isnull(),:].copy()\n",
    "all_folders = noNANdf.index\n",
    "#generate a tmp folder for the parallized analysis\n",
    "tmp_path = os.path.join(BASE_OUT_PATH, 'tmp')\n",
    "dist_suffix = 'distdf_tmp'\n",
    "distdf_header = ['protein_hamming', 'protein_levenshtein', 'cds_hamming',\n",
    "       'cds_levenshtein']\n",
    "distdf = combineTmpToDf(distdf_header, dist_suffix, tmp_path, clean=False)\n",
    "distdf['Index'] = distdf.index\n",
    "noNANdf['Index'] = noNANdf.index\n",
    "tmp_df = pd.merge(noNANdf, distdf,how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:50:08.696257Z",
     "start_time": "2018-02-26T13:50:08.017847Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "203\n"
     ]
    }
   ],
   "source": [
    "dNdS_suffix = 'dNdSdf_tmp'\n",
    "dNdS_header = ['yn00_dN/dS','LWL85_dN/dS','LWL85m_dN/dS','LPB93_dN/dS' ]\n",
    "dNdSdf = combineTmpToDf(dNdS_header, dNdS_suffix, tmp_path, clean=False)\n",
    "dNdSdf['Index'] = dNdSdf.index\n",
    "tmp_df = pd.merge(tmp_df, dNdSdf,how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:57:48.301166Z",
     "start_time": "2018-02-26T13:57:48.263076Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "      <th>allele_source</th>\n",
       "      <th>matchType</th>\n",
       "      <th>aQuery</th>\n",
       "      <th>aTarget</th>\n",
       "      <th>Index</th>\n",
       "      <th>protein_hamming</th>\n",
       "      <th>protein_levenshtein</th>\n",
       "      <th>cds_hamming</th>\n",
       "      <th>cds_levenshtein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>25.00</td>\n",
       "      <td>240.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>330.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>evm.model.pcontig_000.100_evm.model.hcontig_07...</td>\n",
       "      <td>0.8363</td>\n",
       "      <td>0.8183</td>\n",
       "      <td>0.7138</td>\n",
       "      <td>0.6213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>70.27</td>\n",
       "      <td>37.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>evm.model.pcontig_000.1001_evm.model.hcontig_0...</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.6979</td>\n",
       "      <td>0.6550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>96.88</td>\n",
       "      <td>128.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>evm.model.pcontig_000.1002_evm.model.hcontig_0...</td>\n",
       "      <td>0.4115</td>\n",
       "      <td>0.4115</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.4035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>96.95</td>\n",
       "      <td>164.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>evm.model.pcontig_000.1003_evm.model.hcontig_0...</td>\n",
       "      <td>0.6884</td>\n",
       "      <td>0.6884</td>\n",
       "      <td>0.6838</td>\n",
       "      <td>0.6838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>evm.model.pcontig_000.1005</td>\n",
       "      <td>evm.model.hcontig_000_078.13</td>\n",
       "      <td>39.13</td>\n",
       "      <td>92.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1005</td>\n",
       "      <td>evm.model.hcontig_000_078.13</td>\n",
       "      <td>evm.model.pcontig_000.1005_evm.model.hcontig_0...</td>\n",
       "      <td>0.7097</td>\n",
       "      <td>0.7016</td>\n",
       "      <td>0.6344</td>\n",
       "      <td>0.6022</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Query                        Target  PctID  AlnLgth  \\\n",
       "0   evm.model.pcontig_000.100   evm.model.hcontig_072_002.3  25.00    240.0   \n",
       "1  evm.model.pcontig_000.1001   evm.model.hcontig_029_007.1  70.27     37.0   \n",
       "2  evm.model.pcontig_000.1002  evm.model.hcontig_003_006.10  96.88    128.0   \n",
       "3  evm.model.pcontig_000.1003   evm.model.hcontig_003_006.9  96.95    164.0   \n",
       "4  evm.model.pcontig_000.1005  evm.model.hcontig_000_078.13  39.13     92.0   \n",
       "\n",
       "   NumMis  NumGap  StartQuery  StopQuery  StartTarget  StopTarget  \\\n",
       "0   155.0     8.0       110.0      330.0        125.0       358.0   \n",
       "1    11.0     0.0       130.0      166.0         52.0        88.0   \n",
       "2     4.0     0.0         1.0      128.0         83.0       210.0   \n",
       "3     5.0     0.0         1.0      164.0        345.0       508.0   \n",
       "4    56.0     0.0        25.0      116.0          1.0        92.0   \n",
       "\n",
       "        ...        t_contig == h_contig_overlap  allele_source  matchType  \\\n",
       "0       ...                               False          BLAST      BLAST   \n",
       "1       ...                               False          BLAST      BLAST   \n",
       "2       ...                               False          BLAST      BLAST   \n",
       "3       ...                               False          BLAST      BLAST   \n",
       "4       ...                                True          BLAST      BLAST   \n",
       "\n",
       "                       aQuery                       aTarget  \\\n",
       "0   evm.model.pcontig_000.100   evm.model.hcontig_072_002.3   \n",
       "1  evm.model.pcontig_000.1001   evm.model.hcontig_029_007.1   \n",
       "2  evm.model.pcontig_000.1002  evm.model.hcontig_003_006.10   \n",
       "3  evm.model.pcontig_000.1003   evm.model.hcontig_003_006.9   \n",
       "4  evm.model.pcontig_000.1005  evm.model.hcontig_000_078.13   \n",
       "\n",
       "                                               Index protein_hamming  \\\n",
       "0  evm.model.pcontig_000.100_evm.model.hcontig_07...          0.8363   \n",
       "1  evm.model.pcontig_000.1001_evm.model.hcontig_0...          0.7778   \n",
       "2  evm.model.pcontig_000.1002_evm.model.hcontig_0...          0.4115   \n",
       "3  evm.model.pcontig_000.1003_evm.model.hcontig_0...          0.6884   \n",
       "4  evm.model.pcontig_000.1005_evm.model.hcontig_0...          0.7097   \n",
       "\n",
       "  protein_levenshtein cds_hamming cds_levenshtein  \n",
       "0              0.8183      0.7138          0.6213  \n",
       "1              0.7778      0.6979          0.6550  \n",
       "2              0.4115      0.4035          0.4035  \n",
       "3              0.6884      0.6838          0.6838  \n",
       "4              0.7016      0.6344          0.6022  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:51:36.410517Z",
     "start_time": "2018-02-26T13:51:36.371681Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>q_contig</th>\n",
       "      <th>t_contig</th>\n",
       "      <th>q_contig == t_contig</th>\n",
       "      <th>p_protein</th>\n",
       "      <th>h_contig_overlap</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "      <th>allele_source</th>\n",
       "      <th>matchType</th>\n",
       "      <th>aQuery</th>\n",
       "      <th>aTarget</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>folder</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.1004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1004</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1004</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.1025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1025</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1025</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.1080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1080</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>evm.model.pcontig_000.1081</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1081</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Query Target  PctID  AlnLgth  NumMis  NumGap  \\\n",
       "folder                                                                      \n",
       "NaN       evm.model.pcontig_000.10    NaN    NaN      NaN     NaN     NaN   \n",
       "NaN     evm.model.pcontig_000.1004    NaN    NaN      NaN     NaN     NaN   \n",
       "NaN     evm.model.pcontig_000.1025    NaN    NaN      NaN     NaN     NaN   \n",
       "NaN     evm.model.pcontig_000.1080    NaN    NaN      NaN     NaN     NaN   \n",
       "NaN     evm.model.pcontig_000.1081    NaN    NaN      NaN     NaN     NaN   \n",
       "\n",
       "        StartQuery  StopQuery  StartTarget  StopTarget   ...       q_contig  \\\n",
       "folder                                                   ...                  \n",
       "NaN            NaN        NaN          NaN         NaN   ...    pcontig_000   \n",
       "NaN            NaN        NaN          NaN         NaN   ...    pcontig_000   \n",
       "NaN            NaN        NaN          NaN         NaN   ...    pcontig_000   \n",
       "NaN            NaN        NaN          NaN         NaN   ...    pcontig_000   \n",
       "NaN            NaN        NaN          NaN         NaN   ...    pcontig_000   \n",
       "\n",
       "        t_contig  q_contig == t_contig                   p_protein  \\\n",
       "folder                                                               \n",
       "NaN        False                 False                         NaN   \n",
       "NaN        False                 False  evm.model.pcontig_000.1004   \n",
       "NaN        False                 False  evm.model.pcontig_000.1025   \n",
       "NaN        False                 False                         NaN   \n",
       "NaN        False                 False                         NaN   \n",
       "\n",
       "           h_contig_overlap  t_contig == h_contig_overlap allele_source  \\\n",
       "folder                                                                    \n",
       "NaN                   False                         False         BLAST   \n",
       "NaN     ['hcontig_000_078']                         False         BLAST   \n",
       "NaN     ['hcontig_000_078']                         False         BLAST   \n",
       "NaN                   False                         False         BLAST   \n",
       "NaN                   False                         False         BLAST   \n",
       "\n",
       "       matchType                      aQuery aTarget  \n",
       "folder                                                \n",
       "NaN        BLAST    evm.model.pcontig_000.10     NaN  \n",
       "NaN        BLAST  evm.model.pcontig_000.1004     NaN  \n",
       "NaN        BLAST  evm.model.pcontig_000.1025     NaN  \n",
       "NaN        BLAST  evm.model.pcontig_000.1080     NaN  \n",
       "NaN        BLAST  evm.model.pcontig_000.1081     NaN  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf.loc[alleleDf.index.isnull(), :].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:58:03.396752Z",
     "start_time": "2018-02-26T13:58:03.303191Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>q_contig</th>\n",
       "      <th>t_contig</th>\n",
       "      <th>q_contig == t_contig</th>\n",
       "      <th>p_protein</th>\n",
       "      <th>h_contig_overlap</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "      <th>allele_source</th>\n",
       "      <th>matchType</th>\n",
       "      <th>aQuery</th>\n",
       "      <th>aTarget</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>evm.model.pcontig_000.1004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1004</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1004</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>evm.model.pcontig_000.1025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1025</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1025</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>evm.model.pcontig_000.1080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1080</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>evm.model.pcontig_000.1081</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1081</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>evm.model.pcontig_000.1082</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1082</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>evm.model.pcontig_000.1083</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1083</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>evm.model.pcontig_000.1084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1084</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>evm.model.pcontig_000.1095</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.1095</td>\n",
       "      <td>['hcontig_000_055']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1095</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>evm.model.pcontig_000.122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.122</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>evm.model.pcontig_000.123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.123</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>evm.model.pcontig_000.125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.125</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>evm.model.pcontig_000.158</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.158</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>evm.model.pcontig_000.159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.159</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>evm.model.pcontig_000.176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.176</td>\n",
       "      <td>['hcontig_000_019']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>evm.model.pcontig_000.210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.210</td>\n",
       "      <td>['hcontig_000_019']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.210</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>evm.model.pcontig_000.217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.217</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>evm.model.pcontig_000.219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.219</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>evm.model.pcontig_000.224</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.224</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>evm.model.pcontig_000.289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.289</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>evm.model.pcontig_000.308</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.308</td>\n",
       "      <td>['hcontig_000_067']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.308</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>evm.model.pcontig_000.321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.321</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>evm.model.pcontig_000.323</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.323</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>evm.model.pcontig_000.324</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.324</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>evm.model.pcontig_000.341</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.341</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>evm.model.pcontig_000.342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.342</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>evm.model.pcontig_000.360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.360</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>evm.model.pcontig_000.370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>evm.model.pcontig_000.370</td>\n",
       "      <td>['hcontig_000_002']</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.370</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>evm.model.pcontig_000.371</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.371</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>evm.model.pcontig_000.372</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.372</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2374</th>\n",
       "      <td>evm.model.hcontig_056_011.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_056_011.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2375</th>\n",
       "      <td>evm.model.hcontig_017_016.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_017_016.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2376</th>\n",
       "      <td>evm.model.hcontig_026_005.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_026_005.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377</th>\n",
       "      <td>evm.model.hcontig_000_080.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_000_080.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2378</th>\n",
       "      <td>evm.model.hcontig_039_015.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_039_015.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2379</th>\n",
       "      <td>evm.model.hcontig_011_006.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_011_006.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2380</th>\n",
       "      <td>evm.model.hcontig_048_001.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_048_001.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2381</th>\n",
       "      <td>evm.model.hcontig_005_039.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_005_039.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2382</th>\n",
       "      <td>evm.model.hcontig_056_002.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_056_002.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2383</th>\n",
       "      <td>evm.model.hcontig_001_019.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_001_019.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2384</th>\n",
       "      <td>evm.model.hcontig_047_001.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_047_001.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2385</th>\n",
       "      <td>evm.model.hcontig_001_017.29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_001_017.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2386</th>\n",
       "      <td>evm.model.hcontig_056_006.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_056_006.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2387</th>\n",
       "      <td>evm.model.hcontig_015_017.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_015_017.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2388</th>\n",
       "      <td>evm.model.hcontig_003_003.40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_003_003.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2389</th>\n",
       "      <td>evm.model.hcontig_048_006.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_048_006.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2390</th>\n",
       "      <td>evm.model.hcontig_016_002.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_016_002.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2391</th>\n",
       "      <td>evm.model.hcontig_003_017.28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_003_017.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2392</th>\n",
       "      <td>evm.model.hcontig_085_001.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_085_001.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2393</th>\n",
       "      <td>evm.model.hcontig_004_026.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_004_026.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2394</th>\n",
       "      <td>evm.model.hcontig_039_007.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_039_007.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2395</th>\n",
       "      <td>evm.model.hcontig_002_018.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_002_018.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2396</th>\n",
       "      <td>evm.model.hcontig_056_011.78</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_056_011.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2397</th>\n",
       "      <td>evm.model.hcontig_057_003.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_057_003.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2398</th>\n",
       "      <td>evm.model.hcontig_056_003.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_056_003.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2399</th>\n",
       "      <td>evm.model.hcontig_035_017.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_035_017.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2400</th>\n",
       "      <td>evm.model.hcontig_028_017.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_028_017.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2401</th>\n",
       "      <td>evm.model.hcontig_056_011.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_056_011.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2402</th>\n",
       "      <td>evm.model.hcontig_002_028.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_002_028.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2403</th>\n",
       "      <td>evm.model.hcontig_019_004.52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>evm.model.hcontig_019_004.52</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2404 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Query Target  PctID  AlnLgth  NumMis  NumGap  \\\n",
       "0         evm.model.pcontig_000.10    NaN    NaN      NaN     NaN     NaN   \n",
       "1       evm.model.pcontig_000.1004    NaN    NaN      NaN     NaN     NaN   \n",
       "2       evm.model.pcontig_000.1025    NaN    NaN      NaN     NaN     NaN   \n",
       "3       evm.model.pcontig_000.1080    NaN    NaN      NaN     NaN     NaN   \n",
       "4       evm.model.pcontig_000.1081    NaN    NaN      NaN     NaN     NaN   \n",
       "5       evm.model.pcontig_000.1082    NaN    NaN      NaN     NaN     NaN   \n",
       "6       evm.model.pcontig_000.1083    NaN    NaN      NaN     NaN     NaN   \n",
       "7       evm.model.pcontig_000.1084    NaN    NaN      NaN     NaN     NaN   \n",
       "8       evm.model.pcontig_000.1095    NaN    NaN      NaN     NaN     NaN   \n",
       "9        evm.model.pcontig_000.122    NaN    NaN      NaN     NaN     NaN   \n",
       "10       evm.model.pcontig_000.123    NaN    NaN      NaN     NaN     NaN   \n",
       "11       evm.model.pcontig_000.125    NaN    NaN      NaN     NaN     NaN   \n",
       "12       evm.model.pcontig_000.158    NaN    NaN      NaN     NaN     NaN   \n",
       "13       evm.model.pcontig_000.159    NaN    NaN      NaN     NaN     NaN   \n",
       "14       evm.model.pcontig_000.176    NaN    NaN      NaN     NaN     NaN   \n",
       "15       evm.model.pcontig_000.210    NaN    NaN      NaN     NaN     NaN   \n",
       "16       evm.model.pcontig_000.217    NaN    NaN      NaN     NaN     NaN   \n",
       "17       evm.model.pcontig_000.219    NaN    NaN      NaN     NaN     NaN   \n",
       "18       evm.model.pcontig_000.224    NaN    NaN      NaN     NaN     NaN   \n",
       "19       evm.model.pcontig_000.289    NaN    NaN      NaN     NaN     NaN   \n",
       "20       evm.model.pcontig_000.308    NaN    NaN      NaN     NaN     NaN   \n",
       "21       evm.model.pcontig_000.321    NaN    NaN      NaN     NaN     NaN   \n",
       "22       evm.model.pcontig_000.323    NaN    NaN      NaN     NaN     NaN   \n",
       "23       evm.model.pcontig_000.324    NaN    NaN      NaN     NaN     NaN   \n",
       "24       evm.model.pcontig_000.341    NaN    NaN      NaN     NaN     NaN   \n",
       "25       evm.model.pcontig_000.342    NaN    NaN      NaN     NaN     NaN   \n",
       "26       evm.model.pcontig_000.360    NaN    NaN      NaN     NaN     NaN   \n",
       "27       evm.model.pcontig_000.370    NaN    NaN      NaN     NaN     NaN   \n",
       "28       evm.model.pcontig_000.371    NaN    NaN      NaN     NaN     NaN   \n",
       "29       evm.model.pcontig_000.372    NaN    NaN      NaN     NaN     NaN   \n",
       "...                            ...    ...    ...      ...     ...     ...   \n",
       "2374  evm.model.hcontig_056_011.39    NaN    NaN      NaN     NaN     NaN   \n",
       "2375   evm.model.hcontig_017_016.3    NaN    NaN      NaN     NaN     NaN   \n",
       "2376   evm.model.hcontig_026_005.5    NaN    NaN      NaN     NaN     NaN   \n",
       "2377   evm.model.hcontig_000_080.3    NaN    NaN      NaN     NaN     NaN   \n",
       "2378   evm.model.hcontig_039_015.7    NaN    NaN      NaN     NaN     NaN   \n",
       "2379   evm.model.hcontig_011_006.2    NaN    NaN      NaN     NaN     NaN   \n",
       "2380   evm.model.hcontig_048_001.3    NaN    NaN      NaN     NaN     NaN   \n",
       "2381   evm.model.hcontig_005_039.8    NaN    NaN      NaN     NaN     NaN   \n",
       "2382   evm.model.hcontig_056_002.6    NaN    NaN      NaN     NaN     NaN   \n",
       "2383  evm.model.hcontig_001_019.39    NaN    NaN      NaN     NaN     NaN   \n",
       "2384   evm.model.hcontig_047_001.4    NaN    NaN      NaN     NaN     NaN   \n",
       "2385  evm.model.hcontig_001_017.29    NaN    NaN      NaN     NaN     NaN   \n",
       "2386   evm.model.hcontig_056_006.3    NaN    NaN      NaN     NaN     NaN   \n",
       "2387   evm.model.hcontig_015_017.5    NaN    NaN      NaN     NaN     NaN   \n",
       "2388  evm.model.hcontig_003_003.40    NaN    NaN      NaN     NaN     NaN   \n",
       "2389   evm.model.hcontig_048_006.1    NaN    NaN      NaN     NaN     NaN   \n",
       "2390   evm.model.hcontig_016_002.4    NaN    NaN      NaN     NaN     NaN   \n",
       "2391  evm.model.hcontig_003_017.28    NaN    NaN      NaN     NaN     NaN   \n",
       "2392   evm.model.hcontig_085_001.1    NaN    NaN      NaN     NaN     NaN   \n",
       "2393  evm.model.hcontig_004_026.11    NaN    NaN      NaN     NaN     NaN   \n",
       "2394  evm.model.hcontig_039_007.17    NaN    NaN      NaN     NaN     NaN   \n",
       "2395   evm.model.hcontig_002_018.3    NaN    NaN      NaN     NaN     NaN   \n",
       "2396  evm.model.hcontig_056_011.78    NaN    NaN      NaN     NaN     NaN   \n",
       "2397   evm.model.hcontig_057_003.3    NaN    NaN      NaN     NaN     NaN   \n",
       "2398   evm.model.hcontig_056_003.1    NaN    NaN      NaN     NaN     NaN   \n",
       "2399   evm.model.hcontig_035_017.5    NaN    NaN      NaN     NaN     NaN   \n",
       "2400   evm.model.hcontig_028_017.4    NaN    NaN      NaN     NaN     NaN   \n",
       "2401  evm.model.hcontig_056_011.32    NaN    NaN      NaN     NaN     NaN   \n",
       "2402   evm.model.hcontig_002_028.5    NaN    NaN      NaN     NaN     NaN   \n",
       "2403  evm.model.hcontig_019_004.52    NaN    NaN      NaN     NaN     NaN   \n",
       "\n",
       "      StartQuery  StopQuery  StartTarget  StopTarget  \\\n",
       "0            NaN        NaN          NaN         NaN   \n",
       "1            NaN        NaN          NaN         NaN   \n",
       "2            NaN        NaN          NaN         NaN   \n",
       "3            NaN        NaN          NaN         NaN   \n",
       "4            NaN        NaN          NaN         NaN   \n",
       "5            NaN        NaN          NaN         NaN   \n",
       "6            NaN        NaN          NaN         NaN   \n",
       "7            NaN        NaN          NaN         NaN   \n",
       "8            NaN        NaN          NaN         NaN   \n",
       "9            NaN        NaN          NaN         NaN   \n",
       "10           NaN        NaN          NaN         NaN   \n",
       "11           NaN        NaN          NaN         NaN   \n",
       "12           NaN        NaN          NaN         NaN   \n",
       "13           NaN        NaN          NaN         NaN   \n",
       "14           NaN        NaN          NaN         NaN   \n",
       "15           NaN        NaN          NaN         NaN   \n",
       "16           NaN        NaN          NaN         NaN   \n",
       "17           NaN        NaN          NaN         NaN   \n",
       "18           NaN        NaN          NaN         NaN   \n",
       "19           NaN        NaN          NaN         NaN   \n",
       "20           NaN        NaN          NaN         NaN   \n",
       "21           NaN        NaN          NaN         NaN   \n",
       "22           NaN        NaN          NaN         NaN   \n",
       "23           NaN        NaN          NaN         NaN   \n",
       "24           NaN        NaN          NaN         NaN   \n",
       "25           NaN        NaN          NaN         NaN   \n",
       "26           NaN        NaN          NaN         NaN   \n",
       "27           NaN        NaN          NaN         NaN   \n",
       "28           NaN        NaN          NaN         NaN   \n",
       "29           NaN        NaN          NaN         NaN   \n",
       "...          ...        ...          ...         ...   \n",
       "2374         NaN        NaN          NaN         NaN   \n",
       "2375         NaN        NaN          NaN         NaN   \n",
       "2376         NaN        NaN          NaN         NaN   \n",
       "2377         NaN        NaN          NaN         NaN   \n",
       "2378         NaN        NaN          NaN         NaN   \n",
       "2379         NaN        NaN          NaN         NaN   \n",
       "2380         NaN        NaN          NaN         NaN   \n",
       "2381         NaN        NaN          NaN         NaN   \n",
       "2382         NaN        NaN          NaN         NaN   \n",
       "2383         NaN        NaN          NaN         NaN   \n",
       "2384         NaN        NaN          NaN         NaN   \n",
       "2385         NaN        NaN          NaN         NaN   \n",
       "2386         NaN        NaN          NaN         NaN   \n",
       "2387         NaN        NaN          NaN         NaN   \n",
       "2388         NaN        NaN          NaN         NaN   \n",
       "2389         NaN        NaN          NaN         NaN   \n",
       "2390         NaN        NaN          NaN         NaN   \n",
       "2391         NaN        NaN          NaN         NaN   \n",
       "2392         NaN        NaN          NaN         NaN   \n",
       "2393         NaN        NaN          NaN         NaN   \n",
       "2394         NaN        NaN          NaN         NaN   \n",
       "2395         NaN        NaN          NaN         NaN   \n",
       "2396         NaN        NaN          NaN         NaN   \n",
       "2397         NaN        NaN          NaN         NaN   \n",
       "2398         NaN        NaN          NaN         NaN   \n",
       "2399         NaN        NaN          NaN         NaN   \n",
       "2400         NaN        NaN          NaN         NaN   \n",
       "2401         NaN        NaN          NaN         NaN   \n",
       "2402         NaN        NaN          NaN         NaN   \n",
       "2403         NaN        NaN          NaN         NaN   \n",
       "\n",
       "                  ...                  q_contig  t_contig  \\\n",
       "0                 ...               pcontig_000     False   \n",
       "1                 ...               pcontig_000     False   \n",
       "2                 ...               pcontig_000     False   \n",
       "3                 ...               pcontig_000     False   \n",
       "4                 ...               pcontig_000     False   \n",
       "5                 ...               pcontig_000     False   \n",
       "6                 ...               pcontig_000     False   \n",
       "7                 ...               pcontig_000     False   \n",
       "8                 ...               pcontig_000     False   \n",
       "9                 ...               pcontig_000     False   \n",
       "10                ...               pcontig_000     False   \n",
       "11                ...               pcontig_000     False   \n",
       "12                ...               pcontig_000     False   \n",
       "13                ...               pcontig_000     False   \n",
       "14                ...               pcontig_000     False   \n",
       "15                ...               pcontig_000     False   \n",
       "16                ...               pcontig_000     False   \n",
       "17                ...               pcontig_000     False   \n",
       "18                ...               pcontig_000     False   \n",
       "19                ...               pcontig_000     False   \n",
       "20                ...               pcontig_000     False   \n",
       "21                ...               pcontig_000     False   \n",
       "22                ...               pcontig_000     False   \n",
       "23                ...               pcontig_000     False   \n",
       "24                ...               pcontig_000     False   \n",
       "25                ...               pcontig_000     False   \n",
       "26                ...               pcontig_000     False   \n",
       "27                ...               pcontig_000     False   \n",
       "28                ...               pcontig_000     False   \n",
       "29                ...               pcontig_000     False   \n",
       "...               ...                       ...       ...   \n",
       "2374              ...                       NaN       NaN   \n",
       "2375              ...                       NaN       NaN   \n",
       "2376              ...                       NaN       NaN   \n",
       "2377              ...                       NaN       NaN   \n",
       "2378              ...                       NaN       NaN   \n",
       "2379              ...                       NaN       NaN   \n",
       "2380              ...                       NaN       NaN   \n",
       "2381              ...                       NaN       NaN   \n",
       "2382              ...                       NaN       NaN   \n",
       "2383              ...                       NaN       NaN   \n",
       "2384              ...                       NaN       NaN   \n",
       "2385              ...                       NaN       NaN   \n",
       "2386              ...                       NaN       NaN   \n",
       "2387              ...                       NaN       NaN   \n",
       "2388              ...                       NaN       NaN   \n",
       "2389              ...                       NaN       NaN   \n",
       "2390              ...                       NaN       NaN   \n",
       "2391              ...                       NaN       NaN   \n",
       "2392              ...                       NaN       NaN   \n",
       "2393              ...                       NaN       NaN   \n",
       "2394              ...                       NaN       NaN   \n",
       "2395              ...                       NaN       NaN   \n",
       "2396              ...                       NaN       NaN   \n",
       "2397              ...                       NaN       NaN   \n",
       "2398              ...                       NaN       NaN   \n",
       "2399              ...                       NaN       NaN   \n",
       "2400              ...                       NaN       NaN   \n",
       "2401              ...                       NaN       NaN   \n",
       "2402              ...                       NaN       NaN   \n",
       "2403              ...                       NaN       NaN   \n",
       "\n",
       "      q_contig == t_contig                   p_protein     h_contig_overlap  \\\n",
       "0                    False                         NaN                False   \n",
       "1                    False  evm.model.pcontig_000.1004  ['hcontig_000_078']   \n",
       "2                    False  evm.model.pcontig_000.1025  ['hcontig_000_078']   \n",
       "3                    False                         NaN                False   \n",
       "4                    False                         NaN                False   \n",
       "5                    False                         NaN                False   \n",
       "6                    False                         NaN                False   \n",
       "7                    False                         NaN                False   \n",
       "8                    False  evm.model.pcontig_000.1095  ['hcontig_000_055']   \n",
       "9                    False                         NaN                False   \n",
       "10                   False                         NaN                False   \n",
       "11                   False                         NaN                False   \n",
       "12                   False                         NaN                False   \n",
       "13                   False                         NaN                False   \n",
       "14                   False   evm.model.pcontig_000.176  ['hcontig_000_019']   \n",
       "15                   False   evm.model.pcontig_000.210  ['hcontig_000_019']   \n",
       "16                   False                         NaN                False   \n",
       "17                   False                         NaN                False   \n",
       "18                   False                         NaN                False   \n",
       "19                   False                         NaN                False   \n",
       "20                   False   evm.model.pcontig_000.308  ['hcontig_000_067']   \n",
       "21                   False                         NaN                False   \n",
       "22                   False                         NaN                False   \n",
       "23                   False                         NaN                False   \n",
       "24                   False                         NaN                False   \n",
       "25                   False                         NaN                False   \n",
       "26                   False                         NaN                False   \n",
       "27                   False   evm.model.pcontig_000.370  ['hcontig_000_002']   \n",
       "28                   False                         NaN                False   \n",
       "29                   False                         NaN                False   \n",
       "...                    ...                         ...                  ...   \n",
       "2374                   NaN                         NaN                  NaN   \n",
       "2375                   NaN                         NaN                  NaN   \n",
       "2376                   NaN                         NaN                  NaN   \n",
       "2377                   NaN                         NaN                  NaN   \n",
       "2378                   NaN                         NaN                  NaN   \n",
       "2379                   NaN                         NaN                  NaN   \n",
       "2380                   NaN                         NaN                  NaN   \n",
       "2381                   NaN                         NaN                  NaN   \n",
       "2382                   NaN                         NaN                  NaN   \n",
       "2383                   NaN                         NaN                  NaN   \n",
       "2384                   NaN                         NaN                  NaN   \n",
       "2385                   NaN                         NaN                  NaN   \n",
       "2386                   NaN                         NaN                  NaN   \n",
       "2387                   NaN                         NaN                  NaN   \n",
       "2388                   NaN                         NaN                  NaN   \n",
       "2389                   NaN                         NaN                  NaN   \n",
       "2390                   NaN                         NaN                  NaN   \n",
       "2391                   NaN                         NaN                  NaN   \n",
       "2392                   NaN                         NaN                  NaN   \n",
       "2393                   NaN                         NaN                  NaN   \n",
       "2394                   NaN                         NaN                  NaN   \n",
       "2395                   NaN                         NaN                  NaN   \n",
       "2396                   NaN                         NaN                  NaN   \n",
       "2397                   NaN                         NaN                  NaN   \n",
       "2398                   NaN                         NaN                  NaN   \n",
       "2399                   NaN                         NaN                  NaN   \n",
       "2400                   NaN                         NaN                  NaN   \n",
       "2401                   NaN                         NaN                  NaN   \n",
       "2402                   NaN                         NaN                  NaN   \n",
       "2403                   NaN                         NaN                  NaN   \n",
       "\n",
       "      t_contig == h_contig_overlap allele_source matchType  \\\n",
       "0                            False         BLAST     BLAST   \n",
       "1                            False         BLAST     BLAST   \n",
       "2                            False         BLAST     BLAST   \n",
       "3                            False         BLAST     BLAST   \n",
       "4                            False         BLAST     BLAST   \n",
       "5                            False         BLAST     BLAST   \n",
       "6                            False         BLAST     BLAST   \n",
       "7                            False         BLAST     BLAST   \n",
       "8                            False         BLAST     BLAST   \n",
       "9                            False         BLAST     BLAST   \n",
       "10                           False         BLAST     BLAST   \n",
       "11                           False         BLAST     BLAST   \n",
       "12                           False         BLAST     BLAST   \n",
       "13                           False         BLAST     BLAST   \n",
       "14                           False         BLAST     BLAST   \n",
       "15                           False         BLAST     BLAST   \n",
       "16                           False         BLAST     BLAST   \n",
       "17                           False         BLAST     BLAST   \n",
       "18                           False         BLAST     BLAST   \n",
       "19                           False         BLAST     BLAST   \n",
       "20                           False         BLAST     BLAST   \n",
       "21                           False         BLAST     BLAST   \n",
       "22                           False         BLAST     BLAST   \n",
       "23                           False         BLAST     BLAST   \n",
       "24                           False         BLAST     BLAST   \n",
       "25                           False         BLAST     BLAST   \n",
       "26                           False         BLAST     BLAST   \n",
       "27                           False         BLAST     BLAST   \n",
       "28                           False         BLAST     BLAST   \n",
       "29                           False         BLAST     BLAST   \n",
       "...                            ...           ...       ...   \n",
       "2374                           NaN           NaN       NaN   \n",
       "2375                           NaN           NaN       NaN   \n",
       "2376                           NaN           NaN       NaN   \n",
       "2377                           NaN           NaN       NaN   \n",
       "2378                           NaN           NaN       NaN   \n",
       "2379                           NaN           NaN       NaN   \n",
       "2380                           NaN           NaN       NaN   \n",
       "2381                           NaN           NaN       NaN   \n",
       "2382                           NaN           NaN       NaN   \n",
       "2383                           NaN           NaN       NaN   \n",
       "2384                           NaN           NaN       NaN   \n",
       "2385                           NaN           NaN       NaN   \n",
       "2386                           NaN           NaN       NaN   \n",
       "2387                           NaN           NaN       NaN   \n",
       "2388                           NaN           NaN       NaN   \n",
       "2389                           NaN           NaN       NaN   \n",
       "2390                           NaN           NaN       NaN   \n",
       "2391                           NaN           NaN       NaN   \n",
       "2392                           NaN           NaN       NaN   \n",
       "2393                           NaN           NaN       NaN   \n",
       "2394                           NaN           NaN       NaN   \n",
       "2395                           NaN           NaN       NaN   \n",
       "2396                           NaN           NaN       NaN   \n",
       "2397                           NaN           NaN       NaN   \n",
       "2398                           NaN           NaN       NaN   \n",
       "2399                           NaN           NaN       NaN   \n",
       "2400                           NaN           NaN       NaN   \n",
       "2401                           NaN           NaN       NaN   \n",
       "2402                           NaN           NaN       NaN   \n",
       "2403                           NaN           NaN       NaN   \n",
       "\n",
       "                          aQuery                       aTarget  \n",
       "0       evm.model.pcontig_000.10                           NaN  \n",
       "1     evm.model.pcontig_000.1004                           NaN  \n",
       "2     evm.model.pcontig_000.1025                           NaN  \n",
       "3     evm.model.pcontig_000.1080                           NaN  \n",
       "4     evm.model.pcontig_000.1081                           NaN  \n",
       "5     evm.model.pcontig_000.1082                           NaN  \n",
       "6     evm.model.pcontig_000.1083                           NaN  \n",
       "7     evm.model.pcontig_000.1084                           NaN  \n",
       "8     evm.model.pcontig_000.1095                           NaN  \n",
       "9      evm.model.pcontig_000.122                           NaN  \n",
       "10     evm.model.pcontig_000.123                           NaN  \n",
       "11     evm.model.pcontig_000.125                           NaN  \n",
       "12     evm.model.pcontig_000.158                           NaN  \n",
       "13     evm.model.pcontig_000.159                           NaN  \n",
       "14     evm.model.pcontig_000.176                           NaN  \n",
       "15     evm.model.pcontig_000.210                           NaN  \n",
       "16     evm.model.pcontig_000.217                           NaN  \n",
       "17     evm.model.pcontig_000.219                           NaN  \n",
       "18     evm.model.pcontig_000.224                           NaN  \n",
       "19     evm.model.pcontig_000.289                           NaN  \n",
       "20     evm.model.pcontig_000.308                           NaN  \n",
       "21     evm.model.pcontig_000.321                           NaN  \n",
       "22     evm.model.pcontig_000.323                           NaN  \n",
       "23     evm.model.pcontig_000.324                           NaN  \n",
       "24     evm.model.pcontig_000.341                           NaN  \n",
       "25     evm.model.pcontig_000.342                           NaN  \n",
       "26     evm.model.pcontig_000.360                           NaN  \n",
       "27     evm.model.pcontig_000.370                           NaN  \n",
       "28     evm.model.pcontig_000.371                           NaN  \n",
       "29     evm.model.pcontig_000.372                           NaN  \n",
       "...                          ...                           ...  \n",
       "2374                         NaN  evm.model.hcontig_056_011.39  \n",
       "2375                         NaN   evm.model.hcontig_017_016.3  \n",
       "2376                         NaN   evm.model.hcontig_026_005.5  \n",
       "2377                         NaN   evm.model.hcontig_000_080.3  \n",
       "2378                         NaN   evm.model.hcontig_039_015.7  \n",
       "2379                         NaN   evm.model.hcontig_011_006.2  \n",
       "2380                         NaN   evm.model.hcontig_048_001.3  \n",
       "2381                         NaN   evm.model.hcontig_005_039.8  \n",
       "2382                         NaN   evm.model.hcontig_056_002.6  \n",
       "2383                         NaN  evm.model.hcontig_001_019.39  \n",
       "2384                         NaN   evm.model.hcontig_047_001.4  \n",
       "2385                         NaN  evm.model.hcontig_001_017.29  \n",
       "2386                         NaN   evm.model.hcontig_056_006.3  \n",
       "2387                         NaN   evm.model.hcontig_015_017.5  \n",
       "2388                         NaN  evm.model.hcontig_003_003.40  \n",
       "2389                         NaN   evm.model.hcontig_048_006.1  \n",
       "2390                         NaN   evm.model.hcontig_016_002.4  \n",
       "2391                         NaN  evm.model.hcontig_003_017.28  \n",
       "2392                         NaN   evm.model.hcontig_085_001.1  \n",
       "2393                         NaN  evm.model.hcontig_004_026.11  \n",
       "2394                         NaN  evm.model.hcontig_039_007.17  \n",
       "2395                         NaN   evm.model.hcontig_002_018.3  \n",
       "2396                         NaN  evm.model.hcontig_056_011.78  \n",
       "2397                         NaN   evm.model.hcontig_057_003.3  \n",
       "2398                         NaN   evm.model.hcontig_056_003.1  \n",
       "2399                         NaN   evm.model.hcontig_035_017.5  \n",
       "2400                         NaN   evm.model.hcontig_028_017.4  \n",
       "2401                         NaN  evm.model.hcontig_056_011.32  \n",
       "2402                         NaN   evm.model.hcontig_002_028.5  \n",
       "2403                         NaN  evm.model.hcontig_019_004.52  \n",
       "\n",
       "[2404 rows x 26 columns]"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf.loc[alleleDf.index.isnull(), :].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T14:03:28.453055Z",
     "start_time": "2018-02-26T14:03:28.424010Z"
    }
   },
   "outputs": [],
   "source": [
    "test =pd.concat([tmp_df, alleleDf.loc[alleleDf.index.isnull(), :]], axis=0, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T14:03:33.391951Z",
     "start_time": "2018-02-26T14:03:33.342543Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>BitScore</th>\n",
       "      <th>Index</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>PctID</th>\n",
       "      <th>QCov</th>\n",
       "      <th>QLgth</th>\n",
       "      <th>Query</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>...</th>\n",
       "      <th>e-value</th>\n",
       "      <th>h_contig_overlap</th>\n",
       "      <th>matchType</th>\n",
       "      <th>p_protein</th>\n",
       "      <th>protein_hamming</th>\n",
       "      <th>protein_levenshtein</th>\n",
       "      <th>q_contig</th>\n",
       "      <th>q_contig == t_contig</th>\n",
       "      <th>t_contig</th>\n",
       "      <th>t_contig == h_contig_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>240.0</td>\n",
       "      <td>50.4</td>\n",
       "      <td>evm.model.pcontig_000.100_evm.model.hcontig_07...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>25.00</td>\n",
       "      <td>27.180068</td>\n",
       "      <td>883</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>110.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e-06</td>\n",
       "      <td>['hcontig_000_058']</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>0.8363</td>\n",
       "      <td>0.8183</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>hcontig_072_002</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37.0</td>\n",
       "      <td>53.5</td>\n",
       "      <td>evm.model.pcontig_000.1001_evm.model.hcontig_0...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>70.27</td>\n",
       "      <td>21.511628</td>\n",
       "      <td>172</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>130.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.000000e-10</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>hcontig_029_007</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>128.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>evm.model.pcontig_000.1002_evm.model.hcontig_0...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>96.88</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>128</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e-88</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>0.4115</td>\n",
       "      <td>0.4115</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>164.0</td>\n",
       "      <td>315.0</td>\n",
       "      <td>evm.model.pcontig_000.1003_evm.model.hcontig_0...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>96.95</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>164</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.000000e-106</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>0.6884</td>\n",
       "      <td>0.6884</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>False</td>\n",
       "      <td>hcontig_003_006</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92.0</td>\n",
       "      <td>58.2</td>\n",
       "      <td>evm.model.pcontig_000.1005_evm.model.hcontig_0...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>39.13</td>\n",
       "      <td>78.632479</td>\n",
       "      <td>117</td>\n",
       "      <td>evm.model.pcontig_000.1005</td>\n",
       "      <td>25.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.000000e-12</td>\n",
       "      <td>['hcontig_000_078']</td>\n",
       "      <td>BLAST</td>\n",
       "      <td>evm.model.pcontig_000.1005</td>\n",
       "      <td>0.7097</td>\n",
       "      <td>0.7016</td>\n",
       "      <td>pcontig_000</td>\n",
       "      <td>True</td>\n",
       "      <td>hcontig_000_078</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   AlnLgth  BitScore                                              Index  \\\n",
       "0    240.0      50.4  evm.model.pcontig_000.100_evm.model.hcontig_07...   \n",
       "1     37.0      53.5  evm.model.pcontig_000.1001_evm.model.hcontig_0...   \n",
       "2    128.0     258.0  evm.model.pcontig_000.1002_evm.model.hcontig_0...   \n",
       "3    164.0     315.0  evm.model.pcontig_000.1003_evm.model.hcontig_0...   \n",
       "4     92.0      58.2  evm.model.pcontig_000.1005_evm.model.hcontig_0...   \n",
       "\n",
       "   NumGap  NumMis  PctID        QCov  QLgth                       Query  \\\n",
       "0     8.0   155.0  25.00   27.180068    883   evm.model.pcontig_000.100   \n",
       "1     0.0    11.0  70.27   21.511628    172  evm.model.pcontig_000.1001   \n",
       "2     0.0     4.0  96.88  100.000000    128  evm.model.pcontig_000.1002   \n",
       "3     0.0     5.0  96.95  100.000000    164  evm.model.pcontig_000.1003   \n",
       "4     0.0    56.0  39.13   78.632479    117  evm.model.pcontig_000.1005   \n",
       "\n",
       "   StartQuery              ...                     e-value  \\\n",
       "0       110.0              ...                1.000000e-06   \n",
       "1       130.0              ...                2.000000e-10   \n",
       "2         1.0              ...                1.000000e-88   \n",
       "3         1.0              ...               2.000000e-106   \n",
       "4        25.0              ...                3.000000e-12   \n",
       "\n",
       "      h_contig_overlap  matchType                   p_protein  \\\n",
       "0  ['hcontig_000_058']      BLAST   evm.model.pcontig_000.100   \n",
       "1  ['hcontig_000_078']      BLAST  evm.model.pcontig_000.1001   \n",
       "2  ['hcontig_000_078']      BLAST  evm.model.pcontig_000.1002   \n",
       "3  ['hcontig_000_078']      BLAST  evm.model.pcontig_000.1003   \n",
       "4  ['hcontig_000_078']      BLAST  evm.model.pcontig_000.1005   \n",
       "\n",
       "   protein_hamming protein_levenshtein     q_contig q_contig == t_contig  \\\n",
       "0           0.8363              0.8183  pcontig_000                False   \n",
       "1           0.7778              0.7778  pcontig_000                False   \n",
       "2           0.4115              0.4115  pcontig_000                False   \n",
       "3           0.6884              0.6884  pcontig_000                False   \n",
       "4           0.7097              0.7016  pcontig_000                 True   \n",
       "\n",
       "          t_contig  t_contig == h_contig_overlap  \n",
       "0  hcontig_072_002                         False  \n",
       "1  hcontig_029_007                         False  \n",
       "2  hcontig_003_006                         False  \n",
       "3  hcontig_003_006                         False  \n",
       "4  hcontig_000_078                          True  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:48:52.889960Z",
     "start_time": "2018-02-26T13:48:52.881045Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery',\n",
       "       'StopQuery', 'StartTarget', 'StopTarget', 'e-value', 'BitScore',\n",
       "       'QLgth', 'QCov', 'TLgth', 'TCov', 'q_contig', 't_contig',\n",
       "       'q_contig == t_contig', 'p_protein', 'h_contig_overlap',\n",
       "       't_contig == h_contig_overlap', 'allele_source', 'matchType', 'aQuery',\n",
       "       'aTarget', 'unphased', 'PO_single', 'PO_interhaplotype_paralogs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleleDf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:38:52.427046Z",
     "start_time": "2018-02-26T13:38:52.369019Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>...</th>\n",
       "      <th>aTarget</th>\n",
       "      <th>Index</th>\n",
       "      <th>LPB93_dN/dS</th>\n",
       "      <th>LWL85_dN/dS</th>\n",
       "      <th>LWL85m_dN/dS</th>\n",
       "      <th>cds_hamming</th>\n",
       "      <th>cds_levenshtein</th>\n",
       "      <th>protein_hamming</th>\n",
       "      <th>protein_levenshtein</th>\n",
       "      <th>yn00_dN/dS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>evm.model.pcontig_000.100</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>25.00</td>\n",
       "      <td>240.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>330.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>...</td>\n",
       "      <td>evm.model.hcontig_072_002.3</td>\n",
       "      <td>evm.model.pcontig_000.100_evm.model.hcontig_07...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>evm.model.pcontig_000.1001</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>70.27</td>\n",
       "      <td>37.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>...</td>\n",
       "      <td>evm.model.hcontig_029_007.1</td>\n",
       "      <td>evm.model.pcontig_000.1001_evm.model.hcontig_0...</td>\n",
       "      <td>0.5620</td>\n",
       "      <td>0.4593</td>\n",
       "      <td>0.5822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>evm.model.pcontig_000.1002</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>96.88</td>\n",
       "      <td>128.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>evm.model.hcontig_003_006.10</td>\n",
       "      <td>evm.model.pcontig_000.1002_evm.model.hcontig_0...</td>\n",
       "      <td>0.4985</td>\n",
       "      <td>0.3733</td>\n",
       "      <td>0.4469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>evm.model.pcontig_000.1003</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>96.95</td>\n",
       "      <td>164.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>...</td>\n",
       "      <td>evm.model.hcontig_003_006.9</td>\n",
       "      <td>evm.model.pcontig_000.1003_evm.model.hcontig_0...</td>\n",
       "      <td>0.6129</td>\n",
       "      <td>0.4767</td>\n",
       "      <td>0.3212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>evm.model.pcontig_000.1005</td>\n",
       "      <td>evm.model.hcontig_000_078.13</td>\n",
       "      <td>39.13</td>\n",
       "      <td>92.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>...</td>\n",
       "      <td>evm.model.hcontig_000_078.13</td>\n",
       "      <td>evm.model.pcontig_000.1005_evm.model.hcontig_0...</td>\n",
       "      <td>0.4567</td>\n",
       "      <td>0.4674</td>\n",
       "      <td>0.3141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Query                        Target  PctID  AlnLgth  \\\n",
       "0   evm.model.pcontig_000.100   evm.model.hcontig_072_002.3  25.00    240.0   \n",
       "1  evm.model.pcontig_000.1001   evm.model.hcontig_029_007.1  70.27     37.0   \n",
       "2  evm.model.pcontig_000.1002  evm.model.hcontig_003_006.10  96.88    128.0   \n",
       "3  evm.model.pcontig_000.1003   evm.model.hcontig_003_006.9  96.95    164.0   \n",
       "4  evm.model.pcontig_000.1005  evm.model.hcontig_000_078.13  39.13     92.0   \n",
       "\n",
       "   NumMis  NumGap  StartQuery  StopQuery  StartTarget  StopTarget    ...      \\\n",
       "0   155.0     8.0       110.0      330.0        125.0       358.0    ...       \n",
       "1    11.0     0.0       130.0      166.0         52.0        88.0    ...       \n",
       "2     4.0     0.0         1.0      128.0         83.0       210.0    ...       \n",
       "3     5.0     0.0         1.0      164.0        345.0       508.0    ...       \n",
       "4    56.0     0.0        25.0      116.0          1.0        92.0    ...       \n",
       "\n",
       "                        aTarget  \\\n",
       "0   evm.model.hcontig_072_002.3   \n",
       "1   evm.model.hcontig_029_007.1   \n",
       "2  evm.model.hcontig_003_006.10   \n",
       "3   evm.model.hcontig_003_006.9   \n",
       "4  evm.model.hcontig_000_078.13   \n",
       "\n",
       "                                               Index  LPB93_dN/dS  \\\n",
       "0  evm.model.pcontig_000.100_evm.model.hcontig_07...          NaN   \n",
       "1  evm.model.pcontig_000.1001_evm.model.hcontig_0...       0.5620   \n",
       "2  evm.model.pcontig_000.1002_evm.model.hcontig_0...       0.4985   \n",
       "3  evm.model.pcontig_000.1003_evm.model.hcontig_0...       0.6129   \n",
       "4  evm.model.pcontig_000.1005_evm.model.hcontig_0...       0.4567   \n",
       "\n",
       "   LWL85_dN/dS  LWL85m_dN/dS  cds_hamming cds_levenshtein protein_hamming  \\\n",
       "0          NaN           NaN          NaN             NaN             NaN   \n",
       "1       0.4593        0.5822          NaN             NaN             NaN   \n",
       "2       0.3733        0.4469          NaN             NaN             NaN   \n",
       "3       0.4767        0.3212          NaN             NaN             NaN   \n",
       "4       0.4674        0.3141          NaN             NaN             NaN   \n",
       "\n",
       "  protein_levenshtein yn00_dN/dS  \n",
       "0                 NaN     0.2315  \n",
       "1                 NaN     0.6262  \n",
       "2                 NaN     0.3870  \n",
       "3                 NaN     0.8289  \n",
       "4                 NaN     0.4835  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:31:03.259712Z",
     "start_time": "2018-02-26T13:31:03.250806Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[os.path.join(tmp_path, file) for file in os.listdir(tmp_path)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T13:30:44.303516Z",
     "start_time": "2018-02-26T13:30:44.293335Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[os.path.join(tmp_path, file) for file in os.listdir(tmp_path) if file.endswith(dist_suffix) == True ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-20T02:49:47.072255Z",
     "start_time": "2018-02-20T02:49:46.220172Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "########## FIGURE PLOTTING ##########\n",
    "\n",
    "def make_autopct(values):\n",
    "    def my_autopct(pct):\n",
    "        total = sum(values)\n",
    "        val = int(round(pct*total/100.0))\n",
    "        return '{p:.2f}%\\n({v:d})'.format(p=pct,v=val)\n",
    "    return my_autopct\n",
    "\n",
    "def autolabel(rects, labels, ax, fontsize):\n",
    "    \"\"\"\n",
    "    Attach a text label above each bar displaying its height\n",
    "    \"\"\"\n",
    "    for i, rect in enumerate(rects):\n",
    "        height = rect.get_height()\n",
    "        ax.text(rect.get_x() + rect.get_width()/2., height, str(labels[i]), ha='center', va='bottom', fontsize=fontsize)\n",
    "\n",
    "def getNumNoAlleles(pProteinFastaFile, alleleDf):\n",
    "    with open(pProteinFastaFile) as pProteinFasta:\n",
    "        pProteinList = []\n",
    "        for line in pProteinFasta:\n",
    "            if line.startswith('>'):\n",
    "                pProteinList.append(line[1:].strip())\n",
    "\n",
    "    assert(len(pProteinList) == len(set(pProteinList)))\n",
    "\n",
    "\n",
    "    pairedPProteinList = list(alleleDf['Query'])\n",
    "    pairedPProteinList += list(alleleDf['Target'])\n",
    "    pairedPProteinList = set(pairedPProteinList)\n",
    "    \n",
    "    for pairedPProtein in pairedPProteinList:\n",
    "        if pairedPProtein in pProteinList:\n",
    "            pProteinList.remove(pairedPProtein)\n",
    "    \n",
    "    return len(pProteinList)\n",
    "\n",
    "def plotAlleleTypesPie(ax, alleleDf, colors, includeNoAlleles=True):\n",
    "    '''Plots a pie chart of allele types, with the option of also including \n",
    "    primary proteins with no alleles. Strictly, this is not an accurate representation\n",
    "    of the distribution of primary proteins as the reciprocal BLAST-identified (h on p) alleles\n",
    "    may result in double-counting of primary proteins.\n",
    "    '''\n",
    "    # OrderedDict to preserve order, so that plots are coloured with same key as the distance \n",
    "    # bar graphs. This is a bit of a hack-fix; must enter these by hand again in the same order \n",
    "    # as 'matchType' occurs in the alleleAveragesByMatchType DataFrame.\n",
    "    alleleTypeCountDict = collections.OrderedDict()\n",
    "    \n",
    "    for matchType in alleleDf['matchType'].unique():\n",
    "        alleleTypeCountDict[matchType] = len(alleleDf[alleleDf['matchType'] == matchType])\n",
    "    \n",
    "    if includeNoAlleles==True:\n",
    "        numNoAlleles = getNumNoAlleles(P_PROTEINS_FASTA, alleleDf)\n",
    "        alleleTypeCountDict['no_allele'] = numNoAlleles\n",
    "\n",
    "    patches, texts, autotexts = ax.pie(list(alleleTypeCountDict.values()), labels=alleleTypeCountDict.keys(), autopct=make_autopct(list(alleleTypeCountDict.values())), colors=colors)\n",
    "    ax.axis('equal')\n",
    "    ax.set_title('Allele Types', loc='center', fontsize=TITLE_SIZE, position=(0.5, 1.1))\n",
    "\n",
    "def plotLevenshteinBar(alleleAverages, ax, colors):\n",
    "    '''Plots a bar graph of normalised Levenshtein distances on ax from DataFrame alleleAverages.'''\n",
    "    \n",
    "    ind = np.arange(len(alleleAverages.protein_levenshtein))\n",
    "    rects = ax.bar(ind, alleleAverages.protein_levenshtein, 0.35, color=colors, align='center') \n",
    "    \n",
    "    sns.despine(top=True, right=True)\n",
    "    \n",
    "    barLabels = []\n",
    "    for levDist in alleleAverages.protein_levenshtein:\n",
    "        barLabels.append(str(int((1-levDist)*100)) + '%')\n",
    "    autolabel(rects, barLabels, ax, INLINE_LABEL_SIZE)\n",
    "\n",
    "    ax.set_xticks(ind)\n",
    "    ax.set_xticklabels(alleleAverages.index, rotation=45)\n",
    "\n",
    "    # ax.set_xlabel('Allele Types', fontsize=AXIS_LABEL_SIZE)\n",
    "    ax.set_ylabel('Normalised Levenshtein Distance', fontsize=AXIS_LABEL_SIZE)\n",
    "\n",
    "    ax.tick_params(axis='both', which='major', labelsize=AXIS_TICK_SIZE, pad=3)\n",
    "\n",
    "    for tick in ax.get_xaxis().get_major_ticks():\n",
    "        tick.set_pad(2*tick.get_pad())\n",
    "        tick.label1 = tick._get_text1()\n",
    "        \n",
    "def plotAlleles(alleleDf, qCovFilters, tCovFilters, pctIdFilters, levSimFilters, leavePO):\n",
    "    '''Makes a 3x2 plot with normalised Levenshtein distance plots in column 1 and\n",
    "    a pie chart representing the distribution of allele types in column 2.\n",
    "    Each row shows different levels of filtering.\n",
    "    \n",
    "    leavePO is a boolean that determines whether only BLAST hits will be filtered (leavePO=True)\n",
    "    or both BLAST and PO alleles should be filtered (leavePO=False)'''\n",
    "    cmap = plt.cm.Greens\n",
    "    colors = cmap(np.linspace(0.0, 0.6, len(alleleDf['matchType'].unique())))\n",
    "    \n",
    "    assert(len(qCovFilters) == len(pctIdFilters) == len(levSimFilters))\n",
    "    \n",
    "    fig, ax = plt.subplots(len(qCovFilters), 2, figsize=(30, 12*len(qCovFilters)))\n",
    "    \n",
    "    for i in range(len(qCovFilters)):\n",
    "        \n",
    "        filteredAlleleDf = filterAlleleDf(alleleDf, qCovFilters[i], tCovFilters[i], pctIdFilters[i], levSimFilters[i], True)\n",
    "        print(filteredAlleleDf[filteredAlleleDf.allele_source == 'PO'].shape[0])\n",
    "        # levenshtein distance plot\n",
    "        alleleAveragesByMatchType = filteredAlleleDf.groupby(['matchType']).mean()\n",
    "        plotLevenshteinBar(alleleAveragesByMatchType, ax[i, 0], colors)\n",
    "        ax[i, 0].set_xticklabels(alleleAveragesByMatchType.index, rotation=45, ha='right')\n",
    "        \n",
    "        # pie plot\n",
    "        plotAlleleTypesPie(ax[i, 1], filteredAlleleDf, colors)\n",
    "        \n",
    "        # include filtering criteria in title\n",
    "        qCovFilter = qCovFilters[i]\n",
    "        tCovFilter = tCovFilters[i]\n",
    "        pctIdFilter = pctIdFilters[i]\n",
    "        levSimFilter = levSimFilters[i]\n",
    "        if qCovFilter < BASE_QCOV_CUTOFF:\n",
    "            qCovFilter = BASE_QCOV_CUTOFF\n",
    "            print('Base QCov cut-off is 70%; if you desire to filter below this value, decrease BASE_QCOV_CUTOFF.')\n",
    "        if tCovFilter < BASE_TCOV_CUTOFF:\n",
    "            tCovFilter = BASE_TCOV_CUTOFF\n",
    "            print('Base TCov cut-off is 70%; if you desire to filter below this value, decrease BASE_TCOV_CUTOFF.')\n",
    "        if pctIdFilter < BASE_PCTID_CUTOFF:\n",
    "            pctIdFilter = BASE_PCTID_CUTOFF\n",
    "            print('Base %ID cut-off is 70%; if you desire to filter below this value, decrease BASE_PCTID_CUTOFF.')\n",
    "        if not levSimFilter:\n",
    "            levSimFilter = 0\n",
    "    \n",
    "        ax[i, 0].set_title('QCov > %s%%, TCov > %s%%, ID > %s%%, L. sim. > %s%%, PO Filtered: %s' % (qCovFilter, tCovFilter, pctIdFilter, levSimFilter, not leavePO), position=(0.5, 0.85))\n",
    "        \n",
    "    fig.tight_layout()\n",
    "    fig.savefig(os.path.join(FIGURE_PATH, 'fig'), bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-20T02:50:17.346988Z",
     "start_time": "2018-02-20T02:49:49.617825Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Used in the pie chart for all text except title\n",
    "# the ax.pie plotting interface is weird - cannot set other font sizes properly?\n",
    "mpl.rcParams['font.size'] = 24\n",
    "\n",
    "TITLE_SIZE = 32\n",
    "AXIS_LABEL_SIZE = 28\n",
    "AXIS_TICK_SIZE = 24\n",
    "INLINE_LABEL_SIZE = 24\n",
    "\n",
    "# These lists must all be the same length\n",
    "qCovFilters = [False, 80, 90, 95]\n",
    "pctIdFilters = [False, 80, 90, 95]\n",
    "tCovFilters = [False, 80, 90, 95]\n",
    "levSimFilters = [False, False, False, False]\n",
    "\n",
    "plotAlleles(alleleDf, qCovFilters, tCovFilters, pctIdFilters, levSimFilters, True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
