{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "def getIdToLocusTagDict(df):\n",
    "    df = df[df['type'] == 'mRNA']\n",
    "    \n",
    "    locusSearch = re.compile(r'^.*locus_tag=(.*?)(;|$)')\n",
    "    idSearch = re.compile(r'ID=(.*?);')\n",
    "    \n",
    "    d = {}\n",
    "    \n",
    "    for attr in df['attributes']:\n",
    "        val = locusSearch.match(attr).group(1)\n",
    "        key = idSearch.match(attr).group(1)\n",
    "        if key in d.keys():\n",
    "            print(\"Unexpected: id key: %s already in dictionary.\" % key)\n",
    "        d[key] = val\n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import sys\n",
    "import subprocess\n",
    "\n",
    "BASE_OUT_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/'\n",
    "GENOME_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/genome_v03/'\n",
    "PROTEINORTHO_OUT_PATH = os.path.join(BASE_OUT_PATH, 'proteinortho')\n",
    "\n",
    "ORIG_H_CTG_GFF_PATH = os.path.join(GENOME_PATH, 'DK_0911_v03_h_ctg.anno.gff3')\n",
    "ORIG_P_CTG_GFF_PATH = os.path.join(GENOME_PATH, 'DK_0911_v03_p_ctg.anno.gff3')\n",
    "\n",
    "PROTEINORTHO_PATH = '/home/gamran/anaconda3/proteinortho_v5.16b/proteinortho5.pl'\n",
    "PROJECT_NAME = 'ph_ctg_516'\n",
    "H_CTG_FAA = 'DK_0911_v03_h_ctg.protein.faa'\n",
    "P_CTG_FAA = 'DK_0911_v03_p_ctg.protein.faa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "def rewriteGffForProteinortho(gffInLoc, gffOutLoc):\n",
    "    '''Takes a gff file at gffInLoc and rewrites it to conform to\n",
    "    the format required for proteinortho analysis.'''\n",
    "    ctgDf = pd.read_table(gffInLoc, skiprows = 1, header = None, \\\n",
    "                           names = ['seqid', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase', 'attributes'])\n",
    "    \n",
    "    # get dict mapping ID to locus_tag from mRNA entries\n",
    "    # need attribute to have 'ID=locus_tag' because .faa file has ID=locus_tag\n",
    "    # will be used in this way: CDS.parent = mRNA.id =(dict)> locus_tag\n",
    "    d = getIdToLocusTagDict(ctgDf)\n",
    "    \n",
    "    # subset DataFrame to only CDS type (only type required for proteinortho analysis)\n",
    "    ctgDf = ctgDf[ctgDf['type'] == 'CDS']\n",
    "    \n",
    "    # turn attribute column into parent value\n",
    "    parentSearch = re.compile(r'^.*Parent=(.*?)(;|$)')\n",
    "    ctgDf['attributes'] = ctgDf['attributes'].apply(lambda x: parentSearch.match(x).group(1))\n",
    "    \n",
    "    # map CDS.parent values (= mRNA.id) in attribute column to locus_tag\n",
    "    ctgDf['attributes'] = ctgDf['attributes'].map(d)\n",
    "    \n",
    "    # make attribute column into format 'ID=locus_tag' to be readable by proteinortho\n",
    "    ctgDf['attributes'] = ctgDf['attributes'].apply(lambda x: 'ID=%s;'%x)\n",
    "    \n",
    "    with open(gffOutLoc, 'w') as outfile:\n",
    "        ctgDf.to_csv(outfile, sep='\\t', header = False, index = False)\n",
    "    return ctgDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def getProteinorthoScript():\n",
    "    '''Returns a string containing the commands required to run\n",
    "    proteinortho on the two .faa files in synteny mode'''\n",
    "    s = ''\n",
    "    s += 'cd %s\\n' % PROTEINORTHO_OUT_PATH\n",
    "    s += 'source activate py27\\n'\n",
    "    s += '%s -project=%s -synteny %s/%s %s/%s'\\\n",
    "        % (PROTEINORTHO_PATH, PROJECT_NAME, PROJECT_NAME, H_CTG_FAA, PROJECT_NAME, P_CTG_FAA)\n",
    "    return s\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "def exitIfFileNotInCwd(fileName):\n",
    "    '''Checks if a file is in the current working directory (cwd)\n",
    "    and if not, alerts the user to take action.'''\n",
    "    if not os.path.exists(fileName):\n",
    "        print(\"Please copy %s file to %s\" \\\n",
    "              % (fileName, os.getcwd()))   \n",
    "        sys.exit()\n",
    "    return True\n",
    "\n",
    "def writeGffIfDoesntExist(new_gff_file_path, orig_gff_file_path):\n",
    "    '''Writes the new proteinortho-compatible gff file at the\n",
    "    new_gff_file_path if it does not already exist. If the gff at\n",
    "    new_gff_file_path already exists, it will not be re-written.'''\n",
    "    if not os.path.exists(new_gff_file_path):\n",
    "        rewriteGffForProteinortho(orig_gff_file_path, new_gff_file_path)\n",
    "    else:\n",
    "        print('gff file at: %s already exists... no new gff file was generated.' % new_gff_file_path)\n",
    "\n",
    "def setUp():\n",
    "    print(\"Checking for correct files and directories...\")\n",
    "    os.chdir(PROTEINORTHO_OUT_PATH)\n",
    "    if not os.path.isdir(PROJECT_NAME):\n",
    "        print(\"Created directory: \" + os.path.join(os.getcwd(), PROJECT_NAME))\n",
    "        os.mkdir(PROJECT_NAME)\n",
    "    os.chdir(PROJECT_NAME)\n",
    "    \n",
    "    exitIfFileNotInCwd(H_CTG_FAA)\n",
    "    exitIfFileNotInCwd(P_CTG_FAA)\n",
    "    \n",
    "    NEW_H_CTG_GFF_PATH = os.path.join(os.getcwd(), H_CTG_FAA[:-3] + \"gff\")\n",
    "    NEW_P_CTG_GFF_PATH = os.path.join(os.getcwd(), P_CTG_FAA[:-3] + \"gff\")\n",
    "    \n",
    "    writeGffIfDoesntExist(NEW_H_CTG_GFF_PATH, ORIG_H_CTG_GFF_PATH)\n",
    "    writeGffIfDoesntExist(NEW_P_CTG_GFF_PATH, ORIG_P_CTG_GFF_PATH)\n",
    "\n",
    "    exitIfFileNotInCwd(P_CTG_FAA[:-3] + \"gff\")\n",
    "    exitIfFileNotInCwd(H_CTG_FAA[:-3] + \"gff\")\n",
    "    \n",
    "    print(\"Files and directories required for proteinortho analysis exist.\\n\")\n",
    "    \n",
    "    return True\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "\n",
    "def writeBashScript(bashScriptName, folderPath):\n",
    "    '''Writes the proteinortho bash script with name bashScriptName\n",
    "    and in the folder folderPath'''\n",
    "    os.chdir(folderPath)\n",
    "\n",
    "    outFile = open(bashScriptName, 'w')\n",
    "    outFile.write('#!/bin/bash\\n')\n",
    "    outFile.write(getProteinorthoScript())\n",
    "    outFile.close()\n",
    "\n",
    "def runBashScript(bashScript, folderPath):\n",
    "    '''Runs bashScript and output is piped in real-time to Python'''\n",
    "    os.chdir(folderPath)\n",
    "    cmd = ['bash', bashScript]\n",
    "    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)\n",
    "\n",
    "    for line in iter(p.stdout.readline, b''):\n",
    "        print(line.rstrip())\n",
    "    \n",
    "    return True\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/home/gamran/genome_analysis/Warrior/Richard/scripts')\n",
    "%run 'file_counting.ipynb'\n",
    "\n",
    "PROTEINORTHO_OUT_PATH_DICT = {'blast-graph': 1,\n",
    "                             'faa': 2,\n",
    "                             'ffadj-graph': 1,\n",
    "                             'gff': 2,\n",
    "                             'phr': 2,\n",
    "                             'pin': 2,\n",
    "                             'poff': 1,\n",
    "                             'poff-graph': 1,\n",
    "                             'proteinortho': 1,\n",
    "                             'proteinortho-graph': 1,\n",
    "                             'psq': 2}\n",
    "\n",
    "def reportDiscrepancies(folderPath, refDict, ignoreExts = []):\n",
    "    '''Checks whether the proteinortho files already exist, according\n",
    "    to the refDict'''\n",
    "    print(\"Checking whether proteinortho files already exist in %s...\" % folderPath)\n",
    "    folder_name = getFolderName(folderPath)\n",
    "    print('Folder reference dictionary:\\n%s' % refDict)\n",
    "    \n",
    "    discrepancies = getDiscrepancies(folderPath, refDict, ignoreExts)\n",
    "    \n",
    "    if discrepancies == '':\n",
    "        print(\"All proteinortho files, according to the reference dictionary, appear to already exist.\")\n",
    "        return False\n",
    "    \n",
    "    print(discrepancies)\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "def main():\n",
    "    assert(setUp())\n",
    "    fileDiscrepancies = reportDiscrepancies(PROTEINORTHO_OUT_PATH, PROTEINORTHO_OUT_PATH_DICT, ['sh']) \n",
    "    if fileDiscrepancies:\n",
    "        print(\"Not all files supposed to be generated by proteinortho appear to be present. Running proteinortho now (this may take some time)...\")\n",
    "        writeBashScript('run_proteinortho.sh', PROTEINORTHO_OUT_PATH)\n",
    "        runBashScript('run_proteinortho.sh', PROTEINORTHO_OUT_PATH)\n",
    "        print(\"Proteinortho finished running.\")\n",
    "    else:\n",
    "        print(\"\\nProteinortho appears to have been ran previously, therefore it was not run this time.\")\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking for correct files and directories...\n",
      "gff file at: /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/proteinortho/ph_ctg_516/DK_0911_v03_h_ctg.protein.gff already exists... no new gff file was generated.\n",
      "gff file at: /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/proteinortho/ph_ctg_516/DK_0911_v03_p_ctg.protein.gff already exists... no new gff file was generated.\n",
      "Files and directories required for proteinortho analysis exist.\n",
      "\n",
      "Checking whether proteinortho files already exist in /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles/proteinortho...\n",
      "Folder reference dictionary:\n",
      "{'blast-graph': 1, 'poff-graph': 1, 'ffadj-graph': 1, 'gff': 2, 'phr': 2, 'proteinortho': 1, 'poff': 1, 'pin': 2, 'faa': 2, 'psq': 2, 'proteinortho-graph': 1}\n",
      "proteinortho: Discrepancy in *.poff-graph files: 0 [FOLDER] vs 1 [REF]\n",
      "proteinortho: Discrepancy in *.proteinortho files: 0 [FOLDER] vs 1 [REF]\n",
      "proteinortho: Discrepancy in *.poff files: 0 [FOLDER] vs 1 [REF]\n",
      "proteinortho: Discrepancy in *.proteinortho-graph files: 0 [FOLDER] vs 1 [REF]\n",
      "\n",
      "Not all files supposed to be generated by proteinortho appear to be present. Running proteinortho now (this may take some time)...\n",
      "b'*****************************************************************'\n",
      "b'Proteinortho with PoFF version 5.16b - An orthology detection tool'\n",
      "b'*****************************************************************'\n",
      "b'Detected 56 available CPU threads, Detected NCBI BLAST version 2.2.31+'\n",
      "b'Checking input files'\n",
      "b'Checking ph_ctg_516/DK_0911_v03_h_ctg.protein.faa... ph_ctg_516/DK_0911_v03_h_ctg.protein.faa\\t13103 genes\\tok'\n",
      "b'Checking ph_ctg_516/DK_0911_v03_p_ctg.protein.faa... ph_ctg_516/DK_0911_v03_p_ctg.protein.faa\\t18181 genes\\tok'\n",
      "b''\n",
      "b'**Step 1**'\n",
      "b'Generating indices'\n",
      "b\"Building database for 'ph_ctg_516/DK_0911_v03_p_ctg.protein.faa'\\t(18181 sequences)\"\n",
      "b\"Building database for 'ph_ctg_516/DK_0911_v03_h_ctg.protein.faa'\\t(13103 sequences)\"\n",
      "b''\n",
      "b'**Step 2**'\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-95-89026442c8a6>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0m__name__\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"__main__\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-92-7714943e8d71>\u001b[0m in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Not all files supposed to be generated by proteinortho appear to be present. Running proteinortho now (this may take some time)...\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m         \u001b[0mwriteBashScript\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'run_proteinortho.sh'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mPROTEINORTHO_OUT_PATH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m         \u001b[0mrunBashScript\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'run_proteinortho.sh'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mPROTEINORTHO_OUT_PATH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Proteinortho finished running.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-90-b9db2c5ba509>\u001b[0m in \u001b[0;36mrunBashScript\u001b[0;34m(bashScript, folderPath)\u001b[0m\n\u001b[1;32m     17\u001b[0m     \u001b[0mp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstdout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPIPE\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSTDOUT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 19\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32min\u001b[0m \u001b[0miter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mb''\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     20\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
