{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook is aimed at identifying unphased regions of the priamry contigs based on short read illumina mapping data. Parts of it are really slow.\n",
    "\n",
    "\n",
    "This notebook was only designed for the purpose of analyzing the Pst-104E genome and now was adapted to the DK09-11 genome. No gurantees it works in any other situtation. It will have spelling errors due to the lack of autocorrection."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Coverage analysis script\n",
    "The original idea of the script is to simply use the coverage dataframe generated by \n",
    "samtools depth -aa ${x} > ${x}.aa.cov and pull out 1x homozygous and unique coverage areas. This is fr now done with the 'DK_0911_v03_p_ctg.bwamem.PRI_NTKN_DK0911.sam.sorted.bam.aa.cov' files.\n",
    "Think about use DeepTools in future when comparing mulitple mapping experiments.\n",
    "\n",
    "Future version of this script should considered tilling sliding windows instead of per base callculation. The choosen cut off seem reasonable based on the observed values of coverage. This may not be the case for all genomes though. Hence a sliding window approach maybe better."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### preserved comments and ideas from the origianl version\n",
    "do a groupby and the mean on this. get ride of the high coverage contigs. calcutate standard statistics. Do a rolling window on each group and convert to a gff file kind of format.\n",
    "also look at the coverage on p only and calcualte coverage there in the same way this should be the diploid coverage really (kind off). \n",
    "Think about combining the two dataframes for pcontigs and do a covariation analysis on the sliding windows look for areas where there is no change for p vs ph mapping and high coverage in both cases. This should be a homozygous region."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-15T03:00:16.589715Z",
     "start_time": "2018-02-15T03:00:15.961084Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-15T03:00:17.243175Z",
     "start_time": "2018-02-15T03:00:16.594043Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "#import seaborn\n",
    "import matplotlib\n",
    "from Bio import SeqIO, SeqUtils\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-15T03:00:17.264368Z",
     "start_time": "2018-02-15T03:00:17.249709Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Define the PATH\n",
    "BASE_AA_PATH = '/home/benjamin/genome_assembly/Warrior'\n",
    "BASE_A_PATH = '/home/benjamin/genome_assembly/Warrior/genome_v04'\n",
    "#for now use the previous mapping that still included high coverage regions\n",
    "COV_IN_PATH = '/home/benjamin/genome_assembly/Warrior/SRM'\n",
    "BAM_IN_PATH = '/home/benjamin/genome_assembly/Warrior/SRM'\n",
    "#apply analysis restricted to final assembly Pst_104E_v12\n",
    "COV_OUT_PATH = os.path.join(BASE_AA_PATH, 'COV')\n",
    "if not os.path.isdir(COV_OUT_PATH):\n",
    "    os.mkdir(COV_OUT_PATH)\n",
    "pwh_list_path = os.path.join(BASE_A_PATH, 'DK_0911_v04_pwh.list')\n",
    "pwoh_list_path = os.path.join(BASE_A_PATH, 'DK_0911_v04_pwoh.list')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-15T03:00:17.274116Z",
     "start_time": "2018-02-15T03:00:17.267986Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "input_genome = 'DK0911_v03'\n",
    "coverage_file_suffix = 'bwamem.PRI_NTKN_DK0911.sam.sorted.bam.aa.cov'\n",
    "output_genome = 'DK0911_v04'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-15T03:00:17.284371Z",
     "start_time": "2018-02-15T03:00:17.277716Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get all the cov files with this coverage_file_suffix. Expacts to get all three p, h and ph mapping\n",
    "cov_files = [os.path.join(COV_IN_PATH, x) for x in os.listdir(COV_IN_PATH) if x.endswith(coverage_file_suffix)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:15.977Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cov_header = [\"contig\", \"position\", 'coverage']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:15.983Z"
    }
   },
   "outputs": [],
   "source": [
    "ph_cov = pd.read_csv([y for y in cov_files if 'ph_ctg' in y][0], sep='\\t', header=None, names=cov_header)\n",
    "print('Read in following file as ph_coverage produced with samtools depth -aa feature: %s' %[y for y in cov_files if 'ph_ctg' in y][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:15.989Z"
    }
   },
   "outputs": [],
   "source": [
    "p_cov = pd.read_csv([y for y in cov_files if 'p_ctg' in y][0], sep='\\t', header=None, names=cov_header)\n",
    "print('Read in following file as p_coverage produced with samtools depth -aa feature: %s' %[y for y in cov_files if 'p_ctg' in y][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:15.993Z"
    }
   },
   "outputs": [],
   "source": [
    "h_cov = pd.read_csv([y for y in cov_files if '_h_ctg' in y][0], sep='\\t', header=None, names=cov_header)\n",
    "print('Read in following file as h_coverage produced with samtools depth -aa feature: %s' %[y for y in cov_files if '_h_ctg' in y][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.000Z"
    }
   },
   "outputs": [],
   "source": [
    "#summarize the mean coverage by contigs for p_contigs when mapping against p_contigs only\n",
    "mean_cov_per_contig = p_cov.groupby('contig').mean()\n",
    "\n",
    "mean_cov_per_contig['coverage'].plot.box()\n",
    "\n",
    "overall_mean = p_cov['coverage'].mean()\n",
    "\n",
    "overall_std = p_cov['coverage'].std()\n",
    "\n",
    "print(\"The mean overall coverage is %.2f and the std is %.2f for p\" % (overall_mean, overall_std))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.005Z"
    }
   },
   "outputs": [],
   "source": [
    "per_contig_mean_p = mean_cov_per_contig['coverage'].mean()\n",
    "per_contig_std_p = mean_cov_per_contig['coverage'].std()\n",
    "print(\"The per contig mean on p is %.2f and the std is %.2f for p\" % (per_contig_mean_p , per_contig_std_p))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.010Z"
    }
   },
   "outputs": [],
   "source": [
    "#summarize the mean coverage by contigs for all contigs when mapping against p and h contigs. Plot only coverage plot for p_contigs\n",
    "mean_cov_per_contig_ph = ph_cov.groupby('contig').mean()\n",
    "mean_cov_per_contig_ph['contig'] = mean_cov_per_contig_ph.index\n",
    "mean_cov_per_contig_ph['coverage'].plot.box(sym ='r+')\n",
    "overall_mean_ph = ph_cov['coverage'].mean()\n",
    "\n",
    "overall_std_ph = ph_cov['coverage'].std()\n",
    "\n",
    "print(\"The mean overall per base coverage is %.2f and the std is %.2f for ph mapping\" % (overall_mean_ph, overall_std_ph))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.014Z"
    }
   },
   "outputs": [],
   "source": [
    "#summarize the mean coverage by contigs for all contigs when mapping against p and h contigs. Plot only coverage plot for p_contigs\n",
    "mean_cov_per_contig_ph_p = mean_cov_per_contig_ph[mean_cov_per_contig_ph['contig'].str.contains('pcontig')]\n",
    "mean_cov_per_contig_ph_p['coverage'].plot.box(sym ='r+')\n",
    "overall_mean_ph_p = mean_cov_per_contig_ph_p['coverage'].mean()\n",
    "\n",
    "overall_std_ph_p = mean_cov_per_contig_ph_p['coverage'].std()\n",
    "\n",
    "print(\"The mean per contig coverage on p is %.2f and the std is %.2f for ph mapping\" % (overall_mean_ph_p, overall_std_ph_p))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.018Z"
    }
   },
   "outputs": [],
   "source": [
    "mean_cov_per_contig_ph[mean_cov_per_contig_ph['contig'].str.contains('hcontig')]['coverage'].plot.box(sym ='r+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.021Z"
    }
   },
   "outputs": [],
   "source": [
    "#summarize the mean coverage by contigs for h contigs when mapping against h contigs. Plot only coverage plot for h_contigs\n",
    "mean_cov_per_contig_h = h_cov.groupby('contig').mean()\n",
    "mean_cov_per_contig_h['contig'] = mean_cov_per_contig_h.index\n",
    "mean_cov_per_contig_h['coverage'].plot.box(sym ='r+')\n",
    "\n",
    "overall_mean_h = h_cov['coverage'].mean()\n",
    "\n",
    "overall_std_h = h_cov['coverage'].std()\n",
    "\n",
    "print(\"The mean per base coverage is %.2f and the std is %.2f for ph mapping\" % (overall_mean_h, overall_std_h))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.025Z"
    }
   },
   "outputs": [],
   "source": [
    "mean_cov_per_contig.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.028Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#these are the primary contigs with a coverage greater than 50\n",
    "pcongtig_greater_50 = list(mean_cov_per_contig[mean_cov_per_contig.coverage > 50 ].index.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.033Z"
    }
   },
   "outputs": [],
   "source": [
    "pcongtig_greater_50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.038Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "congtig_ph_greater_50 = list(mean_cov_per_contig_ph[mean_cov_per_contig_ph.coverage > 50 ].index.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.041Z"
    }
   },
   "outputs": [],
   "source": [
    "congtig_ph_greater_50"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Look at contigs 52, 101, and 106 if this is something particular"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.045Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hcongtig_greater_50 = list(mean_cov_per_contig_h[mean_cov_per_contig_h.coverage > 50 ].index.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.052Z"
    }
   },
   "outputs": [],
   "source": [
    "hcongtig_greater_50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.060Z"
    }
   },
   "outputs": [],
   "source": [
    "{x[:-4] for x in hcongtig_greater_50}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.064Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get a ph coverage dataframe that does not included the pcontigs greater 50x coverage\n",
    "ph_cov_smaller_50 = ph_cov[~ph_cov['contig'].isin(pcongtig_greater_50)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.068Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#claculate the mean contig coverage\n",
    "mean_cov_ph_contig_smaller_50 = ph_cov_smaller_50.groupby('contig').mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.071Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#fix this up by removing hcontigs associated with pcontigs that have greater 50x coverage\n",
    "mean_cov_ph_contig_smaller_50 = mean_cov_ph_contig_smaller_50[~mean_cov_ph_contig_smaller_50.index.str[:-4].isin(\\\n",
    "                              [x.replace('pc', 'hc') for x in pcongtig_greater_50])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.076Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get the list of p contigs with haplotig [pwh] and without haplotig [pwoh]\n",
    "pwh_list = pd.read_csv(pwh_list_path, sep='\\t', header=None)[0].tolist()\n",
    "pwoh_list = pd.read_csv(pwoh_list_path, sep='\\t', header=None)[0].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.080Z"
    }
   },
   "outputs": [],
   "source": [
    "#mean_cov_contig_s2000_ph_pwh represents all contigs that are below 2000x coverage and are pwh contigs\n",
    "mean_cov_ph_contig_smaller_50_pwh = mean_cov_ph_contig_smaller_50[mean_cov_ph_contig_smaller_50.index.isin(pwh_list)]\n",
    "mean_cov_ph_contig_smaller_50_pwh['coverage'].plot.box()\n",
    "plt.title('mean_cov_contig_s50_ph_pwh')\n",
    "mean_s50_ph_pwh = mean_cov_ph_contig_smaller_50_pwh['coverage'].mean()\n",
    "std_s50_ph_pwh = mean_cov_ph_contig_smaller_50_pwh['coverage'].std()\n",
    "print(\"The mean per contig coverage for s50 contigs of pwh contigs while ph mapping is %.2f and the std is %.2f\" % (mean_s50_ph_pwh, std_s50_ph_pwh))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.084Z"
    }
   },
   "outputs": [],
   "source": [
    "#mean_cov_contig_s2000_ph_pwh represents all contigs that are below 2000x coverage and are pwh contigs\n",
    "mean_cov_ph_contig_smaller_50_pwoh = mean_cov_ph_contig_smaller_50[mean_cov_ph_contig_smaller_50.index.isin(pwoh_list)]\n",
    "mean_cov_ph_contig_smaller_50_pwoh['coverage'].plot.box()\n",
    "plt.title('mean_cov_contig_s50_ph_pwoh')\n",
    "mean_s50_ph_pwoh = mean_cov_ph_contig_smaller_50_pwoh['coverage'].mean()\n",
    "std_s50_ph_pwoh = mean_cov_ph_contig_smaller_50_pwoh['coverage'].std()\n",
    "print(\"The mean per contig coverage for s50 contigs of pwoh contigs while ph mapping is %.2f and the std is %.2f\" % (mean_s50_ph_pwoh, std_s50_ph_pwoh))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.087Z"
    }
   },
   "outputs": [],
   "source": [
    "#mean_cov_contig_s2000_ph_pwh represents all contigs that are below 2000x coverage and are pwh contigs\n",
    "mean_cov_ph_contig_smaller_50_h = mean_cov_ph_contig_smaller_50[mean_cov_ph_contig_smaller_50.index.str.startswith('hcontig')]\n",
    "mean_cov_ph_contig_smaller_50_h['coverage'].plot.box()\n",
    "plt.title('mean_cov_contig_s50_ph_h')\n",
    "mean_s50_ph_h = mean_cov_ph_contig_smaller_50_h['coverage'].mean()\n",
    "std_s50_ph_h = mean_cov_ph_contig_smaller_50_h['coverage'].std()\n",
    "print(\"The mean per contig coverage for s50 contigs of h contigs while ph mapping is %.2f and the std is %.2f\" % (mean_s50_ph_h, std_s50_ph_h))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.092Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mean_cov_ph_contig_smaller_50_p = mean_cov_ph_contig_smaller_50[mean_cov_ph_contig_smaller_50.index.str.startswith('pcon')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.097Z"
    }
   },
   "outputs": [],
   "source": [
    "mean_cov_ph_contig_smaller_50_p['coverage'].plot.box()\n",
    "plt.title('mean_cov_contig_s50_ph_p')\n",
    "mean_s50_ph_p = mean_cov_ph_contig_smaller_50_p['coverage'].mean()\n",
    "std_s50_ph_p = mean_cov_ph_contig_smaller_50_p['coverage'].std()\n",
    "print(\"The mean per contig coverage for s50 contigs of p contigs while ph mapping is %.2f and the std is %.2f\" % (mean_s50_ph_p, std_s50_ph_p))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.102Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#think about what thresholds to pick in the long run.\n",
    "threshold_up_ph_p = mean_s50_ph_p + 2*std_s50_ph_p \n",
    "threshold_down_ph_p = mean_s50_ph_p - 2*std_s50_ph_p \n",
    "\n",
    "#threshold_up = mean_s2000 + 2*std_s2000\n",
    "#threshold_down = mean_s2000 - 2*std_s2000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.109Z"
    }
   },
   "outputs": [],
   "source": [
    "#potnetial fully homozygous contigs\n",
    "mean_cov_per_contig_ph_pwoh = mean_cov_per_contig_ph[mean_cov_per_contig_ph.index.isin(pwoh_list)]\n",
    "mean_cov_per_contig_ph_pwoh[mean_cov_per_contig_ph_pwoh.coverage > threshold_up_ph_p]['coverage']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.113Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mean_file_name = os.path.join(COV_OUT_PATH, '%s_mean_cov.txt' % output_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.117Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(mean_file_name,'w') as outfn:\n",
    "    print('mean_s50_ph_p: %f' % mean_s50_ph_p,file=outfn)\n",
    "    print('std_s50_ph_p: %f' % std_s50_ph_p, file=outfn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.121Z"
    }
   },
   "outputs": [],
   "source": [
    "!cat {mean_file_name}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.125Z"
    }
   },
   "outputs": [],
   "source": [
    "ph_cov.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.130Z"
    }
   },
   "outputs": [],
   "source": [
    "print(ph_cov[ph_cov.contig.str.startswith('pcont')]['coverage'].mean())\n",
    "print(ph_cov[ph_cov.contig.str.startswith('pcont')]['coverage'].std())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.135Z"
    }
   },
   "outputs": [],
   "source": [
    "ph_cov_smaller_50[ph_cov_smaller_50.contig.str.startswith('pcont')].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.138Z"
    }
   },
   "outputs": [],
   "source": [
    "#potnetial fully homozygous contigs if we are not considering the presence of haplotigs. \n",
    "#some of these haplotigs may also be collapsed compared to there primary contigs.\n",
    "mean_cov_per_contig_ph[mean_cov_per_contig_ph.coverage > threshold_up_ph_p]['coverage']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Looking a bit into the nucmer plots of these contigs for visualization\n",
    "* the hcontigs_000_xxx are fairly small sequences and maybe repeats.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.141Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-02-15T03:00:16.145Z"
    }
   },
   "outputs": [],
   "source": [
    "%run DK_0911_coverage_analysis_submission_part2_parallel.ipynb"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
