{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "from Bio import SeqIO\n",
    "from Bio import AlignIO\n",
    "import distance\n",
    "import editdistance\n",
    "import math\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "BASE_OUT_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/post_analysis/'\n",
    "ALLELE_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles_v031_np/allele_analysis/alleles_proteinortho_graph516/'\n",
    "GENOME_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/genome_v031/'\n",
    "\n",
    "GENOME = 'DK_0911_v031'\n",
    "P_GENOME = GENOME + '_p_ctg'\n",
    "H_GENOME = GENOME + '_h_ctg'\n",
    "\n",
    "PAML_PATH = os.path.join(BASE_OUT_PATH, 'paml')\n",
    "if not os.path.exists(PAML_PATH):\n",
    "    os.mkdir(PAML_PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create DataFrame with all alleles\n",
    "overlapDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.h_contig_overlap.alleles' % P_GENOME), header=None, names=['alleleOne', 'alleleTwo'], sep='\\t')\n",
    "noOverlapDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.no_specific_h_contig_overlap.alleles' % P_GENOME), header=None, names=['alleleOne', 'alleleTwo'], sep='\\t')\n",
    "diffContigDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.no_respective_h_contig_overlap.alleles' % P_GENOME), header=None, names=['alleleOne', 'alleleTwo'], sep='\\t')\n",
    "manualAssignDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.manual_assigned.alleles' % H_GENOME), header=None, names=['alleleOne', 'alleleTwo'], sep='\\t')\n",
    "\n",
    "overlapDf['matchType'] = 'overlap'\n",
    "noOverlapDf['matchType'] = 'no_overlap'\n",
    "diffContigDf['matchType'] = 'different_pcontig'\n",
    "manualAssignDf['matchType'] = 'manual_assigned'\n",
    "\n",
    "alleleDf = overlapDf.append([noOverlapDf, diffContigDf, manualAssignDf])\n",
    "\n",
    "alleleDf['folder'] = alleleDf.alleleOne + '_' + alleleDf.alleleTwo\n",
    "alleleDf.set_index('folder', inplace=True)\n",
    "\n",
    "assert(len(alleleDf) == len(overlapDf) + len(noOverlapDf) + len(diffContigDf) + len(manualAssignDf))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine haplotig & primary contig fasta files for each allele type\n",
    "# (protein, gene, CDS), and change headings so that they use id \n",
    "# (e.g. evm.model.pcontig_057.39) instead of locus tag (e.g. DK0911_16805).\n",
    "\n",
    "## obsolete in v031. All relevant fasta files are generated in DK_0911_v03_generate_fa_files_from_gff3.ipynb\n",
    "# os.chdir('/home/gamran/genome_analysis/Warrior/Richard/scripts')\n",
    "# %run DK_0911_v03_dictionaries.ipynb\n",
    "# locusToIdDict = getLocusToIdDict()\n",
    "\n",
    "# def combineCtgFastaFiles(fastaFiles, fOut):\n",
    "#     '''combines fasta files and rewrites headings from locus tags\n",
    "#     (DK0911_16805) to id tags (evm.model.pcontig_057.39)'''\n",
    "#     with open(fOut, 'w') as outFile:\n",
    "#         for fastaFile in fastaFiles:\n",
    "#             with open(fastaFile, 'r') as inFile:\n",
    "#                 for line in inFile:\n",
    "#                     if '>' in line:\n",
    "#                         line = '>' + locusToIdDict[line[1:-1]] + '\\n'\n",
    "#                     outFile.write(line)\n",
    "#     return True\n",
    "\n",
    "# hProteinFasta = os.path.join(GENOME_PATH, 'DK_0911_v03_h_ctg.protein.fa')\n",
    "# pProteinFasta = os.path.join(GENOME_PATH, 'DK_0911_v03_p_ctg.protein.fa')\n",
    "# hGeneFasta = os.path.join(GENOME_PATH, 'DK_0911_v03_h_ctg.gene.fa')\n",
    "# pGeneFasta = os.path.join(GENOME_PATH, 'DK_0911_v03_p_ctg.gene.fa')\n",
    "# hCDSFasta = os.path.join(GENOME_PATH, 'DK_0911_v03_h_ctg.CDS.fa')\n",
    "# pCDSFasta = os.path.join(GENOME_PATH, 'DK_0911_v03_p_ctg.CDS.fa')\n",
    "\n",
    "PH_PROTEIN_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.protein.fa')\n",
    "PH_GENE_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.gene.fa')\n",
    "PH_CDS_FASTA = os.path.join(GENOME_PATH, GENOME + '_ph_ctg.cds.fa')\n",
    "\n",
    "# combineCtgFastaFiles([hProteinFasta, pProteinFasta], PH_PROTEIN_FASTA)\n",
    "# combineCtgFastaFiles([hGeneFasta, pGeneFasta], PH_GENE_FASTA)\n",
    "# combineCtgFastaFiles([hCDSFasta, pCDSFasta], PH_CDS_FASTA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getFastaDict(fastaFile):\n",
    "    d = {}\n",
    "    for gene in SeqIO.parse(fastaFile, 'fasta'):\n",
    "        d[gene.id] = gene\n",
    "    return d\n",
    "\n",
    "SEQRECORD_PROTEIN_DICT = getFastaDict(PH_PROTEIN_FASTA)\n",
    "SEQRECORD_GENE_DICT = getFastaDict(PH_GENE_FASTA)\n",
    "SEQRECORD_CDS_DICT = getFastaDict(PH_CDS_FASTA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# def getSpecificFasta(geneList, fastaFile):\n",
    "#     '''Returns the fasta of gene in list as SeqIO object '''\n",
    "#     # Ben's original code had CDS with '.model.' and \n",
    "#     # protein as '.TU.'\n",
    "#     l = []\n",
    "#     for gene in SeqIO.parse(fastaFile, 'fasta'):\n",
    "#         if gene.id in geneList:\n",
    "#             l.append(gene)\n",
    "#     return l\n",
    "# \n",
    "# def writeAllelicFasta(alleleOne, alleleTwo, alleleType, outPath):\n",
    "#     '''writes fasta file containing fasta information for two alleles\n",
    "#     in the outPath'''\n",
    "#     assert(alleleType.upper() in ['CDS', 'GENE', 'PROTEIN'])\n",
    "#     fastaFilePath = globals()['PH_' + alleleType.upper() + '_FASTA'] # e.g. alleleType = 'GENE', then fastaFile = PH_GENE_FASTA\n",
    "#     specificFastaList = getSpecificFasta([alleleOne, alleleTwo], fastaFilePath)\n",
    "#     with open(os.path.join(outPath, alleleType.lower() + '.fa'), 'w') as outFile:\n",
    "#         SeqIO.write(specificFastaList, outFile, 'fasta')\n",
    "#     return True\n",
    "\n",
    "def writeAllelicFasta(alleleOne, alleleTwo, alleleType, outPath):\n",
    "    '''writes fasta file containing fasta information for two alleles\n",
    "    in the outPath'''\n",
    "    assert(alleleType.upper() in ['CDS', 'GENE', 'PROTEIN'])\n",
    "    \n",
    "    seqRecordDict = globals()['SEQRECORD_' + alleleType.upper() + '_DICT']\n",
    "    \n",
    "    alleleSeqRecords = [seqRecordDict[alleleOne], seqRecordDict[alleleTwo]]\n",
    "    with open(os.path.join(outPath, alleleType.lower() + '.fa'), 'w') as outFile:\n",
    "        SeqIO.write(alleleSeqRecords, outFile, 'fasta')\n",
    "    return True\n",
    "\n",
    "def writeAlignmentScript(alleleOutPath, scriptLoc = os.path.join(PAML_PATH, 'paml_script.sh')):\n",
    "    with open(scriptLoc, 'a') as outFile:\n",
    "        print('cd %s' % alleleOutPath, file=outFile)\n",
    "        print('/home/gamran/anaconda3/muscle3.8.31_i86linux64 -clwstrict -in protein.fa -out protein.aln', file=outFile)\n",
    "        print('perl /home/gamran/anaconda3/pal2nal.v14/pal2nal.pl -output paml protein.aln cds.fa > cds_codon.aln', file=outFile)\n",
    "        print('perl /home/gamran/anaconda3/pal2nal.v14/pal2nal.pl protein.aln cds.fa > cds_codon.clustal', file=outFile)\n",
    "        print('cp %s/yn00.ctl ./' % PAML_PATH, file=outFile)\n",
    "        print('/home/gamran/anaconda3/paml4.9g/bin/yn00', file=outFile)\n",
    "    return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepareAlignmentBashScript(scriptLoc = os.path.join(PAML_PATH, 'paml_script.sh')):\n",
    "    with open(scriptLoc, 'w') as pamlScript:\n",
    "        print('#!/bin/bash', file=pamlScript)\n",
    "\n",
    "    for index, [alleleOne, alleleTwo, alleleType] in alleleDf.iterrows():\n",
    "\n",
    "        alleleOutPath = os.path.join(PAML_PATH, '%s_%s' % (alleleOne, alleleTwo))\n",
    "        if not os.path.exists(alleleOutPath):\n",
    "            os.mkdir(os.path.join(PAML_PATH, '%s_%s' % (alleleOne, alleleTwo)))\n",
    "\n",
    "        writeAllelicFasta(alleleOne, alleleTwo, 'CDS', alleleOutPath)\n",
    "        writeAllelicFasta(alleleOne, alleleTwo, 'PROTEIN', alleleOutPath)\n",
    "\n",
    "        writeAlignmentScript(alleleOutPath, os.path.join(PAML_PATH, 'paml_script.sh'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# alleleDf['folder'] = alleleDf.alleleOne + '_' + alleleDf.alleleTwo\n",
    "# alleleDf.index = alleleDf['folder']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def assignDistancesToAlleles(df, folder, alignmentFile, alleleType):\n",
    "    '''Adds hamming and levenshtein distance columns to an allele pair\n",
    "    (indexed by 'folder' name) in df'''\n",
    "    assert(alleleType.upper() in ['PROTEIN', 'CDS', 'GENE'])\n",
    "    seq1, seq2 = AlignIO.read(open(alignmentFile, 'r'), format='clustal', seq_count=2)\n",
    "    seq1 = str(seq1.seq).upper()\n",
    "    seq2 = str(seq2.seq).upper()\n",
    "    assert(len(seq1) == len(seq2))\n",
    "    df.loc[folder, alleleType.lower() + '_hamming'] = distance.hamming(seq1, seq2, normalized=True)\n",
    "    df.loc[folder, alleleType.lower() + '_levenshtein'] = editdistance.eval(seq1, seq2)/len(seq1)\n",
    "    return df\n",
    "\n",
    "def assignDistancesToAllAlleles(alleleDf):\n",
    "    count = 0\n",
    "    total = len(alleleDf)\n",
    "    percentDone = 0\n",
    "    \n",
    "    print(\"Calculating distances and adding them to the allele DataFrame...\")\n",
    "    \n",
    "    for folder in alleleDf.index:\n",
    "\n",
    "        proteinAlignmentFile = os.path.join(PAML_PATH, folder, 'protein.aln')\n",
    "        alleleDf = assignDistancesToAlleles(alleleDf, folder, proteinAlignmentFile, 'PROTEIN')\n",
    "\n",
    "        cdsAlignmentFile = os.path.join(PAML_PATH, folder, 'cds_codon.clustal')\n",
    "        alleleDf = assignDistancesToAlleles(alleleDf, folder, cdsAlignmentFile, 'CDS')\n",
    "\n",
    "        count += 1\n",
    "        if round(count/total * 100) > percentDone:\n",
    "            percentDone = round(count/total * 100)\n",
    "            print(\"%s%% complete\" % percentDone)\n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_dNdS_to_df(line, alleleDf, folder, dNdS_label):\n",
    "    dN = re.findall(r'dN = [-| ]?(.*) w', line)[0]\n",
    "    dS = re.findall(r'dS = [-| ]?(.*) dN', line)[0]\n",
    "    return assign_dNdS(dN, dS, alleleDf, folder, dNdS_label)\n",
    "\n",
    "def assign_dNdS(dN, dS, alleleDf, folder, dNdS_label):\n",
    "    if float(dS) > 0:\n",
    "        alleleDf.loc[folder, dNdS_label] = float(dN)/float(dS)\n",
    "    else:\n",
    "        alleleDf.loc[folder, dNdS_label] = np.nan\n",
    "    return alleleDf\n",
    "\n",
    "def assign_dNdS_to_all_alleles(alleleDf):\n",
    "    for folder in alleleDf.index:\n",
    "        alleleYn = os.path.join(PAML_PATH, folder,'yn.out')\n",
    "        with open(alleleYn, 'r') as ynOut:\n",
    "            #now loop over the lines and parse out stuff\n",
    "            for i, line in enumerate(ynOut):\n",
    "                if line.startswith('seq. seq. ') and i > 0:\n",
    "                    next(ynOut) # we want the line that is two after the line starting with 'seq. seq '\n",
    "                    dataLine = next(ynOut)\n",
    "                    dN = dataLine.split('+-')[0].rstrip().split(' ')[-1]\n",
    "                    dS = dataLine.split('+-')[1].rstrip().split(' ')[-1]\n",
    "                    alleleDf = assign_dNdS(dN, dS, alleleDf, folder, 'yn00_dN/dS')\n",
    "                elif line.startswith('LWL85:') and 'nan' not in line:\n",
    "                    alleleDf = parse_dNdS_to_df(line, alleleDf, folder, 'LWL85_dN/dS')\n",
    "                elif line.startswith('LWL85m:') and 'nan' not in line:\n",
    "                    alleleDf = parse_dNdS_to_df(line, alleleDf, folder, 'LWL85m_dN/dS')\n",
    "                elif line.startswith('LPB93:') and 'nan' not in line:\n",
    "                    alleleDf = parse_dNdS_to_df(line, alleleDf, folder, 'LPB93_dN/dS')\n",
    "                else:\n",
    "                    continue\n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def main(alleleDf = alleleDf):\n",
    "    prepareAlignmentBashScript(os.path.join(PAML_PATH, 'paml_script.sh'))\n",
    "    \n",
    "    # !bash {os.path.join(PAML_PATH, 'paml_script.sh')}\n",
    "    \n",
    "    distDfPath = os.path.join(BASE_OUT_PATH, GENOME+'_distanceDf.df')\n",
    "    if os.path.exists(distDfPath) and os.path.getsize(distDfPath) > 0:\n",
    "        print(\"DataFrame with distance calculations at %s appears to have already been generated. Reading in this dataframe instead of re-generating it.\" % distDfPath)\n",
    "        alleleDf = pd.read_csv(distDfPath, sep='\\t', index_col=0)\n",
    "    else:\n",
    "        alleleDf = assignDistancesToAllAlleles(alleleDf)\n",
    "        alleleDf.to_csv(distDfPath, sep='\\t')\n",
    "        pd.util.testing.assert_frame_equal(alleleDf, pd.read_csv(distDfPath, sep='\\t', index_col=0))\n",
    "    \n",
    "    alleleDf = assign_dNdS_to_all_alleles(alleleDf)\n",
    "    alleleDf.to_csv(distDfPath, sep='\\t')\n",
    "    \n",
    "    return alleleDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DataFrame with distance calculations at /home/gamran/genome_analysis/Warrior/Richard/output/post_analysis/DK_0911_v031_distanceDf.df appears to have already been generated. Reading in this dataframe instead of re-generating it.\n"
     ]
    }
   ],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    alleleDf = main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
