{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Defining Alleles v02"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook is aimed at doing allelic comparison between primary contigs and hapltotigs. This was used to generate DataFrames used during the analysis and evolved during the analysis. So it might be a bit more complicated than it needs to be, but it works.\n",
    "\n",
    "This notebook does NOT use proteinortho in its identification of allelic pairs\n",
    "\n",
    "Based on original code by Benjamin Schwessinger."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook defines alleles based on a Falcon-Unzip assembly where relational information between two contigs are available.\n",
    "\n",
    "#### The input as follows:\n",
    "* Assemblytics alingment of all haplotigs onto their corresponding primary contigs. The *.oriented_coords.csv will be converted to a gff file of haplotig alingments onto their respective primary contigs\n",
    "* gff file of all features. This will be filtered down to genes only and separated out for each contig\n",
    "* protein and gene fa files for reciprocal blast of haplotig sequences onto primary sequences\n",
    "\n",
    "#### What happens:\n",
    "* for each primary gene/protein the the overlapping haplotig is pulled out and added to the blast dataframe\n",
    "* this blast dataframe of primary proteins onto haplotig proteins is than used for filtering to provide the following files\n",
    "* best blast hits are first filtered on e-value minimums followed by BitScore maximum\n",
    "\n",
    "#### What comes out in the alleles folder:\n",
    "##### for primary genes/proteins\n",
    "* a set of files that contain two proteins IDs identifying the two alleles.  \n",
    "* *_p_ctg.h_contig_overlap.alleles contains best blast hits of the primary protein on haplotig proteins of the overlapping haplotig (truely linked alleles).\n",
    "* *_p_ctg.no_respective_h_contig_overlap.alleles contains all primary protein that don't have a blast hit on the overlapping haplotig but another haplotig associated with its primary contig.\n",
    "* *_p_ctg.no_specific_h_contig_overlap.alleles contains all primary proteins that don't have a blast hit on an associated haplotig but another haplotig.\n",
    "* *_p_ctg.no_alleles contains all primary proteins without blast hit on haplotigs.\n",
    "\n",
    "##### for haplotig genes/proteins\n",
    "* *_h_ctg.no.alleles contains all haplotig proteins that are not associated with a primary allele in the above analysis.\n",
    "* *_h_ctg.no.best_p_hits.alleles contains all haplotig proteins that are not associated with a primary allele in the above analysis but have a blast hit non-withstanding. This might be duplications in the haplotigs.\n",
    "* *_h_ctg.no.no_p_hits.alleles contains all haplotig proteins that have no blast hit.\n",
    "\n",
    "##### filtering of blast output\n",
    "* filtering of blast output is possible on Query coverage of the blast hit [Qcov] and the percentage identity of the aligenment [PctID]. This generates additional outfiles with *.QcovXX.PctIDYY.alleles. The default values are set to Qcov80 and PctID70.\n",
    "\n",
    "#### What else to consider:\n",
    "* the script was tested for outputing the right number of protein sequences for primary and haplotig sequences.\n",
    "* this is a script and not a program. No warrenties.\n",
    "* Feedback always welcome.\n",
    "* Path and other variables are defined at the top of the script.\n",
    "* It assumes that primary contig sequences are labled as e.g. pcontig_000 and associated haplotigs as hcontig_000_0xx.\n",
    "\n",
    "#### Downstream considerations:\n",
    "* Why have some proteins no blast hits? Is it simple that their gene models are missing? Seen this and working on this.\n",
    "* In case of primary genes without blast hits do those lie in homozygous coverage areas? Working on this.\n",
    "* Are some of the h genes/proteins without alleles but with blast hits duplications that diverget?\n",
    "* Do some of the primary proteins/genes without blast hit on haplotigs have a good blast hit on other primary contigs? Is this reciprocal?\n",
    "* Can we use some of this information to talk about linkage of contigs or does this go to far without fully phased genomes using loooong reads or single nucleii sequencing? Would say yes so far.\n",
    "* Some primary proteins have two equal got hits are those duplications in the haplotig?\n",
    "* If we perform downstream genetic variation analysis between alleles do we see any GO, KEGG, effector or other functional domains enriched?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "import pysam\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from pybedtools import BedTool\n",
    "import numpy as np\n",
    "import pybedtools\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import sys\n",
    "import subprocess\n",
    "import shutil\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define the PATH\n",
    "BASE_OUT_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles_v031_np'\n",
    "GENOME_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/genome_v031/'\n",
    "ASSEMBLYTICS_IN_PATH = '/home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/Assemblytics'\n",
    "\n",
    "BLAST_DB = os.path.join(BASE_OUT_PATH, 'blast_DB')\n",
    "OUT_PATH = os.path.join(BASE_OUT_PATH, 'allele_analysis')\n",
    "OUT_tmp = os.path.join(OUT_PATH, 'tmp')\n",
    "PROTEINORTHO_OUT_PATH = os.path.join(BASE_OUT_PATH, 'proteinortho')\n",
    "\n",
    "#renamed the allele path to reflect the proteinortho results\n",
    "ALLELE_PATH = os.path.join(OUT_PATH, 'alleles_proteinortho_graph516')\n",
    "if not os.path.isdir(OUT_PATH):\n",
    "    os.mkdir(OUT_PATH)\n",
    "if not os.path.isdir(OUT_tmp):\n",
    "    os.mkdir(OUT_tmp)\n",
    "if not os.path.exists(ALLELE_PATH):\n",
    "    os.mkdir(ALLELE_PATH)\n",
    "if not os.path.exists(BLAST_DB):\n",
    "    os.mkdir(BLAST_DB)\n",
    "if not os.path.exists(PROTEINORTHO_OUT_PATH):\n",
    "    os.mkdir(PROTEINORTHO_OUT_PATH)\n",
    "\n",
    "#Define your p and h genome and move it into the allele analysis folder\n",
    "P_GENOME = 'DK_0911_v031_p_ctg'\n",
    "H_GENOME = 'DK_0911_v031_h_ctg'\n",
    "for x in (x + '.fa' for x in [P_GENOME, H_GENOME]):\n",
    "    shutil.copy2(GENOME_PATH+'/'+x, OUT_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define ENV parameters for blast hits and threads used in blast analysis\n",
    "n_threads = 16\n",
    "e_value = 1e-3\n",
    "Qcov_cut_off = 0 #this defines the mimimum coverage of the Query to be required for filtering. Will become part of name.\n",
    "PctID_cut_off = 70 #this defines the mimimum PctID accross the alignment to be required for filtering. Will become part of name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def att_column_p(x, y, z, a, b):\n",
    "    '''Generate attribute column of h on p p_gff dataframe.\n",
    "    '''\n",
    "    string = 'h_contig=%s;query_start=%s;query_stop=%s;query_length=%s;query_aln_ln=%s' % (x, str(y), str(z), str(a), str(b))\n",
    "    return string\n",
    "\n",
    "def att_column_h(x, y, z, a, b):\n",
    "    '''Generate attribute column of h on p h_gff dataframe.\n",
    "    '''\n",
    "    string = 'p_contig=%s;ref_start=%s;ref_stop=%s;ref_length=%s;ref_aln_ln=%s' % (x, str(y), str(z), str(a), str(b))\n",
    "    return string\n",
    "\n",
    "def gene_contig_subset(df, contig, feature):\n",
    "    '''Input the gff and subset the feature columne by gene.\n",
    "        Return data frame subset by contig and gene'''\n",
    "    tmp_df = df[(df[0].str.contains(contig)) & (df[2]==feature)]\n",
    "    tmp_df.sort_values(3,inplace=True)\n",
    "    tmp_df.reset_index(drop=True, inplace=True)\n",
    "    return tmp_df\n",
    "\n",
    "def same_contig_blast(x,y):\n",
    "    '''Function that checks if the blast hit in column y is on the same contig as the the query sequence in\n",
    "    column y.\n",
    "    '''\n",
    "    q_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', x).group(1)\n",
    "    if y != 'NaN':\n",
    "        hit_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', y).group(1)\n",
    "    else:\n",
    "        hit_contig = 'NaN'\n",
    "    return q_contig == hit_contig\n",
    "\n",
    "def target_on_mapped_haplotig(t_contig, h_contig_overlap):\n",
    "    '''Simple function that checks if the target contig is in the list of h overlapping contigs'''\n",
    "    if t_contig == False or h_contig_overlap == False:\n",
    "        return False\n",
    "    return t_contig in h_contig_overlap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Considerations for blast analysis\n",
    "Do gene blast and protein blast. Initially go off protein blast results. They should be more imformative as it also includes the coding region and the frame.\n",
    "> Pull this in as a dataframe and filter it down as well to each contig. Do blasts both ways as well.\n",
    "> Copy over primary & haplotig .protein.fa and .gene.fa files from the GENOME_PATH, if not already present."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All databases generated and ready to go!\n"
     ]
    }
   ],
   "source": [
    "#generate the blast databases if not already present\n",
    "os.chdir(BLAST_DB)\n",
    "# copy over .protein.fa and .gene.fa files from the GENOME_PATH folder\n",
    "# if not already present, for both primary and haplotigs\n",
    "shutil.copy2(os.path.join(GENOME_PATH, P_GENOME + '.protein.fa'), BLAST_DB)\n",
    "shutil.copy2(os.path.join(GENOME_PATH, P_GENOME + '.gene.fa'), BLAST_DB)\n",
    "shutil.copy2(os.path.join(GENOME_PATH, H_GENOME + '.protein.fa'), BLAST_DB)\n",
    "shutil.copy2(os.path.join(GENOME_PATH, H_GENOME + '.gene.fa'), BLAST_DB)\n",
    "\n",
    "blast_dir_content = os.listdir(BLAST_DB)\n",
    "for x in blast_dir_content:\n",
    "    if x.endswith('.fa') and ({os.path.isfile(x + e) for e in ['.psq', '.phr', '.pin'] } != {True}\\\n",
    "           and {os.path.isfile(x + e) for e in ['.nin', '.nhr', '.nsq'] } != {True} ):\n",
    "\n",
    "        make_DB_options = ['-in']\n",
    "        make_DB_options.append(x)\n",
    "        make_DB_options.append('-dbtype')\n",
    "        if 'protein' in x:\n",
    "            make_DB_options.append('prot')\n",
    "        else:\n",
    "            make_DB_options.append('nucl')\n",
    "        make_DB_command = 'makeblastdb %s' % ' '.join(make_DB_options)\n",
    "        make_DB_stderr = subprocess.check_output(make_DB_command, shell=True, stderr=subprocess.STDOUT)\n",
    "        print('%s is done!' % make_DB_command)\n",
    "print(\"All databases generated and ready to go!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Blasting DK_0911_v031_h_ctg.gene.fa ...\n",
      "Blast command generated:\n",
      "blastn -query DK_0911_v031_p_ctg.gene.fa -db DK_0911_v031_h_ctg.gene.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles_v031_np/allele_analysis/DK_0911_v031_p_ctg.DK_0911_v031_h_ctg.0.001.blastn.outfmt6\n",
      "Blast command previously completed!\n",
      "\n",
      "Blasting DK_0911_v031_p_ctg.gene.fa ...\n",
      "Blast command generated:\n",
      "blastn -query DK_0911_v031_h_ctg.gene.fa -db DK_0911_v031_p_ctg.gene.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles_v031_np/allele_analysis/DK_0911_v031_h_ctg.DK_0911_v031_p_ctg.0.001.blastn.outfmt6\n",
      "Blast command previously completed!\n",
      "\n",
      "Blasting DK_0911_v031_p_ctg.protein.fa ...\n",
      "Blast command generated:\n",
      "blastp -query DK_0911_v031_h_ctg.protein.fa -db DK_0911_v031_p_ctg.protein.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles_v031_np/allele_analysis/DK_0911_v031_h_ctg.DK_0911_v031_p_ctg.0.001.blastp.outfmt6\n",
      "Blast command previously completed!\n",
      "\n",
      "Blasting DK_0911_v031_h_ctg.protein.fa ...\n",
      "Blast command generated:\n",
      "blastp -query DK_0911_v031_p_ctg.protein.fa -db DK_0911_v031_h_ctg.protein.fa -outfmt 6 -evalue 0.001 -num_threads 16 > /home/gamran/genome_analysis/Warrior/Richard/output/defining_alleles_v031_np/allele_analysis/DK_0911_v031_p_ctg.DK_0911_v031_h_ctg.0.001.blastp.outfmt6\n",
      "Blast command previously completed!\n"
     ]
    }
   ],
   "source": [
    "blast_dict = {}\n",
    "blast_dict['gene_fa'] = [x for x  in os.listdir(BLAST_DB) if x.endswith('gene.fa') and 'ph_ctg' not in x]\n",
    "blast_dict['protein_fa'] = [x for x  in os.listdir(BLAST_DB) if x.endswith('protein.fa') and 'ph_ctg' not in x]\n",
    "blast_stderr_dict = {}\n",
    "for key in blast_dict.keys():\n",
    "    for n, query in enumerate(blast_dict[key]):\n",
    "        tmp_list = blast_dict[key][:]\n",
    "        tmp_stderr_list = []\n",
    "        #this loops through all remaining files and does blast against all those\n",
    "        del tmp_list[n]\n",
    "        for db in tmp_list:\n",
    "            print(\"\\nBlasting %s ...\" %db)\n",
    "            blast_options = ['-query']\n",
    "            blast_options.append(query)\n",
    "            blast_options.append('-db')\n",
    "\n",
    "            blast_options.append(db)\n",
    "            blast_options.append('-outfmt 6')\n",
    "            blast_options.append('-evalue')\n",
    "            blast_options.append(str(e_value))\n",
    "            blast_options.append('-num_threads')\n",
    "            blast_options.append(str(n_threads))\n",
    "            blast_options.append('>')\n",
    "            if 'gene' in query and 'gene' in db:\n",
    "                out_name_list = [ query.split('.')[0], db.split('.')[0], str(e_value), 'blastn.outfmt6']\n",
    "                out_name = os.path.join(OUT_PATH,'.'.join(out_name_list))\n",
    "                blast_options.append(out_name)\n",
    "                blast_command = 'blastn %s' % ' '.join(blast_options)\n",
    "            elif 'protein' in query and 'protein' in db:\n",
    "                out_name_list = [ query.split('.')[0], db.split('.')[0], str(e_value), 'blastp.outfmt6']\n",
    "                out_name = os.path.join(OUT_PATH,'.'.join(out_name_list))\n",
    "                blast_options.append(out_name)\n",
    "                blast_command = 'blastp %s' % ' '.join(blast_options)\n",
    "            print(\"Blast command generated:\\n%s\" % blast_command)\n",
    "            if not os.path.exists(out_name):\n",
    "                print(\"Executing new blast command...\")\n",
    "                blast_stderr_dict[blast_command] = subprocess.check_output(blast_command, shell=True, stderr=subprocess.STDOUT)\n",
    "                print(\"New blast executed!\")\n",
    "            else:\n",
    "                blast_stderr_dict[blast_command] = 'Previously done already!'\n",
    "                print(\"Blast command previously completed!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make a sequence length dict for all genes and proteins for which blast was performed\n",
    "seq_list = []\n",
    "length_list =[]\n",
    "for key in blast_dict.keys():\n",
    "    for file in blast_dict[key]:\n",
    "        for seq in SeqIO.parse(open(file), 'fasta'):\n",
    "            seq_list.append(seq.id)\n",
    "            length_list.append(len(seq.seq))\n",
    "length_dict = dict(zip(seq_list, length_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get assemblytics folders\n",
    "ass_folders = [os.path.join(ASSEMBLYTICS_IN_PATH, x) for x in os.listdir(ASSEMBLYTICS_IN_PATH) if x.endswith('_php_8kbp')]\n",
    "ass_folders.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def same_contig_blast(x,y):\n",
    "    '''Function that checks if the blast hit in columne y is on the same contig as the the query sequence in\n",
    "    column y.\n",
    "    '''\n",
    "    try:\n",
    "        q_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', x).group(1)\n",
    "        if y != 'NaN':\n",
    "            hit_contig = re.search(r'[p|h]([a-z]*_[0-9]*)_?', y).group(1)\n",
    "        else:\n",
    "            hit_contig = 'NaN'\n",
    "        return q_contig == hit_contig\n",
    "    except AttributeError:\n",
    "        print(\"AttributeError\\nx: %s\\ny: %s\" % (x, y))\n",
    "        sys.exit()\n",
    "    except TypeError:\n",
    "        print(\"x: %s\\ny: %s\" % (x, y))\n",
    "        print(\"TypeError\\ntype(x): %s\\ntype(y): %s\" % (type(x), type(y)))\n",
    "        sys.exit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in the gff files\n",
    "p_gff = pd.read_csv(GENOME_PATH+'/'+P_GENOME+'.anno.gff3', sep='\\t', header=None)\n",
    "h_gff = pd.read_csv(GENOME_PATH+'/'+H_GENOME+'.anno.gff3', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in the blast df and add QLgth and QCov columns\n",
    "blast_out_dict = {}\n",
    "blast_header = ['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery', 'StopQuery', 'StartTarget',\\\n",
    "              'StopTarget', 'e-value','BitScore']\n",
    "for key in blast_stderr_dict.keys():\n",
    "    file_loc = key.split('>')[-1][1:]\n",
    "    # file_loc = .../output/defining_alleles/allele_analysis/DK_0911_v03_h_ctg.DK_0911_v03_p_ctg.0.001.blastp.outfmt6\n",
    "    \n",
    "    file_name = file_loc.split('/')[-1]\n",
    "    # DK_0911_v03_h_ctg.DK_0911_v03_p_ctg.0.001.blastp.outfmt6\n",
    "    \n",
    "    tmp_df = pd.read_csv(file_loc, sep='\\t', header=None, names=blast_header)\n",
    "    \n",
    "    tmp_df['QLgth'] = tmp_df['Query'].apply(lambda x: length_dict[x])\n",
    "    tmp_df['QCov'] = tmp_df['AlnLgth']/tmp_df['QLgth']*100\n",
    "    \n",
    "    tmp_df.sort_values(by=['Query', 'e-value','BitScore', ],ascending=[True, True, False], inplace=True)\n",
    "    \n",
    "    # assert(len(tmp_df.loc[tmp_df['Query'] == 'pcontig_000:10023-10824']) == 0)\n",
    "    \n",
    "    #now make sure to add proteins/genes without blast hit to the DataFrame\n",
    "    #check if gene blast or protein blast and pull out respective query file\n",
    "    if file_loc.split('.')[-2] == 'blastp':\n",
    "        tmp_blast_seq = [x for x in blast_dict['protein_fa'] if x.startswith(file_name.split('.')[0])][0]\n",
    "    elif file_loc.split('.')[-2] == 'blastn':\n",
    "        tmp_blast_seq = [x for x in blast_dict['gene_fa'] if x.startswith(file_name.split('.')[0])][0]\n",
    "    tmp_all_blast_seq = []\n",
    "    #make list of all query sequences\n",
    "    for seq in SeqIO.parse(os.path.join(BLAST_DB, tmp_blast_seq), 'fasta'):\n",
    "        tmp_all_blast_seq.append(str(seq.id))\n",
    "    tmp_all_queries_w_hit = tmp_df[\"Query\"].unique()\n",
    "    tmp_queries_no_hit = set(tmp_all_blast_seq) - set(tmp_all_queries_w_hit)\n",
    "    no_hit_list = []\n",
    "    \n",
    "    #loop over the quieres with no hit and make list of list out of them the first element being the query id\n",
    "    for x in tmp_queries_no_hit:\n",
    "        NA_list = ['NaN'] * len(tmp_df.columns)\n",
    "        NA_list[0] = x\n",
    "        no_hit_list.append(NA_list)\n",
    "    \n",
    "    tmp_no_hit_df = pd.DataFrame(no_hit_list)\n",
    "    tmp_no_hit_df.columns = tmp_df.columns\n",
    "    tmp_no_hit_df['QLgth'] = tmp_no_hit_df.Query.apply(lambda x: length_dict[x])\n",
    "    #map stuff at the tmp level\n",
    "    \n",
    "    tmp_df = tmp_df.append(tmp_no_hit_df)\n",
    "    assert(len(tmp_df.loc[tmp_df['Query'] == 'NaN']) == 0)\n",
    "    \n",
    "    \n",
    "    tmp_df['q_contig'] = tmp_df['Query'].str.extract(r'([p|h][a-z]*_[^.]*).?', expand = False)\n",
    "    tmp_df['t_contig'] = tmp_df['Target'].str.extract(r'([p|h][a-z]*_[^.]*).?', expand = False)\n",
    "    # print(tmp_df['q_contig'])\n",
    "    \n",
    "    #fix that if you don't extract anything return False and not 'nan'\n",
    "    tmp_df['t_contig'].fillna(False, inplace=True)\n",
    "    \n",
    "    # print(tmp_df.apply(lambda row: print(row['Query'])), axis = 1)\n",
    "    \n",
    "    tmp_df['q_contig == t_contig'] = tmp_df.apply(lambda row: same_contig_blast(row['Query'], row['Target']), axis = 1)\n",
    "    tmp_df.reset_index(inplace=True, drop=True)\n",
    "    blast_out_dict[file_name] = tmp_df.iloc[:,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Check on /home/gamran/genome_analysis/Warrior/Richard/output/nucmer_assemblytics/Assemblytics/DK_0911_v031_pcontig_075_php_8kbp assemblytics\n"
     ]
    }
   ],
   "source": [
    "for folder in ass_folders:\n",
    "    '''\n",
    "    From oriented_coords.csv files generate gff files with the following set up.\n",
    "    h_contig overlap\n",
    "    'query', 'ref', 'h_feature', 'query_start', 'query_end', 'query_aln_len', 'strand', 'frame', 'attribute_h'\n",
    "    p_contig overlap\n",
    "    'ref', 'query', 'p_feature','ref_start','ref_end', 'alignment_length', 'strand', 'frame', 'attribute_p'.\n",
    "    Save those file in a new folder for further downstream analysis. The same folder should include the contig and gene filtered\n",
    "    gffs. \n",
    "    '''\n",
    "    orient_coords_file = [os.path.join(folder, x) for x in os.listdir(folder) if x.endswith('oriented_coords.csv')][0]\n",
    "    #load in df and generate several additions columns\n",
    "    tmp_df = pd.read_csv(orient_coords_file, sep=',')\n",
    "    #check if there is any resonable alignment if not go to the next one\n",
    "    if len(tmp_df) == 0:\n",
    "        print('Check on %s assemblytics' % (folder))\n",
    "        continue\n",
    "    tmp_df['p_feature'] = \"haplotig\"\n",
    "    tmp_df['h_feature'] ='primary_contig'\n",
    "    tmp_df['strand'] = \"+\"\n",
    "    tmp_df['frame'] = 0\n",
    "    tmp_df['query_aln_len'] = abs(tmp_df['query_end']-tmp_df['query_start'])\n",
    "    tmp_df['alignment_length'] = abs(tmp_df['ref_end'] - tmp_df['ref_start'])\n",
    "    tmp_df.reset_index(drop=True, inplace=True)\n",
    "    tmp_df.sort_values('query', inplace=True)\n",
    "    tmp_df['attribute_p'] = tmp_df.apply(lambda row: att_column_p(row['query'],row['query_start'], row['query_end'],\\\n",
    "                                                         row['query_length'], row['query_aln_len']), axis=1)\n",
    "    \n",
    "    tmp_df['attribute_h'] = tmp_df.apply(lambda row: att_column_h(row['ref'], row['ref_start'], row['ref_end'],\\\n",
    "                                                                  row['ref_length'], row['alignment_length']), axis=1)\n",
    "    \n",
    "    #generate tmp gff dataframe\n",
    "    tmp_p_gff_df = tmp_df.loc[:, ['ref', 'query', 'p_feature','ref_start','ref_end', 'alignment_length', 'strand', 'frame', 'attribute_p']]\n",
    "    #now filter added if start < stop swap round. And if start == 0. This was mostly?! only? the case for haplotigs\n",
    "    tmp_p_gff_df['comp'] = tmp_p_gff_df['ref_start'] - tmp_p_gff_df['ref_end']\n",
    "    tmp_swap_index  = tmp_p_gff_df['comp'] > 0\n",
    "    tmp_p_gff_df.loc[tmp_swap_index, 'ref_start'] , tmp_p_gff_df.loc[tmp_swap_index, 'ref_end'] = tmp_p_gff_df['ref_end'], tmp_p_gff_df['ref_start']\n",
    "    #and if 'ref_start' == g\n",
    "    is_null_index  = tmp_p_gff_df['ref_start'] == 0\n",
    "    tmp_p_gff_df.loc[is_null_index, 'ref_start'] = 1\n",
    "    tmp_p_gff_df = tmp_p_gff_df.iloc[:, 0:9]\n",
    "    tmp_p_gff_df.sort_values('ref_start',inplace=True)\n",
    "    tmp_h_gff_df = tmp_df.loc[:, ['query', 'ref', 'h_feature', 'query_start', 'query_end', 'query_aln_len', 'strand', 'frame', 'attribute_h']]\n",
    "    #same fix for tmp_h_gff\n",
    "    tmp_h_gff_df['comp'] = tmp_h_gff_df['query_start'] - tmp_h_gff_df['query_end']\n",
    "    tmp_swap_index  = tmp_h_gff_df['comp'] > 0\n",
    "    tmp_h_gff_df.loc[tmp_swap_index, 'query_start'] , tmp_h_gff_df.loc[tmp_swap_index, 'query_end'] = tmp_h_gff_df['query_end'], tmp_h_gff_df['query_start']\n",
    "    #and if 'query_start' == g\n",
    "    is_null_index  = tmp_h_gff_df['query_start'] == 0\n",
    "    tmp_h_gff_df.loc[is_null_index, 'query_start'] = 1\n",
    "    tmp_h_gff_df = tmp_h_gff_df.iloc[:, 0:9]\n",
    "    tmp_h_gff_df.sort_values('query_start', inplace=True)\n",
    "    #make the outfolder and save gff files of overlap and filtered gene files\n",
    "    folder_suffix = folder.split('/')[-1]\n",
    "    #get the contig\n",
    "    contig = re.search(r'p([a-z]*_[0-9]*)_', folder_suffix).group(1)\n",
    "    out_folder = os.path.join(OUT_PATH, folder_suffix)\n",
    "    if not os.path.exists(out_folder):\n",
    "        os.mkdir(out_folder)\n",
    "    out_name_p = os.path.join(out_folder, folder_suffix )\n",
    "    tmp_p_gff_df.to_csv(out_name_p+'.p_by_h_cov.gff', sep='\\t' ,header=None, index =None)\n",
    "    out_name_h =  os.path.join(out_folder, folder_suffix)\n",
    "    tmp_h_gff_df.to_csv(out_name_h+'.h_by_p_cov.gff' , sep='\\t' ,header=None, index =None)\n",
    "    #write those out to new folder for php together with the p and h gff of genes\n",
    "    p_gene_gff = gene_contig_subset(p_gff, contig, 'gene')\n",
    "    p_gene_gff.to_csv(out_name_p+'.p_gene.gff', sep='\\t' ,header=None, index =None)\n",
    "    h_gene_gff = gene_contig_subset(h_gff, contig, 'gene')\n",
    "    h_gene_gff.to_csv(out_name_h+'.h_gene.gff', sep='\\t' ,header=None, index =None)\n",
    "    \n",
    "    #next would be to do overlap between the gene gff and the corresponding alignment gff. Write this out. Dict for each gene and its corresponding h_contig/p_contig\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now loop over the outfolders\n",
    "out_folders = [os.path.join(OUT_PATH, x) for x in os.listdir(OUT_PATH) if x.endswith('_php_8kbp')]\n",
    "out_folders.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 786\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 472\n",
      "This is the length of the intersect 552\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 500\n",
      "This is the length of the intersect 498\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 457\n",
      "This is the length of the intersect 358\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 334\n",
      "This is the length of the intersect 417\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 302\n",
      "This is the length of the intersect 365\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 344\n",
      "This is the length of the intersect 402\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 95\n",
      "This is the length of the intersect 336\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 291\n",
      "This is the length of the intersect 302\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 241\n",
      "This is the length of the intersect 241\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 275\n",
      "This is the length of the intersect 246\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 135\n",
      "This is the length of the intersect 250\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 205\n",
      "This is the length of the intersect 0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 207\n",
      "This is the length of the intersect 89\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 107\n",
      "This is the length of the intersect 173\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 219\n",
      "This is the length of the intersect 115\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 224\n",
      "This is the length of the intersect 196\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 213\n",
      "This is the length of the intersect 165\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 185\n",
      "This is the length of the intersect 148\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 89\n",
      "This is the length of the intersect 144\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 154\n",
      "This is the length of the intersect 115\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 114\n",
      "This is the length of the intersect 6\n",
      "This is the length of the intersect 131\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 74\n",
      "This is the length of the intersect 71\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 81\n",
      "This is the length of the intersect 73\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 77\n",
      "This is the length of the intersect 104\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 43\n",
      "This is the length of the intersect 72\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 92\n",
      "This is the length of the intersect 82\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 108\n",
      "This is the length of the intersect 84\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 79\n",
      "This is the length of the intersect 0\n",
      "This is the length of the intersect 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 52\n",
      "This is the length of the intersect 42\n",
      "This is the length of the intersect 35\n",
      "This is the length of the intersect 40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 13\n",
      "This is the length of the intersect 64\n",
      "This is the length of the intersect 26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 21\n",
      "This is the length of the intersect 17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 21\n",
      "This is the length of the intersect 23\n",
      "This is the length of the intersect 31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 10\n",
      "This is the length of the intersect 14\n",
      "This is the length of the intersect 20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 7\n",
      "This is the length of the intersect 7\n",
      "This is the length of the intersect 8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 12\n",
      "This is the length of the intersect 8\n",
      "This is the length of the intersect 7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n",
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is the length of the intersect 25\n",
      "This is the length of the intersect 4\n",
      "This is the length of the intersect 0\n",
      "This is the length of the intersect 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gamran/anaconda3/lib/python3.5/site-packages/pybedtools/bedtool.py:3287: UserWarning: Default names for filetype gff are:\n",
      "['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attributes']\n",
      "but file has 18 fields; you can supply custom names with the `names` kwarg\n",
      "  % (self.file_type, _names, self.field_count()))\n"
     ]
    }
   ],
   "source": [
    "p_gene_h_contig_overlap_dict = {}\n",
    "for out_folder in out_folders:\n",
    "    tmp_folder_content = os.listdir(out_folder)\n",
    "    tmp_p_gene_gff = [os.path.join(out_folder, x) for x in tmp_folder_content if x.endswith('p_gene.gff') ][0]\n",
    "    tmp_p_gene_bed = pybedtools.BedTool(tmp_p_gene_gff).remove_invalid().saveas()\n",
    "    tmp_p_by_h_gff = [os.path.join(out_folder, x) for x in tmp_folder_content if x.endswith('p_by_h_cov.gff') ][0]\n",
    "    tmp_p_by_h_bed = pybedtools.BedTool(tmp_p_by_h_gff).remove_invalid().saveas()\n",
    "    #generate a bed intersect\n",
    "    p_gene_and_p_by_h = tmp_p_gene_bed.intersect(tmp_p_by_h_bed, wb=True)\n",
    "    #check the length of the obtained intersect is \n",
    "    print(\"This is the length of the intersect %i\" %len(p_gene_and_p_by_h ))\n",
    "    if len(p_gene_and_p_by_h) == 0:\n",
    "        continue\n",
    "    #turn this into a dataframe \n",
    "    p_gene_and_p_by_h_df = p_gene_and_p_by_h.to_dataframe()\n",
    "    p_gene_and_p_by_h_df['p_gene'] = p_gene_and_p_by_h_df[8].str.extract(r'ID=([^;]*);', expand=False)\n",
    "    p_gene_and_p_by_h_df['p_protein'] = p_gene_and_p_by_h_df['p_gene'].str.replace('TU', 'model')\n",
    "    #save this in the same folder as dataframe\n",
    "    p_gene_and_p_by_h_df.to_csv(tmp_p_gene_gff.replace('p_gene.gff', 'gene_haplotig_intersect.df'), sep='\\t', index=None)\n",
    "    p_gene_and_p_by_h_df_2 = p_gene_and_p_by_h_df.loc[:, [10, 'p_protein']]\n",
    "    #make a dict for the overlap of each gene with it's correspoding haplotig\n",
    "    tmp_p_gene_h_overlap_df = p_gene_and_p_by_h_df_2.groupby('p_protein')[10].apply(list)\n",
    "    tmp_p_gene_h_overlap_dict = dict(zip(tmp_p_gene_h_overlap_df.index, tmp_p_gene_h_overlap_df.values))\n",
    "    p_gene_h_contig_overlap_dict.update(tmp_p_gene_h_overlap_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get out the p_protein on h_protein blast out dataframe and append the h_contig overlap for each protein\n",
    "_key = [x for x in blast_out_dict.keys() if x.startswith(P_GENOME) and x.split('.')[-2] == 'blastp' and x.endswith('outfmt6')][0]\n",
    "p_gene_h_contig_overlap_df = pd.DataFrame([p_gene_h_contig_overlap_dict.keys(), p_gene_h_contig_overlap_dict.values()],  index = ['p_protein', 'h_contig_overlap']).T\n",
    "_tmp_df = blast_out_dict[_key]\n",
    "_tmp_df = _tmp_df.merge(p_gene_h_contig_overlap_df, how='outer' ,left_on='Query', right_on='p_protein')\n",
    "_tmp_df['h_contig_overlap'].fillna(False, inplace=True)\n",
    "#check if the protein hit resites on the overlapping haplotig\n",
    "_tmp_df['t_contig == h_contig_overlap'] = _tmp_df.apply(lambda row: target_on_mapped_haplotig(row['t_contig'], row['h_contig_overlap']), axis =1 )\n",
    "#blast_out_dict[_key] = _tmp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# p_gene_h_contig_overlap_df['p_protein'] = p_gene_h_contig_overlap_df['p_protein'].apply(lambda x: x.replace('model', 'TU'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write the dataframe again\n",
    "_tmp_df.to_csv(os.path.join(OUT_PATH, _key+'.allele_analysis'), index=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now mangle the data a bit\n",
    "#write out t_contig == h_contig_overlap best hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No cut off defined! Do you want to define it. Please go ahead above.\n"
     ]
    }
   ],
   "source": [
    "#Qcov_cut_off = ''\n",
    "#PctID_cut_off = ''\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    pass\n",
    "else: \n",
    "    print('No cut off defined! Do you want to define it. Please go ahead above.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "# Finding:\n",
    "# • (htg & pCtg on same contig and overlap)\n",
    "\n",
    "_tmp_grouped = _tmp_df[(_tmp_df['t_contig == h_contig_overlap'] == True)].groupby('Query')\n",
    "# .groupby('Query') creates a GroupBy object that can be thought of mini-DataFrames,\n",
    "# each with the same Query. This groups are then reduced according to the\n",
    "# minimum e-value and maximum BitScore of each group (i.e. retaining the\n",
    "# best hit for each Query).\n",
    "\n",
    "# filter on the e-value, then on the BitScore\n",
    "best_hits_t_contig_and_h_contig = _tmp_grouped.apply(lambda g: g[(g['e-value'] == g['e-value'].min())])\n",
    "best_hits_t_contig_and_h_contig = best_hits_t_contig_and_h_contig.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "\n",
    "out_name = P_GENOME + '.h_contig_overlap.alleles'\n",
    "\n",
    "best_hits_t_contig_and_h_contig.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#now save including the filter as well\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    out_name = P_GENOME + '.h_contig_overlap.Qcov%s.PctID%s.alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    best_hits_t_contig_and_h_contig[(best_hits_t_contig_and_h_contig['QCov'] >= Qcov_cut_off) & (best_hits_t_contig_and_h_contig['PctID'] >= PctID_cut_off)]\\\n",
    "    .loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering is working; proceed.\n"
     ]
    }
   ],
   "source": [
    "# test if filtering like this works\n",
    "if set(_tmp_df[(_tmp_df['t_contig == h_contig_overlap'] == True)]['Query'].unique()) == set(best_hits_t_contig_and_h_contig['Query'].unique().tolist()):\n",
    "    print(\"Filtering is working; proceed.\")\n",
    "else:\n",
    "    print('The filter threw an error. Not the expected output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop everything that has already been matched well\n",
    "# already matched:\n",
    "# • (htg belongs to pCtg and matching sequence overlaps)\n",
    "\n",
    "query_ids_to_drop = best_hits_t_contig_and_h_contig['Query'].unique().tolist()\n",
    "target_ids_to_drop = best_hits_t_contig_and_h_contig['Target'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "# finding\n",
    "# • (htg belongs to pCtg but matching sequence does not overlap)\n",
    "_tmp_groupbed = _tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop))&\\\n",
    "                        (_tmp_df['q_contig == t_contig'] == True)].groupby('Query')\n",
    "best_hits_same_contig_no_overlap = _tmp_groupbed.apply(lambda g: g[(g['e-value'] == g['e-value'].min())]) \n",
    "best_hits_same_contig_no_overlap = best_hits_same_contig_no_overlap.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "\n",
    "out_name = P_GENOME + '.no_specific_h_contig_overlap.alleles'\n",
    "best_hits_same_contig_no_overlap.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#now save including the filter as well\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    out_name = P_GENOME + '.no_specific_h_contig_overlap.Qcov%s.PctID%s.alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    best_hits_same_contig_no_overlap[(best_hits_same_contig_no_overlap['QCov'] >= Qcov_cut_off) & (best_hits_same_contig_no_overlap['PctID'] >= PctID_cut_off)]\\\n",
    "    .loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All good please proceed!\n"
     ]
    }
   ],
   "source": [
    "#test if filtering like this works\n",
    "if set(_tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop)) & (_tmp_df['q_contig == t_contig'] == True)]['Query'].unique())\\\n",
    "        == set(best_hits_same_contig_no_overlap['Query'].unique().tolist()):\n",
    "    print(\"All good please proceed!\")\n",
    "else:\n",
    "    print('The filter threw an error. Not the expected output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop everything that has already been matched well\n",
    "# already matched:\n",
    "# • (htg belongs to pCtg and matching sequence overlaps)\n",
    "# • (htg belongs to pCtg but matching sequence does not overlap)\n",
    "\n",
    "query_ids_to_drop = query_ids_to_drop + best_hits_same_contig_no_overlap['Query'].unique().tolist()\n",
    "target_ids_to_drop = target_ids_to_drop + best_hits_same_contig_no_overlap['Target'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "# now get all remaining matches, i.e.\n",
    "# • (htg does not belong to pCtg)\n",
    "\n",
    "_tmp_groupbed = _tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop))&\\\n",
    "                        (_tmp_df['q_contig == t_contig'] == False)\\\n",
    "                        & (_tmp_df['Target'] != 'NaN' )].groupby('Query')\n",
    "best_hits_different_contig = _tmp_groupbed.apply(lambda g: g[(g['e-value'] == g['e-value'].min()) ])\n",
    "best_hits_different_contig = best_hits_different_contig .reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "out_name = P_GENOME + '.no_respective_h_contig_overlap.alleles'\n",
    "best_hits_different_contig.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#now save including the filter as well\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    out_name = P_GENOME + '.no_respective_h_contig_overlap.Qcov%s.PctID%s.alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    best_hits_different_contig[(best_hits_different_contig['QCov'] >= Qcov_cut_off) & (best_hits_different_contig['PctID'] >= PctID_cut_off)]\\\n",
    "    .loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All good please proceed!\n"
     ]
    }
   ],
   "source": [
    "if set(_tmp_df[(~_tmp_df['Query'].isin(query_ids_to_drop))& (_tmp_df['q_contig == t_contig'] == False) &\\\n",
    "               (_tmp_df['Target'] != 'NaN' )]['Query'].unique().tolist())\\\n",
    "        == set(best_hits_different_contig['Query'].unique().tolist()):\n",
    "    print(\"All good please proceed!\")\n",
    "else:\n",
    "    print('The filter threw an error. Not the expected output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop everything that has already been matched well\n",
    "# already matched:\n",
    "# • (htg belongs to pCtg and matching sequence overlaps)\n",
    "# • (htg belongs to pCtg but matching sequence does not overlap)\n",
    "# • (htg does not belong to pCtg)\n",
    "\n",
    "query_ids_to_drop = query_ids_to_drop + best_hits_different_contig['Query'].unique().tolist()\n",
    "target_ids_to_drop = target_ids_to_drop + best_hits_different_contig['Target'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "#now print out the p_proteins without alleles (i.e. no good match)\n",
    "out_name = P_GENOME + '.no_alleles'\n",
    "_tmp_df[~_tmp_df['Query'].isin(query_ids_to_drop)].drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#for filtering different query and targets to drop are required to be dropped\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "    query_ids_to_drop_filtered = \\\n",
    "    best_hits_t_contig_and_h_contig[(best_hits_t_contig_and_h_contig['QCov'] >= Qcov_cut_off) & (best_hits_t_contig_and_h_contig['PctID']>= PctID_cut_off)]['Query'].unique().tolist() \\\n",
    "    + best_hits_same_contig_no_overlap[(best_hits_same_contig_no_overlap['QCov'] >= Qcov_cut_off) & (best_hits_same_contig_no_overlap['PctID'] >= PctID_cut_off)]['Query'].unique().tolist()\\\n",
    "    + best_hits_different_contig[(best_hits_different_contig['QCov'] >= Qcov_cut_off) & (best_hits_different_contig['PctID'] >= PctID_cut_off)]['Query'].unique().tolist()\n",
    "    \n",
    "    target_ids_to_drop_filtered = \\\n",
    "    best_hits_t_contig_and_h_contig[(best_hits_t_contig_and_h_contig['QCov'] >= Qcov_cut_off) & (best_hits_t_contig_and_h_contig['PctID']>= PctID_cut_off)]['Target'].unique().tolist()\\\n",
    "    + best_hits_same_contig_no_overlap[(best_hits_same_contig_no_overlap['QCov'] >= Qcov_cut_off) & (best_hits_same_contig_no_overlap['PctID'] >= PctID_cut_off)]['Target'].unique().tolist() \\\n",
    "    + best_hits_different_contig[(best_hits_different_contig['QCov'] >= Qcov_cut_off) & (best_hits_different_contig['PctID'] >= PctID_cut_off)]['Target'].unique().tolist()\n",
    "    out_name = P_GENOME + '.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    #save the dataframe\n",
    "    _tmp_df[~_tmp_df['Query'].isin(query_ids_to_drop_filtered)].drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)    \n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")                                                                                                                             "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save back the dataframe with '.analyzed' added to the key\n",
    "blast_out_dict[_key+'.analyzed'] = _tmp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pull out the h on p blast dataframe\n",
    "_key = [x for x in blast_out_dict.keys() if x.startswith(H_GENOME) and x.split('.')[-2] == 'blastp' and x.endswith('outfmt6')][0]\n",
    "_tmp_df = blast_out_dict[_key]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pull out all h_contigs without alleles\n",
    "_tmp_df_no_alleles = _tmp_df[~_tmp_df['Query'].isin(target_ids_to_drop)]\n",
    "out_name = H_GENOME + '.no_alleles'\n",
    "_tmp_df_no_alleles.drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now pull out all h_proteins that have a blast hit but were left behind with selection \n",
    "#based on proteinortho and best blast hits above. No QCov and PctID filtering\n",
    "_tmp_df_no_alleles_grouped = _tmp_df_no_alleles[_tmp_df_no_alleles[\"Target\"] != 'NaN'].groupby('Query')\n",
    "h_protein_no_alleles = _tmp_df_no_alleles_grouped.apply(lambda g: g[(g['e-value'] == g['e-value'].min())]) \n",
    "h_protein_no_alleles = h_protein_no_alleles.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "out_name = H_GENOME + '.best_p_hits.no_alleles'\n",
    "h_protein_no_alleles.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "#write done the hits with no blast hit at all. Meaning blasting h on p gave no results\n",
    "out_name = H_GENOME + '.no_p_hits.no_alleles'\n",
    "_tmp_df_no_alleles[_tmp_df_no_alleles[\"Target\"] == 'NaN']['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\n"
     ]
    }
   ],
   "source": [
    "#now pull out all h_proteins that have a blast hit but were left behind with selection of best blast hits above. With QCov and PctID filtering.\n",
    "if Qcov_cut_off and PctID_cut_off:\n",
    "#pull out all h proteins without alleles when filtering was applied\n",
    "    _tmp_df_no_alleles_filtered = _tmp_df[~_tmp_df['Query'].isin(target_ids_to_drop_filtered)]\n",
    "    out_name = H_GENOME + '.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    _tmp_df_no_alleles_filtered.drop_duplicates(subset='Query')['Query'].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "\n",
    "#pull out all the h proteins with blast hit when filtering was applied and filter those as well.\n",
    "    #first get ride of all NAs in future version maybe convert QCov and PctID to floats or ints so this is not neccessary anymore\n",
    "    _tmp_df_no_alleles_filtered_no_NA = _tmp_df_no_alleles_filtered[(_tmp_df_no_alleles_filtered[\"Target\"] != 'NaN')]\n",
    "    _tmp_df_no_alleles_filtered_grouped = \\\n",
    "        _tmp_df_no_alleles_filtered_no_NA[(_tmp_df_no_alleles_filtered_no_NA['QCov'] >= Qcov_cut_off) &\\\n",
    "                                    (_tmp_df_no_alleles_filtered_no_NA['PctID'] >= PctID_cut_off)].groupby('Query') \n",
    "\n",
    "    h_protein_no_alleles_filtered = _tmp_df_no_alleles_filtered_grouped.apply(lambda g: g[(g['e-value'] == g['e-value'].min())]) \n",
    "    h_protein_no_alleles_filtered = h_protein_no_alleles_filtered.reset_index(drop=True).groupby('Query').apply(lambda g: g[(g['BitScore'] == g['BitScore'].max())])\n",
    "    out_name = H_GENOME + '.best_p_hits.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    h_protein_no_alleles_filtered.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "    target_ids_to_drop_filtered = target_ids_to_drop_filtered + h_protein_no_alleles_filtered['Query'].unique().tolist()\n",
    "    out_name = H_GENOME + '.no_p_hits.Qcov%s.PctID%s.no_alleles' % (Qcov_cut_off,PctID_cut_off)\n",
    "    _tmp_df_no_alleles_filtered[~_tmp_df_no_alleles_filtered['Query'].isin(target_ids_to_drop_filtered)].drop_duplicates(subset='Query')['Query']\\\n",
    "    .to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)\n",
    "else:\n",
    "    print(\"No filter selected for Qcov and PctID cut-off. If single filter desired set other filter to 1000.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "#write the dataframe again\n",
    "_tmp_df.to_csv(os.path.join(OUT_PATH, _key+'.allele_analysis'), index=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery',\n",
       "       'StopQuery', 'StartTarget', 'StopTarget', 'e-value', 'BitScore',\n",
       "       'QLgth', 'QCov', 'q_contig', 't_contig', 'q_contig == t_contig'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_tmp_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Query', 'Target', 'PctID', 'AlnLgth', 'NumMis', 'NumGap', 'StartQuery',\n",
       "       'StopQuery', 'StartTarget', 'StopTarget', 'e-value', 'BitScore',\n",
       "       'QLgth', 'QCov', 'q_contig', 't_contig', 'q_contig == t_contig',\n",
       "       'p_protein', 'h_contig_overlap', 't_contig == h_contig_overlap'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#might be good to write these analyzed dataframes out as well to make life a bit easier\n",
    "blast_out_dict['%s.%s.0.001.blastp.outfmt6.analyzed' % (P_GENOME, H_GENOME)].columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2278 haplotigs have no allele\n",
      "Of these haplotigs, 2015 haplotigs have a BLAST hit\n",
      "Thus, 263 haplotigs have both no allele and no BLAST hit\n"
     ]
    }
   ],
   "source": [
    "htgOnPctgBlast = pd.read_csv(os.path.join(OUT_PATH, '%s.%s.0.001.blastp.outfmt6' % (H_GENOME, P_GENOME)), sep='\\t', index_col=None, header=None, names=blast_header)\n",
    "noAllelesDf = pd.read_csv(os.path.join(ALLELE_PATH, '%s.no_alleles' % H_GENOME), names=['Query'])\n",
    "\n",
    "htgNoAllelesBlastDf = htgOnPctgBlast[htgOnPctgBlast['Query'].isin(noAllelesDf.Query)]\n",
    "print('%s haplotigs have no allele' % len(noAllelesDf))\n",
    "print('Of these haplotigs, %s haplotigs have a BLAST hit' % len(htgNoAllelesBlastDf['Query'].unique()))\n",
    "print('Thus, %s haplotigs have both no allele and no BLAST hit' % (len(noAllelesDf['Query']) - len(htgNoAllelesBlastDf['Query'].unique())))\n",
    "\n",
    "# Look at haplotig on primary contig blast, and take the best hit. If the best hit is >=90%\n",
    "# assign the haplotig.\n",
    "\n",
    "htgNoAllelesBlastDf_grouped = htgNoAllelesBlastDf.groupby('Query')\n",
    "htgNoAllelesBlastDf = htgNoAllelesBlastDf_grouped.apply(lambda g: g[g['PctID'] == g['PctID'].max()])\n",
    "\n",
    "htgNoAllelesBlastDf = htgNoAllelesBlastDf[htgNoAllelesBlastDf['PctID'] >= 90]\n",
    "out_name = H_GENOME + '.manual_assigned.alleles'\n",
    "htgNoAllelesBlastDf.loc[:, ['Query', 'Target']].to_csv(os.path.join(ALLELE_PATH, out_name), sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Query</th>\n",
       "      <th>Target</th>\n",
       "      <th>PctID</th>\n",
       "      <th>AlnLgth</th>\n",
       "      <th>NumMis</th>\n",
       "      <th>NumGap</th>\n",
       "      <th>StartQuery</th>\n",
       "      <th>StopQuery</th>\n",
       "      <th>StartTarget</th>\n",
       "      <th>StopTarget</th>\n",
       "      <th>e-value</th>\n",
       "      <th>BitScore</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Query</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">evm.model.hcontig_000_001.2</th>\n",
       "      <th>611289</th>\n",
       "      <td>evm.model.hcontig_000_001.2</td>\n",
       "      <td>evm.model.pcontig_002.624</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1416</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1416</td>\n",
       "      <td>1</td>\n",
       "      <td>1416</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2958.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611290</th>\n",
       "      <td>evm.model.hcontig_000_001.2</td>\n",
       "      <td>evm.model.pcontig_010.134</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1416</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1416</td>\n",
       "      <td>1</td>\n",
       "      <td>1416</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2958.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611337</th>\n",
       "      <td>evm.model.hcontig_000_001.2</td>\n",
       "      <td>evm.model.pcontig_032.198</td>\n",
       "      <td>100.00</td>\n",
       "      <td>354</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1063</td>\n",
       "      <td>1416</td>\n",
       "      <td>1</td>\n",
       "      <td>354</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>737.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_001.8</th>\n",
       "      <th>611953</th>\n",
       "      <td>evm.model.hcontig_000_001.8</td>\n",
       "      <td>evm.model.pcontig_000.1077</td>\n",
       "      <td>100.00</td>\n",
       "      <td>167</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "      <td>6.000000e-120</td>\n",
       "      <td>351.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">evm.model.hcontig_000_003.1</th>\n",
       "      <th>749535</th>\n",
       "      <td>evm.model.hcontig_000_003.1</td>\n",
       "      <td>evm.model.pcontig_021.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>112</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>112</td>\n",
       "      <td>183</td>\n",
       "      <td>294</td>\n",
       "      <td>2.000000e-75</td>\n",
       "      <td>236.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749536</th>\n",
       "      <td>evm.model.hcontig_000_003.1</td>\n",
       "      <td>evm.model.pcontig_000.534</td>\n",
       "      <td>100.00</td>\n",
       "      <td>112</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>112</td>\n",
       "      <td>183</td>\n",
       "      <td>294</td>\n",
       "      <td>7.000000e-72</td>\n",
       "      <td>234.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_007.8</th>\n",
       "      <th>116406</th>\n",
       "      <td>evm.model.hcontig_000_007.8</td>\n",
       "      <td>evm.model.pcontig_000.501</td>\n",
       "      <td>95.40</td>\n",
       "      <td>261</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>261</td>\n",
       "      <td>899</td>\n",
       "      <td>1159</td>\n",
       "      <td>2.000000e-172</td>\n",
       "      <td>509.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_013.2</th>\n",
       "      <th>113158</th>\n",
       "      <td>evm.model.hcontig_000_013.2</td>\n",
       "      <td>evm.model.pcontig_000.916</td>\n",
       "      <td>100.00</td>\n",
       "      <td>327</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>92</td>\n",
       "      <td>418</td>\n",
       "      <td>2</td>\n",
       "      <td>328</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>667.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_013.4</th>\n",
       "      <th>113160</th>\n",
       "      <td>evm.model.hcontig_000_013.4</td>\n",
       "      <td>evm.model.pcontig_000.917</td>\n",
       "      <td>100.00</td>\n",
       "      <td>74</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>101</td>\n",
       "      <td>109</td>\n",
       "      <td>182</td>\n",
       "      <td>7.000000e-49</td>\n",
       "      <td>155.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_014.1</th>\n",
       "      <th>210275</th>\n",
       "      <td>evm.model.hcontig_000_014.1</td>\n",
       "      <td>evm.model.pcontig_000.106</td>\n",
       "      <td>99.79</td>\n",
       "      <td>477</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>477</td>\n",
       "      <td>1</td>\n",
       "      <td>477</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>979.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_014.2</th>\n",
       "      <th>210276</th>\n",
       "      <td>evm.model.hcontig_000_014.2</td>\n",
       "      <td>evm.model.pcontig_000.107</td>\n",
       "      <td>98.85</td>\n",
       "      <td>349</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>349</td>\n",
       "      <td>1</td>\n",
       "      <td>349</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>722.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_015.3</th>\n",
       "      <th>163157</th>\n",
       "      <td>evm.model.hcontig_000_015.3</td>\n",
       "      <td>evm.model.pcontig_000.873</td>\n",
       "      <td>99.75</td>\n",
       "      <td>408</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>90</td>\n",
       "      <td>497</td>\n",
       "      <td>109</td>\n",
       "      <td>516</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>836.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_015.4</th>\n",
       "      <th>163158</th>\n",
       "      <td>evm.model.hcontig_000_015.4</td>\n",
       "      <td>evm.model.pcontig_000.873</td>\n",
       "      <td>100.00</td>\n",
       "      <td>263</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>263</td>\n",
       "      <td>617</td>\n",
       "      <td>879</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_016.2</th>\n",
       "      <th>320490</th>\n",
       "      <td>evm.model.hcontig_000_016.2</td>\n",
       "      <td>evm.model.pcontig_004.580</td>\n",
       "      <td>94.74</td>\n",
       "      <td>57</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>78</td>\n",
       "      <td>77</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000e-31</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_016.5</th>\n",
       "      <th>320497</th>\n",
       "      <td>evm.model.hcontig_000_016.5</td>\n",
       "      <td>evm.model.pcontig_000.186</td>\n",
       "      <td>100.00</td>\n",
       "      <td>654</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>654</td>\n",
       "      <td>120</td>\n",
       "      <td>773</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1332.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_016.6</th>\n",
       "      <th>320500</th>\n",
       "      <td>evm.model.hcontig_000_016.6</td>\n",
       "      <td>evm.model.pcontig_000.186</td>\n",
       "      <td>100.00</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>81</td>\n",
       "      <td>1</td>\n",
       "      <td>81</td>\n",
       "      <td>4.000000e-49</td>\n",
       "      <td>165.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_019.13</th>\n",
       "      <th>344690</th>\n",
       "      <td>evm.model.hcontig_000_019.13</td>\n",
       "      <td>evm.model.pcontig_004.580</td>\n",
       "      <td>94.74</td>\n",
       "      <td>57</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>78</td>\n",
       "      <td>77</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000e-31</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_019.41</th>\n",
       "      <th>347610</th>\n",
       "      <td>evm.model.hcontig_000_019.41</td>\n",
       "      <td>evm.model.pcontig_000.214</td>\n",
       "      <td>100.00</td>\n",
       "      <td>94</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>94</td>\n",
       "      <td>1</td>\n",
       "      <td>94</td>\n",
       "      <td>8.000000e-59</td>\n",
       "      <td>190.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_021.2</th>\n",
       "      <th>634686</th>\n",
       "      <td>evm.model.hcontig_000_021.2</td>\n",
       "      <td>evm.model.pcontig_000.998</td>\n",
       "      <td>100.00</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>93</td>\n",
       "      <td>351</td>\n",
       "      <td>443</td>\n",
       "      <td>2.000000e-57</td>\n",
       "      <td>189.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_023.2</th>\n",
       "      <th>456184</th>\n",
       "      <td>evm.model.hcontig_000_023.2</td>\n",
       "      <td>evm.model.pcontig_000.695</td>\n",
       "      <td>100.00</td>\n",
       "      <td>252</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>252</td>\n",
       "      <td>386</td>\n",
       "      <td>637</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>523.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_025.3</th>\n",
       "      <th>116644</th>\n",
       "      <td>evm.model.hcontig_000_025.3</td>\n",
       "      <td>evm.model.pcontig_000.77</td>\n",
       "      <td>100.00</td>\n",
       "      <td>149</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>149</td>\n",
       "      <td>121</td>\n",
       "      <td>269</td>\n",
       "      <td>9.000000e-107</td>\n",
       "      <td>308.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_028.1</th>\n",
       "      <th>837500</th>\n",
       "      <td>evm.model.hcontig_000_028.1</td>\n",
       "      <td>evm.model.pcontig_000.240</td>\n",
       "      <td>99.20</td>\n",
       "      <td>125</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>125</td>\n",
       "      <td>2.000000e-84</td>\n",
       "      <td>246.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_030.3</th>\n",
       "      <th>467487</th>\n",
       "      <td>evm.model.hcontig_000_030.3</td>\n",
       "      <td>evm.model.pcontig_000.818</td>\n",
       "      <td>99.35</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>2.000000e-108</td>\n",
       "      <td>308.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_030.4</th>\n",
       "      <th>467494</th>\n",
       "      <td>evm.model.hcontig_000_030.4</td>\n",
       "      <td>evm.model.pcontig_000.819</td>\n",
       "      <td>99.49</td>\n",
       "      <td>389</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>389</td>\n",
       "      <td>1</td>\n",
       "      <td>389</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>786.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_031.30</th>\n",
       "      <th>590064</th>\n",
       "      <td>evm.model.hcontig_000_031.30</td>\n",
       "      <td>evm.model.pcontig_000.465</td>\n",
       "      <td>100.00</td>\n",
       "      <td>401</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>401</td>\n",
       "      <td>1</td>\n",
       "      <td>401</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>833.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_033.1</th>\n",
       "      <th>581132</th>\n",
       "      <td>evm.model.hcontig_000_033.1</td>\n",
       "      <td>evm.model.pcontig_000.516</td>\n",
       "      <td>99.41</td>\n",
       "      <td>170</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>170</td>\n",
       "      <td>1</td>\n",
       "      <td>170</td>\n",
       "      <td>5.000000e-120</td>\n",
       "      <td>352.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_034.1</th>\n",
       "      <th>723200</th>\n",
       "      <td>evm.model.hcontig_000_034.1</td>\n",
       "      <td>evm.model.pcontig_000.83</td>\n",
       "      <td>99.65</td>\n",
       "      <td>286</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>286</td>\n",
       "      <td>1</td>\n",
       "      <td>286</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>590.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_036.13</th>\n",
       "      <th>788014</th>\n",
       "      <td>evm.model.hcontig_000_036.13</td>\n",
       "      <td>evm.model.pcontig_000.919</td>\n",
       "      <td>97.04</td>\n",
       "      <td>135</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>135</td>\n",
       "      <td>1</td>\n",
       "      <td>135</td>\n",
       "      <td>8.000000e-80</td>\n",
       "      <td>251.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_037.8</th>\n",
       "      <th>788136</th>\n",
       "      <td>evm.model.hcontig_000_037.8</td>\n",
       "      <td>evm.model.pcontig_000.51</td>\n",
       "      <td>97.30</td>\n",
       "      <td>889</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>866</td>\n",
       "      <td>1</td>\n",
       "      <td>889</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1793.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_000_039.3</th>\n",
       "      <th>566835</th>\n",
       "      <td>evm.model.hcontig_000_039.3</td>\n",
       "      <td>evm.model.pcontig_000.108</td>\n",
       "      <td>100.00</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>158</td>\n",
       "      <td>329</td>\n",
       "      <td>471</td>\n",
       "      <td>3.000000e-98</td>\n",
       "      <td>294.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_090_002.3</th>\n",
       "      <th>553752</th>\n",
       "      <td>evm.model.hcontig_090_002.3</td>\n",
       "      <td>evm.model.pcontig_090.4</td>\n",
       "      <td>99.75</td>\n",
       "      <td>400</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>400</td>\n",
       "      <td>204</td>\n",
       "      <td>603</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>833.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_091_003.1</th>\n",
       "      <th>559371</th>\n",
       "      <td>evm.model.hcontig_091_003.1</td>\n",
       "      <td>evm.model.pcontig_029.3</td>\n",
       "      <td>97.70</td>\n",
       "      <td>174</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>66</td>\n",
       "      <td>239</td>\n",
       "      <td>15</td>\n",
       "      <td>188</td>\n",
       "      <td>8.000000e-120</td>\n",
       "      <td>347.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_091_003.2</th>\n",
       "      <th>559376</th>\n",
       "      <td>evm.model.hcontig_091_003.2</td>\n",
       "      <td>evm.model.pcontig_029.2</td>\n",
       "      <td>97.06</td>\n",
       "      <td>68</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>502</td>\n",
       "      <td>568</td>\n",
       "      <td>5.000000e-35</td>\n",
       "      <td>125.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">evm.model.hcontig_092_001.2</th>\n",
       "      <th>811325</th>\n",
       "      <td>evm.model.hcontig_092_001.2</td>\n",
       "      <td>evm.model.pcontig_001.355</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>184</td>\n",
       "      <td>199</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811326</th>\n",
       "      <td>evm.model.hcontig_092_001.2</td>\n",
       "      <td>evm.model.pcontig_000.68</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>704</td>\n",
       "      <td>719</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811327</th>\n",
       "      <td>evm.model.hcontig_092_001.2</td>\n",
       "      <td>evm.model.pcontig_073.21</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>250</td>\n",
       "      <td>265</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811331</th>\n",
       "      <td>evm.model.hcontig_092_001.2</td>\n",
       "      <td>evm.model.pcontig_078.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>9.000000e-04</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811332</th>\n",
       "      <td>evm.model.hcontig_092_001.2</td>\n",
       "      <td>evm.model.pcontig_003.92</td>\n",
       "      <td>100.00</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000e-03</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_092_001.4</th>\n",
       "      <th>811382</th>\n",
       "      <td>evm.model.hcontig_092_001.4</td>\n",
       "      <td>evm.model.pcontig_016.293</td>\n",
       "      <td>96.93</td>\n",
       "      <td>976</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>995</td>\n",
       "      <td>1</td>\n",
       "      <td>967</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1949.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_092_001.5</th>\n",
       "      <th>811600</th>\n",
       "      <td>evm.model.hcontig_092_001.5</td>\n",
       "      <td>evm.model.pcontig_087.11</td>\n",
       "      <td>100.00</td>\n",
       "      <td>201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>201</td>\n",
       "      <td>1</td>\n",
       "      <td>201</td>\n",
       "      <td>2.000000e-150</td>\n",
       "      <td>419.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_092_002.1</th>\n",
       "      <th>611970</th>\n",
       "      <td>evm.model.hcontig_092_002.1</td>\n",
       "      <td>evm.model.pcontig_087.11</td>\n",
       "      <td>99.50</td>\n",
       "      <td>201</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>201</td>\n",
       "      <td>1</td>\n",
       "      <td>201</td>\n",
       "      <td>9.000000e-150</td>\n",
       "      <td>417.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">evm.model.hcontig_092_003.1</th>\n",
       "      <th>139254</th>\n",
       "      <td>evm.model.hcontig_092_003.1</td>\n",
       "      <td>evm.model.pcontig_057.45</td>\n",
       "      <td>99.11</td>\n",
       "      <td>112</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>112</td>\n",
       "      <td>297</td>\n",
       "      <td>408</td>\n",
       "      <td>1.000000e-72</td>\n",
       "      <td>234.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139255</th>\n",
       "      <td>evm.model.hcontig_092_003.1</td>\n",
       "      <td>evm.model.pcontig_008.479</td>\n",
       "      <td>99.11</td>\n",
       "      <td>112</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>112</td>\n",
       "      <td>450</td>\n",
       "      <td>561</td>\n",
       "      <td>2.000000e-72</td>\n",
       "      <td>238.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_092_003.2</th>\n",
       "      <th>139370</th>\n",
       "      <td>evm.model.hcontig_092_003.2</td>\n",
       "      <td>evm.model.pcontig_010.150</td>\n",
       "      <td>99.55</td>\n",
       "      <td>220</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>185</td>\n",
       "      <td>404</td>\n",
       "      <td>1</td>\n",
       "      <td>220</td>\n",
       "      <td>1.000000e-164</td>\n",
       "      <td>471.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">evm.model.hcontig_092_003.4</th>\n",
       "      <th>139603</th>\n",
       "      <td>evm.model.hcontig_092_003.4</td>\n",
       "      <td>evm.model.pcontig_001.355</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>184</td>\n",
       "      <td>199</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139604</th>\n",
       "      <td>evm.model.hcontig_092_003.4</td>\n",
       "      <td>evm.model.pcontig_000.68</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>704</td>\n",
       "      <td>719</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139605</th>\n",
       "      <td>evm.model.hcontig_092_003.4</td>\n",
       "      <td>evm.model.pcontig_073.21</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>250</td>\n",
       "      <td>265</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139609</th>\n",
       "      <td>evm.model.hcontig_092_003.4</td>\n",
       "      <td>evm.model.pcontig_078.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>9.000000e-04</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139610</th>\n",
       "      <td>evm.model.hcontig_092_003.4</td>\n",
       "      <td>evm.model.pcontig_003.92</td>\n",
       "      <td>100.00</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000e-03</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_092_003.6</th>\n",
       "      <th>139659</th>\n",
       "      <td>evm.model.hcontig_092_003.6</td>\n",
       "      <td>evm.model.pcontig_016.293</td>\n",
       "      <td>96.72</td>\n",
       "      <td>976</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>995</td>\n",
       "      <td>1</td>\n",
       "      <td>967</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1947.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">evm.model.hcontig_092_004.3</th>\n",
       "      <th>437531</th>\n",
       "      <td>evm.model.hcontig_092_004.3</td>\n",
       "      <td>evm.model.pcontig_001.355</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>184</td>\n",
       "      <td>199</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437532</th>\n",
       "      <td>evm.model.hcontig_092_004.3</td>\n",
       "      <td>evm.model.pcontig_000.68</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>704</td>\n",
       "      <td>719</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437533</th>\n",
       "      <td>evm.model.hcontig_092_004.3</td>\n",
       "      <td>evm.model.pcontig_073.21</td>\n",
       "      <td>100.00</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>493</td>\n",
       "      <td>508</td>\n",
       "      <td>250</td>\n",
       "      <td>265</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>43.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437537</th>\n",
       "      <td>evm.model.hcontig_092_004.3</td>\n",
       "      <td>evm.model.pcontig_078.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>9.000000e-04</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437538</th>\n",
       "      <td>evm.model.hcontig_092_004.3</td>\n",
       "      <td>evm.model.pcontig_003.92</td>\n",
       "      <td>100.00</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000e-03</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_093_001.1</th>\n",
       "      <th>106110</th>\n",
       "      <td>evm.model.hcontig_093_001.1</td>\n",
       "      <td>evm.model.pcontig_053.34</td>\n",
       "      <td>96.52</td>\n",
       "      <td>115</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>124</td>\n",
       "      <td>622</td>\n",
       "      <td>736</td>\n",
       "      <td>2.000000e-71</td>\n",
       "      <td>231.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_093_002.4</th>\n",
       "      <th>127630</th>\n",
       "      <td>evm.model.hcontig_093_002.4</td>\n",
       "      <td>evm.model.pcontig_093.2</td>\n",
       "      <td>99.55</td>\n",
       "      <td>440</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>440</td>\n",
       "      <td>1</td>\n",
       "      <td>440</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>907.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_095_001.2</th>\n",
       "      <th>390804</th>\n",
       "      <td>evm.model.hcontig_095_001.2</td>\n",
       "      <td>evm.model.pcontig_095.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>88</td>\n",
       "      <td>141</td>\n",
       "      <td>228</td>\n",
       "      <td>2.000000e-55</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_095_002.1</th>\n",
       "      <th>347812</th>\n",
       "      <td>evm.model.hcontig_095_002.1</td>\n",
       "      <td>evm.model.pcontig_095.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>88</td>\n",
       "      <td>141</td>\n",
       "      <td>228</td>\n",
       "      <td>2.000000e-55</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evm.model.hcontig_095_002.11</th>\n",
       "      <th>347835</th>\n",
       "      <td>evm.model.hcontig_095_002.11</td>\n",
       "      <td>evm.model.pcontig_095.10</td>\n",
       "      <td>99.16</td>\n",
       "      <td>119</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>119</td>\n",
       "      <td>1</td>\n",
       "      <td>119</td>\n",
       "      <td>2.000000e-81</td>\n",
       "      <td>244.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1618 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            Query  \\\n",
       "Query                                                               \n",
       "evm.model.hcontig_000_001.2  611289   evm.model.hcontig_000_001.2   \n",
       "                             611290   evm.model.hcontig_000_001.2   \n",
       "                             611337   evm.model.hcontig_000_001.2   \n",
       "evm.model.hcontig_000_001.8  611953   evm.model.hcontig_000_001.8   \n",
       "evm.model.hcontig_000_003.1  749535   evm.model.hcontig_000_003.1   \n",
       "                             749536   evm.model.hcontig_000_003.1   \n",
       "evm.model.hcontig_000_007.8  116406   evm.model.hcontig_000_007.8   \n",
       "evm.model.hcontig_000_013.2  113158   evm.model.hcontig_000_013.2   \n",
       "evm.model.hcontig_000_013.4  113160   evm.model.hcontig_000_013.4   \n",
       "evm.model.hcontig_000_014.1  210275   evm.model.hcontig_000_014.1   \n",
       "evm.model.hcontig_000_014.2  210276   evm.model.hcontig_000_014.2   \n",
       "evm.model.hcontig_000_015.3  163157   evm.model.hcontig_000_015.3   \n",
       "evm.model.hcontig_000_015.4  163158   evm.model.hcontig_000_015.4   \n",
       "evm.model.hcontig_000_016.2  320490   evm.model.hcontig_000_016.2   \n",
       "evm.model.hcontig_000_016.5  320497   evm.model.hcontig_000_016.5   \n",
       "evm.model.hcontig_000_016.6  320500   evm.model.hcontig_000_016.6   \n",
       "evm.model.hcontig_000_019.13 344690  evm.model.hcontig_000_019.13   \n",
       "evm.model.hcontig_000_019.41 347610  evm.model.hcontig_000_019.41   \n",
       "evm.model.hcontig_000_021.2  634686   evm.model.hcontig_000_021.2   \n",
       "evm.model.hcontig_000_023.2  456184   evm.model.hcontig_000_023.2   \n",
       "evm.model.hcontig_000_025.3  116644   evm.model.hcontig_000_025.3   \n",
       "evm.model.hcontig_000_028.1  837500   evm.model.hcontig_000_028.1   \n",
       "evm.model.hcontig_000_030.3  467487   evm.model.hcontig_000_030.3   \n",
       "evm.model.hcontig_000_030.4  467494   evm.model.hcontig_000_030.4   \n",
       "evm.model.hcontig_000_031.30 590064  evm.model.hcontig_000_031.30   \n",
       "evm.model.hcontig_000_033.1  581132   evm.model.hcontig_000_033.1   \n",
       "evm.model.hcontig_000_034.1  723200   evm.model.hcontig_000_034.1   \n",
       "evm.model.hcontig_000_036.13 788014  evm.model.hcontig_000_036.13   \n",
       "evm.model.hcontig_000_037.8  788136   evm.model.hcontig_000_037.8   \n",
       "evm.model.hcontig_000_039.3  566835   evm.model.hcontig_000_039.3   \n",
       "...                                                           ...   \n",
       "evm.model.hcontig_090_002.3  553752   evm.model.hcontig_090_002.3   \n",
       "evm.model.hcontig_091_003.1  559371   evm.model.hcontig_091_003.1   \n",
       "evm.model.hcontig_091_003.2  559376   evm.model.hcontig_091_003.2   \n",
       "evm.model.hcontig_092_001.2  811325   evm.model.hcontig_092_001.2   \n",
       "                             811326   evm.model.hcontig_092_001.2   \n",
       "                             811327   evm.model.hcontig_092_001.2   \n",
       "                             811331   evm.model.hcontig_092_001.2   \n",
       "                             811332   evm.model.hcontig_092_001.2   \n",
       "evm.model.hcontig_092_001.4  811382   evm.model.hcontig_092_001.4   \n",
       "evm.model.hcontig_092_001.5  811600   evm.model.hcontig_092_001.5   \n",
       "evm.model.hcontig_092_002.1  611970   evm.model.hcontig_092_002.1   \n",
       "evm.model.hcontig_092_003.1  139254   evm.model.hcontig_092_003.1   \n",
       "                             139255   evm.model.hcontig_092_003.1   \n",
       "evm.model.hcontig_092_003.2  139370   evm.model.hcontig_092_003.2   \n",
       "evm.model.hcontig_092_003.4  139603   evm.model.hcontig_092_003.4   \n",
       "                             139604   evm.model.hcontig_092_003.4   \n",
       "                             139605   evm.model.hcontig_092_003.4   \n",
       "                             139609   evm.model.hcontig_092_003.4   \n",
       "                             139610   evm.model.hcontig_092_003.4   \n",
       "evm.model.hcontig_092_003.6  139659   evm.model.hcontig_092_003.6   \n",
       "evm.model.hcontig_092_004.3  437531   evm.model.hcontig_092_004.3   \n",
       "                             437532   evm.model.hcontig_092_004.3   \n",
       "                             437533   evm.model.hcontig_092_004.3   \n",
       "                             437537   evm.model.hcontig_092_004.3   \n",
       "                             437538   evm.model.hcontig_092_004.3   \n",
       "evm.model.hcontig_093_001.1  106110   evm.model.hcontig_093_001.1   \n",
       "evm.model.hcontig_093_002.4  127630   evm.model.hcontig_093_002.4   \n",
       "evm.model.hcontig_095_001.2  390804   evm.model.hcontig_095_001.2   \n",
       "evm.model.hcontig_095_002.1  347812   evm.model.hcontig_095_002.1   \n",
       "evm.model.hcontig_095_002.11 347835  evm.model.hcontig_095_002.11   \n",
       "\n",
       "                                                         Target   PctID  \\\n",
       "Query                                                                     \n",
       "evm.model.hcontig_000_001.2  611289   evm.model.pcontig_002.624  100.00   \n",
       "                             611290   evm.model.pcontig_010.134  100.00   \n",
       "                             611337   evm.model.pcontig_032.198  100.00   \n",
       "evm.model.hcontig_000_001.8  611953  evm.model.pcontig_000.1077  100.00   \n",
       "evm.model.hcontig_000_003.1  749535     evm.model.pcontig_021.1  100.00   \n",
       "                             749536   evm.model.pcontig_000.534  100.00   \n",
       "evm.model.hcontig_000_007.8  116406   evm.model.pcontig_000.501   95.40   \n",
       "evm.model.hcontig_000_013.2  113158   evm.model.pcontig_000.916  100.00   \n",
       "evm.model.hcontig_000_013.4  113160   evm.model.pcontig_000.917  100.00   \n",
       "evm.model.hcontig_000_014.1  210275   evm.model.pcontig_000.106   99.79   \n",
       "evm.model.hcontig_000_014.2  210276   evm.model.pcontig_000.107   98.85   \n",
       "evm.model.hcontig_000_015.3  163157   evm.model.pcontig_000.873   99.75   \n",
       "evm.model.hcontig_000_015.4  163158   evm.model.pcontig_000.873  100.00   \n",
       "evm.model.hcontig_000_016.2  320490   evm.model.pcontig_004.580   94.74   \n",
       "evm.model.hcontig_000_016.5  320497   evm.model.pcontig_000.186  100.00   \n",
       "evm.model.hcontig_000_016.6  320500   evm.model.pcontig_000.186  100.00   \n",
       "evm.model.hcontig_000_019.13 344690   evm.model.pcontig_004.580   94.74   \n",
       "evm.model.hcontig_000_019.41 347610   evm.model.pcontig_000.214  100.00   \n",
       "evm.model.hcontig_000_021.2  634686   evm.model.pcontig_000.998  100.00   \n",
       "evm.model.hcontig_000_023.2  456184   evm.model.pcontig_000.695  100.00   \n",
       "evm.model.hcontig_000_025.3  116644    evm.model.pcontig_000.77  100.00   \n",
       "evm.model.hcontig_000_028.1  837500   evm.model.pcontig_000.240   99.20   \n",
       "evm.model.hcontig_000_030.3  467487   evm.model.pcontig_000.818   99.35   \n",
       "evm.model.hcontig_000_030.4  467494   evm.model.pcontig_000.819   99.49   \n",
       "evm.model.hcontig_000_031.30 590064   evm.model.pcontig_000.465  100.00   \n",
       "evm.model.hcontig_000_033.1  581132   evm.model.pcontig_000.516   99.41   \n",
       "evm.model.hcontig_000_034.1  723200    evm.model.pcontig_000.83   99.65   \n",
       "evm.model.hcontig_000_036.13 788014   evm.model.pcontig_000.919   97.04   \n",
       "evm.model.hcontig_000_037.8  788136    evm.model.pcontig_000.51   97.30   \n",
       "evm.model.hcontig_000_039.3  566835   evm.model.pcontig_000.108  100.00   \n",
       "...                                                         ...     ...   \n",
       "evm.model.hcontig_090_002.3  553752     evm.model.pcontig_090.4   99.75   \n",
       "evm.model.hcontig_091_003.1  559371     evm.model.pcontig_029.3   97.70   \n",
       "evm.model.hcontig_091_003.2  559376     evm.model.pcontig_029.2   97.06   \n",
       "evm.model.hcontig_092_001.2  811325   evm.model.pcontig_001.355  100.00   \n",
       "                             811326    evm.model.pcontig_000.68  100.00   \n",
       "                             811327    evm.model.pcontig_073.21  100.00   \n",
       "                             811331     evm.model.pcontig_078.3  100.00   \n",
       "                             811332    evm.model.pcontig_003.92  100.00   \n",
       "evm.model.hcontig_092_001.4  811382   evm.model.pcontig_016.293   96.93   \n",
       "evm.model.hcontig_092_001.5  811600    evm.model.pcontig_087.11  100.00   \n",
       "evm.model.hcontig_092_002.1  611970    evm.model.pcontig_087.11   99.50   \n",
       "evm.model.hcontig_092_003.1  139254    evm.model.pcontig_057.45   99.11   \n",
       "                             139255   evm.model.pcontig_008.479   99.11   \n",
       "evm.model.hcontig_092_003.2  139370   evm.model.pcontig_010.150   99.55   \n",
       "evm.model.hcontig_092_003.4  139603   evm.model.pcontig_001.355  100.00   \n",
       "                             139604    evm.model.pcontig_000.68  100.00   \n",
       "                             139605    evm.model.pcontig_073.21  100.00   \n",
       "                             139609     evm.model.pcontig_078.3  100.00   \n",
       "                             139610    evm.model.pcontig_003.92  100.00   \n",
       "evm.model.hcontig_092_003.6  139659   evm.model.pcontig_016.293   96.72   \n",
       "evm.model.hcontig_092_004.3  437531   evm.model.pcontig_001.355  100.00   \n",
       "                             437532    evm.model.pcontig_000.68  100.00   \n",
       "                             437533    evm.model.pcontig_073.21  100.00   \n",
       "                             437537     evm.model.pcontig_078.3  100.00   \n",
       "                             437538    evm.model.pcontig_003.92  100.00   \n",
       "evm.model.hcontig_093_001.1  106110    evm.model.pcontig_053.34   96.52   \n",
       "evm.model.hcontig_093_002.4  127630     evm.model.pcontig_093.2   99.55   \n",
       "evm.model.hcontig_095_001.2  390804     evm.model.pcontig_095.1  100.00   \n",
       "evm.model.hcontig_095_002.1  347812     evm.model.pcontig_095.1  100.00   \n",
       "evm.model.hcontig_095_002.11 347835    evm.model.pcontig_095.10   99.16   \n",
       "\n",
       "                                     AlnLgth  NumMis  NumGap  StartQuery  \\\n",
       "Query                                                                      \n",
       "evm.model.hcontig_000_001.2  611289     1416       0       0           1   \n",
       "                             611290     1416       0       0           1   \n",
       "                             611337      354       0       0        1063   \n",
       "evm.model.hcontig_000_001.8  611953      167       0       0           1   \n",
       "evm.model.hcontig_000_003.1  749535      112       0       0           1   \n",
       "                             749536      112       0       0           1   \n",
       "evm.model.hcontig_000_007.8  116406      261      12       0           1   \n",
       "evm.model.hcontig_000_013.2  113158      327       0       0          92   \n",
       "evm.model.hcontig_000_013.4  113160       74       0       0          28   \n",
       "evm.model.hcontig_000_014.1  210275      477       1       0           1   \n",
       "evm.model.hcontig_000_014.2  210276      349       4       0           1   \n",
       "evm.model.hcontig_000_015.3  163157      408       1       0          90   \n",
       "evm.model.hcontig_000_015.4  163158      263       0       0           1   \n",
       "evm.model.hcontig_000_016.2  320490       57       3       0          22   \n",
       "evm.model.hcontig_000_016.5  320497      654       0       0           1   \n",
       "evm.model.hcontig_000_016.6  320500       81       0       0           1   \n",
       "evm.model.hcontig_000_019.13 344690       57       3       0          22   \n",
       "evm.model.hcontig_000_019.41 347610       94       0       0           1   \n",
       "evm.model.hcontig_000_021.2  634686       93       0       0           1   \n",
       "evm.model.hcontig_000_023.2  456184      252       0       0           1   \n",
       "evm.model.hcontig_000_025.3  116644      149       0       0           1   \n",
       "evm.model.hcontig_000_028.1  837500      125       1       0          30   \n",
       "evm.model.hcontig_000_030.3  467487      154       1       0           1   \n",
       "evm.model.hcontig_000_030.4  467494      389       2       0           1   \n",
       "evm.model.hcontig_000_031.30 590064      401       0       0           1   \n",
       "evm.model.hcontig_000_033.1  581132      170       1       0           1   \n",
       "evm.model.hcontig_000_034.1  723200      286       1       0           1   \n",
       "evm.model.hcontig_000_036.13 788014      135       4       0           1   \n",
       "evm.model.hcontig_000_037.8  788136      889       1       1           1   \n",
       "evm.model.hcontig_000_039.3  566835      143       0       0          16   \n",
       "...                                      ...     ...     ...         ...   \n",
       "evm.model.hcontig_090_002.3  553752      400       1       0           1   \n",
       "evm.model.hcontig_091_003.1  559371      174       4       0          66   \n",
       "evm.model.hcontig_091_003.2  559376       68       1       1           1   \n",
       "evm.model.hcontig_092_001.2  811325       16       0       0         493   \n",
       "                             811326       16       0       0         493   \n",
       "                             811327       16       0       0         493   \n",
       "                             811331       28       0       0         429   \n",
       "                             811332       28       0       0         429   \n",
       "evm.model.hcontig_092_001.4  811382      976      21       2          20   \n",
       "evm.model.hcontig_092_001.5  811600      201       0       0           1   \n",
       "evm.model.hcontig_092_002.1  611970      201       1       0           1   \n",
       "evm.model.hcontig_092_003.1  139254      112       1       0           1   \n",
       "                             139255      112       1       0           1   \n",
       "evm.model.hcontig_092_003.2  139370      220       1       0         185   \n",
       "evm.model.hcontig_092_003.4  139603       16       0       0         493   \n",
       "                             139604       16       0       0         493   \n",
       "                             139605       16       0       0         493   \n",
       "                             139609       28       0       0         429   \n",
       "                             139610       28       0       0         429   \n",
       "evm.model.hcontig_092_003.6  139659      976      23       2          20   \n",
       "evm.model.hcontig_092_004.3  437531       16       0       0         493   \n",
       "                             437532       16       0       0         493   \n",
       "                             437533       16       0       0         493   \n",
       "                             437537       28       0       0         429   \n",
       "                             437538       28       0       0         429   \n",
       "evm.model.hcontig_093_001.1  106110      115       4       0          10   \n",
       "evm.model.hcontig_093_002.4  127630      440       2       0           1   \n",
       "evm.model.hcontig_095_001.2  390804       88       0       0           1   \n",
       "evm.model.hcontig_095_002.1  347812       88       0       0           1   \n",
       "evm.model.hcontig_095_002.11 347835      119       1       0           1   \n",
       "\n",
       "                                     StopQuery  StartTarget  StopTarget  \\\n",
       "Query                                                                     \n",
       "evm.model.hcontig_000_001.2  611289       1416            1        1416   \n",
       "                             611290       1416            1        1416   \n",
       "                             611337       1416            1         354   \n",
       "evm.model.hcontig_000_001.8  611953        167            1         167   \n",
       "evm.model.hcontig_000_003.1  749535        112          183         294   \n",
       "                             749536        112          183         294   \n",
       "evm.model.hcontig_000_007.8  116406        261          899        1159   \n",
       "evm.model.hcontig_000_013.2  113158        418            2         328   \n",
       "evm.model.hcontig_000_013.4  113160        101          109         182   \n",
       "evm.model.hcontig_000_014.1  210275        477            1         477   \n",
       "evm.model.hcontig_000_014.2  210276        349            1         349   \n",
       "evm.model.hcontig_000_015.3  163157        497          109         516   \n",
       "evm.model.hcontig_000_015.4  163158        263          617         879   \n",
       "evm.model.hcontig_000_016.2  320490         78           77         133   \n",
       "evm.model.hcontig_000_016.5  320497        654          120         773   \n",
       "evm.model.hcontig_000_016.6  320500         81            1          81   \n",
       "evm.model.hcontig_000_019.13 344690         78           77         133   \n",
       "evm.model.hcontig_000_019.41 347610         94            1          94   \n",
       "evm.model.hcontig_000_021.2  634686         93          351         443   \n",
       "evm.model.hcontig_000_023.2  456184        252          386         637   \n",
       "evm.model.hcontig_000_025.3  116644        149          121         269   \n",
       "evm.model.hcontig_000_028.1  837500        154            1         125   \n",
       "evm.model.hcontig_000_030.3  467487        154            1         154   \n",
       "evm.model.hcontig_000_030.4  467494        389            1         389   \n",
       "evm.model.hcontig_000_031.30 590064        401            1         401   \n",
       "evm.model.hcontig_000_033.1  581132        170            1         170   \n",
       "evm.model.hcontig_000_034.1  723200        286            1         286   \n",
       "evm.model.hcontig_000_036.13 788014        135            1         135   \n",
       "evm.model.hcontig_000_037.8  788136        866            1         889   \n",
       "evm.model.hcontig_000_039.3  566835        158          329         471   \n",
       "...                                        ...          ...         ...   \n",
       "evm.model.hcontig_090_002.3  553752        400          204         603   \n",
       "evm.model.hcontig_091_003.1  559371        239           15         188   \n",
       "evm.model.hcontig_091_003.2  559376         68          502         568   \n",
       "evm.model.hcontig_092_001.2  811325        508          184         199   \n",
       "                             811326        508          704         719   \n",
       "                             811327        508          250         265   \n",
       "                             811331        456            1          28   \n",
       "                             811332        456            1          28   \n",
       "evm.model.hcontig_092_001.4  811382        995            1         967   \n",
       "evm.model.hcontig_092_001.5  811600        201            1         201   \n",
       "evm.model.hcontig_092_002.1  611970        201            1         201   \n",
       "evm.model.hcontig_092_003.1  139254        112          297         408   \n",
       "                             139255        112          450         561   \n",
       "evm.model.hcontig_092_003.2  139370        404            1         220   \n",
       "evm.model.hcontig_092_003.4  139603        508          184         199   \n",
       "                             139604        508          704         719   \n",
       "                             139605        508          250         265   \n",
       "                             139609        456            1          28   \n",
       "                             139610        456            1          28   \n",
       "evm.model.hcontig_092_003.6  139659        995            1         967   \n",
       "evm.model.hcontig_092_004.3  437531        508          184         199   \n",
       "                             437532        508          704         719   \n",
       "                             437533        508          250         265   \n",
       "                             437537        456            1          28   \n",
       "                             437538        456            1          28   \n",
       "evm.model.hcontig_093_001.1  106110        124          622         736   \n",
       "evm.model.hcontig_093_002.4  127630        440            1         440   \n",
       "evm.model.hcontig_095_001.2  390804         88          141         228   \n",
       "evm.model.hcontig_095_002.1  347812         88          141         228   \n",
       "evm.model.hcontig_095_002.11 347835        119            1         119   \n",
       "\n",
       "                                           e-value  BitScore  \n",
       "Query                                                         \n",
       "evm.model.hcontig_000_001.2  611289   0.000000e+00    2958.0  \n",
       "                             611290   0.000000e+00    2958.0  \n",
       "                             611337   0.000000e+00     737.0  \n",
       "evm.model.hcontig_000_001.8  611953  6.000000e-120     351.0  \n",
       "evm.model.hcontig_000_003.1  749535   2.000000e-75     236.0  \n",
       "                             749536   7.000000e-72     234.0  \n",
       "evm.model.hcontig_000_007.8  116406  2.000000e-172     509.0  \n",
       "evm.model.hcontig_000_013.2  113158   0.000000e+00     667.0  \n",
       "evm.model.hcontig_000_013.4  113160   7.000000e-49     155.0  \n",
       "evm.model.hcontig_000_014.1  210275   0.000000e+00     979.0  \n",
       "evm.model.hcontig_000_014.2  210276   0.000000e+00     722.0  \n",
       "evm.model.hcontig_000_015.3  163157   0.000000e+00     836.0  \n",
       "evm.model.hcontig_000_015.4  163158   0.000000e+00     555.0  \n",
       "evm.model.hcontig_000_016.2  320490   1.000000e-31     112.0  \n",
       "evm.model.hcontig_000_016.5  320497   0.000000e+00    1332.0  \n",
       "evm.model.hcontig_000_016.6  320500   4.000000e-49     165.0  \n",
       "evm.model.hcontig_000_019.13 344690   1.000000e-31     112.0  \n",
       "evm.model.hcontig_000_019.41 347610   8.000000e-59     190.0  \n",
       "evm.model.hcontig_000_021.2  634686   2.000000e-57     189.0  \n",
       "evm.model.hcontig_000_023.2  456184   0.000000e+00     523.0  \n",
       "evm.model.hcontig_000_025.3  116644  9.000000e-107     308.0  \n",
       "evm.model.hcontig_000_028.1  837500   2.000000e-84     246.0  \n",
       "evm.model.hcontig_000_030.3  467487  2.000000e-108     308.0  \n",
       "evm.model.hcontig_000_030.4  467494   0.000000e+00     786.0  \n",
       "evm.model.hcontig_000_031.30 590064   0.000000e+00     833.0  \n",
       "evm.model.hcontig_000_033.1  581132  5.000000e-120     352.0  \n",
       "evm.model.hcontig_000_034.1  723200   0.000000e+00     590.0  \n",
       "evm.model.hcontig_000_036.13 788014   8.000000e-80     251.0  \n",
       "evm.model.hcontig_000_037.8  788136   0.000000e+00    1793.0  \n",
       "evm.model.hcontig_000_039.3  566835   3.000000e-98     294.0  \n",
       "...                                            ...       ...  \n",
       "evm.model.hcontig_090_002.3  553752   0.000000e+00     833.0  \n",
       "evm.model.hcontig_091_003.1  559371  8.000000e-120     347.0  \n",
       "evm.model.hcontig_091_003.2  559376   5.000000e-35     125.0  \n",
       "evm.model.hcontig_092_001.2  811325   1.000000e-04      43.1  \n",
       "                             811326   1.000000e-04      43.9  \n",
       "                             811327   1.000000e-04      43.1  \n",
       "                             811331   9.000000e-04      40.4  \n",
       "                             811332   1.000000e-03      40.4  \n",
       "evm.model.hcontig_092_001.4  811382   0.000000e+00    1949.0  \n",
       "evm.model.hcontig_092_001.5  811600  2.000000e-150     419.0  \n",
       "evm.model.hcontig_092_002.1  611970  9.000000e-150     417.0  \n",
       "evm.model.hcontig_092_003.1  139254   1.000000e-72     234.0  \n",
       "                             139255   2.000000e-72     238.0  \n",
       "evm.model.hcontig_092_003.2  139370  1.000000e-164     471.0  \n",
       "evm.model.hcontig_092_003.4  139603   1.000000e-04      43.1  \n",
       "                             139604   1.000000e-04      43.9  \n",
       "                             139605   1.000000e-04      43.1  \n",
       "                             139609   9.000000e-04      40.4  \n",
       "                             139610   1.000000e-03      40.4  \n",
       "evm.model.hcontig_092_003.6  139659   0.000000e+00    1947.0  \n",
       "evm.model.hcontig_092_004.3  437531   1.000000e-04      43.1  \n",
       "                             437532   1.000000e-04      43.9  \n",
       "                             437533   1.000000e-04      43.1  \n",
       "                             437537   9.000000e-04      40.4  \n",
       "                             437538   1.000000e-03      40.4  \n",
       "evm.model.hcontig_093_001.1  106110   2.000000e-71     231.0  \n",
       "evm.model.hcontig_093_002.4  127630   0.000000e+00     907.0  \n",
       "evm.model.hcontig_095_001.2  390804   2.000000e-55     173.0  \n",
       "evm.model.hcontig_095_002.1  347812   2.000000e-55     173.0  \n",
       "evm.model.hcontig_095_002.11 347835   2.000000e-81     244.0  \n",
       "\n",
       "[1618 rows x 12 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "htgNoAllelesBlastDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
