{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Notebook to get synteny measures between two assemblies\n",
    "The goal is to get a tool that takes in orthofinder results and a CDS files to check fof the length of synteny with different gene categories.\n",
    "\n",
    "The idea is that BUSCOs are more syntenous then effectors or such. \n",
    "\n",
    "The program tries to count how many neighbours of an allele pair are in the same orthogroup. This requires first to anchor the allele pairing e.g. find out what the 1:1 allele is and then walk outward to see if each others neighbours are in the same orthogroup. For now we won't allow for any skips and just go one by one.\n",
    "\n",
    "GENOMEA_orthoarray =  [1,2,3,4]\n",
    "GENOMEB_orthoarray =  [1,2,3,4]\n",
    "\n",
    "should result into a tuple that where the first element is the number of observed matches and the second is the number of possible matches. here this would be (8,8).\n",
    "\n",
    "If this analysis is performed on two different set of gene groups this may tell something if microsynteny is more conserved within one gene group then another."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### initial outline\n",
    "\n",
    "get a gene ID -> get its neightbours n in two different arrays up and down stream -> get the orthogroup arrays for the neighbours \n",
    "get a gene ID  -> get orthogroup of the gene -> get all members of the orthogroup that belong to the other genome -> get their neighbours +/- 1 -> get the best seed where both neighbours add up,  \n",
    "* if we have only one best match that is easy. Use this seed as 1:1 match get the neighbourhood array up and down -> compare the arrays one element at a time and safe the tuple in a dictonary.\n",
    "* if multiple hits. Move one further out with each hit and look at those with the same ideas.\n",
    "\n",
    "Things also to generate:\n",
    "A match dictionary for allele pairing.  \n",
    "A unqiue gene dictionary.  \n",
    "A paraloge dictionray. Meaning where there is no allele pairing but thing are in the same ortho group."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:32.523757Z",
     "start_time": "2019-02-12T03:33:30.940365Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/benjamin/anaconda3/lib/python3.5/site-packages/Bio/SearchIO/__init__.py:211: BiopythonExperimentalWarning: Bio.SearchIO is an experimental submodule which may undergo significant changes prior to its future official release.\n",
      "  BiopythonExperimentalWarning)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "from Bio import SeqUtils\n",
    "import pysam\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from pybedtools import BedTool\n",
    "import numpy as np\n",
    "import pybedtools\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import sys\n",
    "import subprocess\n",
    "import shutil\n",
    "from Bio.Seq import Seq\n",
    "import pysam\n",
    "from Bio import SearchIO\n",
    "import json\n",
    "import glob\n",
    "import scipy.stats as stats\n",
    "import statsmodels as sms\n",
    "import statsmodels.sandbox.stats.multicomp\n",
    "import distance\n",
    "import seaborn as sns\n",
    "from pybedtools import BedTool\n",
    "import matplotlib\n",
    "from sklearn.externals.joblib import Parallel, delayed\n",
    "import itertools as it\n",
    "import tempfile\n",
    "from scipy.signal import argrelextrema\n",
    "import scipy\n",
    "from IPython.display import Image\n",
    "from PIL import Image\n",
    "from collections import OrderedDict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:32.533499Z",
     "start_time": "2019-02-12T03:33:32.527044Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "### define some path that at the end should come as args\n",
    "ORTHOFINDER_FILE_NAME = '/home/benjamin/genome_assembly/Warrior/DK0911_v04/comp_orthology/orthofinder/OrthoFinder_all/Orthogroups_2.txt'\n",
    "QUERY_GENOME_GENE_BED6 = '/home/benjamin/genome_assembly/Warrior/DK0911_v04/comp_orthology/Pst104E_annotations/Pst_104E_v13_p_ctg.gene.bed'\n",
    "SUBJECT_GENOME_GENE_BED6 = '/home/benjamin/genome_assembly/Warrior/DK0911_v04/comp_orthology/DK0911_annotations/DK_0911_v04_p_ctg.genes.gene.bed'\n",
    "QUERY_SELECT_GENE_BED6 = '/home/benjamin/genome_assembly/Warrior/DK0911_v04/comp_orthology/orthofinder/test_folder/Pst_104E_v13_p_ctg.gene1000.bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:32.620169Z",
     "start_time": "2019-02-12T03:33:32.535883Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#now write some functions\n",
    "def get_neighbours(gene_id, bed_filename, n=5, direction='up'):\n",
    "    \"\"\"A function that either takes a filename to return an array of downstream and upstream neighbouring genes.\n",
    "    Input: \n",
    "        gene_id\n",
    "        bed6 filename\n",
    "        n being the number of neighbours we want to get\n",
    "        direction being up or down.\n",
    "    Output:\n",
    "        returns the largest possible array of neighbours up to n.\n",
    "        The 0 element is always the closest neighbour now matter if you return and 'up' or 'down' array.\"\"\"\n",
    "    bed_6_header = ['chrom', 'start', 'stop', 'gene_id', 'phase', 'strand']\n",
    "    try:\n",
    "        bed_df = pd.read_csv(bed_filename, sep='\\t', header=None, names=bed_6_header)\n",
    "    except:\n",
    "        print('Check if the bedfiles are bed6')\n",
    "    if not direction in ['up', 'down']:\n",
    "        print('Ensure direction is up or down.')\n",
    "        \n",
    "    #fix to make sure the gene id is actually in the bed_file if not just return an empty list    \n",
    "    if gene_id not in bed_df['gene_id'].unique():\n",
    "        return []\n",
    "    gene_index = bed_df[bed_df['gene_id'] == gene_id].index[0]\n",
    "    contig = bed_df.loc[gene_index, ['chrom']]['chrom']\n",
    "    contig_index = bed_df[bed_df['chrom']== contig].index\n",
    "    if direction == 'up':\n",
    "        index_list = []\n",
    "        if (gene_index+(n)) in contig_index:\n",
    "            for i in range(gene_index+1, gene_index+(n+1)):\n",
    "                index_list.append(i)\n",
    "        else:\n",
    "            for i in range(gene_index+1, contig_index[-1]+1):\n",
    "                index_list.append(i)\n",
    "        return bed_df.loc[index_list, ['gene_id']]['gene_id'].tolist()\n",
    "    if direction == 'down':\n",
    "        index_list = []\n",
    "        if (gene_index-n) in contig_index:\n",
    "            for i in range(gene_index-(n), gene_index):\n",
    "                index_list.append(i)\n",
    "        else:\n",
    "            for i in range(contig_index[0], gene_index):\n",
    "                index_list.append(i)\n",
    "        index_list.reverse()\n",
    "        return bed_df.loc[index_list, ['gene_id']]['gene_id'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:32.630269Z",
     "start_time": "2019-02-12T03:33:32.622656Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_orthologs(gene_id, gene_to_ortho_dict, orthofinder_dict, other_id):\n",
    "    \"\"\"The function returns all the potential orthologs of the comparative species\n",
    "    Input:\n",
    "        gene_id\n",
    "        gene_to_ortho_dict, the ortho dict\n",
    "        orthofinder_dict, the dictionary of the orthgroups to get all genes belonging to the orthogroup in question.\n",
    "        other_id, is the identifier of the comparative species.\"\"\"\n",
    "\n",
    "    return [x for x in orthofinder_dict[gene_to_ortho_dict[gene_id]] if x.startswith(other_id)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:33.004406Z",
     "start_time": "2019-02-12T03:33:32.926288Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def count_pairings(query_ortho, subject_ortho, simple = True):\n",
    "    \"\"\"Function that counts the pair wise overlap of two lists.\n",
    "    In the simple = True mode this is a stricked 1:1 pairing.\n",
    "    In the simple = Flase mode this is a sliding window of 1 vs [0, 1 , 2] elements.\n",
    "    Input: \n",
    "        query_ortho list\n",
    "        subject_orth list\n",
    "        simple True or False\n",
    "    output:\n",
    "        obs the number of observed pairings\n",
    "        max_len the number of potential pairings\"\"\"\n",
    "    if len(query_ortho) <= len(subject_ortho):\n",
    "        max_len = len(query_ortho)\n",
    "    else:\n",
    "        max_len = len(subject_ortho)\n",
    "    obs = 0\n",
    "    if simple == True:\n",
    "        for i in range(0, max_len):\n",
    "            if query_ortho[i] == subject_ortho[i]:\n",
    "                obs = obs + 1\n",
    "        return obs, max_len\n",
    "            \n",
    "        \n",
    "    if simple == False:\n",
    "        for i in range(0, max_len):\n",
    "            if i > 0 and i < max_len -1:    \n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i-1] \\\n",
    "                or query_ortho[i] == subject_ortho[i+1]:\n",
    "                    obs = obs + 1\n",
    "            elif i == max_len -1:\n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i-1]:\n",
    "                    obs = obs + 1\n",
    "            elif query_ortho[i] == subject_ortho[i]:\n",
    "                obs = obs + 1\n",
    "            elif i == 0:\n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i+1]:\n",
    "                    obs = obs + 1\n",
    "            \n",
    "        return obs, max_len"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:33.734597Z",
     "start_time": "2019-02-12T03:33:33.563934Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_up_and_down(gene_id, query_downstream_ortho,query_upstream_ortho, subject_bed_fn, n, simple):\n",
    "    \"\"\"A function that returns a list of two tuples. The first tuple is the for the downstream pairing the second for the\n",
    "    upstream pairing of obs, max possible matches of orthologs.\n",
    "    Input: \n",
    "           Gene_id to check from the subject genome, e.g. initial orthogroup matches of query gene.\n",
    "           query_downstream_ortho is the list of downstream ortho groups from the gene_ids neighbours.\n",
    "           query_upstream_ortho is the list of upstream ortho groups from the gene_ids neighbours.\n",
    "           subject_bed_fn is the absolute path of the subject_bed_fn to get the neighbouring genes of the gene_id.\n",
    "           n is the numbers of neighbours to search.\n",
    "           simple can be True or False for searching without skiping (window of three) or with skipping enabled.\n",
    "    Output:\n",
    "           [(obs_down, max_down), (obs_up, max_up)]\"\"\"\n",
    "    \n",
    "    \n",
    "    if simple not in [True, False]:\n",
    "        simple = True\n",
    "    \n",
    "    subject_upstream_ortho = [gene_to_ortho_dict[x] for x in get_neighbours(gene_id, subject_bed_fn, n=n, direction='up')]\n",
    "    subject_downstream_ortho = [gene_to_ortho_dict[x] for x in get_neighbours(gene_id, subject_bed_fn, n=n, direction='down')]\n",
    "    \n",
    "    #define in case something has no upstream or downstream hits\n",
    "    obs_up, max_up, obs_down, max_down = 0, 0, 0, 0\n",
    "    \n",
    "    if len(query_upstream_ortho) > 0 and len(subject_upstream_ortho) > 0: \n",
    "        if query_upstream_ortho[0] == subject_upstream_ortho[0]:   \n",
    "            obs_up, max_up = count_pairings(query_upstream_ortho, subject_upstream_ortho, simple=simple)       \n",
    "            #now look at the downstream\n",
    "            obs_down, max_down = count_pairings(query_downstream_ortho, subject_downstream_ortho, simple=simple)\n",
    "            \n",
    "            if len(query_downstream_ortho) > 0 and len(query_upstream_ortho):\n",
    "                if query_downstream_ortho[0] == query_upstream_ortho[0]:\n",
    "                    print(\"Orthogroups of up and downstream are the same. Gene_id: %s\" % gene_id)\n",
    "                    if len(subject_downstream_ortho) > 0 and len(query_upstream_ortho) >0 :\n",
    "                        obs_up_new, max_up_new = count_pairings(query_upstream_ortho, subject_downstream_ortho, simple=simple)\n",
    "                        obs_down_new, max_down_new = count_pairings(query_downstream_ortho, subject_upstream_ortho, simple=simple)\n",
    "                        if up_down_ratio([(obs_down_new, max_down_new), (obs_up_new, max_up_new)])\\\n",
    "                        > up_down_ratio([(obs_down, max_down), (obs_up, max_up)]):\n",
    "                        #think about if we also want to test for \n",
    "                        #(max_down_new+max_up_new) > (max_down, max_up) :\n",
    "                            obs_up, max_up = obs_up_new, max_up_new\n",
    "                            obs_down, max_down = obs_down_new, max_down_new\n",
    "            \n",
    "            \n",
    "            return [(obs_down, max_down), (obs_up, max_up)]\n",
    "        \n",
    "    if len(subject_downstream_ortho) > 0 and len(query_upstream_ortho) >0 :\n",
    "        if query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "            obs_up, max_up = count_pairings(query_upstream_ortho, subject_downstream_ortho, simple=simple)\n",
    "            obs_down, max_down = count_pairings(query_downstream_ortho, subject_upstream_ortho, simple=simple)\n",
    "            return [(obs_down, max_down), (obs_up, max_up)]\n",
    "        \n",
    "    if len(subject_downstream_ortho) > 0 and len(query_downstream_ortho) > 0:\n",
    "        if subject_downstream_ortho[0] == query_downstream_ortho[0]:\n",
    "            obs_up, max_up = count_pairings(query_upstream_ortho, subject_upstream_ortho, simple=simple)       \n",
    "            #now look at the downstream\n",
    "            obs_down, max_down = count_pairings(query_downstream_ortho, subject_downstream_ortho, simple=simple)\n",
    "            return [(obs_down, max_down), (obs_up, max_up)]\n",
    "    \n",
    "    return [(obs_down, max_down), (obs_up, max_up)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T05:47:06.085112Z",
     "start_time": "2019-02-12T05:47:05.930004Z"
    }
   },
   "outputs": [],
   "source": [
    "def count_pairings_array(query_ortho, subject_ortho, n, simple = True):\n",
    "    \"\"\"Function that does a pairwise by element comparison of two arrays.\n",
    "    In the simple = True mode this is a stricked 1:1 pairing.\n",
    "    In the simple = Flase mode this is a sliding window of 1 vs [0, 1 , 2] elements.\n",
    "    Input: \n",
    "        query_ortho list\n",
    "        subject_orth list\n",
    "        simple True or False\n",
    "    output:\n",
    "        result array that has the positional overlap of the array.\n",
    "        e.g. for n = 8 with one list being lenght 8 and the other length 6\n",
    "        [0, 1, 0, 1, 0, 1, nan, nan]\"\"\"\n",
    "    \n",
    "    array = np.empty(n)\n",
    "    array[:] = np.nan\n",
    "    array\n",
    "    \n",
    "    if simple == True:\n",
    "        for i in range(0, n):\n",
    "            if i < len(query_ortho) and i < len(subject_ortho):\n",
    "                if query_ortho[i] == subject_ortho[i]:\n",
    "                    array[i] = int(1)\n",
    "                else:\n",
    "                    array[i] = int(0)\n",
    "            else:\n",
    "                continue\n",
    "        return array\n",
    "            \n",
    "        \n",
    "    if simple == False:\n",
    "\n",
    "        for i in range(0, len(query_ortho)):\n",
    "            \n",
    "            if i == len(subject_ortho)-1:\n",
    "                if query_ortho[i] == subject_ortho[i-1] or query_ortho[i] == subject_ortho[i]:\n",
    "                    array[i] = int(1)\n",
    "                else:\n",
    "                    array[i] = int(0)\n",
    "            elif i > 0 and i < len(subject_ortho):    \n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i-1] \\\n",
    "                or query_ortho[i] == subject_ortho[i+1]:\n",
    "                    array[i] = int(1)\n",
    "                else:\n",
    "                    array[i] = int(0)\n",
    "\n",
    "            elif i == 0:\n",
    "                do_it = False\n",
    "                try:\n",
    "                    if query_ortho[i] == subject_ortho[i+1]:\n",
    "                        do_it = True\n",
    "                except IndexError:\n",
    "                        pass\n",
    "                try:\n",
    "                    if query_ortho[i] == subject_ortho[i]:\n",
    "                        do_it = True\n",
    "                except IndexError:\n",
    "                        pass\n",
    "                if do_it == True:\n",
    "                    array[i] = int(1)\n",
    "                else:\n",
    "                    array[i] = int(0)\n",
    "        return array\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "(array([  1.,   1.,   0.,   1.,   1.,  nan,  nan,  nan]), 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:40.081410Z",
     "start_time": "2019-02-12T03:33:40.068307Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([  1.,   1.,   0.,  nan,  nan])"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_pairings_array([0, 1, 2, 6, 6, 7],[0, 0, 1], n=5, simple = False )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:40:03.166314Z",
     "start_time": "2019-02-12T03:40:02.501843Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_up_and_down_array(gene_id, query_downstream_ortho,query_upstream_ortho, subject_bed_fn, n, simple):\n",
    "    \"\"\"A function that returns a list of two tuples. The first tuple is the for the downstream pairing the second for the\n",
    "    upstream pairing of obs, max possible matches of orthologs.\n",
    "    Input: \n",
    "           Gene_id to check from the subject genome, e.g. initial orthogroup matches of query gene.\n",
    "           query_downstream_ortho is the list of downstream ortho groups from the gene_ids neighbours.\n",
    "           query_upstream_ortho is the list of upstream ortho groups from the gene_ids neighbours.\n",
    "           subject_bed_fn is the absolute path of the subject_bed_fn to get the neighbouring genes of the gene_id.\n",
    "           n is the numbers of neighbours to search.\n",
    "           simple can be True or False for searching without skiping (window of three) or with skipping enabled.\n",
    "    Output:\n",
    "           [(obs_down, max_down), (obs_up, max_up)]\"\"\"\n",
    "    \n",
    "    \n",
    "    if simple not in [True, False]:\n",
    "        simple = True\n",
    "    \n",
    "    subject_upstream_ortho = [gene_to_ortho_dict[x] for x in get_neighbours(gene_id, subject_bed_fn, n=n, direction='up')]\n",
    "    subject_downstream_ortho = [gene_to_ortho_dict[x] for x in get_neighbours(gene_id, subject_bed_fn, n=n, direction='down')]\n",
    "    \n",
    "    #now move to empty arrays\n",
    "    \n",
    "    \n",
    "    #define in case something has no upstream or downstream hits\n",
    "    array = array = np.empty(n)\n",
    "    array[:] = np.nan\n",
    "    array\n",
    "    up_array, down_array = array, array\n",
    "    \n",
    "    if simple == True:\n",
    "    \n",
    "        if len(query_upstream_ortho) > 0 and len(subject_upstream_ortho) > 0: \n",
    "\n",
    "            if query_upstream_ortho[0] == subject_upstream_ortho[0]:   \n",
    "                up_array = count_pairings_array(query_upstream_ortho, subject_upstream_ortho, n, simple=simple)       \n",
    "                #now look at the downstream\n",
    "                down_array = count_pairings_array(query_downstream_ortho, subject_downstream_ortho,n, simple=simple)\n",
    "\n",
    "                if len(query_downstream_ortho) > 0 and len(query_upstream_ortho) > 0:\n",
    "                    if query_downstream_ortho[0] == query_upstream_ortho[0]:\n",
    "                        print(\"Orthogroups of up and downstream are the same. Gene_id: %s\" % gene_id)\n",
    "                        up_array_new = count_pairings_array(query_upstream_ortho, subject_downstream_ortho, n, simple=simple)\n",
    "                        down_array_new = count_pairings(query_downstream_ortho, subject_upstream_ortho, n, simple=simple)\n",
    "                        if up_down_ratio_array([down_array_new, up_array_new])\\\n",
    "                            > up_down_ratio_array([down_array, up_array]):\n",
    "                            #think about if we also want to test for \n",
    "                            #(max_down_new+max_up_new) > (max_down, max_up) :\n",
    "                            down_array, up_array = down_array_new, up_array_new\n",
    "\n",
    "\n",
    "                return [down_array, up_array]\n",
    "\n",
    "        if len(subject_downstream_ortho) > 0 and len(query_upstream_ortho) >0 :\n",
    "            if query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "                up_array = count_pairings_array(query_upstream_ortho, subject_downstream_ortho, n, simple=simple)\n",
    "                down_array = count_pairings_array(query_downstream_ortho, subject_upstream_ortho, n, simple=simple)\n",
    "                return [down_array, up_array]\n",
    "\n",
    "        if len(subject_downstream_ortho) > 0 and len(query_downstream_ortho) > 0:\n",
    "            if subject_downstream_ortho[0] == query_downstream_ortho[0]:\n",
    "                up_array = count_pairings_array(query_upstream_ortho, subject_upstream_ortho,n, simple=simple)       \n",
    "                #now look at the downstream\n",
    "                down_array = count_pairings(query_downstream_ortho, subject_downstream_ortho,n, simple=simple)\n",
    "                return [down_array, up_array]\n",
    "\n",
    "        return [down_array, up_array]\n",
    "    \n",
    "    elif simple == False:\n",
    "        \n",
    "        do_it = False\n",
    "        one_exists = False\n",
    "        \n",
    "        if len(query_upstream_ortho) > 0 and len(subject_upstream_ortho) > 0: \n",
    "            try:\n",
    "                if query_upstream_ortho[0] == subject_upstream_ortho[0] or \\\n",
    "                query_upstream_ortho[0] == subject_upstream_ortho[1]:\n",
    "                    do_it = True\n",
    "            \n",
    "            except IndexError:\n",
    "                if query_upstream_ortho[0] == subject_upstream_ortho[0]:\n",
    "                    do_it = True\n",
    "                    \n",
    "            if do_it == True:        \n",
    "                up_array = count_pairings_array(query_upstream_ortho, subject_upstream_ortho, n, simple=simple)       \n",
    "                #now look at the downstream\n",
    "                down_array = count_pairings_array(query_downstream_ortho, subject_downstream_ortho,n, simple=simple)\n",
    "\n",
    "                if len(query_downstream_ortho) > 0 and len(query_upstream_ortho) > 0:\n",
    "                    do_it_too = False\n",
    "                    try:\n",
    "                        if query_downstream_ortho[0] == query_upstream_ortho[0] or \\\n",
    "                        query_upstream_ortho[0] == subject_upstream_ortho[1]:\n",
    "                            do_it_too = True\n",
    "                    except IndexError:\n",
    "                        if query_downstream_ortho[0] == query_upstream_ortho[0]:\n",
    "                            do_it_too = True\n",
    "                            \n",
    "                        if do_it_too == True:\n",
    "                            print(\"Orthogroups of up and downstream are the same. Gene_id: %s\" % gene_id)\n",
    "                            up_array_new = count_pairings_array(query_upstream_ortho, subject_downstream_ortho, n, simple=simple)\n",
    "                            down_array_new = count_pairings(query_downstream_ortho, subject_upstream_ortho, n, simple=simple)\n",
    "                            if up_down_ratio_array([down_array_new, up_array_new])\\\n",
    "                                > up_down_ratio_array([down_array, up_array]):\n",
    "                                #think about if we also want to test for \n",
    "                                #(max_down_new+max_up_new) > (max_down, max_up) :\n",
    "                                down_array, up_array = down_array_new, up_array_new\n",
    "\n",
    "\n",
    "                    return [down_array, up_array]\n",
    "\n",
    "        if len(subject_downstream_ortho) > 0 and len(query_upstream_ortho) >0 :\n",
    "            try:\n",
    "                \n",
    "                if query_upstream_ortho[0] == subject_downstream_ortho[0] or \\\n",
    "                query_upstream_ortho[0] == subject_upstream_ortho[1]:\n",
    "                    do_it = True\n",
    "            except IndexError:\n",
    "                if query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "                    do_it = True\n",
    "            if do_it == True:\n",
    "                up_array = count_pairings_array(query_upstream_ortho, subject_downstream_ortho, n, simple=simple)\n",
    "                down_array = count_pairings_array(query_downstream_ortho, subject_upstream_ortho, n, simple=simple)\n",
    "                return [down_array, up_array]\n",
    "\n",
    "        if len(subject_downstream_ortho) > 0 and len(query_downstream_ortho) > 0:\n",
    "            try: \n",
    "                if subject_downstream_ortho[0] == query_downstream_ortho[0] or \\\n",
    "                subject_downstream_ortho[0] == query_downstream_ortho[1]:\n",
    "                    do_it = True\n",
    "            except IndexError:\n",
    "                if subject_downstream_ortho[0] == query_downstream_ortho[0]:\n",
    "                    do_it = True\n",
    "            if do_it == True:\n",
    "                up_array = count_pairings_array(query_upstream_ortho, subject_upstream_ortho,n, simple=simple)       \n",
    "                    #now look at the downstream\n",
    "                down_array = count_pairings_array(query_downstream_ortho, subject_downstream_ortho,n, simple=simple)\n",
    "                return [down_array, up_array]\n",
    "\n",
    "        return [down_array, up_array]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:18:12.445502Z",
     "start_time": "2019-02-12T04:18:12.425885Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def array_to_tuple(array_list):\n",
    "    if len(array_list) != 2:\n",
    "        print('The length of the list is not 2.')\n",
    "        return 0\n",
    "    tuple_list = []\n",
    "    for array in array_list:\n",
    "        tuple_list.append(((np.nan_to_num(array)).sum(),np.count_nonzero(~np.isnan(array)) ))\n",
    "    return tuple_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:43.602788Z",
     "start_time": "2019-02-12T03:33:43.581314Z"
    }
   },
   "outputs": [],
   "source": [
    "def up_down_ratio_array(up_down_list):\n",
    "    if len(up_down_list) != 2:\n",
    "        print('The length of the list is not 2.')\n",
    "        return 0\n",
    "    sum_observed = np.nan_to_num(up_down_list[0]).sum() + np.nan_to_num(up_down_list[1]).sum()\n",
    "    sum_possible = np.count_nonzero(~np.isnan(up_down_list[0])) + np.count_nonzero(~np.isnan(up_down_list[1]))\n",
    "    if sum_possible > 0:\n",
    "        \n",
    "        return sum_observed/sum_possible\n",
    "    else:\n",
    "        return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:44.283451Z",
     "start_time": "2019-02-12T03:33:44.268519Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def up_down_ratio(up_down_list):\n",
    "    if len(up_down_list) == 0:\n",
    "        return 0\n",
    "    if (up_down_list[0][1] + up_down_list[1][1]) > 0:\n",
    "        ratio = (up_down_list[0][0] + up_down_list[1][0]) / (up_down_list[0][1] + up_down_list[1][1])\n",
    "        return ratio\n",
    "    else:\n",
    "        return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:45:47.617903Z",
     "start_time": "2019-02-12T04:45:47.610023Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#generate dicts that will be used to track things\n",
    "allele_dict = {}\n",
    "singleton_dict = {}\n",
    "paralog_dict = {}\n",
    "up_match_dict = {}\n",
    "down_match_dict = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:45:49.523600Z",
     "start_time": "2019-02-12T04:45:49.468586Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get the identifiers if not provided.\n",
    "query_id = pd.read_csv(QUERY_GENOME_GENE_BED6, sep='\\t', header=None).loc[0,[3]][3].split('_')[0]\n",
    "subject_id = pd.read_csv(SUBJECT_GENOME_GENE_BED6, sep='\\t', header=None).loc[0,[3]][3].split('_')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:45:51.802472Z",
     "start_time": "2019-02-12T04:45:50.986569Z"
    }
   },
   "outputs": [],
   "source": [
    "#get the orthofinder results read in\n",
    "#this is the dict for all orthogroups and what members they have\n",
    "orthofinder_dict = {}\n",
    "with open(ORTHOFINDER_FILE_NAME) as fh:\n",
    "    for line in fh:\n",
    "        line.strip()\n",
    "        orthofinder_dict[int(line.split(':')[0].strip('OG'))] = line.split(':')[1].strip().split(' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:45:54.281399Z",
     "start_time": "2019-02-12T04:45:53.739255Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#this is the dict for proteins and what orthogroup they have\n",
    "gene_to_ortho_dict = {}\n",
    "for key,value in orthofinder_dict.items():\n",
    "    for item in value:\n",
    "        gene_to_ortho_dict[item] = key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:25:11.650826Z",
     "start_time": "2019-02-12T04:25:11.640639Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['DK0911_15056']\n"
     ]
    }
   ],
   "source": [
    "#gene_id = 'Pst104E_00003'\n",
    "gene_id = 'Pst104E_00000'\n",
    "print([x for x in orthofinder_dict[gene_to_ortho_dict[gene_id]]if x.startswith('DK0911')])\n",
    "n=8\n",
    "simple = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:25:12.349544Z",
     "start_time": "2019-02-12T04:25:12.273371Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#populate the query arrays for neighbours and their corresponding ortho group ids\n",
    "query_upstream = get_neighbours(gene_id, QUERY_GENOME_GENE_BED6, n=n, direction='up')\n",
    "query_downstream = get_neighbours(gene_id, QUERY_GENOME_GENE_BED6, n=n, direction='down')\n",
    "query_upstream_ortho = [gene_to_ortho_dict[x] for x in query_upstream]\n",
    "query_downstream_ortho = [gene_to_ortho_dict[x] for x in query_downstream]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:25:13.119420Z",
     "start_time": "2019-02-12T04:25:12.899255Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ 1.,  1.,  1.,  1.,  1.,  0.,  1.,  1.])]\n"
     ]
    }
   ],
   "source": [
    "#get the first orthologs (aka potential allele pairings)\n",
    "first_orthologs = get_orthologs(gene_id, gene_to_ortho_dict, orthofinder_dict, subject_id)\n",
    "if len(first_orthologs) == 0:\n",
    "    singleton_dict[gene_id] = True\n",
    "elif len(first_orthologs) == 1:\n",
    "    down_up_list = get_up_and_down_array(first_orthologs[0],\\\n",
    "                        query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6,\\\n",
    "                                         n=n, simple=simple)\n",
    "    down_match_dict[gene_id], up_match_dict[gene_id] = down_up_list[0], down_up_list[1]\n",
    "    if sum((~np.isnan(down_up_list[0]))) > 0 and sum((~np.isnan(down_up_list[1]))) > 0:\n",
    "        allele_dict[gene_id] = first_orthologs[0]\n",
    "    else:\n",
    "        paralog_dict[gene_id] = first_orthologs[0]\n",
    "elif len(first_orthologs) > 1:\n",
    "    ortho_dict = {}\n",
    "    for ortho in first_orthologs:\n",
    "        down_up_list = get_up_and_down_array(ortho,query_downstream_ortho,query_upstream_ortho,\\\n",
    "                                           SUBJECT_GENOME_GENE_BED6, n=n, simple=simple)\n",
    "        if sum((~np.isnan(down_up_list[0]))) == 0 and sum((~np.isnan(down_up_list[1]))) == 0:\n",
    "            continue\n",
    "        else:\n",
    "            ortho_dict[ortho] = down_up_list\n",
    "    \n",
    "    \n",
    "    max_list = [(0,0), (0,0)]\n",
    "    real_ortho = ''\n",
    "    real_ortho_list = []\n",
    "    \n",
    "    for ortho, ortho_list in ortho_dict.items():\n",
    "        ortho_tuple_list = array_to_tuple(ortho_list)\n",
    "        if up_down_ratio(ortho_tuple_list) > up_down_ratio(max_list):\n",
    "            print(ortho_tuple_list)\n",
    "            max_list = ortho_list\n",
    "            real_ortho = ortho\n",
    "            real_ortho_list = ortho_list\n",
    "            down_up_list = real_ortho_list\n",
    "    if max_list != [(0,0), (0,0)]:\n",
    "        allele_dict[gene_id] = real_ortho\n",
    "        down_match_dict[gene_id], up_match_dict[gene_id] = ortho_list[0], ortho_list[1]\n",
    "    else:\n",
    "        paralog_dict[gene_id] = list(ortho_dict.keys())\n",
    "\n",
    "print(down_up_list)\n",
    "#think about catching paralogs meaning that +1/-1 are not in the same orthogroup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:25:13.870343Z",
     "start_time": "2019-02-12T04:25:13.859282Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Pst104E_00000': array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]),\n",
       " 'Pst104E_00001': array([  1.,  nan,  nan,  nan,  nan,  nan,  nan,  nan]),\n",
       " 'Pst104E_00002': array([ 1.,  1.,  1.,  0.,  1.,  1.,  1.,  0.]),\n",
       " 'Pst104E_00022': array([ 1.,  1.,  1.,  0.,  0.,  0.,  0.,  0.]),\n",
       " 'Pst104E_00033': array([  1.,   1.,   0.,   0.,   0.,  nan,  nan,  nan])}"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "down_match_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:25:17.363507Z",
     "start_time": "2019-02-12T04:25:17.352479Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Pst104E_00000': array([ 1.,  1.,  1.,  1.,  1.,  0.,  1.,  1.]),\n",
       " 'Pst104E_00001': array([ 1.,  1.,  1.,  1.,  0.,  1.,  1.,  1.]),\n",
       " 'Pst104E_00022': array([ 0.,  1.,  1.,  1.,  1.,  0.,  0.,  0.]),\n",
       " 'Pst104E_00033': array([ 0.,  1.,  0.,  0.,  0.,  0.,  0.,  0.])}"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "up_match_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:52.866689Z",
     "start_time": "2019-02-12T03:33:52.860367Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[3456, 34, 1077, 6914, 1007, 14, 3819, 8542] [94582, 685, 3115, 18428, 39658, 22733, 262, 15049]\n"
     ]
    }
   ],
   "source": [
    "print(query_downstream_ortho, query_upstream_ortho)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:33:53.933977Z",
     "start_time": "2019-02-12T03:33:53.820973Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[3456, 34, 1077, 9190, 4, 4, 2, 9189] [685, 39657, 3115, 18428, 39658, 14638, 28956, 22733]\n"
     ]
    }
   ],
   "source": [
    "if len(first_orthologs) == 1:\n",
    "    subject_ortho_up = [gene_to_ortho_dict[x] for x in get_neighbours(first_orthologs[0], SUBJECT_GENOME_GENE_BED6, n=n, direction='up')]\n",
    "    subject_ortho_down = [gene_to_ortho_dict[x] for x in get_neighbours(first_orthologs[0], SUBJECT_GENOME_GENE_BED6, n=n, direction='down')]\n",
    "elif len(first_orthologs) > 1:\n",
    "    subject_ortho_up = [gene_to_ortho_dict[x] for x in get_neighbours(real_ortho, SUBJECT_GENOME_GENE_BED6, n=n, direction='up')]\n",
    "    subject_ortho_down = [gene_to_ortho_dict[x] for x in get_neighbours(real_ortho, SUBJECT_GENOME_GENE_BED6, n=n, direction='down')]\n",
    "\n",
    "print(subject_ortho_down, subject_ortho_up)    \n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:36:08.351520Z",
     "start_time": "2019-02-12T03:36:08.258927Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]),\n",
       " array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_up_and_down_array(first_orthologs[0],\\\n",
    "                        query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6,\\\n",
    "                                         n=n, simple=simple)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T04:23:39.211639Z",
     "start_time": "2019-02-12T04:23:33.124923Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n",
      "[array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan]), array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])]\n"
     ]
    }
   ],
   "source": [
    "for x in first_orthologs:\n",
    "    print(get_up_and_down_array(x,\\\n",
    "                        query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6,\\\n",
    "                                         n=n, simple=simple))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T05:58:30.650598Z",
     "start_time": "2019-02-12T05:47:53.359473Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(3.0, 3), (5.0, 8)]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/benjamin/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:48: DeprecationWarning: elementwise == comparison failed; this will raise an error in the future.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(0.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (5.0, 8)]\n",
      "[(3.0, 8), (4.0, 8)]\n",
      "[(4.0, 8), (3.0, 8)]\n",
      "[(6.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (6.0, 8)]\n",
      "[(1.0, 8), (8.0, 8)]\n",
      "[(0.0, 3), (3.0, 8)]\n",
      "[(1.0, 3), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 2), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 5)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (3.0, 7)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 4), (2.0, 8)]\n",
      "[(1.0, 2), (2.0, 8)]\n",
      "[(0.0, 8), (4.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 5), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 5), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 6), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 4), (4.0, 8)]\n",
      "[(1.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 1), (3.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (2.0, 7)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 3), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (6.0, 8)]\n",
      "[(1.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (3.0, 7)]\n",
      "[(0.0, 1), (3.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (4.0, 8)]\n",
      "[(1.0, 6), (1.0, 8)]\n",
      "[(2.0, 3), (1.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(2.0, 3), (2.0, 8)]\n",
      "[(3.0, 8), (1.0, 7)]\n",
      "[(3.0, 6), (0.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(1.0, 2), (2.0, 8)]\n",
      "[(3.0, 8), (0.0, 4)]\n",
      "[(1.0, 8), (3.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(2.0, 3), (2.0, 8)]\n",
      "[(2.0, 8), (2.0, 8)]\n",
      "[(3.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (3.0, 8)]\n",
      "[(3.0, 7), (0.0, 8)]\n",
      "[(3.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (2.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 4), (1.0, 8)]\n",
      "[(1.0, 2), (2.0, 8)]\n",
      "[(3.0, 7), (0.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(0.0, 3), (4.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(0.0, 2), (5.0, 8)]\n",
      "[(3.0, 8), (1.0, 5)]\n",
      "[(1.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (4.0, 8)]\n",
      "[(4.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(1.0, 3), (4.0, 8)]\n",
      "[(0.0, 8), (4.0, 8)]\n",
      "[(2.0, 3), (3.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(2.0, 3), (3.0, 8)]\n",
      "[(2.0, 3), (2.0, 8)]\n",
      "[(3.0, 3), (1.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(4.0, 8), (8.0, 8)]\n",
      "[(4.0, 8), (8.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (7.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 3)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 3), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 4)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 4), (0.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(0.0, 3), (2.0, 8)]\n",
      "[(1.0, 3), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 2), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 5)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 7)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(6.0, 8), (4.0, 8)]\n",
      "[(2.0, 3), (0.0, 8)]\n",
      "[(2.0, 8), (7.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 7)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 6), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(1.0, 8), (5.0, 8)]\n",
      "[(6.0, 8), (4.0, 7)]\n",
      "[(6.0, 8), (0.0, 3)]\n",
      "[(1.0, 4), (8.0, 8)]\n",
      "[(3.0, 6), (8.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (2.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (6.0, 8)]\n",
      "[(4.0, 8), (4.0, 8)]\n",
      "[(4.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(4.0, 8), (3.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (8.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (4.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(8.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(5.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(1.0, 8), (1.0, 3)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(5.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 4), (1.0, 8)]\n",
      "[(0.0, 4), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 5)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 1)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 2)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 1)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 5)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 2), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 1)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(1.0, 5), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 1)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 4)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 1), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 4)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 1)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 1)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 4)]\n",
      "[(0.0, 3), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 3)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(3.0, 8), (8.0, 8)]\n",
      "[(4.0, 8), (8.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(3.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(3.0, 8), (8.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(6.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 5)]\n",
      "[(0.0, 4), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 5), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(2.0, 8), (5.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(8.0, 8), (2.0, 8)]\n",
      "[(8.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (5.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 3)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(4.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(4.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (6.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(3.0, 8), (5.0, 8)]\n",
      "[(4.0, 8), (4.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (8.0, 8)]\n",
      "[(2.0, 8), (8.0, 8)]\n",
      "[(4.0, 8), (6.0, 8)]\n",
      "[(5.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(5.0, 8), (6.0, 8)]\n",
      "[(5.0, 8), (5.0, 8)]\n",
      "[(6.0, 8), (4.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (5.0, 8)]\n",
      "[(0.0, 5), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 6), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 2), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 3)]\n",
      "[(0.0, 8), (1.0, 2)]\n",
      "[(2.0, 8), (8.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 6), (0.0, 8)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 2)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (8.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (3.0, 6)]\n",
      "[(1.0, 8), (3.0, 5)]\n",
      "[(2.0, 8), (2.0, 4)]\n",
      "[(2.0, 8), (1.0, 3)]\n",
      "[(4.0, 8), (0.0, 2)]\n",
      "[(3.0, 3), (5.0, 8)]\n",
      "[(7.0, 7), (1.0, 8)]\n",
      "[(0.0, 8), (5.0, 8)]\n",
      "[(1.0, 8), (4.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (3.0, 8)]\n",
      "[(3.0, 8), (2.0, 8)]\n",
      "[(4.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (8.0, 8)]\n",
      "[(3.0, 8), (8.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (7.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(0.0, 7), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 1), (3.0, 8)]\n",
      "[(0.0, 3), (2.0, 8)]\n",
      "[(1.0, 4), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 5)]\n",
      "[(1.0, 8), (1.0, 4)]\n",
      "[(2.0, 5), (0.0, 8)]\n",
      "[(0.0, 1), (8.0, 8)]\n",
      "[(0.0, 3), (1.0, 8)]\n",
      "[(1.0, 4), (0.0, 8)]\n",
      "[(2.0, 5), (0.0, 8)]\n",
      "[(3.0, 3), (6.0, 8)]\n",
      "[(1.0, 8), (1.0, 4)]\n",
      "[(2.0, 8), (0.0, 3)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(7.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (5.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (8.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(6.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (5.0, 8)]\n",
      "[(1.0, 8), (5.0, 8)]\n",
      "[(2.0, 8), (5.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (6.0, 8)]\n",
      "[(7.0, 8), (2.0, 8)]\n",
      "[(7.0, 8), (1.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(0.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 4)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 3)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(8.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (4.0, 8)]\n",
      "[(1.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(1.0, 8), (2.0, 8)]\n",
      "[(2.0, 8), (1.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(3.0, 8), (1.0, 8)]\n",
      "[(8.0, 8), (3.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(3.0, 8), (6.0, 8)]\n",
      "[(4.0, 8), (5.0, 8)]\n",
      "[(5.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (8.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(5.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (6.0, 8)]\n",
      "[(8.0, 8), (4.0, 8)]\n",
      "[(8.0, 8), (3.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 4)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 3)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 2), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (8.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(6.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (8.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (3.0, 8)]\n",
      "[(2.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(4.0, 8), (0.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (2.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 1)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 5), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 1), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (7.0, 8)]\n",
      "[(2.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (5.0, 8)]\n",
      "[(5.0, 8), (8.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (7.0, 8)]\n",
      "[(8.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (3.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(3.0, 8), (0.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(7.0, 8), (8.0, 8)]\n",
      "[(8.0, 8), (5.0, 8)]\n",
      "[(8.0, 8), (4.0, 8)]\n",
      "[(4.0, 8), (7.0, 8)]\n",
      "[(6.0, 8), (6.0, 8)]\n",
      "[(7.0, 8), (6.0, 8)]\n",
      "[(3.0, 8), (8.0, 8)]\n",
      "[(7.0, 8), (4.0, 8)]\n",
      "[(6.0, 8), (2.0, 8)]\n",
      "[(1.0, 8), (1.0, 8)]\n",
      "[(2.0, 8), (0.0, 8)]\n",
      "[(3.0, 8), (7.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 7), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 7), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(1.0, 8), (0.0, 7)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(5.0, 8), (6.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(6.0, 8), (5.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(5.0, 8), (6.0, 8)]\n",
      "[(5.0, 8), (6.0, 8)]\n",
      "[(0.0, 1), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(0.0, 8), (1.0, 8)]\n",
      "[(6.0, 8), (6.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 1)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 1)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(6.0, 8), (6.0, 8)]\n",
      "[(5.0, 8), (6.0, 8)]\n",
      "[(1.0, 8), (0.0, 8)]\n",
      "[(6.0, 8), (6.0, 8)]\n"
     ]
    }
   ],
   "source": [
    "gene_done_list = []\n",
    "for x in range(0,1000):\n",
    "    if x < 10:\n",
    "        gene_id = 'Pst104E_0000%s' % x\n",
    "    elif x < 100:\n",
    "        gene_id = 'Pst104E_000%s' % x\n",
    "    elif x < 1000:\n",
    "        gene_id = 'Pst104E_00%s' % x\n",
    "    gene_done_list.append(gene_id)\n",
    "    #now some testing here    \n",
    "    query_upstream = get_neighbours(gene_id, QUERY_GENOME_GENE_BED6, n=n, direction='up')\n",
    "    query_downstream = get_neighbours(gene_id, QUERY_GENOME_GENE_BED6, n=n, direction='down')\n",
    "    query_upstream_ortho = [gene_to_ortho_dict[x] for x in query_upstream]\n",
    "    query_downstream_ortho = [gene_to_ortho_dict[x] for x in query_downstream]\n",
    "    \n",
    "    first_orthologs = get_orthologs(gene_id, gene_to_ortho_dict, orthofinder_dict, subject_id)\n",
    "    if len(first_orthologs) == 0:\n",
    "        singleton_dict[gene_id] = True\n",
    "    elif len(first_orthologs) == 1:\n",
    "        down_up_list = get_up_and_down_array(first_orthologs[0],\\\n",
    "                            query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6,\\\n",
    "                                             n=n, simple=simple)\n",
    "        down_match_dict[gene_id], up_match_dict[gene_id] = down_up_list[0], down_up_list[1]\n",
    "        if sum((~np.isnan(down_up_list[0]))) > 0 and sum((~np.isnan(down_up_list[1]))) > 0:\n",
    "            allele_dict[gene_id] = first_orthologs[0]\n",
    "        else:\n",
    "            paralog_dict[gene_id] = first_orthologs[0]\n",
    "    elif len(first_orthologs) > 1:\n",
    "        ortho_dict = {}\n",
    "        for ortho in first_orthologs:\n",
    "            down_up_list = get_up_and_down_array(ortho,query_downstream_ortho,query_upstream_ortho,\\\n",
    "                                               SUBJECT_GENOME_GENE_BED6, n=n, simple=simple)\n",
    "            if sum((~np.isnan(down_up_list[0]))) == 0 and sum((~np.isnan(down_up_list[1]))) == 0:\n",
    "                continue\n",
    "            else:\n",
    "                ortho_dict[ortho] = down_up_list\n",
    "\n",
    "\n",
    "        max_list = [(0,0), (0,0)]\n",
    "        real_ortho = ''\n",
    "        real_ortho_list = []\n",
    "\n",
    "        for ortho, ortho_list in ortho_dict.items():\n",
    "            ortho_tuple_list = array_to_tuple(ortho_list)\n",
    "            if up_down_ratio(ortho_tuple_list) > up_down_ratio(max_list):\n",
    "                print(ortho_tuple_list)\n",
    "                max_list = ortho_list\n",
    "                real_ortho = ortho\n",
    "                real_ortho_list = ortho_list\n",
    "                down_up_list = real_ortho_list\n",
    "        if max_list != [(0,0), (0,0)]:\n",
    "            allele_dict[gene_id] = real_ortho\n",
    "            down_match_dict[gene_id], up_match_dict[gene_id] = ortho_list[0], ortho_list[1]\n",
    "        else:\n",
    "            paralog_dict[gene_id] = list(ortho_dict.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T05:47:27.724198Z",
     "start_time": "2019-02-12T05:47:27.719315Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pst104E_00136\n"
     ]
    }
   ],
   "source": [
    "#problem gene_id\n",
    "print(gene_id)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "<ipython-input-72-cc707f5a2a2e> in count_pairings_array(query_ortho, subject_ortho, n, simple)\n",
    "     47                 try:\n",
    "---> 48                     if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i+1]:\n",
    "     49                         do_it = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:31:48.001457Z",
     "start_time": "2019-02-12T03:31:47.995273Z"
    }
   },
   "outputs": [],
   "source": [
    "print(query_downstream_ortho, query_upstream_ortho)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:22:57.516613Z",
     "start_time": "2019-02-12T03:22:57.508534Z"
    }
   },
   "outputs": [],
   "source": [
    "down_match_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:23:13.274290Z",
     "start_time": "2019-02-12T03:23:13.268939Z"
    }
   },
   "outputs": [],
   "source": [
    "query_upstream"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "'DK0911_15063'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-12T03:25:47.085405Z",
     "start_time": "2019-02-12T03:25:47.077762Z"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_done_list = []\n",
    "for x in range(0,1000):\n",
    "    if x < 10:\n",
    "        gene_id = 'Pst104E_0000%s' % x\n",
    "    elif x < 100:\n",
    "        gene_id = 'Pst104E_000%s' % x\n",
    "    elif x < 1000:\n",
    "        gene_id = 'Pst104E_00%s' % x\n",
    "    gene_done_list.append(gene_id)\n",
    "    #now some testing here    \n",
    "    query_upstream = get_neighbours(gene_id, QUERY_GENOME_GENE_BED6, n=n, direction='up')\n",
    "    query_downstream = get_neighbours(gene_id, QUERY_GENOME_GENE_BED6, n=n, direction='down')\n",
    "    query_upstream_ortho = [gene_to_ortho_dict[x] for x in query_upstream]\n",
    "    query_downstream_ortho = [gene_to_ortho_dict[x] for x in query_downstream]\n",
    "    \n",
    "    first_orthologs = get_orthologs(gene_id, gene_to_ortho_dict, orthofinder_dict, subject_id)\n",
    "    if len(first_orthologs) == 0:\n",
    "        singleton_dict[gene_id] = True\n",
    "    elif len(first_orthologs) == 1:\n",
    "        up_and_down_list = get_up_and_down(first_orthologs[0],query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6, n=n, simple=simple)\n",
    "        up_down_dict[gene_id] = up_and_down_list\n",
    "        if up_and_down_list[0][0] > 0 and up_and_down_list[1][0] > 0:\n",
    "            allele_dict[gene_id] = first_orthologs[0]\n",
    "        else:\n",
    "            paralog_dict[gene_id] = first_orthologs[0]\n",
    "    elif len(first_orthologs) > 1:\n",
    "        ortho_dict = {}\n",
    "        for ortho in first_orthologs:\n",
    "            up_and_down_list = get_up_and_down(ortho,query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6, n=n, simple=simple)\n",
    "            if up_and_down_list[0][0] == 0 and up_and_down_list[1][0] == 0:\n",
    "                continue\n",
    "            else:\n",
    "                ortho_dict[ortho] = up_and_down_list\n",
    "        max_list = [(0,0), (0,0)]\n",
    "        for ortho, ortho_list in ortho_dict.items():\n",
    "            if up_down_ratio(ortho_list) > up_down_ratio(max_list) and ortho_list[0][0] > 0 and ortho_list[1][0] >0:\n",
    "                max_list = ortho_list\n",
    "        if max_list[0][0] > 0 and max_list[1][0] > 0:\n",
    "            allele_dict[gene_id] = first_orthologs[0]\n",
    "        else:\n",
    "            paralog_dict[gene_id] = first_orthologs[0] "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get in the ideas of np arrarys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-11T06:12:43.014789Z",
     "start_time": "2019-02-11T06:12:43.004681Z"
    }
   },
   "outputs": [],
   "source": [
    "array = np.empty(8)\n",
    "array[:] = np.nan\n",
    "array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-11T06:10:06.451580Z",
     "start_time": "2019-02-11T06:10:06.439046Z"
    }
   },
   "outputs": [],
   "source": [
    "array = np.empty(8)\n",
    "array[:] = np.nan\n",
    "array\n",
    "array[2] = int(1)\n",
    "\n",
    "np.count_nonzero(np.isnan(array))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-11T06:10:11.998769Z",
     "start_time": "2019-02-11T06:10:11.989895Z"
    }
   },
   "outputs": [],
   "source": [
    "[array, array]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-07T06:34:15.108102Z",
     "start_time": "2019-01-07T06:34:15.100410Z"
    }
   },
   "outputs": [],
   "source": [
    "np.concatenate((array,array),axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-07T06:31:44.657692Z",
     "start_time": "2019-01-07T06:31:44.650590Z"
    }
   },
   "outputs": [],
   "source": [
    "array + array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "singleton_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "allele_list = list(allele_dict.keys())\n",
    "singleton_list = list(singleton_dict.keys())\n",
    "paralogs_list = list(paralog_dict.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "set(gene_done_list) - set(allele_list) - set(singleton_list) - set(paralogs_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_down_dict.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_and_down_list[0][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "first_orthologs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#now we need to work on the case when we have multiple orthologs. This will be done on a 1:1 basis moving outward if the first neighbours match return a 1.\n",
    "#if in the first round nothing matches don't move forward. Just return all possible paraloge pairings.\n",
    "#if multiple match move one further out and add half a point for each match either side.\n",
    "#still need to think about missing orthologs from the original search and how to pull those in  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def up_down_ratio(up_down_list):\n",
    "    if len(up_and_down_list) == 0:\n",
    "        return 0\n",
    "    if (up_and_down_list[0][1] + up_and_down_list[1][1]) > 0:\n",
    "        ratio = (up_and_down_list[0][0] + up_and_down_list[1][0]) / (up_and_down_list[0][1] + up_and_down_list[1][1])\n",
    "        return ratio\n",
    "    else:\n",
    "        return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho_dict = {}\n",
    "for ortho in first_orthologs:\n",
    "    print(ortho)\n",
    "    up_and_down_list = get_up_and_down(ortho,query_downstream_ortho,query_upstream_ortho, SUBJECT_GENOME_GENE_BED6, n=n, simple=simple)\n",
    "    if up_and_down_list[0][0] == 0 and up_and_down_list[1][0] == 0:\n",
    "        continue\n",
    "    else:\n",
    "        ortho_dict[ortho] = up_and_down_list\n",
    "max_list = [(0,0), (0,0)]\n",
    "for ortho, ortho_list in ortho_dict.items():\n",
    "    if up_down_ratio(max_list) < up_down_ratio(ortho_list):\n",
    "        max_list = ortho_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "broken = ortho\n",
    "gene_to_ortho_dict[ortho]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print([gene_to_ortho_dict[x] for x in get_neighbours(ortho, SUBJECT_GENOME_GENE_BED6, n=n, direction='up')])\n",
    "print([gene_to_ortho_dict[x] for x in get_neighbours(ortho, SUBJECT_GENOME_GENE_BED6, n=n, direction='down')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get the first orthologs (aka potential allele pairings)\n",
    "first_orthologs = get_orthologs(gene_id, gene_to_ortho_dict, orthofinder_dict, subject_id)\n",
    "if len(first_orthologs) == 0:\n",
    "    singleton_dict[gene_id] = True\n",
    "elif len(first_orthologs) == 1:\n",
    "    subject_upstream_ortho = [gene_to_ortho_dict[x] for x in get_neighbours(first_orthologs[0], SUBJECT_GENOME_GENE_BED6, n=n, direction='up')]\n",
    "    subject_downstream_ortho = [gene_to_ortho_dict[x] for x in get_neighbours(first_orthologs[0], SUBJECT_GENOME_GENE_BED6, n=n, direction='down')]\n",
    "    if query_upstream_ortho[0] == subject_upstream_ortho[0]:   \n",
    "        obs_up, max_up = count_pairings(query_upstream_ortho, subject_upstream_ortho, simple=simple)       \n",
    "        #now look at the downstream\n",
    "        obs_down, max_down = count_pairings(query_downstream_ortho, subject_downstream_ortho, simple=simple)\n",
    "    elif query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "        obs_up, max_up = count_pairings(query_upstream_ortho, subject_downstream_ortho, simple=simple)\n",
    "        obs_down, max_down = count_pairings(query_downstream_ortho, subject_upstream_ortho, simple=simple)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now we need to look at the pairing of the directions\n",
    "#and add stuff up\n",
    "if query_upstream_ortho[0] == subject_upstream_ortho[0]:\n",
    "    obs_up = 0\n",
    "    obs_down = 0 \n",
    "    if len(query_upstream_ortho) <= len(subject_upstream_ortho):\n",
    "        \n",
    "        max_up = len(query_upstream_ortho)\n",
    "    else:\n",
    "        max_up = len(subject_upstream_ortho)\n",
    "        \n",
    "    for i in range(0, max_up):\n",
    "        if query_upstream_ortho[i] == subject_upstream_ortho[i]:\n",
    "            obs_up = obs_up + 1\n",
    "            \n",
    "    #now look at the downstream\n",
    "    if len(query_downstream_ortho) <= len(subject_downstream_ortho):\n",
    "        \n",
    "        max_down = len(query_downstream_ortho)\n",
    "    else:\n",
    "        max_down = len(subject_downstream_ortho)\n",
    "        \n",
    "    for i in range(0, max_down):\n",
    "        if query_downstream_ortho[i] == subject_downstream_ortho[i]:\n",
    "            obs_down = obs_down + 1\n",
    "    \n",
    "elif query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "    obs_up = 0\n",
    "    obs_down = 0\n",
    "    if len(query_upstream_ortho) <= len(subject_downstream_ortho):\n",
    "        max_up = len(query_upstream_ortho)\n",
    "    else:\n",
    "        max_up = len(subject_downstream_ortho)\n",
    "    for i in range(0, obs_down):\n",
    "        if query_upstream_ortho[i] == subject_downstream_ortho[i]:\n",
    "            obs_up = obs_up + 1\n",
    "            \n",
    "    #now look at the downstream\n",
    "    if len(query_downstream_ortho) <= len(subject_upstream_ortho):\n",
    "        \n",
    "        max_down = len(query_downstream_ortho)\n",
    "    else:\n",
    "        max_down = len(subject_upstream_ortho)\n",
    "        \n",
    "    for i in range(0, max_down):\n",
    "        if query_downstream_ortho[i] == subject_upstream_ortho[i]:\n",
    "            obs_down = obs_down + 1\n",
    "print('obs_up %i, max_up %i, obs_down %i, max_down %i' % (obs_up, max_up, obs_down, max_down))\n",
    "print('sub_up %s\\nquery_up %s\\nsub_down %s\\nquery_down %s' % (subject_upstream_ortho, query_upstream_ortho,subject_downstream_ortho,query_downstream_ortho ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def count_pairings(query_ortho, subject_ortho, simple = True):\n",
    "    \"\"\"Function that counts the pair wise overlap of two lists.\n",
    "    In the simple = True mode this is a stricked 1:1 pairing.\n",
    "    In the simple = Flase mode this is a sliding window of 1 vs [0, 1 , 2] elements.\n",
    "    Input: \n",
    "        query_ortho list\n",
    "        subject_orth list\n",
    "        simple True or False\n",
    "    output:\n",
    "        obs the number of observed pairings\n",
    "        max_len the number of potential pairings\"\"\"\n",
    "    if len(query_ortho) <= len(subject_ortho):\n",
    "        max_len = len(query_ortho)\n",
    "    else:\n",
    "        max_len = len(subject_ortho)\n",
    "    obs = 0\n",
    "    if simple == True:\n",
    "        for i in range(0, max_len):\n",
    "            if query_ortho[i] == subject_ortho[i]:\n",
    "                obs = obs + 1\n",
    "        return obs, max_len\n",
    "            \n",
    "        \n",
    "    if simple == False:\n",
    "        for i in range(0, max_len):\n",
    "            if i > 0 and i < max_len -1:    \n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i-1] \\\n",
    "                or query_ortho[i] == subject_ortho[i+1]:\n",
    "                    obs = obs + 1\n",
    "            elif i == max_up -1:\n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i-1]:\n",
    "                    obs = obs + 1\n",
    "            elif query_ortho[i] == subject_ortho[i]:\n",
    "                obs = obs + 1\n",
    "            elif i == 0:\n",
    "                if query_ortho[i] == subject_ortho[i] or query_ortho[i] == subject_ortho[i+1]:\n",
    "                    obs = obs + 1\n",
    "            \n",
    "        return obs, max_len"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#second iterations that allows for +1/=-1 difference once the first neighbour is the same\n",
    "#now we need to look at the pairing of the directions\n",
    "#and add stuff up\n",
    "if query_upstream_ortho[0] == subject_upstream_ortho[0]:   \n",
    "    obs_up, max_up = count_pairings(query_upstream_ortho, subject_upstream_ortho)       \n",
    "    #now look at the downstream\n",
    "    obs_down, max_down = count_pairings(query_downstream_ortho, subject_downstream_ortho)\n",
    "elif query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "    obs_up, max_up = count_pairings(query_upstream_ortho, subject_downstream_ortho)\n",
    "    obs_down, max_down = count_pairings(query_downstream_ortho, subject_upstream_ortho)\n",
    "print('obs_up %i, max_up %i, obs_down %i, max_down %i' % (obs_up, max_up, obs_down, max_down))\n",
    "print('sub_up %s\\nquery_up %s\\nsub_down %s\\nquery_down %s' % (subject_upstream_ortho, query_upstream_ortho,subject_downstream_ortho,query_downstream_ortho ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#second iterations that allows for +1/=-1 difference once the first neighbour is the same\n",
    "#now we need to look at the pairing of the directions\n",
    "#and add stuff up\n",
    "if query_upstream_ortho[0] == subject_upstream_ortho[0]:   \n",
    "    obs_up, max_up = count_pairings(query_upstream_ortho, subject_upstream_ortho, False)       \n",
    "    #now look at the downstream\n",
    "    obs_down, max_down = count_pairings(query_downstream_ortho, subject_downstream_ortho, False)\n",
    "elif query_upstream_ortho[0] == subject_downstream_ortho[0]:\n",
    "    obs_up, max_up = count_pairings(query_upstream_ortho, subject_downstream_ortho, False)\n",
    "    obs_down, max_down = count_pairings(query_downstream_ortho, subject_upstream_ortho, False)\n",
    "print('obs_up %i, max_up %i, obs_down %i, max_down %i' % (obs_up, max_up, obs_down, max_down))\n",
    "print('sub_up %s\\nquery_up %s\\nsub_down %s\\nquery_down %s' % (subject_upstream_ortho, query_upstream_ortho,subject_downstream_ortho,query_downstream_ortho ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_neighbours(first_orthologs[0], SUBJECT_GENOME_GENE_BED6, n=n, direction='down')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "first_orthologs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_upstream_ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "subject_upstream_ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_downstream_ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "subject_downstream_ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "line.split(':')[1].strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_neighbours('Pst104E_00002', QUERY_GENOME_GENE_BED6, n=10, direction = 'down')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "bed_6_header = ['chrom', 'start', 'stop', 'gene_id', 'phase', 'strand']\n",
    "bed_df = pd.read_csv(QUERY_GENOME_GENE_BED6, sep='\\t', header=None, names=bed_6_header)\n",
    "gene_id = pd.read_csv(QUERY_SELECT_GENE_BED6, sep='\\t', header=None, names=bed_6_header).loc[0,['gene_id']]['gene_id']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "n=5\n",
    "direction = 'up'\n",
    "gene_index = bed_df[bed_df['gene_id'] == gene_id].index[0]\n",
    "contig = bed_df.loc[gene_index, ['chrom']]['chrom']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "contig_index = bed_df[bed_df['chrom']== contig].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_id = 'Pst104E_00002'\n",
    "gene_index = bed_df[bed_df['gene_id'] == gene_id].index[0]\n",
    "contig = bed_df.loc[gene_index, ['chrom']]['chrom']\n",
    "contig_index = bed_df[bed_df['chrom']== contig].index\n",
    "direction = 'down'\n",
    "if direction == 'up':\n",
    "    index_list = []\n",
    "    if (gene_index+(n)) in contig_index:\n",
    "        for i in range(gene_index+1, gene_index+(n+1)):\n",
    "            index_list.append(i)\n",
    "    else:\n",
    "        for i in range(gene_index+1, contig_index[-1]+1):\n",
    "            index_list.append(i)\n",
    "    print(bed_df.loc[index_list, ['gene_id']]['gene_id'].tolist())\n",
    "if direction == 'down':\n",
    "    index_list = []\n",
    "    if (gene_index-n) in contig_index:\n",
    "        for i in range(gene_index-(n), gene_index):\n",
    "            index_list.append(i)\n",
    "    else:\n",
    "        for i in range(contig_index[0], gene_index):\n",
    "            index_list.append(i)\n",
    "    index_list.reverse()\n",
    "    print(bed_df.loc[index_list, ['gene_id']]['gene_id'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_index-n in contig_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "index_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_index[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "[gene_index: gene_index+6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
